<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:53 NZDT 2016 -->
<title>CombinedAlignmentAndSequenceDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="CombinedAlignmentAndSequenceDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/CombinedAlignmentAndSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="CombinedAlignmentAndSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations.sequence</div>
<h2 title="Class CombinedAlignmentAndSequenceDocument" class="title">Class CombinedAlignmentAndSequenceDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.sequence.CombinedAlignmentAndSequenceDocument</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndAminoAcidSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndAminoAcidSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndNucleotideGraphSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndNucleotideSequenceDocument</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="strong">CombinedAlignmentAndSequenceDocument</span>
extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">A wrapper to both an alignment and a stand-alone sequence that appears as a single EditableSequenceDocument. This is useful
 in cases where we have an alignment, but want it to appear as either a sequence or an alignment to other areas
 of Geneious. For example, a blast summary pairwse alignment document should appear as a sequence (for situations
 where the user wants to align their blast summaries), but it should also appear as a pairwise alignment in the sequence viewer.
 Once a full document has been downloaded from blast, the sequence viewer can view this as both the pairwise alignment
 of the subsequence, or as the full downloaded sequence.

 This class should not be instantiated directly. Instead, one of <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndAminoAcidSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>CombinedAlignmentAndAminoAcidSequenceDocument</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>CombinedAlignmentAndNucleotideSequenceDocument</code></a> should be used.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD">GENOME_SEQUENCE_THRESHOLD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier</th>
<th class="colLast" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#CombinedAlignmentAndSequenceDocument()">CombinedAlignmentAndSequenceDocument</a></strong>()</code>
<div class="block">XML Deserialization constructor.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#CombinedAlignmentAndSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.EditableSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean, boolean)">CombinedAlignmentAndSequenceDocument</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>&nbsp;sequenceDocument,
                                    <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignmentDocument,
                                    boolean&nbsp;obtainSharedPropertiesFromSequence,
                                    boolean&nbsp;showSequenceView)</code>
<div class="block">Construct a new CombinedAlignmentAndSequenceDocument.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Add a sequence to the end of the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#asJeblAlignment(boolean)">asJeblAlignment</a></strong>(boolean&nbsp;silentlyStripDuplicates)</code>
<div class="block">Get alignment as a Jebl alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a></strong>()</code>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canSetAnnotations()">canSetAnnotations</a></strong>()</code>
<div class="block">Returns true if this alignment can have its annotations on sequences (including the consensus sequence) edited alone via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>
 Note that some alignment implementations may return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a> and allow updating their annotations via
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>SequenceAlignmentDocument.updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a> even if they return false from this method.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canSetAnnotationsOnConsensus()">canSetAnnotationsOnConsensus</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canSetSequenceNames()">canSetSequenceNames</a></strong>()</code>
<div class="block">Specifies whether this alignment supports changing the sequence names.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getAlignmentDocument()">getAlignmentDocument</a></strong>()</code>
<div class="block">Get the alignment document contained within this document</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getAnnotationsOnConsensus()">getAnnotationsOnConsensus</a></strong>()</code>
<div class="block">Gets annotations that are associated with the consensus sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getCharSequence()">getCharSequence</a></strong>()</code>
<div class="block">Get the sequence as a CharSequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getContigReferenceSequenceIndex()">getContigReferenceSequenceIndex</a></strong>()</code>
<div class="block">Get the index (if any) of the reference sequence in a contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getCreationDate()">getCreationDate</a></strong>()</code>
<div class="block">Get the date this document was created.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getDescription()">getDescription</a></strong>()</code>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getDisplayableFields()">getDisplayableFields</a></strong>()</code>
<div class="block">A list of columns for display in the main document panel.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getFieldValue(java.lang.String)">getFieldValue</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fieldCodeName)</code>
<div class="block">Get the value of a particular field.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getName()">getName</a></strong>()</code>
<div class="block">A short name (preferably one or two words) for the document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>
<div class="block">Number of sequences in this alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getReferencedDocument(int)">getReferencedDocument</a></strong>(int&nbsp;sequenceIndexInAlignment)</code>
<div class="block">Get an original sequence document that this alignment was made from.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getReferencedSequence(int)">getReferencedSequence</a></strong>(int&nbsp;sequenceIndexInAlignment)</code>
<div class="block">Get an original sequence document that this alignment was made from.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequence(int)">getSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Gets one row from the alignment as a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a></strong>()</code>
<div class="block">Gets sequence annotations which are directly on the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceAsJeblSequence(int)">getSequenceAsJeblSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Gets a sequence as a jebl sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceDocument()">getSequenceDocument</a></strong>()</code>
<div class="block">Get the sequence document contained within this document</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceLength()">getSequenceLength</a></strong>()</code>
<div class="block">Returns the length of the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequences()">getSequences</a></strong>()</code>
<div class="block">Get the alignment as a list of sequences (Convenience function).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceString()">getSequenceString</a></strong>()</code>
<div class="block">Get the sequence as a string.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getURN()">getURN</a></strong>()</code>
<div class="block">A Universal Resource Name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isCircular()">isCircular</a></strong>()</code>
<div class="block">Returns true if this sequence is circular.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isContig()">isContig</a></strong>()</code>
<div class="block">Returns false by default.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isEditable()">isEditable</a></strong>()</code>
<div class="block">true if this alignment is editable.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a></strong>()</code>
<div class="block">True if this alignment was built using free end gaps.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isReferencedDocumentReversed(int)">isReferencedDocumentReversed</a></strong>(int&nbsp;index)</code>
<div class="block">Sequence direction.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#removeSequence(int)">removeSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Removes the specified sequence from the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#reorderSequences(int[])">reorderSequences</a></strong>(int[]&nbsp;ordering)</code>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setAnnotations(int, java.util.List, boolean)">setAnnotations</a></strong>(int&nbsp;index,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
              boolean&nbsp;allowModifyingReferencedDocuments)</code>
<div class="block">Updates just the annotations on a sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setAnnotations(java.util.List)">setAnnotations</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets new annotations on this document, replacing the existing ones.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setAnnotationsOnConsensus(java.util.List)">setAnnotationsOnConsensus</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets the annotations on the consensus sequence that will be returned from a later call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.getAnnotationsOnConsensus()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setCircular(boolean)">setCircular</a></strong>(boolean&nbsp;circular)</code>
<div class="block">Set whether or not this sequence is circular.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">setReferencedDocument</a></strong>(int&nbsp;sequenceIndexInAlignment,
                     <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</code>
<div class="block">Sets the sequence at the given sequenceIndexInAlignment in this alignment to refer to the given document as if it is the original source.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">setReferencedSequence</a></strong>(int&nbsp;sequenceIndexInAlignment,
                     <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence)</code>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)"><code>setReferencedSequence(sequenceIndexInAlignment, referencedSequence, -1</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets the sequence and annotations of this document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a></strong>(int&nbsp;index,
               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
               boolean&nbsp;allowModifyingReferencedDocuments)</code>
<div class="block">change the name of a sequence within the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#showSequenceView()">showSequenceView</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#toHTML()">toHTML</a></strong>()</code>
<div class="block">Provide a textual view of document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></strong>(int&nbsp;index,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
              <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
              boolean&nbsp;allowModifyingReferencedDocuments)</code>
<div class="block">Set new residues and related data for one sequence.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignmentIfMemoryAllows(boolean)">asJeblAlignmentIfMemoryAllows</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">canSetFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()">getCircularLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateExpectedDistance(int)">getMateExpectedDistance</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateIndex(int)">getMateIndex</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRealignedName(java.lang.String)">getRealignedName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocuments()">getReferencedDocuments</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequences()">getReferencedSequences</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRefinedName(java.lang.String)">getRefinedName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()">getSequencesInImmutableList</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceType()">getSequenceType</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isQueryCentricAlignment()">isQueryCentricAlignment</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">setFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setMates(int, int, int)">setMates</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">setReferencedSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="CombinedAlignmentAndSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.EditableSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CombinedAlignmentAndSequenceDocument</h4>
<pre>protected&nbsp;CombinedAlignmentAndSequenceDocument(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>&nbsp;sequenceDocument,
                                    <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignmentDocument,
                                    boolean&nbsp;obtainSharedPropertiesFromSequence,
                                    boolean&nbsp;showSequenceView)</pre>
<div class="block">Construct a new CombinedAlignmentAndSequenceDocument. This method should not be called directly. Instead
 use either <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndAminoAcidSequenceDocument.html#CombinedAlignmentAndAminoAcidSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean, boolean)"><code>CombinedAlignmentAndAminoAcidSequenceDocument</code></a>
 or
 <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html#CombinedAlignmentAndNucleotideSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean, boolean)"><code>CombinedAlignmentAndNucleotideSequenceDocument</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceDocument</code> - the sequenceDocument to wrap.</dd><dd><code>alignmentDocument</code> - the alignmentDocument to wrap.</dd><dd><code>obtainSharedPropertiesFromSequence</code> - for properties which exist on both documents
 (e.g. name or description), if only one provides a non-null property, that is used.
 If both provide a non-null value, when this paremeter is true, the value from the sequence is used.
 If both provide a non-null value, when this paremeter is false, the value from the alignment is used.</dd><dd><code>showSequenceView</code> - True if the sequence viewer should display a sequence view on this document. A situation
 where this would be false would be for a Blast Summary result document, where the sequence document is identical
 to the sequence contained in the alignment, and so a sequence view would provide no further useful information.</dd></dl>
</li>
</ul>
<a name="CombinedAlignmentAndSequenceDocument()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>CombinedAlignmentAndSequenceDocument</h4>
<pre>protected&nbsp;CombinedAlignmentAndSequenceDocument()</pre>
<div class="block">XML Deserialization constructor. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>XMLSerializable</code></a></div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="showSequenceView()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>showSequenceView</h4>
<pre>public&nbsp;boolean&nbsp;showSequenceView()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if a when this document is viewed, a sequence view should be shown to the user in addition to the alignment view.</dd></dl>
</li>
</ul>
<a name="getSequenceDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceDocument</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>&nbsp;getSequenceDocument()</pre>
<div class="block">Get the sequence document contained within this document</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sequence document contained within this document</dd></dl>
</li>
</ul>
<a name="getAlignmentDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlignmentDocument</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;getAlignmentDocument()</pre>
<div class="block">Get the alignment document contained within this document</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the alignment document contained within this document</dd></dl>
</li>
</ul>
<a name="toHTML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toHTML</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toHTML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">PluginDocument</a></code></strong></div>
<div class="block">Provide a textual view of document. The contents should be at most 30K-60K characters. One reason is that this
 text is rendered using JTextPane, which is horrendously slow when handling large text. The more fundamental
 reason is that large amounts of text are generally not useful for a user. The whole idea of using a tool such as
 Geneious is to provide specialized viewers which make the large amount of bioinformatics data perceivable and
 comprehensible to a human user.
 <p/>
 Return an HTML body fragment (i.e. a fragment which sits inside a &lt;body&gt;&lt;/body&gt; tag). (Header,stylesheet and
 annotations are automatically added by GENEious code).
 <p/>
 Can optionally return an entire html tag set, as long as the first six characters are <code>&lt;html></code></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#toHTML()">toHTML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an HTML fragment of tag set, or null to have no text view.</dd></dl>
</li>
</ul>
<a name="getDisplayableFields()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisplayableFields</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&gt;&nbsp;getDisplayableFields()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">PluginDocument</a></code></strong></div>
<div class="block">A list of columns for display in the main document panel.
 This function only defines the columns available to display,
 not the particular values of these columns, which are returned by
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)"><code>PluginDocument.getFieldValue(String)</code></a>.
 <p/>
 Each column is specified by a Field. The fields should be obtained from a call to the appropriate
 Field.createXXXXField, where XXXX is the appropriate type.
 <p/>
 While it is not required, it is recommended that a PluginDocument
 handle returning a complete list of its displayable fields before
 the document has been properly constructed via a call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a>

 The returned list need not and should not be modifiable.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a list of document fields (or null to indicate no displayable fields)</dd></dl>
</li>
</ul>
<a name="getFieldValue(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFieldValue</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;getFieldValue(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fieldCodeName)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">PluginDocument</a></code></strong></div>
<div class="block">Get the value of a particular field.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>fieldCodeName</code> - field code. This should be the
      <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#getCode()"><code>code</code></a> of one of the fields
      returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()"><code>PluginDocument.getDisplayableFields()</code></a>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>value for a field or null if this document does not have a field with the given field code.
          The class of the returned value must be the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#getValueType()"><code>DocumentField.getValueType()</code></a> (or a subclass) of the corresponding DocumentField returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()"><code>PluginDocument.getDisplayableFields()</code></a>.</dd></dl>
</li>
</ul>
<a name="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getName()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">PluginDocument</a></code></strong></div>
<div class="block">A short name (preferably one or two words) for the document.
 <p/>
 If this PluginDocument is wrapped inside an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a> then this method
 will be overridden and delegate back to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getName()"><code>AnnotatedPluginDocument.getName()</code></a>. One way to look at it is
 that when you call PluginDocument.getName() you will get the current name of the document. If you are implementing PluginDocument.getName() then you are
 returning the default name for the document.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>document name. Cannot be null but may be an empty string.</dd></dl>
</li>
</ul>
<a name="getURN()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getURN</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;getURN()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">PluginDocument</a></code></strong></div>
<div class="block">A Universal Resource Name.
 <p/>
 Documents from external sources should set this to a unique name inside that service. Documents generated locally
 (such as from operations) and documents that don't have a unique identifier should return null since
 Geneious handles URNs for these.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the urn, or null if this document doesn't have a unique identifier</dd></dl>
</li>
</ul>
<a name="getCreationDate()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCreationDate</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;getCreationDate()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">PluginDocument</a></code></strong></div>
<div class="block">Get the date this document was created. This method may return null.
 Generally clients should use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getCreationDate()"><code>AnnotatedPluginDocument.getCreationDate()</code></a>
 instead as that will return either the date returned from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()"><code>PluginDocument.getCreationDate()</code></a> or if that returned null,
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getCreationDate()"><code>AnnotatedPluginDocument.getCreationDate()</code></a> will return the date the
 document was first instantiated.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the date this document was created or null to use the date the document was first instantiated.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD"><code>PluginDocument.MODIFIED_DATE_FIELD</code></a></dd></dl>
</li>
</ul>
<a name="getDescription()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">PluginDocument</a></code></strong></div>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a one line simple text description of the document, or null if no description is available.</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;element)
             throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - representation from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="getSequenceString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSequenceString()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">SequenceDocument</a></code></strong></div>
<div class="block">Get the sequence as a string. For nucleotide or amino acid sequences,
 each residue or ambiguity symbol is represented by one uppercase
 or lowercase character as per the IUPAC recommendations.
 <p/>
 For nucleotide sequences, any of the following are valid characters "ACGTURYMWSKBDHVN?-acgturymwskbdhvn"
 <p/>
 For amino acid sequences, any of the following are valid characters "ACDEFGHIKLMNPQRSTVWY*OUBJZX?-acdefghiklmnpqrstvwyoubjzx"

 If <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a> doesn't call this method, then one possible implementation
 for this method is the following:

 <p><code>return getCharSequence().toString();</code></p>

 <p>WARNING: It is generally a bad idea to use this method on anything but very short sequences. On large sequences (e.g. whole chromosomes) this method
 may take some time and may use significant amounts of memory. e.g. A String representation of human chr1 requires 500 MB of memory. It is preferable to use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a> instead.</p></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">getSequenceString</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence as a string; must not return null.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLength</h4>
<pre>public&nbsp;int&nbsp;getSequenceLength()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">SequenceDocument</a></code></strong></div>
<div class="block">Returns the length of the sequence. This returns exactly the same as
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a>.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>length()</code></a>,
 but may be more performant on some implementations.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">getSequenceLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>The length of this document's sequence.

 Note that if a class implements both this interface and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>,
 the return value of this method must be consistent with the contracts from both
 interfaces.</dd></dl>
</li>
</ul>
<a name="getCharSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCharSequence</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&nbsp;getCharSequence()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">SequenceDocument</a></code></strong></div>
<div class="block">Get the sequence as a CharSequence. Used as an alternative to
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a>
 in situations where a String is unnecessary and could potentially
 use too much memory, or where we want to efficiently calculate the terminal gap lengths.

 The returned CharSequence must always return consistent
 values from any of the CharSequence methods
 (e.g. <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>CharSequence.length()</code></a> and  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#charAt(int)" title="class or interface in java.lang"><code>CharSequence.charAt(int)</code></a>.
 If a SequenceDocument is editable and changes are made to it,
 it is required that the original CharSequence obtained from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a>
 not change. A client of the SequenceDocument is only able to obtain
 the changes by making a fresh call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a>.

 If <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a> doesn't call this method, then it is possible to
 implement this method as follows:

 <p><code>return SequenceCharSequence.valueOf(getSequenceString());</code></p>

 <p>However, it is recommended that a more memory efficient representation is used
 where possible; in that case, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a> can be implemented
 based on this method (see its javadoc).</p>
 <p>Some implementations of this method may choose to throw a RuntimeException whose cause is an XMLSerializationException if there is
 insufficient memory to load the sequence. In most situations (e.g. from a DocumentOperation or DocumentFileImporter) catching
 this exception is unnecessary as core Geneious will have either pre-loaded the sequence or will catch and unwrap this exception itself gracefully.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">getCharSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence as a SequenceCharSequence; must not return null.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceAnnotations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAnnotations</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getSequenceAnnotations()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">SequenceDocument</a></code></strong></div>
<div class="block">Gets sequence annotations which are directly on the sequence.
 <b>This does not include annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>sequence tracks</code></a>.</b>
 <br><br>
 To get all annotations, including those on sequence tracks, use
 <a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAndTrackAnnotations</code></a>

 This method may return an unmodifiable list and must not return the list which is used internally by the SequenceDocument to store the annotations
 (ie modifying the returned list should not modify the annotations on the SequenceDocument directly)<br/><br/>

 <b>WARNING:</b> this list may not include all SequenceAnnotations represented as annotations in the sequence viewer.
 One such case is with trim annotations, which should be found using
 <a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAnnotationsIncludingImmutableSequencesTrims(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAnnotationsIncludingImmutableSequencesTrims(SequenceDocument)</code></a>.<br/><br/>

 This may also return annotations that are not visible in the sequence viewer, such as <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION"><code>SequenceAnnotation.TYPE_EXTRACTED_REGION</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a list of SequenceAnnotations; must not return null, but may return an
         empty list (e.g. <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collections.html?is-external=true#emptyList()" title="class or interface in java.util"><code>Collections.emptyList()</code></a>).</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAndTrackAnnotations(SequenceDocument)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)"><code>SequenceDocumentWithEditableAnnotations.setAnnotations(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="isCircular()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isCircular</h4>
<pre>public&nbsp;boolean&nbsp;isCircular()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()">SequenceDocument</a></code></strong></div>
<div class="block">Returns true if this sequence is circular.
 <p/>
 A circular sequence is one where the end points touch forming one continuous
 sequence without ends.
 For information on how annotations are handled in circular sequences, see.
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getFrom()"><code>SequenceAnnotationInterval.getFrom()</code></a></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()">isCircular</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this sequence is circular.</dd></dl>
</li>
</ul>
<a name="setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceAndAnnotations</h4>
<pre>public&nbsp;void&nbsp;setSequenceAndAnnotations(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)
                               throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">EditableSequenceDocument</a></code></strong></div>
<div class="block">Sets the sequence and annotations of this document. <b>This does not replace annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a></b>
 <br><br>
 If this changes the length of the sequence, this is only allowed if this document is not a NucleotideGraphSequence
 (because the graph site positions and qualities would have to be updated too by calling
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html#setSequenceAndAnnotationsAndGraph(java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)"><code>EditableNucleotideGraphSequenceDocument.setSequenceAndAnnotationsAndGraph(CharSequence, java.util.List, NucleotideGraph)</code></a>.

 Hence, it is illegal to change the length of the sequence through this method if
 this is instanceof NucleotideGraphSequenceDocument;
 Therefore, implementations of this method should start with the following code:

 <pre>
 if (this instanceof NucleotideGraphSequenceDocument && sequence.length() != getSequenceLength()) {
       throw new IllegalArgumentException("Can't change sequence length without updating graph for " + getClass());
 }
 </pre></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - new residues; NOTE: This CharSequence must not change its length or characters after it was passed to this method!</dd><dd><code>annotations</code> - new annotations</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></code></dd></dl>
</li>
</ul>
<a name="setAnnotations(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotations</h4>
<pre>public&nbsp;void&nbsp;setAnnotations(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)">SequenceDocumentWithEditableAnnotations</a></code></strong></div>
<div class="block">Sets new annotations on this document, replacing the existing ones.
 This does not replace annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)">setAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>annotations</code> - New annotations to set on this document.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()"><code>SequenceDocument.getSequenceAnnotations()</code></a></dd></dl>
</li>
</ul>
<a name="setCircular(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCircular</h4>
<pre>public&nbsp;void&nbsp;setCircular(boolean&nbsp;circular)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setCircular(boolean)">EditableSequenceDocument</a></code></strong></div>
<div class="block">Set whether or not this sequence is circular.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setCircular(boolean)">setCircular</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>circular</code> - true to set this sequence as circular.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public&nbsp;int&nbsp;getNumberOfSequences()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Number of sequences in this alignment</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()">getNumberOfSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>number of sequences</dd></dl>
</li>
</ul>
<a name="getSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;getSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Gets one row from the alignment as a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>. As this sequence
 is an alignment row, it will normally contain gaps.

 If implementation stores the URN of the original sequence document, see 
 the note in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)">getSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence at the specified index.</dd></dl>
</li>
</ul>
<a name="getSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequences()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get the alignment as a list of sequences (Convenience function).

 The default implementation returns an unmodifiable list view backed by
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)"><code>SequenceAlignmentDocument.getSequence(int)</code></a>.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()">getSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a list of sequences</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()"><code>SequenceAlignmentDocument.getSequencesInImmutableList()</code></a></dd></dl>
</li>
</ul>
<a name="getReferencedDocument(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedDocument</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;getReferencedDocument(int&nbsp;sequenceIndexInAlignment)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get an original sequence document that this alignment was made from.

 This may be null if the alignment was not generated from
 sequence documents, or if the alignment has been user modified, so the reference
 to the original sequence document would not be relevant.

 Note that when an alignment implements this by storing the sequence URN and loading it
 on request it needs to check that the seqeunce residues has not been changed by a later edit
 operation. In that case the seqeunce document is no longer valid for that alignment and a null should be returned.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)">getReferencedDocument</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the index of the sequence in the alignment to get the original document for. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the original document or null if this alignment does not a reference an original document for this sequence index. The default implementation of this method returns null</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)"><code>SequenceAlignmentDocument.getReferencedSequence(int)</code></a></dd></dl>
</li>
</ul>
<a name="getReferencedSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedSequence</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getReferencedSequence(int&nbsp;sequenceIndexInAlignment)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get an original sequence document that this alignment was made from.

 This may be null if the alignment was not generated from
 sequence documents, or if the alignment has been user modified, so the reference
 to the original sequence document would not be relevant.

 Note that when an alignment implements this by storing the sequence URN and loading it
 on request it needs to check that the seqeunce residues has not been changed by a later edit
 operation. In that case the seqeunce document is no longer valid for that alignment and a null should be returned.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)">getReferencedSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the index of the sequence in the alignment to get the original document for. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the original document or null if this alignment does not a reference an original document for this sequence index. The default implementation of this method forwards to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a></dd></dl>
</li>
</ul>
<a name="getContigReferenceSequenceIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getContigReferenceSequenceIndex</h4>
<pre>public&nbsp;int&nbsp;getContigReferenceSequenceIndex()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get the index (if any) of the reference sequence in a contig. Note - despite the name, this method can also be used to identify
 the query sequence in a query centric alignment.
 <p/>
 The default implementation returns -1.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()">getContigReferenceSequenceIndex</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the index (in the range [0,<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1]) or -1 to if this contig has no reference sequence.</dd></dl>
</li>
</ul>
<a name="setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedSequence</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedSequence(int&nbsp;sequenceIndexInAlignment,
                            <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)"><code>setReferencedSequence(sequenceIndexInAlignment, referencedSequence, -1</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">setReferencedSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the sequenceIndexInAlignment of the sequence referred to. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd><dd><code>referencedSequence</code> - the document to reference</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this document was successfully set as a referenced document</dd></dl>
</li>
</ul>
<a name="isReferencedDocumentReversed(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReferencedDocumentReversed</h4>
<pre>public&nbsp;boolean&nbsp;isReferencedDocumentReversed(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isReferencedDocumentReversed(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sequence direction.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isReferencedDocumentReversed(int)">isReferencedDocumentReversed</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a> is not null and it is in the reverse direction of its sequence string in the alignment.</dd></dl>
</li>
</ul>
<a name="isEditable()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEditable</h4>
<pre>public&nbsp;boolean&nbsp;isEditable()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">true if this alignment is editable.
 if it is, it must implement <br>
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a>, <br>
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)"><code>SequenceAlignmentDocument.removeSequence(int)</code></a> , <br>
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>SequenceAlignmentDocument.updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a>, <br>
 and may want to implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)"><code>SequenceAlignmentDocument.setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)</code></a> <br><br>

 An implementation may also want to return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()"><code>SequenceAlignmentDocument.canSetAnnotations()</code></a> to allow updating just the annotations via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()">isEditable</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment is editable.</dd></dl>
</li>
</ul>
<a name="reorderSequences(int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderSequences</h4>
<pre>public&nbsp;void&nbsp;reorderSequences(int[]&nbsp;ordering)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.
 If ordering[i]==j, then the sequence at index 'i' should be the sequence previously at index 'j'
 For example, if ordering = {0,1,3,4,2}
 Then:
 sequence at index 2 is moved to index 4
 sequence at index 3 is moved to index 2
 sequence at index 4 is moved to index 3

 This function may only be called on alignments that return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])">reorderSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>ordering</code> - Each integer between 0 and ordering.length - 1 inclusive should occur exactly once in this array.</dd></dl>
</li>
</ul>
<a name="removeSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeSequence</h4>
<pre>public&nbsp;void&nbsp;removeSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Removes the specified sequence from the alignment. All other sequences should
 maintain their original ordering.

 This function may only be called on alignments that return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)">removeSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be removed. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd></dl>
</li>
</ul>
<a name="canAddAndRemoveSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canAddAndRemoveSequences</h4>
<pre>public&nbsp;boolean&nbsp;canAddAndRemoveSequences()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canAddAndRemoveSequences()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if adding and removing sequences is supported, false otherwise.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceAlignmentDocument.addSequence(SequenceDocument)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)"><code>SequenceAlignmentDocument.removeSequence(int)</code></a></dd></dl>
</li>
</ul>
<a name="addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequence</h4>
<pre>public&nbsp;void&nbsp;addSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Add a sequence to the end of the alignment. <code>sequence</code> should be the
 gapped (aligned) sequence.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - sequence to add</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)"><code>SequenceAlignmentDocument.setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)</code></a></dd></dl>
</li>
</ul>
<a name="updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>updateSequence</h4>
<pre>public&nbsp;void&nbsp;updateSequence(int&nbsp;index,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                  <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
                  boolean&nbsp;allowModifyingReferencedDocuments)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Set new residues and related data for one sequence.

 This function may only be called on alignments that return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a>
 (otherwise, the method should throw <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang"><code>UnsupportedOperationException</code></a>).</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be updated. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd><dd><code>sequence</code> - the new residues</dd><dd><code>annotations</code> - a list of the annotations</dd><dd><code>graph</code> - Must be null if the original sequence at the specified index is not a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>.
 Otherwise (i.e. if the sequence was a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>), specifies the new
 graph to use for that sequence.</dd><dd><code>allowModifyingReferencedDocuments</code> - if the change being made to the sequence would affect the unaligned version of it
 (for example, the sequence residues are changing rather than just the gaps)
 and the alignment stores a reference an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be modified.
 If this parameter is not true and the change would have otherwise modified the original SequenceDocument,
 then any reference to the original document will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedDocument</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedDocument(int&nbsp;sequenceIndexInAlignment,
                            <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sets the sequence at the given sequenceIndexInAlignment in this alignment to refer to the given document as if it is the original source.
 This document must contain a sequence consisting of residues matching the aligned sequence stored at <code> sequenceIndexInAlignment </code>,
 with gap residues removed.
 This method may be used when the user chooses to extract a sequence from an alignment, so the alignment
 should now be updated to reference the extracted sequence.
 Any alignment implementations may choose to not implement this method, in which case of the default implementation is used which always returns false.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">setReferencedDocument</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the sequenceIndexInAlignment of the sequence referred to. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd><dd><code>document</code> - the document to reference</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this document was successfully set as a referenced document. Note that some implementations may allow setting the reference and return true,
 but then discard the reference next time it is requested if the sequence from the referenced document does not correspond to sequence in the alignment.
 Call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a> to check if it really worked.</dd></dl>
</li>
</ul>
<a name="asJeblAlignment(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblAlignment</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a>&nbsp;asJeblAlignment(boolean&nbsp;silentlyStripDuplicates)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignment(boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get alignment as a Jebl alignment. Throws an OutOfMemoryError if there is not enough memory to do so.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignment(boolean)">asJeblAlignment</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>silentlyStripDuplicates</code> - Unlike Geneious alignments, Jebl alignments cannot contain sequences
 with the same names. If this parameter is true, duplicate sequences will be absent from the returned alignment.
 if this parameter is false, sequences will be given unique names by appending "1", "2", etc .

 This function is implemented for all alignments. Sub-classes need not implement it, but may wish to
 if they internally store their alignment in jebl format already.</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequences as a Jebl alignment</dd></dl>
</li>
</ul>
<a name="getSequenceAsJeblSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAsJeblSequence</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&nbsp;getSequenceAsJeblSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceAsJeblSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Gets a sequence as a jebl sequence</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceAsJeblSequence(int)">getSequenceAsJeblSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to get. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequence as a jebl sequence</dd></dl>
</li>
</ul>
<a name="isContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isContig</h4>
<pre>public&nbsp;boolean&nbsp;isContig()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isContig()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Returns false by default. If true then this alignment may should be
 treated as a contig (may affect e.g. the way consensus sequence
 are calculated).
 <p/>
 If this is overridden to return true by classes other than
 <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocument</code></a>
 then the document will not be given the contig icon (to set non default icons for
 a document see <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentType.html#OVERRIDE_DOCUMENT_TYPE_KEY"><code>DocumentType.OVERRIDE_DOCUMENT_TYPE_KEY</code></a>.
 <p/>
 Use <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)"><code>DefaultAlignmentDocument.setContig(boolean)</code></a>
 to create a contig.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isContig()">isContig</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment should be treated as a contig</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)"><code>DefaultAlignmentDocument.setContig(boolean)</code></a></dd></dl>
</li>
</ul>
<a name="canSetSequenceNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetSequenceNames</h4>
<pre>public&nbsp;boolean&nbsp;canSetSequenceNames()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Specifies whether this alignment supports changing the sequence names.
 If this method returns true, then this alignment must implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)"><code>SequenceAlignmentDocument.setSequenceName(int, String, boolean)</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()">canSetSequenceNames</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment supports changing the sequence names.</dd></dl>
</li>
</ul>
<a name="setSequenceName(int, java.lang.String, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceName</h4>
<pre>public&nbsp;void&nbsp;setSequenceName(int&nbsp;index,
                   <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
                   boolean&nbsp;allowModifyingReferencedDocuments)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">change the name of a sequence within the alignment.
 This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()"><code>SequenceAlignmentDocument.canSetSequenceNames()</code></a> returns true.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - index of the sequence name to change</dd><dd><code>newName</code> - the new name for the sequence.</dd><dd><code>allowModifyingReferencedDocuments</code> - if the alignment stores a reference to an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be renamed too.
 Otherwise,  any existing reference to the original SequenceDocument will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="isFreeEndGapAlignment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isFreeEndGapAlignment</h4>
<pre>public&nbsp;boolean&nbsp;isFreeEndGapAlignment()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isFreeEndGapAlignment()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">True if this alignment was built using free end gaps. This means gaps at either end of the sequences are
 not penalized.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment was built using free end gaps. The default implementation returns true.</dd></dl>
</li>
</ul>
<a name="canSetAnnotations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetAnnotations</h4>
<pre>public&nbsp;boolean&nbsp;canSetAnnotations()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Returns true if this alignment can have its annotations on sequences (including the consensus sequence) edited alone via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>
 Note that some alignment implementations may return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a> and allow updating their annotations via
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>SequenceAlignmentDocument.updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a> even if they return false from this method.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()">canSetAnnotations</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment can have its annotations edited alone via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.canSetAnnotationsOnConsensus()</code></a></dd></dl>
</li>
</ul>
<a name="setAnnotations(int, java.util.List, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotations</h4>
<pre>public&nbsp;void&nbsp;setAnnotations(int&nbsp;index,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                  boolean&nbsp;allowModifyingReferencedDocuments)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Updates just the annotations on a sequence. This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()"><code>SequenceAlignmentDocument.canSetAnnotations()</code></a> returns true<p/>
 This is used for updating non-track annotations only. Track annotations are handled via a different API. To update or add tracks check if
 a sequence in the alignment implements <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)">setAnnotations</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the 0-based index of the sequence to set annotations on, or -1 to set annotations on the consensus sequence.</dd><dd><code>annotations</code> - the new annotations for this sequence, which will replace any existing annotations.</dd><dd><code>allowModifyingReferencedDocuments</code> - if this is true  the original referenced sequence document will be modified to use these annotations
 If this parameter is not true and then any reference to the original document will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="canSetAnnotationsOnConsensus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetAnnotationsOnConsensus</h4>
<pre>public&nbsp;boolean&nbsp;canSetAnnotationsOnConsensus()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()">canSetAnnotationsOnConsensus</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment implementation supports calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="getAnnotationsOnConsensus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationsOnConsensus</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getAnnotationsOnConsensus()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Gets annotations that are associated with the consensus sequence.
 Note - this doesn't include consensus annotations in tracks. Consensus tracks can be obtained by checking if this alignment
 implements <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()">getAnnotationsOnConsensus</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>annotations that are associated with the consensus sequence. Never null but may be empty</dd></dl>
</li>
</ul>
<a name="setAnnotationsOnConsensus(java.util.List)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setAnnotationsOnConsensus</h4>
<pre>public&nbsp;void&nbsp;setAnnotationsOnConsensus(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sets the annotations on the consensus sequence that will be returned from a later call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.getAnnotationsOnConsensus()</code></a>.
 It is only legal to call this method if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.canSetAnnotationsOnConsensus()</code></a> returns true.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)">setAnnotationsOnConsensus</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>annotations</code> - the annotations to store on the consensus. Must not be null. This replaces any existing annotations on the consensus.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.getAnnotationsOnConsensus()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.canSetAnnotationsOnConsensus()</code></a></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/CombinedAlignmentAndSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="CombinedAlignmentAndSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
