<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:53 NZDT 2016 -->
<title>CombinedAlignmentAndNucleotideGraphSequenceDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="CombinedAlignmentAndNucleotideGraphSequenceDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/CombinedAlignmentAndNucleotideGraphSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndAminoAcidSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="CombinedAlignmentAndNucleotideGraphSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations.sequence</div>
<h2 title="Class CombinedAlignmentAndNucleotideGraphSequenceDocument" class="title">Class CombinedAlignmentAndNucleotideGraphSequenceDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">com.biomatters.geneious.publicapi.implementations.sequence.CombinedAlignmentAndSequenceDocument</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.sequence.CombinedAlignmentAndNucleotideGraphSequenceDocument</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableNucleotideGraphSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraphSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">CombinedAlignmentAndNucleotideGraphSequenceDocument</span>
extends <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndSequenceDocument</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableNucleotideGraphSequenceDocument</a></pre>
<div class="block">See documentation at <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>CombinedAlignmentAndSequenceDocument</code></a></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.ImmutableGraphProvider.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph.ImmutableGraphProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html#NICE_NAME">NICE_NAME</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD">GENOME_SEQUENCE_THRESHOLD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#CombinedAlignmentAndNucleotideGraphSequenceDocument()">CombinedAlignmentAndNucleotideGraphSequenceDocument</a></strong>()</code>
<div class="block">XML Deserialization constructor.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#CombinedAlignmentAndNucleotideGraphSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraphSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean, boolean)">CombinedAlignmentAndNucleotideGraphSequenceDocument</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraphSequenceDocument</a>&nbsp;sequenceDocument,
                                                   <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignmentDocument,
                                                   boolean&nbsp;obtainSharedPropertiesFromSequence,
                                                   boolean&nbsp;showSequenceView)</code>
<div class="block">Constructs a new combined alignment and sequence document.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#getChromatogramLength()">getChromatogramLength</a></strong>()</code>
<div class="block">The number of points for which this graph has values, which is typically larger than the
 length of the nucleotide sequence because a base isn't called at every position in the graph.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#getChromatogramPositionForResidue(int)">getChromatogramPositionForResidue</a></strong>(int&nbsp;residueIndex)</code>
<div class="block">The graph position (can be used as an argument to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)"><code>NucleotideGraph.getChromatogramValue(int, int)</code></a>) where the
 nucleotide at the specified residue index has been called.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#getChromatogramValue(int, int)">getChromatogramValue</a></strong>(int&nbsp;nucleotideStateNumber,
                    int&nbsp;graphPosition)</code>
<div class="block">Returns the graph value for the specified nucleotide and at the specified position.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#getSequenceQuality(int)">getSequenceQuality</a></strong>(int&nbsp;residueIndex)</code>
<div class="block">Returns a measurement of confidence in the residue reported at the specified residueIndex in the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#hasChromatogramPositionsForResidues()">hasChromatogramPositionsForResidues</a></strong>()</code>
<div class="block">Tests whether this NucleotideGraph contains information about the position of the base calls,
 i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#hasChromatogramValues(int)">hasChromatogramValues</a></strong>(int&nbsp;nucleotideStateNumber)</code>
<div class="block">Specifies whether it is legal to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)"><code>NucleotideGraph.getChromatogramValue(int, int)</code></a> for the specified nucleotide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#hasSequenceQualities()">hasSequenceQualities</a></strong>()</code>
<div class="block">Tests whether this graph contains information about the confidence in the associated sequence's
 residues, i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html#setSequenceAndAnnotationsAndGraph(java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)">setSequenceAndAnnotationsAndGraph</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                                 <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                                 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph)</code>
<div class="block">Replaces <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)"><code>EditableSequenceDocument.setSequenceAndAnnotations(CharSequence,java.util.List)</code></a>
 which you must not call on a NucleotideGraphSequenceDocument.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.implementations.sequence.CombinedAlignmentAndSequenceDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.implementations.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndSequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#asJeblAlignment(boolean)">asJeblAlignment</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canSetAnnotations()">canSetAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canSetAnnotationsOnConsensus()">canSetAnnotationsOnConsensus</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#canSetSequenceNames()">canSetSequenceNames</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#fromXML(org.jdom.Element)">fromXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getAlignmentDocument()">getAlignmentDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getAnnotationsOnConsensus()">getAnnotationsOnConsensus</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getCharSequence()">getCharSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getContigReferenceSequenceIndex()">getContigReferenceSequenceIndex</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getDescription()">getDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getName()">getName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getNumberOfSequences()">getNumberOfSequences</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getReferencedDocument(int)">getReferencedDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getReferencedSequence(int)">getReferencedSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequence(int)">getSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceAsJeblSequence(int)">getSequenceAsJeblSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceDocument()">getSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceLength()">getSequenceLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequences()">getSequences</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getSequenceString()">getSequenceString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getURN()">getURN</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isCircular()">isCircular</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isContig()">isContig</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isEditable()">isEditable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#isReferencedDocumentReversed(int)">isReferencedDocumentReversed</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#removeSequence(int)">removeSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#reorderSequences(int[])">reorderSequences</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setAnnotations(int, java.util.List, boolean)">setAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setAnnotations(java.util.List)">setAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setAnnotationsOnConsensus(java.util.List)">setAnnotationsOnConsensus</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setCircular(boolean)">setCircular</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">setReferencedDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">setReferencedSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#showSequenceView()">showSequenceView</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#toHTML()">toHTML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#toXML()">toXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignmentIfMemoryAllows(boolean)">asJeblAlignmentIfMemoryAllows</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">canSetFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()">getCircularLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateExpectedDistance(int)">getMateExpectedDistance</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateIndex(int)">getMateIndex</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRealignedName(java.lang.String)">getRealignedName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocuments()">getReferencedDocuments</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequences()">getReferencedSequences</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRefinedName(java.lang.String)">getRefinedName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()">getSequencesInImmutableList</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceType()">getSequenceType</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isQueryCentricAlignment()">isQueryCentricAlignment</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">setFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setMates(int, int, int)">setMates</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">setReferencedSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceLength()">getSequenceLength</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.EditableSequenceDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setCircular(boolean)">setCircular</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocumentWithEditableAnnotations">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)">setAnnotations</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">getCharSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">getSequenceLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">getSequenceString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()">isCircular</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="CombinedAlignmentAndNucleotideGraphSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraphSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CombinedAlignmentAndNucleotideGraphSequenceDocument</h4>
<pre>public&nbsp;CombinedAlignmentAndNucleotideGraphSequenceDocument(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraphSequenceDocument</a>&nbsp;sequenceDocument,
                                                   <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignmentDocument,
                                                   boolean&nbsp;obtainSharedPropertiesFromSequence,
                                                   boolean&nbsp;showSequenceView)</pre>
<div class="block">Constructs a new combined alignment and sequence document.
 See <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html#CombinedAlignmentAndSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.EditableSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean, boolean)"><code>CombinedAlignmentAndSequenceDocument</code></a>
 for documentation for the parameters to this method.</div>
</li>
</ul>
<a name="CombinedAlignmentAndNucleotideGraphSequenceDocument()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>CombinedAlignmentAndNucleotideGraphSequenceDocument</h4>
<pre>public&nbsp;CombinedAlignmentAndNucleotideGraphSequenceDocument()</pre>
<div class="block">XML Deserialization constructor. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>XMLSerializable</code></a></div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="setSequenceAndAnnotationsAndGraph(java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceAndAnnotationsAndGraph</h4>
<pre>public&nbsp;void&nbsp;setSequenceAndAnnotationsAndGraph(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                                     <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                                     <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph)
                                       throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html#setSequenceAndAnnotationsAndGraph(java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)">EditableNucleotideGraphSequenceDocument</a></code></strong></div>
<div class="block">Replaces <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)"><code>EditableSequenceDocument.setSequenceAndAnnotations(CharSequence,java.util.List)</code></a>
 which you must not call on a NucleotideGraphSequenceDocument. Does the same as that method
 normally would, but also sets a new graph to use (which must be consistent with the sequence,
 i.e. <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceLength()"><code>graph.getSequenceLength()</code></a>
  must be equal to <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>sequence.length()</code></a>
 It is recommended that an implementation call
 <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideGraphSequence.html#checkChromatogramValidity(com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraphSequenceDocument)"><code>DefaultNucleotideGraphSequence.checkChromatogramValidity(NucleotideGraphSequenceDocument)</code></a>
 before returning from this method, to ensure that the graph and sequence are consistent.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html#setSequenceAndAnnotationsAndGraph(java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)">setSequenceAndAnnotationsAndGraph</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableNucleotideGraphSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - New sequence to set on this document; must be consistent with the annotations
        and the graph (see above). Must not change length or characters after being passed to this method
        (as in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)"><code>EditableSequenceDocument.setSequenceAndAnnotations(CharSequence, java.util.List)</code></a>).</dd><dd><code>annotations</code> - New annotations to set on this document, with indices relative to the new sequence.</dd><dd><code>graph</code> - New graph to set on this document; must be consistent with the new sequence. The implementation must store a copy of this graph instead
        of a reference to the specified graph (using for eaxmple <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultNucleotideGraph.html#copyOf(com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)"><code>DefaultNucleotideGraph.copyOf(NucleotideGraph)</code></a>)</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if the sequence and graph are not consistent.</dd></dl>
</li>
</ul>
<a name="getChromatogramValue(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromatogramValue</h4>
<pre>public&nbsp;int&nbsp;getChromatogramValue(int&nbsp;nucleotideStateNumber,
                       int&nbsp;graphPosition)
                         throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IndexOutOfBoundsException.html?is-external=true" title="class or interface in java.lang">IndexOutOfBoundsException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)">NucleotideGraph</a></code></strong></div>
<div class="block">Returns the graph value for the specified nucleotide and at the specified position.
 If the graph is a chromatogram, this is a measure of the support for the specified nucleotide
 at this position. May only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasChromatogramValues(int)"><code>NucleotideGraph.hasChromatogramValues(int)</code></a> returns true for the
 specified nucleotide.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)">getChromatogramValue</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideStateNumber</code> - The nucleotide for which to retrieve the graph value, encoded as
        0 = A, 1 = C, 2 = G, 3 = T (alphabetic order, same as in <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Nucleotides.html?is-external=true" title="class or interface in jebl.evolution.sequences"><code>Nucleotides</code></a>)</dd><dd><code>graphPosition</code> - The position in the sequence for which to query the graph value,
        with 0 &lt;= graphPosition &lt; <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramLength()"><code>graph length</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>the graph value for the specified nucleotide and at the specified position.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IndexOutOfBoundsException.html?is-external=true" title="class or interface in java.lang">IndexOutOfBoundsException</a></code> - if graphPosition &lt; 0 or graphPosition &gt;= <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramLength()"><code>NucleotideGraph.getChromatogramLength()</code></a></dd></dl>
</li>
</ul>
<a name="hasChromatogramValues(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasChromatogramValues</h4>
<pre>public&nbsp;boolean&nbsp;hasChromatogramValues(int&nbsp;nucleotideStateNumber)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasChromatogramValues(int)">NucleotideGraph</a></code></strong></div>
<div class="block">Specifies whether it is legal to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)"><code>NucleotideGraph.getChromatogramValue(int, int)</code></a> for the specified nucleotide.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasChromatogramValues(int)">hasChromatogramValues</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideStateNumber</code> - A nucleotide state number encoded in the same way as for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)"><code>NucleotideGraph.getChromatogramValue(int, int)</code></a>,
        i.e. 0 = A, 1 = C, 2 = G, 3 = T</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if is legal to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)"><code>NucleotideGraph.getChromatogramValue(int, int)</code></a> for the specified nucleotide</dd></dl>
</li>
</ul>
<a name="getChromatogramPositionForResidue(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromatogramPositionForResidue</h4>
<pre>public&nbsp;int&nbsp;getChromatogramPositionForResidue(int&nbsp;residueIndex)
                                      throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IndexOutOfBoundsException.html?is-external=true" title="class or interface in java.lang">IndexOutOfBoundsException</a>,
                                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramPositionForResidue(int)">NucleotideGraph</a></code></strong></div>
<div class="block">The graph position (can be used as an argument to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramValue(int, int)"><code>NucleotideGraph.getChromatogramValue(int, int)</code></a>) where the
 nucleotide at the specified residue index has been called. This method may only be called if
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasChromatogramPositionsForResidues()"><code>NucleotideGraph.hasChromatogramPositionsForResidues()</code></a> returns true.

 The graph sites are guaranteed to have the same order as the sequence nucleotides. Therefore,
 any implementation must ensure that if <code>i &lt; j</code>, then
 <code>getChromatogramPositionForResidue(i) &lt;= getChromatogramPositionForResidue(j)</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramPositionForResidue(int)">getChromatogramPositionForResidue</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - A value <code>0 &lt;= residueIndex &lt; <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceLength()"><code>NucleotideGraph.getSequenceLength()</code></a></code> specifying
        the index in the nucleotide sequence for which to calculate the corresponding position in the graph.</dd>
<dt><span class="strong">Returns:</span></dt><dd>A value in the range 0 &lt;= value &lt; <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramLength()"><code>NucleotideGraph.getChromatogramLength()</code></a>, specifiying the
         graph position where the nucleotide that is at position <code>residueIndex</code> has been
         called. For positions <code>residueIndex</code> where the
         <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>sequence</code></a>
         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>length</code></a> consisting of values
         between 0 and getChromatogramLength()-1 inclusive. For a residueIndex where the sequence <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#charAt(int)" title="class or interface in java.lang"><code>holds</code></a>
         a '-', the returned site position must always be <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Integer.html?is-external=true#MIN_VALUE" title="class or interface in java.lang"><code>Integer.MIN_VALUE</code></a>.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IndexOutOfBoundsException.html?is-external=true" title="class or interface in java.lang">IndexOutOfBoundsException</a></code> - if residueIndex < 0 or residueIndex > <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceLength()"><code>NucleotideGraph.getSequenceLength()</code></a>.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></code> - if !hasChromatogramPositionsForResidues().</dd></dl>
</li>
</ul>
<a name="hasChromatogramPositionsForResidues()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasChromatogramPositionsForResidues</h4>
<pre>public&nbsp;boolean&nbsp;hasChromatogramPositionsForResidues()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasChromatogramPositionsForResidues()">NucleotideGraph</a></code></strong></div>
<div class="block">Tests whether this NucleotideGraph contains information about the position of the base calls,
 i.e. whether it is legal to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramPositionForResidue(int)"><code>NucleotideGraph.getChromatogramPositionForResidue(int)</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasChromatogramPositionsForResidues()">hasChromatogramPositionsForResidues</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this NucleotideGraph contains information about the position of the base calls.</dd></dl>
</li>
</ul>
<a name="getSequenceQuality(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceQuality</h4>
<pre>public&nbsp;int&nbsp;getSequenceQuality(int&nbsp;residueIndex)
                       throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IndexOutOfBoundsException.html?is-external=true" title="class or interface in java.lang">IndexOutOfBoundsException</a>,
                              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)">NucleotideGraph</a></code></strong></div>
<div class="block">Returns a measurement of confidence in the residue reported at the specified residueIndex in the sequence.
 The measure is on a logaritmic scale: P(call correct) = 1 - 10^(-v/10)
 where v is the quality returned from this method.
 <p/>
 If the sequence contains a gap at the specified position, this method should return <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Integer.html?is-external=true#MIN_VALUE" title="class or interface in java.lang"><code>Integer.MIN_VALUE</code></a>
 if it is an end gap. If it is an internal gap, it should return the average quality of the non-gap positions
 on either side of the gap.
 <p/>
 This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()"><code>NucleotideGraph.hasSequenceQualities()</code></a> returns true.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)">getSequenceQuality</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - A value <code>0 &lt;= residueIndex &lt; <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceLength()"><code>NucleotideGraph.getSequenceLength()</code></a></code> specifying
        the index in the nucleotide sequence for which to calculate the quality of the graph.</dd>
<dt><span class="strong">Returns:</span></dt><dd>The quality score v for the residue at the specified index in the nucleotide sequence,
         which indicates the reliability of the nucleotide on a logarithmic scale as described above.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IndexOutOfBoundsException.html?is-external=true" title="class or interface in java.lang">IndexOutOfBoundsException</a></code> - if residueIndex < 0 or residueIndex > <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceLength()"><code>NucleotideGraph.getSequenceLength()</code></a>.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></code> - if !hasSequenceQualities().</dd></dl>
</li>
</ul>
<a name="hasSequenceQualities()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasSequenceQualities</h4>
<pre>public&nbsp;boolean&nbsp;hasSequenceQualities()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()">NucleotideGraph</a></code></strong></div>
<div class="block">Tests whether this graph contains information about the confidence in the associated sequence's
 residues, i.e. whether it is legal to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)"><code>NucleotideGraph.getSequenceQuality(int)</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()">hasSequenceQualities</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if it is legal to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)"><code>NucleotideGraph.getSequenceQuality(int)</code></a> on this graph.</dd></dl>
</li>
</ul>
<a name="getChromatogramLength()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getChromatogramLength</h4>
<pre>public&nbsp;int&nbsp;getChromatogramLength()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramLength()">NucleotideGraph</a></code></strong></div>
<div class="block">The number of points for which this graph has values, which is typically larger than the
 length of the nucleotide sequence because a base isn't called at every position in the graph.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getChromatogramLength()">getChromatogramLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the length of this graph, or 0 if this graph has no graph values for any nucleotide.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/CombinedAlignmentAndNucleotideGraphSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndAminoAcidSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="CombinedAlignmentAndNucleotideGraphSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
