<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:53 NZDT 2016 -->
<title>DefaultSequenceDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="DefaultSequenceDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.Cache.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations.sequence</div>
<h2 title="Class DefaultSequenceDocument" class="title">Class DefaultSequenceDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">com.biomatters.geneious.publicapi.documents.AbstractPluginDocument</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation.Provider</a></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultAminoAcidSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultNucleotideSequence</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="strong">DefaultSequenceDocument</span>
extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation.Provider</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">A default implementation of a sequence document.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.Cache.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument.Cache</a></strong></code>
<div class="block">Used for controlling caching of the char sequence and annotations controlled from this class.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD">GENOME_SEQUENCE_THRESHOLD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier</th>
<th class="colLast" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#DefaultSequenceDocument()">DefaultSequenceDocument</a></strong>()</code>
<div class="block">Construct a DefaultSequenceDocument with no arguments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#DefaultSequenceDocument(jebl.evolution.sequences.Sequence)">DefaultSequenceDocument</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&nbsp;jeblSequence)</code>
<div class="block">Construct a new DefaultSequenceDocument using a Sequence from JEBL.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#DefaultSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.Date)">DefaultSequenceDocument</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created)</code>
<div class="block">Construct a new DefaultSequenceDocument from an existing
 SequenceDocument.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#DefaultSequenceDocument(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)">DefaultSequenceDocument</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#DefaultSequenceDocument(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date, com.biomatters.geneious.publicapi.documents.URN)">DefaultSequenceDocument</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created,
                       <a href="../../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;urn)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#addSequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">addSequenceAnnotation</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>
<div class="block">Add a sequence annotation to this document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#annotationsFromXML(org.jdom.Element)">annotationsFromXML</a></strong>(org.jdom.Element&nbsp;annotationsElement)</code>
<div class="block">Return a List of SequenceAnnotations from a JDOM XML Element.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#copySequenceTracksFrom(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">copySequenceTracksFrom</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceSequence)</code>
<div class="block">Copies all <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a> (if any) from the source sequence on to this sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#createCopy()">createCopy</a></strong>()</code>
<div class="block">Creates a copy of this sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#createCopy(boolean)">createCopy</a></strong>(boolean&nbsp;allowReturningNonStandardImplementations)</code>
<div class="block">Creates a copy of this sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;root)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getAnnotationsRevisionNumber()">getAnnotationsRevisionNumber</a></strong>()</code>
<div class="block">Gets a revision number associated with the annotations.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()">getCharSequence</a></strong>()</code>
<div class="block">Get the sequence as a CharSequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequenceRevisionNumber()">getCharSequenceRevisionNumber</a></strong>()</code>
<div class="block">Gets a revision number associated with the char sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getDescription()">getDescription</a></strong>()</code>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getDisplayableFields()">getDisplayableFields</a></strong>()</code>
<div class="block">A list of columns for display in the main document panel.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getFieldValue(java.lang.String)">getFieldValue</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;code)</code>
<div class="block">Get the value of a particular field.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getName()">getName</a></strong>()</code>
<div class="block">A short name (preferably one or two words) for the document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a></strong>()</code>
<div class="block">Gets sequence annotations which are directly on the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceGapInformation()">getSequenceGapInformation</a></strong>()</code>
<div class="block">Gets SequenceGapInformation associated with the char sequence returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceGapInformation(boolean)">getSequenceGapInformation</a></strong>(boolean&nbsp;createIfNull)</code>
<div class="block">Returns a SequenceGapInformation associated with this sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceLength()">getSequenceLength</a></strong>()</code>
<div class="block">Returns the length of the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceString()">getSequenceString</a></strong>()</code>
<div class="block">Get the sequence as a string.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getTrackManager(boolean)">getTrackManager</a></strong>(boolean&nbsp;mayReturnNullIfNoTracks)</code>
<div class="block">Gets a <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager</code></a> associated with this sequence for providing sequence tracks.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isBestSerializedToXmlRatherThanBinaryFormat()">isBestSerializedToXmlRatherThanBinaryFormat</a></strong>()</code>
<div class="block">Returns true if this sequence would be best serialized using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toXML()"><code>toXML()</code></a> rather than <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)"><code>writeObject(java.io.DataOutput, jebl.util.ProgressListener)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isCircular()">isCircular</a></strong>()</code>
<div class="block">Is this a circular sequence?</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isStandardImplementation()">isStandardImplementation</a></strong>()</code>
<div class="block">Returns true if this sequence implementation is either <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideSequence</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultAminoAcidSequence</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideGraphSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideGraphSequence</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#readObject(java.io.DataInput)">readObject</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataInput.html?is-external=true" title="class or interface in java.io">DataInput</a>&nbsp;in)</code>
<div class="block">Recreates a sequence from a stream previously written to using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)"><code>writeObject(java.io.DataOutput, jebl.util.ProgressListener)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">readObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)</code>
<div class="block">Recreates a sequence from a stream previously written to using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)"><code>writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream , jebl.util.ProgressListener)</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#sequenceAnnotationsToXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.lang.String, java.util.List)">sequenceAnnotationsToXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Return a list of sequence annotations as XML in the specified version of Geneious</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#sequenceAnnotationsToXML(java.lang.String, java.util.List)">sequenceAnnotationsToXML</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Return a list of sequence annotations as XML.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setAnnotations(java.util.List)">setAnnotations</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets new annotations on this document, replacing the existing ones.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setCache(com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument.Cache)">setCache</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.Cache.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument.Cache</a>&nbsp;cache)</code>
<div class="block">Sets the cache used for caching the vaules returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceAnnotations()"><code>getSequenceAnnotations()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setCircular(boolean)">setCircular</a></strong>(boolean&nbsp;isCircular)</code>
<div class="block">Set whether this sequence is circular.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setDescription(java.lang.String)">setDescription</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description)</code>
<div class="block">Sets the description of this sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setFieldValue(java.lang.String, java.lang.Object)">setFieldValue</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;code,
             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;value)</code>
<div class="block">Set value of one data item (field).<br>
 <b>It is important to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addDisplayableField(com.biomatters.geneious.publicapi.documents.DocumentField)"><code>AbstractPluginDocument.addDisplayableField(DocumentField)</code></a> if this field should be displayable and isn't already on this document.</b>
 </b>For non displayable fields, the value must either implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>XMLSerializable</code></a> or be an instance of a simple type that Geneious knows how to serialize (e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setModified(boolean)">setModified</a></strong>(boolean&nbsp;modifiedFromOriginal)</code>
<div class="block">Sets whether or not this sequence has been modified from its original.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setName(java.lang.String)">setName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Change the name of the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets the sequence and annotations of this document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toHTML()">toHTML</a></strong>()</code>
<div class="block">Provide a textual view of document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)"><code>writeObject</code></a>(Geneious.getMajorVersion(), out, progressListener)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
           <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Writes this sequence to a stream as specified by <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io"><code>Serializable</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)">writeObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;out,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Writes this sequence to a stream as specified by <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io"><code>Serializable</code></a>.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.AbstractPluginDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addDisplayableField(com.biomatters.geneious.publicapi.documents.DocumentField)">addDisplayableField</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addReferencedDocumentUrn(com.biomatters.geneious.publicapi.documents.URN)">addReferencedDocumentUrn</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#clearReferencedDocumentUrns()">clearReferencedDocumentUrns</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#createFromBaseElement(org.jdom.Element)">createFromBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getBaseElement()">getBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getBaseElementString()">getBaseElementString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getFieldValueCodes()">getFieldValueCodes</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getReferencedDocumentUrns()">getReferencedDocumentUrns</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getURN()">getURN</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#hasProperties()">hasProperties</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setBaseElement(org.jdom.Element)">setBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setCreated(java.util.Date)">setCreated</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValue(com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object)">setFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValues(java.util.Map)">setFieldValues</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setURN(com.biomatters.geneious.publicapi.documents.URN)">setURN</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="DefaultSequenceDocument(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultSequenceDocument</h4>
<pre>protected&nbsp;DefaultSequenceDocument(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - name of the document, must not be null</dd><dd><code>description</code> - sequence descriptions (may be null or "")</dd><dd><code>sequence</code> - sequence as a string</dd><dd><code>created</code> - sequence creation date. May be null.</dd></dl>
</li>
</ul>
<a name="DefaultSequenceDocument(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date, com.biomatters.geneious.publicapi.documents.URN)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultSequenceDocument</h4>
<pre>protected&nbsp;DefaultSequenceDocument(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created,
                       <a href="../../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;urn)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - name of the document, must not be null</dd><dd><code>description</code> - sequence descriptions (may be null or "")</dd><dd><code>sequence</code> - sequence as a string</dd><dd><code>created</code> - sequence creation date. May be null.</dd><dd><code>urn</code> - URN. May be null.</dd></dl>
</li>
</ul>
<a name="DefaultSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.Date)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultSequenceDocument</h4>
<pre>protected&nbsp;DefaultSequenceDocument(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created)</pre>
<div class="block">Construct a new DefaultSequenceDocument from an existing
 SequenceDocument.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - an existing SequenceDocument</dd><dd><code>created</code> - sequence creation time</dd></dl>
</li>
</ul>
<a name="DefaultSequenceDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultSequenceDocument</h4>
<pre>protected&nbsp;DefaultSequenceDocument()</pre>
<div class="block">Construct a DefaultSequenceDocument with no arguments.</div>
</li>
</ul>
<a name="DefaultSequenceDocument(jebl.evolution.sequences.Sequence)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DefaultSequenceDocument</h4>
<pre>protected&nbsp;DefaultSequenceDocument(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&nbsp;jeblSequence)</pre>
<div class="block">Construct a new DefaultSequenceDocument using a Sequence from JEBL.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>jeblSequence</code> - a JEBL Sequence</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getSequenceAnnotations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAnnotations</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getSequenceAnnotations()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">SequenceDocument</a></code></strong></div>
<div class="block">Gets sequence annotations which are directly on the sequence.
 <b>This does not include annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>sequence tracks</code></a>.</b>
 <br><br>
 To get all annotations, including those on sequence tracks, use
 <a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAndTrackAnnotations</code></a>

 This method may return an unmodifiable list and must not return the list which is used internally by the SequenceDocument to store the annotations
 (ie modifying the returned list should not modify the annotations on the SequenceDocument directly)<br/><br/>

 <b>WARNING:</b> this list may not include all SequenceAnnotations represented as annotations in the sequence viewer.
 One such case is with trim annotations, which should be found using
 <a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAnnotationsIncludingImmutableSequencesTrims(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAnnotationsIncludingImmutableSequencesTrims(SequenceDocument)</code></a>.<br/><br/>

 This may also return annotations that are not visible in the sequence viewer, such as <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION"><code>SequenceAnnotation.TYPE_EXTRACTED_REGION</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a list of SequenceAnnotations; must not return null, but may return an
         empty list (e.g. <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collections.html?is-external=true#emptyList()" title="class or interface in java.util"><code>Collections.emptyList()</code></a>).</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAndTrackAnnotations(SequenceDocument)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)"><code>SequenceDocumentWithEditableAnnotations.setAnnotations(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="addSequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequenceAnnotation</h4>
<pre>public&nbsp;void&nbsp;addSequenceAnnotation(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<div class="block">Add a sequence annotation to this document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotation</code> - a SequenceAnnotation</dd></dl>
</li>
</ul>
<a name="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getName()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">PluginDocument</a></code></strong></div>
<div class="block">A short name (preferably one or two words) for the document.
 <p/>
 If this PluginDocument is wrapped inside an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a> then this method
 will be overridden and delegate back to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getName()"><code>AnnotatedPluginDocument.getName()</code></a>. One way to look at it is
 that when you call PluginDocument.getName() you will get the current name of the document. If you are implementing PluginDocument.getName() then you are
 returning the default name for the document.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>document name. Cannot be null but may be an empty string.</dd></dl>
</li>
</ul>
<a name="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre>public&nbsp;void&nbsp;setName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block">Change the name of the sequence.
 Note that if this sequence is wrapped by an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>
 then a call to this method will not have any effect. This is because AnnotatedPluginDocument overrides
 calls to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()"><code>PluginDocument.getName()</code></a> on the document it wraps.
 Instead you should call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#setName(java.lang.String)"><code>AnnotatedPluginDocument.setName(String)</code></a>
 in most cases.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html#setName(java.lang.String)">setName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - the new sequence name</dd></dl>
</li>
</ul>
<a name="getDescription()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">PluginDocument</a></code></strong></div>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a one line simple text description of the document, or null if no description is available.</dd></dl>
</li>
</ul>
<a name="setDescription(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDescription</h4>
<pre>public&nbsp;void&nbsp;setDescription(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description)</pre>
<div class="block">Sets the description of this sequence. Note that if this sequence is wrapped by an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>
 then (like <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setName(java.lang.String)"><code>setName</code></a>) a call to this method may not have any effect. This is because AnnotatedPluginDocument overrides
 calls to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()"><code>PluginDocument.getDescription()</code></a> on the document it wraps.
 Instead you should call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#setFieldValue(com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object)"><code>AnnotatedPluginDocument.setFieldValue</code></a>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#DESCRIPTION_FIELD"><code>DocumentField.DESCRIPTION_FIELD</code></a>,value)
 in those cases.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>description</code> - the new description for this document</dd></dl>
</li>
</ul>
<a name="setFieldValue(java.lang.String, java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFieldValue</h4>
<pre>public&nbsp;void&nbsp;setFieldValue(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;code,
                 <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;value)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValue(java.lang.String, java.lang.Object)">AbstractPluginDocument</a></code></strong></div>
<div class="block">Set value of one data item (field).<br>
 <b>It is important to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addDisplayableField(com.biomatters.geneious.publicapi.documents.DocumentField)"><code>AbstractPluginDocument.addDisplayableField(DocumentField)</code></a> if this field should be displayable and isn't already on this document.</b>
 </b>For non displayable fields, the value must either implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>XMLSerializable</code></a> or be an instance of a simple type that Geneious knows how to serialize (e.g. Integer, Double, Date, String) or a List or Map of any of these types.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValue(java.lang.String, java.lang.Object)">setFieldValue</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>code</code> - field code</dd><dd><code>value</code> - item value</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValue(com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object)"><code>AbstractPluginDocument.setFieldValue(DocumentField, Object)</code></a></dd></dl>
</li>
</ul>
<a name="getFieldValue(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFieldValue</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;getFieldValue(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;code)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">PluginDocument</a></code></strong></div>
<div class="block">Get the value of a particular field.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>code</code> - field code. This should be the
      <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#getCode()"><code>code</code></a> of one of the fields
      returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()"><code>PluginDocument.getDisplayableFields()</code></a>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>value for a field or null if this document does not have a field with the given field code.
          The class of the returned value must be the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#getValueType()"><code>DocumentField.getValueType()</code></a> (or a subclass) of the corresponding DocumentField returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()"><code>PluginDocument.getDisplayableFields()</code></a>.</dd></dl>
</li>
</ul>
<a name="getSequenceString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceString</h4>
<pre>public final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSequenceString()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">SequenceDocument</a></code></strong></div>
<div class="block">Get the sequence as a string. For nucleotide or amino acid sequences,
 each residue or ambiguity symbol is represented by one uppercase
 or lowercase character as per the IUPAC recommendations.
 <p/>
 For nucleotide sequences, any of the following are valid characters "ACGTURYMWSKBDHVN?-acgturymwskbdhvn"
 <p/>
 For amino acid sequences, any of the following are valid characters "ACDEFGHIKLMNPQRSTVWY*OUBJZX?-acdefghiklmnpqrstvwyoubjzx"

 If <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a> doesn't call this method, then one possible implementation
 for this method is the following:

 <p><code>return getCharSequence().toString();</code></p>

 <p>WARNING: It is generally a bad idea to use this method on anything but very short sequences. On large sequences (e.g. whole chromosomes) this method
 may take some time and may use significant amounts of memory. e.g. A String representation of human chr1 requires 500 MB of memory. It is preferable to use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a> instead.</p></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">getSequenceString</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence as a string; must not return null.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLength</h4>
<pre>public final&nbsp;int&nbsp;getSequenceLength()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">SequenceDocument</a></code></strong></div>
<div class="block">Returns the length of the sequence. This returns exactly the same as
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a>.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>length()</code></a>,
 but may be more performant on some implementations.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">getSequenceLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>The length of this document's sequence.

 Note that if a class implements both this interface and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>,
 the return value of this method must be consistent with the contracts from both
 interfaces.</dd></dl>
</li>
</ul>
<a name="getCharSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCharSequence</h4>
<pre>public final&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&nbsp;getCharSequence()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">SequenceDocument</a></code></strong></div>
<div class="block">Get the sequence as a CharSequence. Used as an alternative to
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a>
 in situations where a String is unnecessary and could potentially
 use too much memory, or where we want to efficiently calculate the terminal gap lengths.

 The returned CharSequence must always return consistent
 values from any of the CharSequence methods
 (e.g. <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>CharSequence.length()</code></a> and  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#charAt(int)" title="class or interface in java.lang"><code>CharSequence.charAt(int)</code></a>.
 If a SequenceDocument is editable and changes are made to it,
 it is required that the original CharSequence obtained from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a>
 not change. A client of the SequenceDocument is only able to obtain
 the changes by making a fresh call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a>.

 If <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a> doesn't call this method, then it is possible to
 implement this method as follows:

 <p><code>return SequenceCharSequence.valueOf(getSequenceString());</code></p>

 <p>However, it is recommended that a more memory efficient representation is used
 where possible; in that case, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a> can be implemented
 based on this method (see its javadoc).</p>
 <p>Some implementations of this method may choose to throw a RuntimeException whose cause is an XMLSerializationException if there is
 insufficient memory to load the sequence. In most situations (e.g. from a DocumentOperation or DocumentFileImporter) catching
 this exception is unnecessary as core Geneious will have either pre-loaded the sequence or will catch and unwrap this exception itself gracefully.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">getCharSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence as a SequenceCharSequence; must not return null.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>SequenceDocument.getSequenceString()</code></a></dd></dl>
</li>
</ul>
<a name="setCache(com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument.Cache)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCache</h4>
<pre>public&nbsp;void&nbsp;setCache(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.Cache.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument.Cache</a>&nbsp;cache)</pre>
<div class="block">Sets the cache used for caching the vaules returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceAnnotations()"><code>getSequenceAnnotations()</code></a>.
 A client may set the same cache on multiple sequences to ensure that only one of them is ever cached in memory at once.
 For example sequence list document of multiple chromosomes might use this to ensure they are not all cached in memory at once.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>cache</code> - the cache to use</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="getSequenceGapInformation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceGapInformation</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a>&nbsp;getSequenceGapInformation()</pre>
<div class="block">Gets SequenceGapInformation associated with the char sequence returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>SequenceGapInformation associated with the char sequence returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="getSequenceGapInformation(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceGapInformation</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a>&nbsp;getSequenceGapInformation(boolean&nbsp;createIfNull)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.Provider.html#getSequenceGapInformation(boolean)">SequenceGapInformation.Provider</a></code></strong></div>
<div class="block">Returns a SequenceGapInformation associated with this sequence.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.Provider.html#getSequenceGapInformation(boolean)">getSequenceGapInformation</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation.Provider</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>createIfNull</code> - true to create a SequenceGapInformation even if the sequence doesn't already have one cached. The implementation may choose to cache the result it returns.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a SequenceGapInformation associated with this sequence or potentially null if createIfNecessary is false</dd></dl>
</li>
</ul>
<a name="setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceAndAnnotations</h4>
<pre>public&nbsp;void&nbsp;setSequenceAndAnnotations(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">EditableSequenceDocument</a></code></strong></div>
<div class="block">Sets the sequence and annotations of this document. <b>This does not replace annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a></b>
 <br><br>
 If this changes the length of the sequence, this is only allowed if this document is not a NucleotideGraphSequence
 (because the graph site positions and qualities would have to be updated too by calling
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html#setSequenceAndAnnotationsAndGraph(java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)"><code>EditableNucleotideGraphSequenceDocument.setSequenceAndAnnotationsAndGraph(CharSequence, java.util.List, NucleotideGraph)</code></a>.

 Hence, it is illegal to change the length of the sequence through this method if
 this is instanceof NucleotideGraphSequenceDocument;
 Therefore, implementations of this method should start with the following code:

 <pre>
 if (this instanceof NucleotideGraphSequenceDocument && sequence.length() != getSequenceLength()) {
       throw new IllegalArgumentException("Can't change sequence length without updating graph for " + getClass());
 }
 </pre></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - new residues; NOTE: This CharSequence must not change its length or characters after it was passed to this method!</dd><dd><code>annotations</code> - new annotations</dd></dl>
</li>
</ul>
<a name="setAnnotations(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotations</h4>
<pre>public&nbsp;void&nbsp;setAnnotations(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)">SequenceDocumentWithEditableAnnotations</a></code></strong></div>
<div class="block">Sets new annotations on this document, replacing the existing ones.
 This does not replace annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)">setAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>annotations</code> - New annotations to set on this document.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()"><code>SequenceDocument.getSequenceAnnotations()</code></a></dd></dl>
</li>
</ul>
<a name="setModified(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setModified</h4>
<pre>public&nbsp;void&nbsp;setModified(boolean&nbsp;modifiedFromOriginal)</pre>
<div class="block">Sets whether or not this sequence has been modified from its original. Core Geneious calls this method when
 necessary so you don't normally need to call it, but you may override it in your own subclass if you want to
 record the modified state of the sequence. The default implementation does nothing.
 <br><br>
 As an example, this is used by Genbank documents to display a note in the Text View informing the user if a
 sequence has been modified since it was downloaded from NCBI.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>modifiedFromOriginal</code> - whether this document is a modified version of the original.
                             May be set to false if changes are made that are not considered as modification or to remove a previously set flag.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.910 (Geneious 9.1.0)</dd></dl>
</li>
</ul>
<a name="toHTML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toHTML</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toHTML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">PluginDocument</a></code></strong></div>
<div class="block">Provide a textual view of document. The contents should be at most 30K-60K characters. One reason is that this
 text is rendered using JTextPane, which is horrendously slow when handling large text. The more fundamental
 reason is that large amounts of text are generally not useful for a user. The whole idea of using a tool such as
 Geneious is to provide specialized viewers which make the large amount of bioinformatics data perceivable and
 comprehensible to a human user.
 <p/>
 Return an HTML body fragment (i.e. a fragment which sits inside a &lt;body&gt;&lt;/body&gt; tag). (Header,stylesheet and
 annotations are automatically added by GENEious code).
 <p/>
 Can optionally return an entire html tag set, as long as the first six characters are <code>&lt;html></code></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an HTML fragment of tag set, or null to have no text view.</dd></dl>
</li>
</ul>
<a name="getDisplayableFields()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisplayableFields</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&gt;&nbsp;getDisplayableFields()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">PluginDocument</a></code></strong></div>
<div class="block">A list of columns for display in the main document panel.
 This function only defines the columns available to display,
 not the particular values of these columns, which are returned by
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)"><code>PluginDocument.getFieldValue(String)</code></a>.
 <p/>
 Each column is specified by a Field. The fields should be obtained from a call to the appropriate
 Field.createXXXXField, where XXXX is the appropriate type.
 <p/>
 While it is not required, it is recommended that a PluginDocument
 handle returning a complete list of its displayable fields before
 the document has been properly constructed via a call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a>

 The returned list need not and should not be modifiable.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getDisplayableFields()">getDisplayableFields</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a list of document fields (or null to indicate no displayable fields)</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#toXML()">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="getAnnotationsRevisionNumber()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationsRevisionNumber</h4>
<pre>public&nbsp;int&nbsp;getAnnotationsRevisionNumber()</pre>
<div class="block">Gets a revision number associated with the annotations. The revision number is incremented every time annotations (those returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceAnnotations()"><code>getSequenceAnnotations()</code></a>) are changed on this sequence.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a revision number associated with the annotations</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="getCharSequenceRevisionNumber()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCharSequenceRevisionNumber</h4>
<pre>public&nbsp;int&nbsp;getCharSequenceRevisionNumber()</pre>
<div class="block">Gets a revision number associated with the char sequence. The revision number is incremented every time char sequence (returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a> is changed on this sequence.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a revision number associated with the char sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;root)
             throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>root</code> - representation from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="sequenceAnnotationsToXML(java.lang.String, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>sequenceAnnotationsToXML</h4>
<pre>public static&nbsp;org.jdom.Element&nbsp;sequenceAnnotationsToXML(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName,
                                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block">Return a list of sequence annotations as XML.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>elementName</code> - the name of the root XML Element</dd><dd><code>annotations</code> - a List of SequenceAnnotations</dd>
<dt><span class="strong">Returns:</span></dt><dd>a JDOM XML Element</dd></dl>
</li>
</ul>
<a name="sequenceAnnotationsToXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.lang.String, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>sequenceAnnotationsToXML</h4>
<pre>public static&nbsp;org.jdom.Element&nbsp;sequenceAnnotationsToXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName,
                                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block">Return a list of sequence annotations as XML in the specified version of Geneious</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to ensure the annotations are readable in.</dd><dd><code>elementName</code> - the name of the root XML Element</dd><dd><code>annotations</code> - a List of SequenceAnnotations</dd>
<dt><span class="strong">Returns:</span></dt><dd>a JDOM XML Element</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="annotationsFromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>annotationsFromXML</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotationsFromXML(org.jdom.Element&nbsp;annotationsElement)
                                                   throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Return a List of SequenceAnnotations from a JDOM XML Element.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotationsElement</code> - a JDOM XML Element</dd>
<dt><span class="strong">Returns:</span></dt><dd>a List of SequenceAnnotations</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the annotations cannot be deserialized</dd></dl>
</li>
</ul>
<a name="isCircular()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isCircular</h4>
<pre>public&nbsp;boolean&nbsp;isCircular()</pre>
<div class="block">Is this a circular sequence?</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()">isCircular</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this a circular sequence.</dd></dl>
</li>
</ul>
<a name="setCircular(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCircular</h4>
<pre>public&nbsp;void&nbsp;setCircular(boolean&nbsp;isCircular)</pre>
<div class="block">Set whether this sequence is circular.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setCircular(boolean)">setCircular</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>isCircular</code> - true if this sequence should be considered circular.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a></dd></dl>
</li>
</ul>
<a name="isStandardImplementation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isStandardImplementation</h4>
<pre>public final&nbsp;boolean&nbsp;isStandardImplementation()</pre>
<div class="block">Returns true if this sequence implementation is either <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideSequence</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultAminoAcidSequence</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideGraphSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideGraphSequence</code></a>.
 Returns false for any implementation that is not one of these and is a sub-class of one of these.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this sequence implementation is either <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideSequence</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultAminoAcidSequence</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideGraphSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideGraphSequence</code></a>.</dd></dl>
</li>
</ul>
<a name="isBestSerializedToXmlRatherThanBinaryFormat()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isBestSerializedToXmlRatherThanBinaryFormat</h4>
<pre>public final&nbsp;boolean&nbsp;isBestSerializedToXmlRatherThanBinaryFormat()</pre>
<div class="block">Returns true if this sequence would be best serialized using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toXML()"><code>toXML()</code></a> rather than <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)"><code>writeObject(java.io.DataOutput, jebl.util.ProgressListener)</code></a>.
 This method is mainly intended for core Geneious usage. Plugin developers probably don't need to use it.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this sequence would be best serialized using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toXML()"><code>toXML()</code></a> rather than <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)"><code>writeObject(java.io.DataOutput, jebl.util.ProgressListener)</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;out,
               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Writes this sequence to a stream as specified by <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io"><code>Serializable</code></a>.
 The object can be deserialized using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)</code></a>
 <p/>
 This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isStandardImplementation()"><code>isStandardImplementation()</code></a> returns true.
 Furthermore, if this document returns a non-empty list from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getReferencedDocumentUrns()"><code>AbstractPluginDocument.getReferencedDocumentUrns()</code></a>, then the caller
 of writeObject is responsible for separately serializing the referenced URNs to XML and reassigning them back
 to this sequence after deserialization. This is because core Geneious can only recognize serialized URNs in XML
 (but not in binary formats)
 and may in some circumstances (e.g. exporting and importing) may need to change the URNs in the serialized XML.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>out</code> - the stream to write to.</dd><dd><code>progressListener</code> - for reporting progress and allowing the write to be canceled.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if the write cannot be completed for any reason (including if the progress listener requests the operation be canceled)</dd></dl>
</li>
</ul>
<a name="writeObject(java.io.DataOutput, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)"><code>writeObject</code></a>(Geneious.getMajorVersion(), out, progressListener)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>out</code> - the stream to write to.</dd><dd><code>progressListener</code> - for reporting progress and allowing the write to be canceled.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if the write cannot be completed for any reason (including if the progress listener requests the operation be canceled)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd></dl>
</li>
</ul>
<a name="writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Writes this sequence to a stream as specified by <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io"><code>Serializable</code></a>.
 The object can be deserialized using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#readObject(java.io.DataInput)"><code>readObject(java.io.DataInput)</code></a>
 <p/>
 This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isStandardImplementation()"><code>isStandardImplementation()</code></a> returns true.
 Furthermore, if this document returns a non-empty list from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getReferencedDocumentUrns()"><code>AbstractPluginDocument.getReferencedDocumentUrns()</code></a>, then the caller
 of writeObject is responsible for separately serializing the referenced URNs to XML and reassigning them back
 to this sequence after deserialization. This is because core Geneious can only recognize serialized URNs in XML
 (but not in binary formats)
 and may in some circumstances (e.g. exporting and importing) may need to change the URNs in the serialized XML.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the Geneious version to ensure the data is compatible with</dd><dd><code>out</code> - the stream to write to.</dd><dd><code>progressListener</code> - for reporting progress and allowing the write to be canceled.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if the write cannot be completed for any reason (including if the progress listener requests the operation be canceled)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readObject</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;readObject(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)
                                          throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Recreates a sequence from a stream previously written to using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)"><code>writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream , jebl.util.ProgressListener)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>in</code> - the stream to read from</dd>
<dt><span class="strong">Returns:</span></dt><dd>the deserialized sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the sequence cannot be deserialized</dd></dl>
</li>
</ul>
<a name="readObject(java.io.DataInput)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readObject</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;readObject(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataInput.html?is-external=true" title="class or interface in java.io">DataInput</a>&nbsp;in)
                                          throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Recreates a sequence from a stream previously written to using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)"><code>writeObject(java.io.DataOutput, jebl.util.ProgressListener)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>in</code> - the stream to read from</dd>
<dt><span class="strong">Returns:</span></dt><dd>the deserialized sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the sequence cannot be deserialized</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd></dl>
</li>
</ul>
<a name="createCopy()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createCopy</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;createCopy()</pre>
<div class="block">Creates a copy of this sequence</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a copy of this sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="createCopy(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createCopy</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;createCopy(boolean&nbsp;allowReturningNonStandardImplementations)</pre>
<div class="block">Creates a copy of this sequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>allowReturningNonStandardImplementations</code> - false to force this method to only return instances for which <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isStandardImplementation()"><code>isStandardImplementation()</code></a> is true. If in doubt, use true.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a copy of this sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.801 (Geneious 8.0.1)</dd></dl>
</li>
</ul>
<a name="getTrackManager(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrackManager</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager</a>&nbsp;getTrackManager(boolean&nbsp;mayReturnNullIfNoTracks)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html#getTrackManager(boolean)">SequenceTrack.Manager.Provider</a></code></strong></div>
<div class="block">Gets a <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager</code></a> associated with this sequence for providing sequence tracks. For detailed information on sequence tracks, see <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack</code></a>.
 If this method is called on an alignment, the track manager is for the consensus sequence. Individual sequences in the alignment may also implement <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a> if they provide tracks.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html#getTrackManager(boolean)">getTrackManager</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>mayReturnNullIfNoTracks</code> - to save on memory usage when dealing with lots of small sequences, some implementations may not wish to construct a TrackManager unless necessary. If this parameter is true the implementation may return null rather than an empty track manager if is has no tracks</dd>
<dt><span class="strong">Returns:</span></dt><dd>a track manager or possibly null if mayReturnNullIfNoTracks is true</dd></dl>
</li>
</ul>
<a name="copySequenceTracksFrom(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>copySequenceTracksFrom</h4>
<pre>public&nbsp;void&nbsp;copySequenceTracksFrom(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceSequence)</pre>
<div class="block">Copies all <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a> (if any) from the source sequence on to this sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sourceSequence</code> - the source sequence to copy tracks from</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.Cache.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
