<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:53 NZDT 2016 -->
<title>OligoSequenceDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="OligoSequenceDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/OligoSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/NucleotideCounter.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="OligoSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations.sequence</div>
<h2 title="Class OligoSequenceDocument" class="title">Class OligoSequenceDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">com.biomatters.geneious.publicapi.documents.AbstractPluginDocument</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">com.biomatters.geneious.publicapi.implementations.sequence.DefaultNucleotideSequence</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.sequence.OligoSequenceDocument</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation.Provider</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">OligoSequenceDocument</span>
extends <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultNucleotideSequence</a></pre>
<div class="block">A sequence document which represents an olgionucleotide such as a Primer or DNA probe. Special properties include displayable
 fields for common oligo stats such as melting point and GC content as well as a special text view which displays the
 oligo properties. The collection of all OligoSequenceDocuments in a user's database makes up their "primer database" which
 is used when testing primers.
 </p>
 The sequence for the class consists of two parts: a binding region and potentially an extension. The binding region
 (which can be retrieved using <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getBindingInterval()"><code>getBindingInterval()</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getBindingSequence()"><code>getBindingSequence()</code></a>) is the part of the sequence
 which must match a target sequence for binding to occur. The extension (which is the remainder of <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceString()"><code>DefaultSequenceDocument.getSequenceString()</code></a>)
 is made optional additional bases (eg. a restriction site) which will be introduced to the end of the PCR product. An
 extension is optional.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence">OligoSequenceDocument.OligoType</a></strong></code>
<div class="block">The possible types of oligos</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;com.biomatters.geneious.publicapi.implementations.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.Cache.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument.Cache</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#BINDING_REGION_ANNOTATION_NAME">BINDING_REGION_ANNOTATION_NAME</a></strong></code>
<div class="block">The name of the annotation returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getPrimerAnnotation()"><code>getPrimerAnnotation()</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#EXTENSION_QUALIFIER_NAME">EXTENSION_QUALIFIER_NAME</a></strong></code>
<div class="block">Name to use for SequenceAnnotationQualifiers that specify any primer extension sequence in addition to the
 primer binding region.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#HAIRPIN_FIELD">HAIRPIN_FIELD</a></strong></code>
<div class="block">Displayable field for the hairpin alignment score of the oligo</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#OLIGO_TYPE_FIELD">OLIGO_TYPE_FIELD</a></strong></code>
<div class="block">Displayable field for the type of oligo (see <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence"><code>OligoSequenceDocument.OligoType</code></a>)
 The value of this field is the toString() value of the OligoType object</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#PERCENT_GC_FIELD">PERCENT_GC_FIELD</a></strong></code>
<div class="block">Displayable field for GC percent of the oligo</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#PRIMER_DIMER_FIELD">PRIMER_DIMER_FIELD</a></strong></code>
<div class="block">Displayable field for the primer-dimer alignment score of the oligo</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#PRIMER_INFO_NOTE_CODE">PRIMER_INFO_NOTE_CODE</a></strong></code>
<div class="block">Code for the "Primer Info" note type.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#PRIMER_PERCENT_GC_ANNOTATION_QUALIFIER_NAME">PRIMER_PERCENT_GC_ANNOTATION_QUALIFIER_NAME</a></strong></code>
<div class="block">The name of the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationQualifier</code></a> on primer annotations that contains the GC%.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#SOURCE_QUALIFIER_NAME">SOURCE_QUALIFIER_NAME</a></strong></code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>from 7.0.0, this field is not added to new primer annotations</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#TM_FIELD">TM_FIELD</a></strong></code>
<div class="block">Displayable field for the melting point of the oligo</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html#NICE_NAME">NICE_NAME</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD">GENOME_SEQUENCE_THRESHOLD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#OligoSequenceDocument()">OligoSequenceDocument</a></strong>()</code>
<div class="block">PluginDocument requirement, do not call directly</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#OligoSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">OligoSequenceDocument</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;sequenceDocument)</code>
<div class="block">Create an OligoSequenceDocument from an existing SequenceDocument.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#OligoSequenceDocument(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)">OligoSequenceDocument</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                     <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                     <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence,
                     <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created)</code>
<div class="block">Create an OligoSequenceDocument "from scratch".</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#addSequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">addSequenceAnnotation</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>
<div class="block">Add a sequence annotation to this document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#createExtensionForPrimerAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">createExtensionForPrimerAnnotation</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;primerAnnotation)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>this method no longer works and there is no replacement in the API. Sorry!</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;root)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getBindingInterval()">getBindingInterval</a></strong>()</code>
<div class="block">Get the interval for the binding region of the oligo sequence (excluding any 5' extension).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getBindingSequence()">getBindingSequence</a></strong>()</code>
<div class="block">Get the sequence of bases making up the binding region of this oligo.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getExtensionInterval()">getExtensionInterval</a></strong>()</code>
<div class="block">Get the interval for the 5' extension region of the oligo sequence .</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getExtensionSequence()">getExtensionSequence</a></strong>()</code>
<div class="block">Get the sequence of bases making up the 5' extension region of this oligo.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence">OligoSequenceDocument.OligoType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getOligoType()">getOligoType</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getPrimerAnnotation()">getPrimerAnnotation</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#setAnnotations(java.util.List)">setAnnotations</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets new annotations on this document, replacing the existing ones.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets the sequence and annotations of this document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#toHTML()">toHTML</a></strong>()</code>
<div class="block">Provide a textual view of document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.implementations.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#annotationsFromXML(org.jdom.Element)">annotationsFromXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#copySequenceTracksFrom(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">copySequenceTracksFrom</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#createCopy()">createCopy</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#createCopy(boolean)">createCopy</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getAnnotationsRevisionNumber()">getAnnotationsRevisionNumber</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequence()">getCharSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getCharSequenceRevisionNumber()">getCharSequenceRevisionNumber</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getDescription()">getDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getName()">getName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceGapInformation()">getSequenceGapInformation</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceGapInformation(boolean)">getSequenceGapInformation</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceLength()">getSequenceLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getSequenceString()">getSequenceString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getTrackManager(boolean)">getTrackManager</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isBestSerializedToXmlRatherThanBinaryFormat()">isBestSerializedToXmlRatherThanBinaryFormat</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isCircular()">isCircular</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#isStandardImplementation()">isStandardImplementation</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#readObject(java.io.DataInput)">readObject</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">readObject</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#sequenceAnnotationsToXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.lang.String, java.util.List)">sequenceAnnotationsToXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#sequenceAnnotationsToXML(java.lang.String, java.util.List)">sequenceAnnotationsToXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setCache(com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument.Cache)">setCache</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setCircular(boolean)">setCircular</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setDescription(java.lang.String)">setDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setFieldValue(java.lang.String, java.lang.Object)">setFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setModified(boolean)">setModified</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setName(java.lang.String)">setName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)">writeObject</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.AbstractPluginDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addDisplayableField(com.biomatters.geneious.publicapi.documents.DocumentField)">addDisplayableField</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addReferencedDocumentUrn(com.biomatters.geneious.publicapi.documents.URN)">addReferencedDocumentUrn</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#clearReferencedDocumentUrns()">clearReferencedDocumentUrns</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#createFromBaseElement(org.jdom.Element)">createFromBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getBaseElement()">getBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getBaseElementString()">getBaseElementString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getFieldValueCodes()">getFieldValueCodes</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getReferencedDocumentUrns()">getReferencedDocumentUrns</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getURN()">getURN</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#hasProperties()">hasProperties</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setBaseElement(org.jdom.Element)">setBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setCreated(java.util.Date)">setCreated</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValue(com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object)">setFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValues(java.util.Map)">setFieldValues</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setURN(com.biomatters.geneious.publicapi.documents.URN)">setURN</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">getCharSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">getSequenceLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">getSequenceString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()">isCircular</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="PRIMER_INFO_NOTE_CODE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PRIMER_INFO_NOTE_CODE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> PRIMER_INFO_NOTE_CODE</pre>
<div class="block">Code for the "Primer Info" note type. To get the note type call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentNoteUtilities.html#getNoteType(java.lang.String)"><code>DocumentNoteUtilities.getNoteType(PRIMER_INFO_NOTE_CODE)</code></a></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.sequence.OligoSequenceDocument.PRIMER_INFO_NOTE_CODE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="BINDING_REGION_ANNOTATION_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>BINDING_REGION_ANNOTATION_NAME</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> BINDING_REGION_ANNOTATION_NAME</pre>
<div class="block">The name of the annotation returned from <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html#getPrimerAnnotation()"><code>getPrimerAnnotation()</code></a></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.sequence.OligoSequenceDocument.BINDING_REGION_ANNOTATION_NAME">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="OLIGO_TYPE_FIELD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OLIGO_TYPE_FIELD</h4>
<pre>public static final&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a> OLIGO_TYPE_FIELD</pre>
<div class="block">Displayable field for the type of oligo (see <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence"><code>OligoSequenceDocument.OligoType</code></a>)
 The value of this field is the toString() value of the OligoType object</div>
</li>
</ul>
<a name="TM_FIELD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TM_FIELD</h4>
<pre>public static final&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a> TM_FIELD</pre>
<div class="block">Displayable field for the melting point of the oligo</div>
</li>
</ul>
<a name="PRIMER_PERCENT_GC_ANNOTATION_QUALIFIER_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PRIMER_PERCENT_GC_ANNOTATION_QUALIFIER_NAME</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> PRIMER_PERCENT_GC_ANNOTATION_QUALIFIER_NAME</pre>
<div class="block">The name of the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationQualifier</code></a> on primer annotations that contains the GC%.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.sequence.OligoSequenceDocument.PRIMER_PERCENT_GC_ANNOTATION_QUALIFIER_NAME">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="PERCENT_GC_FIELD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PERCENT_GC_FIELD</h4>
<pre>public static final&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a> PERCENT_GC_FIELD</pre>
<div class="block">Displayable field for GC percent of the oligo</div>
</li>
</ul>
<a name="HAIRPIN_FIELD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HAIRPIN_FIELD</h4>
<pre>public static final&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a> HAIRPIN_FIELD</pre>
<div class="block">Displayable field for the hairpin alignment score of the oligo</div>
</li>
</ul>
<a name="PRIMER_DIMER_FIELD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PRIMER_DIMER_FIELD</h4>
<pre>public static final&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a> PRIMER_DIMER_FIELD</pre>
<div class="block">Displayable field for the primer-dimer alignment score of the oligo</div>
</li>
</ul>
<a name="SOURCE_QUALIFIER_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SOURCE_QUALIFIER_NAME</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> SOURCE_QUALIFIER_NAME</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>from 7.0.0, this field is not added to new primer annotations</i></div>
<div class="block">Name to use for SequenceAnnotationQualifiers to provide a reference back to the original <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>OligoSequenceDocument</code></a>
 from which an annotation was created.
 <p>
 If an OligoSequenceDocument with an extension has had its binding region annotated onto another sequence
 then the extension is not visible. However the annotation should still hold a reference to the original
 <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>OligoSequenceDocument</code></a> by storing a qualifier that uses this name and has a value of the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents"><code>URN</code></a> of
 the source document.  This qualifier is hidden from the Geneious user interface.

 This, for example, is done by the bundled Primer and Probe Design Plugin when doing Test Primers.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.sequence.OligoSequenceDocument.SOURCE_QUALIFIER_NAME">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="EXTENSION_QUALIFIER_NAME">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>EXTENSION_QUALIFIER_NAME</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> EXTENSION_QUALIFIER_NAME</pre>
<div class="block">Name to use for SequenceAnnotationQualifiers that specify any primer extension sequence in addition to the
 primer binding region.
 <p>
 An <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>OligoSequenceDocument</code></a> may have an extension in addition to the binding sequence.  If the binding
 region is annotated onto a sequence then the extension will appear as a qualifier of the binding annotation.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.sequence.OligoSequenceDocument.EXTENSION_QUALIFIER_NAME">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="OligoSequenceDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OligoSequenceDocument</h4>
<pre>public&nbsp;OligoSequenceDocument()</pre>
<div class="block">PluginDocument requirement, do not call directly</div>
</li>
</ul>
<a name="OligoSequenceDocument(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OligoSequenceDocument</h4>
<pre>public&nbsp;OligoSequenceDocument(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                     <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                     <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence,
                     <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;created)</pre>
<div class="block">Create an OligoSequenceDocument "from scratch". Primer characteristics such as TM and GC% are not added in this
 class, they must be added separately as an annotation by the core Primer plugin.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - name of the document, must not be null</dd><dd><code>description</code> - sequence descriptions (may be null or "")</dd><dd><code>charSequence</code> - sequence bases as a CharSequence</dd><dd><code>created</code> - sequence creation date. May be null.</dd></dl>
</li>
</ul>
<a name="OligoSequenceDocument(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>OligoSequenceDocument</h4>
<pre>public&nbsp;OligoSequenceDocument(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;sequenceDocument)</pre>
<div class="block">Create an OligoSequenceDocument from an existing SequenceDocument. If this is being used to replace an existing
 sequence in the database then the caller should copy any notes across from the
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>. This constructor copies all displayable
 fields and values from the source <code>SequenceDocument</code>.
 <p/>
 Primer characteristics such as TM and GC% are not added in this
 class, they must be added separately as an annotation by the core Primer plugin.

 Primer characteristics will be picked up from a primer_bind annotation if there is one and only one that
 covers the sequence entirely.  If such an annotation exists and the sequence qualifier for the annotation differs
 from the source sequence itself, then the sequence in the qualifier will be used for the sequence of this document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceDocument</code> - the sequence document to copy data from</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="addSequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequenceAnnotation</h4>
<pre>public&nbsp;void&nbsp;addSequenceAnnotation(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#addSequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">DefaultSequenceDocument</a></code></strong></div>
<div class="block">Add a sequence annotation to this document.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#addSequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">addSequenceAnnotation</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>annotation</code> - a SequenceAnnotation</dd></dl>
</li>
</ul>
<a name="setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceAndAnnotations</h4>
<pre>public&nbsp;void&nbsp;setSequenceAndAnnotations(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">EditableSequenceDocument</a></code></strong></div>
<div class="block">Sets the sequence and annotations of this document. <b>This does not replace annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a></b>
 <br><br>
 If this changes the length of the sequence, this is only allowed if this document is not a NucleotideGraphSequence
 (because the graph site positions and qualities would have to be updated too by calling
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html#setSequenceAndAnnotationsAndGraph(java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph)"><code>EditableNucleotideGraphSequenceDocument.setSequenceAndAnnotationsAndGraph(CharSequence, java.util.List, NucleotideGraph)</code></a>.

 Hence, it is illegal to change the length of the sequence through this method if
 this is instanceof NucleotideGraphSequenceDocument;
 Therefore, implementations of this method should start with the following code:

 <pre>
 if (this instanceof NucleotideGraphSequenceDocument && sequence.length() != getSequenceLength()) {
       throw new IllegalArgumentException("Can't change sequence length without updating graph for " + getClass());
 }
 </pre></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setSequenceAndAnnotations(java.lang.CharSequence, java.util.List)">setSequenceAndAnnotations</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - new residues; NOTE: This CharSequence must not change its length or characters after it was passed to this method!</dd><dd><code>annotations</code> - new annotations</dd></dl>
</li>
</ul>
<a name="setAnnotations(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotations</h4>
<pre>public&nbsp;void&nbsp;setAnnotations(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)">SequenceDocumentWithEditableAnnotations</a></code></strong></div>
<div class="block">Sets new annotations on this document, replacing the existing ones.
 This does not replace annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)">setAnnotations</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#setAnnotations(java.util.List)">setAnnotations</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>annotations</code> - New annotations to set on this document.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()"><code>SequenceDocument.getSequenceAnnotations()</code></a></dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toXML()">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;root)
             throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>root</code> - representation from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="toHTML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toHTML</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toHTML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">PluginDocument</a></code></strong></div>
<div class="block">Provide a textual view of document. The contents should be at most 30K-60K characters. One reason is that this
 text is rendered using JTextPane, which is horrendously slow when handling large text. The more fundamental
 reason is that large amounts of text are generally not useful for a user. The whole idea of using a tool such as
 Geneious is to provide specialized viewers which make the large amount of bioinformatics data perceivable and
 comprehensible to a human user.
 <p/>
 Return an HTML body fragment (i.e. a fragment which sits inside a &lt;body&gt;&lt;/body&gt; tag). (Header,stylesheet and
 annotations are automatically added by GENEious code).
 <p/>
 Can optionally return an entire html tag set, as long as the first six characters are <code>&lt;html></code></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html#toHTML()">toHTML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an HTML fragment of tag set, or null to have no text view.</dd></dl>
</li>
</ul>
<a name="getOligoType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOligoType</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence">OligoSequenceDocument.OligoType</a>&nbsp;getOligoType()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>The type of oligonucleotide. Never null.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence"><code>OligoSequenceDocument.OligoType</code></a></dd></dl>
</li>
</ul>
<a name="getPrimerAnnotation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPrimerAnnotation</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;getPrimerAnnotation()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>A copy of the single annotation defining the binding region of this OligoSequenceDocument, or null if
 there is none or there are multiple annotations that could describe the binding region.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.805 (Geneious 8.0.5)</dd></dl>
</li>
</ul>
<a name="getBindingInterval()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBindingInterval</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;getBindingInterval()</pre>
<div class="block">Get the interval for the binding region of the oligo sequence (excluding any 5' extension). This is determined by the interval of the primer_bind
 or probe_bind annotation on the sequence. If there is no bind annotation then the whole sequence is considered
 binding and an interval representing the length of the sequence is returned
 </p>
 The binding region excludes any additional extensions that have been added after the creation of the primer.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The interval of the binding region in this OligoSequenceDocument. Never null.</dd></dl>
</li>
</ul>
<a name="getExtensionInterval()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExtensionInterval</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;getExtensionInterval()</pre>
<div class="block">Get the interval for the 5' extension region of the oligo sequence . This is the region of the sequence not covered by the primer_bind
 or probe_bind annotation on the sequence. If there is no bind annotation then there is considered to be no extension.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The interval of the 5' extension region in this OligoSequenceDocument. Zero-length interval at the start of the binding interval, if there is no extension</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getExtensionSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExtensionSequence</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;getExtensionSequence()</pre>
<div class="block">Get the sequence of bases making up the 5' extension region of this oligo.  This is the region not covered by the primer_bind
 or probe_bind annotation on the sequence. If there is no bind annotation then there is considered to be no extension.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The 5' extension sequence for this oligo.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getBindingSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBindingSequence</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;getBindingSequence()</pre>
<div class="block">Get the sequence of bases making up the binding region of this oligo. This is determined by the primer_bind
 or probe_bind annotation on the sequence. If there is no bind annotation then the whole sequence is considered
 binding the entire sequence is returned from this method.
 </p>
 The binding sequence excludes any additional extensions that have been added after the creation of the primer.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The binding sequence for this oligo.  This is the sequence that will be tested for annealing against a target
 sequence.</dd></dl>
</li>
</ul>
<a name="createExtensionForPrimerAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>createExtensionForPrimerAnnotation</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;createExtensionForPrimerAnnotation(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;primerAnnotation)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>this method no longer works and there is no replacement in the API. Sorry!</i></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>primerAnnotation</code> - An annotation of type <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PRIMER_BIND"><code>SequenceAnnotation.TYPE_PRIMER_BIND</code></a>,
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PRIMER_BIND_REVERSE"><code>SequenceAnnotation.TYPE_PRIMER_BIND_REVERSE</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_DNA_PROBE"><code>SequenceAnnotation.TYPE_DNA_PROBE</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>Will return a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> that represents the extension for the binding region
 described by the primerAnnotation or null if no extension is found.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/OligoSequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/NucleotideCounter.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.OligoType.html" title="enum in com.biomatters.geneious.publicapi.implementations.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html" target="_top">Frames</a></li>
<li><a href="OligoSequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
