<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:55 NZDT 2016 -->
<title>Overview (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Overview (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li class="navBarCell1Rev">Overview</li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-all.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?overview-summary.html" target="_top">Frames</a></li>
<li><a href="overview-summary.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 class="title">Geneious 9.1.2 Public API</h1>
</div>
<div class="header">
<div class="subTitle">
<div class="block"><h3>Contents</h2>
 <ul>
 <li><a href="#apiIntro">Introduction</a></li>
 <li><a href="#apiGettingStartedChoosePluginType">Getting Started - Choosing your plugin type</a></li>
 <li><a href="#gettingStartedCompiling">Getting Started - Compiling and building your plugin</a></li>
 <li><a href="#documentsInGeneious">Documents in Geneious</a></li>
 <li><a href="#importantClasses">Important classes in the API</a></li>
 <li><a href="#apiFaq">FAQ</a>
    <ol>
        <li> <a href="#faqAddMenuBar">How do I add an item to the menu bar or toolbar?</a></li>
        <li> <a href="#faqPopupDialog">How do I popup a dialog for user input?</a></li>
        <li> <a href="#faqGetDocumentsInSameFolder">How do I get all documents in the same folder as a given document?</a></li>
        <li> <a href="#faqGetDocumentsInDatabase">How do I get all documents in the user's local database?</a></li>
        <li> <a href="#faqExportDocument">How do I export to any document format supported by Geneious?</a></li>
        <li> <a href="#faqImportDocument">How do I import any document format supported by Geneious?</a></li>
        <li> <a href="#faqAddDocument">How do I add a document to the user's database?</a></li>
        <li> <a href="#faqAddAnnotations">How do I add annotations to a sequence?</a></li>
        <li> <a href="#faqDependOnExternal">How do I allow my plugin to depend on external libraries or resources?</a></li>
        <li> <a href="#faqLog">How do I create a log file from within my plugin?</a></li>
        <li> <a href="#faqTestCases">How do I write unit test cases for my plugin?</a></li>
        <li> <a href="#faqInvokeOperations">How do I invoke other plugin operations?</a></li>  
    </ol>
 </li>
 <li><a href="#overview_description">API Packages</a></li>
 
 </ul>
 <a name="apiIntro">
 <h3>Introduction</h3>
 The Geneious Public API is a rich and powerful API that allows to create a wide variety of plugins. In
 fact most of the functionality provided by Geneious is actually just a set of plugins bundled
 with the Geneious distribution that interact with the core Geneious platform and with each other via the Public API. Examples of plugins
 in Geneious include the sequence viewer, sequence alignment operation, primer design, the sequence logo graph,
 the NCBI databases, agents, and even the user's local database.
 <p/>
 The architecture of Geneious is centered around the concept of documents which are stored in either
 a local or shared database. Operations can modify documents or create derived documents using existing documents
 as input. Viewers provide a system for visualization and sometimes editing the contents of one or more documents.

 <a name="apiGettingStartedChoosePluginType">
 <h3>Getting Started - Choosing your plugin type</h3>
  The first thing to decide upon when writing a plug-in is what type of plug-in you want. All
 <a href="com/biomatters/geneious/publicapi/plugin/GeneiousPlugin.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>GeneiousPlugins</code></a> fall
 under one of the following categories. There is an example plug-in for many of these categories included with the
 API distribution.
  <ul>
 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerators</code></a>
   integrate into the SequenceViewer to provide an entry in the tools popup menu to
  generate annotations on the currently selected sequence(s). The SequenceViewer plugin handles all
  associated necessary behaviour such as providing undo functionality and saving the generated annotations to disk.
 <p/>
 Examples of SequenceAnnotationGenerator plugins in Geneious include <b>primer design</b>, <b>restriction analysis</b>, <b>ORF finding</b>, and <b>motif</b> finding.
 <br>
 Example Source Code:  <a href="../examples/ExampleSequenceAnnotationGenerator">ExampleSequenceAnnotationGenerator</a>
 </li>
 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperations</code></a>
 allow the creation of a new document(s) from an existing document(s) (or from no documents).
 The DocumentOperation specifies its location in the menu or toolbar along with a signature
 specifying what type of documents it operates on and Geneious handles invoking the DocumentOperation
 and saving any generated documents to disk.
 <p/>
 Examples of DocumentOperation plugins in Geneious include <b>Set Paired Reads</b> and <b>tree building</b>.
 <br>
 Example Source Code: <a href="../examples/ReverseSequence">ReverseSequence</a>,
 <a href="../examples/BackTranslationPlugin">BackTranslationPlugin</a>
 </li>
 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/Assembler.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>Assemblers</code></a>
 allow the implementation of a read mapper, reference assembler or de novo assembler. Assemblers could be implemented
 as DocumentOperations but this API provides a simpler API tailored for assemblers.
 <p/>
 Examples of Assembler plugins in Geneious include <b>Geneious Reference Assembler</b>, <b>Geneious De Novo Assembler</b> and <b>Velvet</b> (on Geneious Server only)
 <br>
 Example Source Code: <a href="../examples/ExampleAssembler">ExampleAssembler</a>
 </li>
 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/AlignmentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AlignmentOperations</code></a>
 allow the implementation of an alignment algorithm. AlignmentOperations could be implemented
 as DocumentOperations but this API provides a simpler API tailored for alignment.
 <p/>
 Examples of AlignmentOperation plugins in Geneious include <b>Geneious Aligner</b>, <b>Muscle</b> and <b>ClustalW</b>
 <br>
 Example Source Code: <a href="../examples/ExampleAlignmentOperation">ExampleAlignmentOperation</a>
 </li>
 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/DocumentViewer.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentViewers</code></a>
 allow the viewing of 1 or more documents. DocumentViewers specify a signature of the types of
 documents they can view and Geneious displays the DocumentViewer each time the user
 selects documents matching it's signature.
 <p/>
 Examples of DocumentViewer plugins in Geneious include the <b>sequence viewer</b>, the <b>tree viewer</b>, and the <b>dotplot viewer</b>.
 <br>
 Example Source Code: <a href="../examples/ExampleDocumentViewer">ExampleDocumentViewer</a>

 </li>
 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/SequenceGraph.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceGraphs</code></a>
 allow displaying of graphs on sequences inside the sequence viewer.
 <p/>
 Examples of SequenceGraph plugins in Geneious include the <b>identity graph</b>, the <b>sequence logo graph</b>, and the <b>chromatogram graph</b>.
 <br>
 Example Source Code: <a href="../examples/ExampleSequenceGraph">ExampleSequenceGraph</a>
 , <a href="../examples/ExampleSequenceGraph2">ExampleSequenceGraph2</a>
 </li>
 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/GeneiousService.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>GeneiousServices</code></a>
 allow entries in the sources panel on the left hand side of the main Geneious window. Typically
 these are databases providing access to local or external sequences.
 <p/>
 Examples of GeneiousService plugins in Geneious include the <b>local database</b> and the <b>NCBI services</b>.
 <br>
 Example Source Code: <a href="../examples/ExampleGeneiousService">ExampleGeneiousService</a>
 </li>

 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/DocumentFileImporter.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentFileImporters</code></a>
 provide support for importing various file formats into Geneious.
 <p/>
 Examples of DocumentFileImporter plugins in Geneious include the <b>fasta importer</b>, <b>GenBank importer</b>, and <b>nexus importer</b>.
 <br>
 Example Source Code: <a href="../examples/exampleFastaImporter">ExampleFastaImporter</a>
 , <a href="../examples/TextFiles">TextFiles</a>
 </li>

 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/DocumentFileExporter.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentFileExporters</code></a>
 provide support for exporting documents from Geneious into various file formats.
 <p/>
 Examples of DocumentFileExporter plugins in Geneious include the <b>fasta exporter</b>, <b>GenBank exporter</b> and <b>nexus exporter</b>.
 <br>
 Example Source Code: <a href="../examples/exampleFastaExporter">ExampleFastaExporter</a>
 , <a href="../examples/TextFiles">TextFiles</a>
 </li>

 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/TreeViewerExtension.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>TreeViewerExtensions</code></a>
 provide support for extending the functionality of the Tree Viewer.
 <p/>
 The standard Geneious distribution does not include plugins of this type, but the 3rd party <b>Species Delimitation</b> plugin available
 from the Preferences->Plugins window is an example of this.
 <br>
 Example Source Code: <a href="../examples/ExampleTreeViewerExtensionPlugin">ExampleTreeViewerExtensionPlugin</a>
 </li>

 <li>
  <a href="com/biomatters/geneious/publicapi/plugin/SequenceViewerExtension.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceViewerExtensions</code></a>
 provide support for extending the functionality of the Sequence/Alignment/Contig Viewer. For example with statistics, toolbar buttons or new components.
 <p/>
 Examples of SequenceViewerExtensions plugins in Geneious include the <b>annotations table</b> and <b>many of the statistics in the sequence viewer</b>.

 <br>
 Example Source Code: <a href="../examples/exampleSequenceViewerStatistics">ExampleSequenceViewerStatistics</a>
 </li>
 </li>

  </ul>
 <p/>
 <a name="gettingStartedCompiling">
 <h3>Getting Started - Compiling and building your plugin</h3>

 The easiest thing to do is run one of the examples using a supported IDE. More information about how to do this
 is available in the <a href="../setup.html">setup file</a> included with this documentation.
 Once you are comfortable with this choose an example plugin that most closely matches the plugin you want to create (see <a href="#apiGettingStartedChoosePluginType">Getting Started - Choosing your plugin type</a>),
 optionally rename the example using the renamePlugin Ant task, and start modifying it to achieve the functionality you want.<br>
 <br>
 More information on building and distributing your plugin can also be found in the setup file.

 <a name="documentsInGeneious">
  <h3>Documents in Geneious</h3>
 A document is an item that appears in the table listing the contents of a folder.
 There are 2 super-classes in the API that are used when handling documents.
 <p/>

 <a href="com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>PluginDocument</code></a> is the interface
 which most other document types
 such as <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>,
 <a href="com/biomatters/geneious/publicapi/documents/types/TreeDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types"><code>TreeDocument</code></a> or
 <a href="com/biomatters/geneious/publicapi/documents/types/PublicationDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types"><code>PublicationDocument</code></a> extend.
 PluginDocuments are however
 not stored directly in the user's database. Instead these are wrapped in the top-level AnnotatedPluginDocument.
 <p/>
 <a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>
  decorates all PluginDocuments with additional functionality, such as allowing custom notes to be added
 to documents, and allowing the user to edit displayed field values without the PluginDocument
 implementations needing to worry about providing this sort of functionality. A plugin never implements
 or subclasses AnnotatedPluginDocument.
 <p/>
 In many situations,
 plugins can ignore the functionality provided by <a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>
 and instead just directly access the wrapped PluginDocument from an AnnotatedPluginDocument using
 <a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getDocument()"><code>AnnotatedPluginDocument.getDocument()</code></a> or
 creating a AnnotatedPluginDocument from a PluginDocument using
 <a href="com/biomatters/geneious/publicapi/documents/DocumentUtilities.html#createAnnotatedPluginDocument(com.biomatters.geneious.publicapi.documents.PluginDocument)"><code>DocumentUtilities.createAnnotatedPluginDocument()</code></a>.

  <h3>Defining your own document types</h3>
 While the Geneious Public API provides a range of document interfaces and implementations, sometimes
 it is necessary to create your own document types or extend the functionality of
 the existing types.
 Usually this is only necessary if you are creating a plugin which provides multiple types of
 functionality - for example you wish to create a document operation that generates your own custom
 data types, and provide a document viewer to view those data types.
 Geneious allows you to do this by directly implementing the <a href="com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>PluginDocument</code></a>
 interface. A plugin can also define additional static properties (such as icons) associated with new
 PluginDocument types using <a href="com/biomatters/geneious/publicapi/plugin/GeneiousPlugin.html#getDocumentTypes()"><code>GeneiousPlugin.getDocumentTypes()</code></a>.

 <a name="importantClasses">
  <h3>Important classes in the API</h3>
 In addition to the GeneiousPlugin and the various plugin type classes (see <a href="#apiGettingStartedChoosePluginType">choosing your plugin type</a>),
 and the Geneious document system (see <a href="#documentsInGeneious">documents in Geneious)</a>,
 there are a number of important or widely used classes in the Geneious API you should become familiar with.
 <ul>

 <li><a href="com/biomatters/geneious/publicapi/plugin/Options.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>Options</code></a> - Many types of plugins will need to use Options as the way of prompting for user input. Make sure
 you read the top-level documentation in this class to see what Options is capable of.
 </li>

 <li><a href="com/biomatters/geneious/publicapi/plugin/GeneiousActionOptions.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>GeneiousActionOptions</code></a> - The description, icon and location of a menu item or button to invoke a plugin or functionality within a plugin.
 </li>

 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a> - Any plugin that deals with nucleotide or amino acid sequences should be aware of this. Also see <a href="com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultSequenceDocument</code></a>
 which is the most widely used SequenceDocument implementation.
 </li>

 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceCharSequence</code></a> - The nucleotides or amino acids in a SequenceDocument are often stored in a SequenceCharSequence instead of a String
 for both more efficient memory usage, and for efficient representation of end gaps in an alignment.</li>

 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotation</code></a> - is used to represent features or annotations on a SequenceDocument</li>

 <li><a href="com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack</code></a> - is used to represent a collection of SequenceAnnotations on a SequenceDocument</li>

 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a> - SequenceDocuments are often grouped into a single SequenceListDocument.
 <a href="com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>DefaultSequenceListDocument</code></a> is the most widely used SequenceListDocument implementation</li>

 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a> - The list of nucleotide sequences returned from
 <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getNucleotideSequences()"><code>SequenceListDocument.getNucleotideSequences()</code></a> is often a SequenceListOnDisk where the
 sequences represent a large number of sequencing reads. The SequenceDocuments obtained from this list
 are loaded on demand and for optimal performance code should iterate over the list in order. Note that collections of chromosome sized sequences are usually not a
 SequenceListOnDisk. Instead the SequenceDocuments
 in these lists are loaded along with the SequenceListDocument, but
 the <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceCharSequence</code></a> and <a href="com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a>
 within those sequences are loaded later on demand.
 </li>

 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a> - Alignments and contigs are both represented using a SequenceAlignmentDocument which
 is most commonly implemented using a <a href="com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocument</code></a></li>
 <li><a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util"><code>ProgressListener</code></a> - although this is from the JEBL package which Geneious depends on, it is very widely used throughout Geneious for operations to report progress and allow the user to cancel them.</li>
 </ul>

 As well as the above classes, some less frequently used classes and functionality that are of interest to more advanced users of the API include
 <ul>
 <li><a href="com/biomatters/geneious/publicapi/components/Dialogs.html" title="class in com.biomatters.geneious.publicapi.components"><code>Dialogs</code></a> - methods for displaying dialogs</li>
 <li><a href="com/biomatters/geneious/publicapi/plugin/Findable.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>Findable</code></a> - Allows <a href="com/biomatters/geneious/publicapi/plugin/DocumentViewer.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentViewer</code></a> to easily implement 'Find' functionality</li>
 <li><a href="com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>EndGapsManager</code></a> - Given a set of aligned sequences, provides methods for quickly finding all sequences
 that intersect a given base number, ignoring those sequences in end gap regions.</li>
 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a> - Provides distance and orientation of paired/mate sequencing reads</li>
 <li><a href="com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>SequenceGapInformation</code></a> - Precalculates information about the location of gaps ('-') in a CharSequence, and can efficiently calculate translate between indices in the gapped and ungapped sequence</li>
 <li><a href="com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>CombinedAlignmentAndSequenceDocument</code></a> - a document which contains both a stand-alone sequence and an alignment. Typically used in sequence search results</li>
 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a> - an very compact (in memory and on disk) implementation of a <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a></li>
 <li><a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListSummary</code></a> - provides details about the sequences and annotations in a large sequence list or alignment/contig without needing to iterate over the entire list</li>
 <li><a href="com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceSelection</code></a> - provides details on the currently selection regions in the sequence viewer to DocumentOperations, AnnotationGenerators and other DocumentViewers</li>
 <li><a href="com/biomatters/geneious/publicapi/plugin/DocumentViewerMessageHandler.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentViewerMessageHandler</code></a> - provides a system for DocumentViewers to interact with each other</li>
 <li><a href="com/biomatters/geneious/publicapi/documents/OperationRecordDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>OperationRecordDocument</code></a> - provides details about the relationships between documents</li>
 <li><a href="com/biomatters/geneious/publicapi/databaseservice/FolderViewDocument.html" title="interface in com.biomatters.geneious.publicapi.databaseservice"><code>FolderViewDocument</code></a> - a hidden document attached to the folder of a sequence search result</li>
 <li><a href="com/biomatters/geneious/publicapi/utilities/StandardIcons.html" title="enum in com.biomatters.geneious.publicapi.utilities"><code>StandardIcons</code></a> - set of standard icons used in Geneious. Use of these in 3rd party plugins is encouraged to keep the look consistent.</li>
 <li><a href="com/biomatters/geneious/publicapi/databaseservice/WritableDatabaseService.html" title="class in com.biomatters.geneious.publicapi.databaseservice"><code>WritableDatabaseService</code></a> - the local and shared Databases in Geneious are both WritableDatabaseServices. Each
 document stored in a database exposes its location via <a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getDatabase()"><code>AnnotatedPluginDocument.getDatabase()</code></a></li>
 <li><a href="com/biomatters/geneious/publicapi/utilities/CallSoon.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>CallSoon</code></a> - Provides a system for invoking a Runnable shortly in the Swing thread with optional coalescing of multiple calls and an optional delay until the runnable is invoked</li>
 <li><a href="com/biomatters/geneious/publicapi/documents/DocumentNoteUtilities.html" title="class in com.biomatters.geneious.publicapi.documents"><code>DocumentNoteUtilities</code></a> - management of custom notes on documents</li>
 <li><a href="com/biomatters/geneious/publicapi/documents/DocumentUtilities.html" title="class in com.biomatters.geneious.publicapi.documents"><code>DocumentUtilities</code></a> - utility methods for handling <a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocuments</code></a></li>
 <li><a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>PluginUtilities</code></a> - utility methods for handling interaction with other plugins</li>
 <li><a href="com/biomatters/geneious/publicapi/utilities/ImportUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>ImportUtilities</code></a> - utility methods for importing documents and various file formats into Geneious</li>
 <li><a href="com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>SequenceUtilities</code></a> and <a href="com/biomatters/geneious/publicapi/utilities/CharSequenceUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>CharSequenceUtilities</code></a> -
 utility methods for handling <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocuments</code></a> and the char sequences (<a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceCharSequence</code></a>) within them
 <li><a href="com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>SequenceExtractionUtilities</code></a> - utility methods for handling extracting sub-sequences from <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocuments</code></a>
 <li><a href="com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME"><code>PluginDocument.FILE_DATA_ATTRIBUTE_NAME</code></a> - provides a system for efficient serialization of binary data in PluginDocuments</li>
 <li><a href="com/biomatters/geneious/publicapi/utilities/FileUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>FileUtilities</code></a>,
  <a href="com/biomatters/geneious/publicapi/utilities/StringUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>StringUtilities</code></a>,
  <a href="com/biomatters/geneious/publicapi/utilities/GuiUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>GuiUtilities</code></a>,
  <a href="com/biomatters/geneious/publicapi/utilities/SystemUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>SystemUtilities</code></a>,
  <a href="com/biomatters/geneious/publicapi/utilities/IconUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>IconUtilities</code></a>,
  <a href="com/biomatters/geneious/publicapi/utilities/GeneralUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>GeneralUtilities</code></a> - various non-bioinformatics utility classes whose names should be self explanatory.
 </li>
 </ul>

 <a name="apiFaq">
 <h3>FAQ</h3>
 <a name="faqAddMenuBar"><h4>How do I add an item to the menu bar or toolbar?</h4>
 This depends on what you will use it for. If it is intended for generating annotations on a sequence, use
 <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerator</code></a>.<a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html#getActionOptions()"><code>getActionOptions()</code></a>.
 Otherwise use <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperation</code></a>.<a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html#getActionOptions()"><code>getActionOptions()</code></a>.
 If your menu item is not intended to apply an action to any documents, you should still use a <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperation</code></a>,
 except it should return an empty selection signature to indicate it accepts no input documents.
 <pre>
  DocumentOperation.getActionOptions() {
     return new <a href="com/biomatters/geneious/publicapi/plugin/GeneiousActionOptions.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>GeneiousActionOptions</code></a>("My Action").setMainMenuLocation(GeneiousActionOptions.MainMenu.Tools).setInMainToolbar(true).setInPopupMenu(true);
  }
 </pre>
 <a name="faqPopupDialog"><h4>How do I popup a dialog for user input?</h4>
 This is usually done as part of a <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerator</code></a>
 or <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperation</code></a> by returning
 <a href="com/biomatters/geneious/publicapi/plugin/Options.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>Options</code></a>  from
 <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html#getOptions(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.plugin.SequenceAnnotationGenerator.SelectionRange)"><code>SequenceAnnotationGenerator.getOptions()</code></a>
 or
 <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html#getOptions(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument...)"><code>DocumentOperation.getOptions()</code></a>.
 <pre>
 public <a href="com/biomatters/geneious/publicapi/plugin/Options.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>Options</code></a> DocumentOperation.getOptions(AnnotatedPluginDocument... documents) throws DocumentOperationException {
     Options options=new Options(getClass());
     options.<a href="com/biomatters/geneious/publicapi/plugin/Options.html#addBooleanOption(java.lang.String, java.lang.String, java.lang.Boolean)"><code>addBooleanOption</code></a>("reverseComplement","Reverse Complement",false);
     options.<a href="com/biomatters/geneious/publicapi/plugin/Options.html#addStringOption(java.lang.String, java.lang.String, java.lang.String)"><code>addStringOption</code></a>("name","Name","Default Name");
     return options;
 }
 </pre>
 Alternatively, if you wish to popup a dialog independent of a <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerator</code></a>
 or <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperation</code></a> then do something like this
 <pre>
 Options options=new Options(getClass());
 Options.BooleanOption reverseComplement = options.addBooleanOption("reverseComplement", "Reverse Complement", false);
 Options.StringOption name = options.addStringOption("name", "Name", "Default Name");
 if (Dialogs.showOptionsDialog(options,"Title",true)) { // returns true if user clicked OK.
     if (reverseComplement.getValue()) {
         System.out.println("Reverse complement selected and name="+name.getValue());
     }

 }
 </pre>
 <a name="faqGetDocumentsInSameFolder"><h4>How do I get all documents in the same folder as a given document?</h4>
 <pre>
 AnnotatedPluginDocument document;
 List&lt;AnnotatedPluginDocument&gt; documentsInSameFolder = document.<a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getDatabase()"><code>getDatabase()</code></a>.<a href="com/biomatters/geneious/publicapi/databaseservice/DatabaseService.html#retrieve(com.biomatters.geneious.publicapi.databaseservice.Query, jebl.util.ProgressListener)"><code>retrieve</code></a>(Query.Factory.<a href="com/biomatters/geneious/publicapi/databaseservice/Query.Factory.html#createBrowseQuery()"><code>createBrowseQuery</code></a>(), ProgressListener.EMPTY);
 </pre>

 <a name="faqGetDocumentsInDatabase"><h4>How do I get all documents in the user's local database?</h4>
 <pre>
 <a href="com/biomatters/geneious/publicapi/databaseservice/WritableDatabaseService.html" title="class in com.biomatters.geneious.publicapi.databaseservice"><code>WritableDatabaseService</code></a> database= (WritableDatabaseService) PluginUtilities.<a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html#getGeneiousService(java.lang.String)"><code>getGeneiousService</code></a>("LocalDocuments");
 List&lt;AnnotatedPluginDocument&gt; documentsInEntireDatabase = database.<a href="com/biomatters/geneious/publicapi/databaseservice/DatabaseService.html#retrieve(com.biomatters.geneious.publicapi.databaseservice.Query, jebl.util.ProgressListener)"><code>retrieve</code></a>(Query.Factory.<a href="com/biomatters/geneious/publicapi/databaseservice/Query.Factory.html#createQuery(java.lang.String)"><code>createQuery</code></a>(""), ProgressListener.EMPTY);
 List&lt;AnnotatedPluginDocument&gt; documentsInRootFolderOnly = database.<a href="com/biomatters/geneious/publicapi/databaseservice/DatabaseService.html#retrieve(com.biomatters.geneious.publicapi.databaseservice.Query, jebl.util.ProgressListener)"><code>retrieve</code></a>(Query.Factory.<a href="com/biomatters/geneious/publicapi/databaseservice/Query.Factory.html#createBrowseQuery()"><code>createBrowseQuery</code></a>(), ProgressListener.EMPTY);
 </pre>

 <a name="faqExportDocument"><h4>How do I export to any document format supported by Geneious?</h4>

 <pre>
 SequenceDocument sequence=new <a href="com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html#DefaultNucleotideSequence(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)"><code>DefaultNucleotideSequence</code></a>("Name","","GATTACA",null);
 AnnotatedPluginDocument document = DocumentUtilities.<a href="com/biomatters/geneious/publicapi/documents/DocumentUtilities.html#createAnnotatedPluginDocument(com.biomatters.geneious.publicapi.documents.PluginDocument)"><code>createAnnotatedPluginDocument</code></a>(sequence);
 PluginUtilities.<a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html#exportDocuments(java.io.File, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument...)"><code>exportDocuments</code></a>(new File("a.fasta"),document);
 </pre>

 <a name="faqImportDocument"><h4>How do I import any document format supported by Geneious?</h4>

 <pre>
 List&lt;AnnotatedPluginDocument&gt; docs = PluginUtilities.<a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html#importDocuments(java.io.File, jebl.util.ProgressListener)"><code>importDocuments</code></a>(new File("a.fasta"), ProgressListener.EMPTY);
 <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a> sequence = (SequenceDocument) docs.get(0).<a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getDocument()"><code>getDocument</code></a>();
 System.out.println("Sequence String="+sequence.<a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>getSequenceString</code></a>());
 </pre>

 <a name="faqAddDocument"><h4>How do I add a document to the user's database?</h4>

 Normally you wouldn't do this directly. Instead use a
 <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperation</code></a> to generate documents. But if
 this doesn't suit your plugin needs, you could do something like this which creates
 a folder in the user's database, adds a new document to it, and selects that document.
 <pre>
 SequenceDocument sequence=new <a href="com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html#DefaultNucleotideSequence(java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)"><code>DefaultNucleotideSequence</code></a>("Name","","GATTACA",null);
 AnnotatedPluginDocument document = DocumentUtilities.<a href="com/biomatters/geneious/publicapi/documents/DocumentUtilities.html#createAnnotatedPluginDocument(com.biomatters.geneious.publicapi.documents.PluginDocument)"><code>createAnnotatedPluginDocument</code></a>(sequence);
 <a href="com/biomatters/geneious/publicapi/databaseservice/WritableDatabaseService.html" title="class in com.biomatters.geneious.publicapi.databaseservice"><code>WritableDatabaseService</code></a> database= (WritableDatabaseService) PluginUtilities.getGeneiousService("LocalDocuments");
 WritableDatabaseService folder = database.<a href="com/biomatters/geneious/publicapi/databaseservice/WritableDatabaseService.html#createChildFolder(java.lang.String)"><code>createChildFolder</code></a>("My Folder");
 document = folder.<a href="com/biomatters/geneious/publicapi/databaseservice/WritableDatabaseService.html#addDocumentCopy(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, jebl.util.ProgressListener)"><code>addDocumentCopy</code></a>(document, ProgressListener.EMPTY);
 DocumentUtilities.<a href="com/biomatters/geneious/publicapi/documents/DocumentUtilities.html#selectDocument(com.biomatters.geneious.publicapi.documents.URN)"><code>selectDocument</code></a>(document.getURN());
 </pre>

 <a name="faqAddAnnotations"><h4>How do I add annotations to a sequence?</h4>
  Normally you wouldn't do this directly. Instead use a
 <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerator</code></a> which integrates creating
 annotations into the sequence viewer and manages undoing and saving. But if you need to do it yourself, here is how:

 <pre>
 AnnotatedPluginDocument document; // This is assumed to be an editable sequence document that is already in the user's database.
 <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotation</code></a> annotationToAdd = new SequenceAnnotation("Name", SequenceAnnotation.TYPE_CDS, new SequenceAnnotationInterval(3,10));
 <a href="com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>EditableSequenceDocument</code></a> sequence = (EditableSequenceDocument) document.<a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getDocument()"><code>getDocument</code></a>();
 List&lt;SequenceAnnotation&gt; annotations= new ArrayList&lt;SequenceAnnotation&gt;(sequence.<a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()"><code>getSequenceAnnotations</code></a>());
 annotations.add(annotationToAdd);
 sequence.<a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)"><code>setAnnotations</code></a>(annotations);
 document.<a href="com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#saveDocument()"><code>saveDocument</code></a>();
 </pre>
 
 <a name="faqDependOnExternal"><h4>How do I allow my plugin to depend on external libraries or resources?</h4>
 For accessing external resources such as data files your plugin should be distributed as a zip file with extension .gplugin
 containing only a single folder named exactly the same as the fully qualified name of your GeneiousPlugin class.
 This folder must then contain your plugin jar file (can have any name) and any other files your plugin uses.
 This folder will be given to your plugin at runtime through the pluginDirectory parameter of the
 <a href="com/biomatters/geneious/publicapi/plugin/GeneiousPlugin.html#initialize(java.io.File, java.io.File)"><code>GeneiousPlugin.initialize</code></a> method.
 See ExampleGeneiousService plugin for an example that bundles a data file.
 <p/>
 To depend on an external library jar file, also just bundle that alongside your plugin's jar file.
 Geneious automatically adds all jar files in your plugin's directory to your plugin's class-loader. However,
 in rare cases you may find this doesn't work as the external library may try to find classes using the
 system or thread class-loader instead of the class-loader they
 were created from. If the external library is using the current thread class-loader, wrapping calls
 to the external library from within your plugin like in the following code fragment may help:
 <pre>
 final ClassLoader oldContextClassLoader = Thread.currentThread().getContextClassLoader();
 try {
     Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
     // Code that calls a problematic external library method goes here.
 }
 finally {
     Thread.currentThread().setContextClassLoader(oldContextClassLoader);
 }
 </pre>
 <a name="faqLog"><h4>How do I create a log file from within my plugin?</h4>
 You can use Java's standard logging system (see
 <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/logging/package-summary.html">java.util.logging</a>).
 You can specify your logging properties file in the Geneious virtual machine options file.<br><br>
 <ul>
 <li>
 On Windows, this is <code>Geneious.vmoptions</code> inside your Geneious installation directory.
 Each command line option should be separated by a new line. For example, your VM options file may contain:
 <pre>
 -Xmx500M
 -Djava.util.logging.config.file=c:\temp\logging.properties
 </pre>
 </li>
 <li>
 On MacOS, this is <code>Info.plist</code> in the <code>Contents</code> folder inside the Geneious.app bundle (Ctrl+Click on
 Geneious and select <code>Show package contents</code>).
 Open <code>Info.plist</code> using the default program (Property List Editor.app) and go in to the <code>Java</code>
 then the <code>Properties</code> node. Change the key <code>java.util.logging.config.file</code> to contain the path
 to your logging properties file, eg:
 <pre>
/Users/richard/logging.properties<br>
 </pre>
 </li></ul>
 Your logging.properties file may contain something like:
 <pre>
.level=WARNING
com.biomatters.level=INFO
your.package.name.level=FINE
handlers=java.util.logging.FileHandler
java.util.logging.FileHandler.append=true
java.util.logging.FileHandler.formatter=java.util.logging.SimpleFormatter
java.util.logging.FileHandler.pattern=c:\\temp\\geneious.log
 </pre>
 Replacing your.package.name with your package name. And in your source code you would use a line like:
 <pre>
 Logger.getLogger(getClass().getName()).fine("Hello World!");
 </pre>

 <a name="faqTestCases"><h4>How do I write unit test cases for my plugin?</h4>
 Geneious provides the class <a href="com/biomatters/geneious/publicapi/plugin/TestGeneious.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>TestGeneious</code></a> to facilitate
 writing test cases. If you add all Geneious libraries to your classpath, then you can use
 <a href="com/biomatters/geneious/publicapi/plugin/TestGeneious.html#initialize()"><code>TestGeneious.initialize</code></a> to initialize
 many of the methods in the public api that would otherwise only work from within the Geneious application.
 For example <a href="com/biomatters/geneious/publicapi/documents/DocumentUtilities.html#createAnnotatedPluginDocument(com.biomatters.geneious.publicapi.documents.PluginDocument)"><code>DocumentUtilities.createAnnotatedPluginDocument</code></a>
 requires TestGeneious.initialize() to have been called. If your plugin test case itself needs to depend on other
 plugins, then <a href="com/biomatters/geneious/publicapi/plugin/TestGeneious.html#initializeAllPlugins()"><code>TestGeneious.initializeAllPlugins</code></a> initializes all
 plugins so that methods such as <a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html#getDocumentOperation(java.lang.String)"><code>PluginUtilities.getDocumentOperation</code></a> 
 can be used to obtain references to other plugins. You may also find <a href="com/biomatters/geneious/publicapi/plugin/TestGeneious.html#isRunningTest()"><code>TestGeneious.isRunningTest</code></a>
 useful.
 <p/>
 In a rare case that you want to test your plugin when it is instantiated from the Geneious plugin loading environment (rather
 than just running from class files), then if you build and copy your plugin to the Geneious plugins folder then your test case
 could use <a href="com/biomatters/geneious/publicapi/plugin/TestGeneious.html#initializePlugins(java.lang.String...)"><code>TestGeneious.initializePlugin</code></a>
 to load your plugin and can then use <a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html#getClass(java.lang.String)"><code>PluginUtilities.getClass</code></a>
 to obtain a reference to your class and call any testing methods on it you need to.

 <a name="faqInvokeOperations"><h4>How do I invoke other plugin operations?</h4>
 Sometimes you may want to make use of functionality provided by other plugins when implementing your own plugin. First
 your plugin needs to get a reference to the functionality provided by another plugin using <a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>PluginUtilities</code></a>
 and it can then proceed to make use of the plugin.

 <ol>
 <li>Getting a reference to the other plugin.<br><br>

 Each <a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperation</code></a> and
 <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerator</code></a>
 has a unique ID. (<a href="com/biomatters/geneious/publicapi/plugin/DocumentOperation.html#getUniqueId()"><code>DocumentOperation.getUniqueId()</code></a> and
 <a href="com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html#getUniqueId()"><code>SequenceAnnotationGenerator.getUniqueId()</code></a>).
 As long as you know the uniqueID of the operation or annotation generator you can use
 <a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html#getDocumentOperation(java.lang.String)"><code>PluginUtilities.getDocumentOperation()</code></a>
 or <a href="com/biomatters/geneious/publicapi/plugin/PluginUtilities.html#getSequenceAnnotationGenerator(java.lang.String)"><code>PluginUtilities.getSequenceAnnotationGenerator()</code></a>
 to get a reference to it.
 <p/>
 To view a list of all available uniqueIds, use the following code fragment: <br><br>

 <pre>
 StringBuilder builder = new StringBuilder();
 builder.append("Document Operations:\n");
 for (DocumentOperation operation : PluginUtilities.getDocumentOperations()) {
     builder.append(operation.getActionOptions().getName()+" has id: "+operation.getUniqueId()+"\n");
 }
 builder.append("\n\nAnnotation Generators:\n");
 for (SequenceAnnotationGenerator generator : PluginUtilities.getSequenceAnnotationGenerators()) {
     builder.append(generator.getActionOptions().getName()+" has id: "+generator.getUniqueId()+"\n");
 }
 String message=builder.toString();
 Dialogs.showMessageDialog(message);
 </pre>

 </li>
 <li>Using the other plugin.<br><br>
 Once you have a reference to your document operation or annotation generator, you could do something like this to use it:<br><br>

 <pre>
 SequenceAnnotationGenerator primerDesign = PluginUtilities.getSequenceAnnotationGenerator("com.biomatters.plugins.primerDesign.PrimerDesignAnnotationGenerator");

 SequenceDocument sequence = new DefaultNucleotideSequence("Name","","GATTACA",null);
 AnnotatedPluginDocument sequenceDocument = DocumentUtilities.createAnnotatedPluginDocument(sequence);
 SequenceAnnotationGenerator.SelectionRange selectionRange = new SequenceAnnotationGenerator.SelectionRange(0, 0, 0, sequence.getSequenceLength() - 1);

 Options options = primerDesign.getOptions(selectionRange, sequenceDocument);
 options.setStringValue("MAXIMUM_NUMBER_OF_NON_MATCHING_SEQUENCES","2");

 List&lt;SequenceAnnotation&gt; annotationsList = primerDesign.generateAnnotations(selectionRange, ProgressListener.EMPTY, options, sequenceDocument).get(0);
 </pre>
 Calling <a href="com/biomatters/geneious/publicapi/plugin/Options.html#setStringValue(java.lang.String, java.lang.String)"><code>Options.setStringValue()</code></a> requires knowledge of what options are available. For a given
 Options instance, calling <a href="com/biomatters/geneious/publicapi/plugin/Options.html#getDescriptionAndState()"><code>Options.getDescriptionAndState()</code></a> lists all supported option values. For example,
 <pre>
 Dialogs.showMessageDialog(options.getDescriptionAndState());
 </pre>
 in the above example would list all the options you could set when generating primers.
 </li>
 </ol>
 <a name="EndOfPublicApiDocumentation"></div>
</div>
<p>See: <a href="#overview_description">Description</a></p>
</div>
<div class="contentContainer">
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Packages table, listing packages, and an explanation">
<caption><span>Packages</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/components/package-summary.html">com.biomatters.geneious.publicapi.components</a></td>
<td class="colLast">
<div class="block">Provides useful GUI (graphical user interface) components, none of which
 are necessary for creating a plugin, but many plugin implentations
 will find these components useful.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/databaseservice/package-summary.html">com.biomatters.geneious.publicapi.databaseservice</a></td>
<td class="colLast">
<div class="block">Provides the interface and associated classes for
 defining a database service which is a service that
 appears on the left-hand side of the main Geneious window
 and provides the user with access to a database, for example
 NCBI or the local database for storing the user's documents.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/documents/package-summary.html">com.biomatters.geneious.publicapi.documents</a></td>
<td class="colLast">
<div class="block">Provides interfaces and classes for defining documents in Geneious and related
 interfaces and classes for dealing with XMLSerialization.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/documents/sequence/package-summary.html">com.biomatters.geneious.publicapi.documents.sequence</a></td>
<td class="colLast">
<div class="block">Provides interfaces specifying the types of
 <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>sequence</code></a> and
 <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>alignment</code></a>
 documents in Geneious, together with concrete classes used by sequence documents
 such as <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotation</code></a>
 and <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceCharSequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/documents/types/package-summary.html">com.biomatters.geneious.publicapi.documents.types</a></td>
<td class="colLast">
<div class="block">Provides interfaces specifying the types of documents available in Geneious.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a></td>
<td class="colLast">
<div class="block">Provides implementations for many of the document interfaces defined
 in the package com.biomatters.geneious.publicapi.documents.types
 together with some utility classes for dealing with some of these
 document types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/implementations/sequence/package-summary.html">com.biomatters.geneious.publicapi.implementations.sequence</a></td>
<td class="colLast">
<div class="block">Provides <a href="com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>
 implementations for nucelotide, amino acid, and nucleotide graph (chromatogram) sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/implementations/structure/package-summary.html">com.biomatters.geneious.publicapi.implementations.structure</a></td>
<td class="colLast">
<div class="block">Provides <a href="com/biomatters/geneious/publicapi/documents/types/MolecularStructureDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types"><code>MolecularStructureDocument</code></a>
 implementations for many commonly used 3D structure documents.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/plugin/package-summary.html">com.biomatters.geneious.publicapi.plugin</a></td>
<td class="colLast">
<div class="block">Provides the <a href="com/biomatters/geneious/publicapi/plugin/GeneiousPlugin.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>GeneiousPlugin</code></a> interface and plugin
 related interfaces.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/utilities/package-summary.html">com.biomatters.geneious.publicapi.utilities</a></td>
<td class="colLast">
<div class="block">Provides various utility methods and classes, none of which
 are necessary for creating a plugin, but many plugins implentations
 will find these methods useful.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="com/biomatters/geneious/publicapi/utilities/xml/package-summary.html">com.biomatters.geneious.publicapi.utilities.xml</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="footer"><a name="overview_description">
<!--   -->
</a>
<div class="subTitle">
<div class="block"><h2></div>
</div>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li class="navBarCell1Rev">Overview</li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-all.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?overview-summary.html" target="_top">Frames</a></li>
<li><a href="overview-summary.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
