<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:53 NZDT 2016 -->
<title>SequenceUtilities (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceUtilities (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceUtilities.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/utilities/ProgressInputStream.ProgressInterruptedIOException.html" title="class in com.biomatters.geneious.publicapi.utilities"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/utilities/StandardIcons.html" title="enum in com.biomatters.geneious.publicapi.utilities"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html" target="_top">Frames</a></li>
<li><a href="SequenceUtilities.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.utilities</div>
<h2 title="Class SequenceUtilities" class="title">Class SequenceUtilities</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.utilities.SequenceUtilities</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public final class <span class="strong">SequenceUtilities</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">A noninstantiable class providing static methods for common tasks associated with nucleotide and
 protein sequences.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>SequenceExtractionUtilities</code></a></dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#alignmentFromJeblSequences(java.lang.String, java.util.List)">alignmentFromJeblSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                          <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;jeblSequences)</code>
<div class="block">Converts the given alignment of Jebl sequences into a DefaultAlignmentDocument</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asDna(java.lang.CharSequence)">asDna</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence)</code>
<div class="block">Views an underlying (nucleotide) CharSequence as DNA by dynamically
 translating 'U's to 'T's and 'u's to 't's

 It is guaranteed that if charSequence instanceof SequenceCharSequence, the returned
 value will also be instanceof SequenceCharSequence (but it may not support log-time modifications).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblAlignment(java.util.List)">asJeblAlignment</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences)</code>
<div class="block">Convert a list of (aligned) Geneious sequences to a jebl alignmnent</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblSequence(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">asJeblSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;referenceDocument,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>asJeblSequence(SequenceAlignmentDocument.ReferencedSequence, SequenceDocument)</code></a></i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">asJeblSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Convert from a Geneious sequence to a jebl sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">asJeblSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Convert from a Geneious sequence to a jebl sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblSequences(java.util.List)">asJeblSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences)</code>
<div class="block">Convert a set of Geneious sequences to jebl sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)">asJeblSequences</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>...&nbsp;sequences)</code>
<div class="block">Convert a set of Geneious sequences to jebl sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asRna(java.lang.CharSequence)">asRna</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence)</code>
<div class="block">Views an underlying (nucleotide) CharSequence as RNA by dynamically
 translating 'T's to 'U's and 't's to 'u's

 It is guaranteed that if charSequence instanceof SequenceCharSequence, the returned
 value will also be instanceof SequenceCharSequence (but it may not support log-time modifications).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asTranslation(java.lang.CharSequence, jebl.evolution.sequences.GeneticCode)">asTranslation</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence,
             <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a>&nbsp;geneticCode)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asTranslation(java.lang.CharSequence, jebl.evolution.sequences.GeneticCode, boolean)"><code>asTranslation(CharSequence, jebl.evolution.sequences.GeneticCode, boolean)</code></a> instead.</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asTranslation(java.lang.CharSequence, jebl.evolution.sequences.GeneticCode, boolean)">asTranslation</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence,
             <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a>&nbsp;geneticCode,
             boolean&nbsp;translateFirstCodonUsingFirstCodonTable)</code>
<div class="block">Views an underlying (nucleotide) CharSequence as its translation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#containsInvalidResidues(java.lang.CharSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean, boolean)">containsInvalidResidues</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceResidues,
                       <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                       boolean&nbsp;allowGaps,
                       boolean&nbsp;fastIncompleteCheck)</code>
<div class="block">Checks if a sequence contain invalid sequence residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#containsInvalidResidues(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)">containsInvalidResidues</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceResidues,
                       <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                       boolean&nbsp;allowGaps,
                       boolean&nbsp;fastIncompleteCheck)</code>
<div class="block">Checks if a sequence contain invalid sequence residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#containsInvalidResidues(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean, boolean)">containsInvalidResidues</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                       boolean&nbsp;allowGaps,
                       boolean&nbsp;fastIncompleteCheck)</code>
<div class="block">Checks if a sequence contain invalid sequence residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createNewDocumentsByTransformingSequences(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener, java.lang.String)">createNewDocumentsByTransformingSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;sourceDocuments,
                                         <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a>&nbsp;transformer,
                                         <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener,
                                         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newSequenceOrDocumentNamePrefix)</code>
<div class="block">Transforms the sequence(s) in each input document and returns a new document corresponding to each input document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createSequenceCopy(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">createSequenceCopy</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;original)</code>
<div class="block">Creates a copy of the original sequence if necessary.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createSequenceCopyAdjustedForGapInsertion(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.CharSequence)">createSequenceCopyAdjustedForGapInsertion</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                                         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;gappedSequenceCharacters)</code>
<div class="block">Creates a copy of the given sequence with annotations, sequence residues, and chromatogram values adjusted to account for gap insertion.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createSequenceCopyEditable(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">createSequenceCopyEditable</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;original)</code>
<div class="block">Creates a copy of the original sequence that is editable.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createSequenceDocument(jebl.evolution.sequences.SequenceType, java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)">createSequenceDocument</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceString,
                      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;creationDate)</code>
<div class="block">Creates a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideSequence</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultAminoAcidSequence</code></a>
 depending on sequenceType.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#generateConsensus(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)">generateConsensus</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
                 <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Generates a consensus sequence for an alignment using default consensus settings.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#generateConsensusSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)">generateConsensusSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
                         <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#generateConsensus(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)"><code>generateConsensus(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getAlphabet(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">getAlphabet</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Get the Alphabet of a sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getAlphabet(jebl.evolution.sequences.SequenceType)">getAlphabet</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType)</code>
<div class="block">Gets a Geneious alphabet type that is equivalent to a jebl library SequenceType.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getAnnotationsOfType(java.util.List, java.lang.String)">getAnnotationsOfType</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>
<div class="block">Get all annotations in list matching the given type</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String)">getAnnotationsOfType</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;document,
                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, boolean)"><code>getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, String, boolean)</code></a> instead.</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, boolean)">getAnnotationsOfType</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;document,
                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type,
                    boolean&nbsp;returnAnnotationsInTracks)</code>
<div class="block">Get all annotations in document matching the given type.<p/>
 <b>WARNING:</b> this list may not include all SequenceAnnotations represented as annotations in the sequence viewer.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getBlastAlignmentText(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean)">getBlastAlignmentText</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
                     boolean&nbsp;geneiousFriendly)</code>
<div class="block">Formats the given alignment in BLAST text format</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean)">getForwardRegexForSequence</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                          boolean&nbsp;interpretAmbiguitiesInQuery)</code>
<div class="block">Equivalent to calling <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)"><code>getForwardRegexForSequence(charSequence, sequenceType, interpretAmbiguities, false, true)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)">getForwardRegexForSequence</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                          boolean&nbsp;interpretAmbiguitiesInQuery,
                          boolean&nbsp;interpretAmbiguitiesInTarget)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)"><code>getForwardRegexForSequence(CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)">getForwardRegexForSequence</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                          boolean&nbsp;interpretAmbiguitiesInQuery,
                          boolean&nbsp;interpretAmbiguitiesInTarget,
                          boolean&nbsp;allowExtraGapsInTarget)</code>
<div class="block">Given a nucleotide or amino acid sequence, returns a regular
 expression that matches forward occurrences of this sequence in a larger
 sequence, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/regex/Pattern.html?is-external=true" title="class or interface in java.util.regex">Pattern</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexPatternForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean)">getForwardRegexPatternForSequence</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                                 <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                 boolean&nbsp;interpretAmbiguitiesInQuery)</code>
<div class="block">Equivalent to calling <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexPatternForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)"><code>getForwardRegexPatternForSequence(charSequence, sequenceType, interpretAmbiguities, false)</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/regex/Pattern.html?is-external=true" title="class or interface in java.util.regex">Pattern</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexPatternForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)">getForwardRegexPatternForSequence</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                                 <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                 boolean&nbsp;interpretAmbiguitiesInQuery,
                                 boolean&nbsp;interpretAmbiguitiesInTarget)</code>
<div class="block">Given a nucleotide or amino acid sequence string, returns a regular
 expression pattern that matches forward occurrences of this search string in a larger
 sequence string,</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getLeadingGapsLength(java.lang.CharSequence)">getLeadingGapsLength</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</code>
<div class="block">Returns the start index of the non-gap regions in the specified charSequence,
 i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getMaximalAmbiguitySymbol(jebl.evolution.sequences.SequenceType)">getMaximalAmbiguitySymbol</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType)</code>
<div class="block">get the code for the state in this sequence type which represents a base/residue that is completely unknown</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getNumberOfSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">getNumberOfSequences</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document,
                    <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</code>
<div class="block">Gets the total number of nucleotide or amino acid sequences contained in the given document which may be an individual sequence, sequence list, or alignment/contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getNumberOfSequences(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">getNumberOfSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents,
                    <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</code>
<div class="block">Gets the total number of nucleotide or amino acid sequences contained in the given documents which may be individual sequences, sequence lists, or alignments/contigs.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getOriginalIndex(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, int)">getOriginalIndex</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                int&nbsp;index)</code>
<div class="block">Gets the original numbering of the given index if it is covered by a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION"><code>SequenceAnnotation.TYPE_EXTRACTED_REGION</code></a> annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">getSequenceAndTrackAnnotations</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">A convenience method to get all annotations on the sequence and all annotations on all <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a> on this sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAnnotationsIncludingImmutableSequencesTrims(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">getSequenceAnnotationsIncludingImmutableSequencesTrims</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Gets all the annotations on the given sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)">getSequences</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;documents,
            <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
            <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">get all the sequences out of a set of AnnotatedPluginDocuments that may wrap SequenceDocuments, SequenceListDocuments or SequenceAlignmentDocuments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequences(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)">getSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents,
            <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
            <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">get all the sequences out of a set of AnnotatedPluginDocuments that may wrap SequenceDocuments, SequenceListDocuments or SequenceAlignmentDocuments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequencesWithoutImmediateLoading(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">getSequencesWithoutImmediateLoading</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;documents,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</code>
<div class="block">Like <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)"><code>getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)</code></a>
 but doesn't require each plugin document to be in memory as long as this Collection is around.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceType(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">getSequenceType</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</code>
<div class="block">Examines a document and determines what the (jebl) sequence type (or types) of the document is (or are), and returns it (or them).<br/><br/>

 Always returns a List&lt;SequenceType&gt; of size 0, 1 or 2.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">getSequenceType</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</code>
<div class="block">Gets a jebl library SequenceType that is equivalent to a Geneious alphabet.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">getSequenceType</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Get the (jebl) sequence type.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getTrailingGapsLength(java.lang.CharSequence)">getTrailingGapsLength</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</code>
<div class="block">Get the number of trailing gap ('-') characters in the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getTrailingGapsStartIndex(java.lang.CharSequence)">getTrailingGapsStartIndex</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</code>
<div class="block">Returns the end index of the non-gap regions in the specified charSequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getValidSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">getValidSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                boolean&nbsp;allowGaps)</code>
<div class="block">Replace any invalid bases/residues in the given sequence document with ambiguity symbols.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getValidSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean, boolean)">getValidSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                boolean&nbsp;allowGaps,
                boolean&nbsp;replaceWithGaps)</code>
<div class="block">Replace any invalid bases/residues in the given sequence document with ambiguity symbols or gaps.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#isPredominantlyRna(java.lang.CharSequence, int)">isPredominantlyRna</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence,
                  int&nbsp;maximumNonGapsToLookAt)</code>
<div class="block">Checks whether a sequence is predominantly RNA (rather than DNA).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#isRna(java.lang.CharSequence)">isRna</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</code>
<div class="block">Checks whether a sequence is RNA (rather than DNA) based on whether the sequence contains either a T/t or a U/u first.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#isStateAssignableFrom(jebl.evolution.sequences.State, jebl.evolution.sequences.State)">isStateAssignableFrom</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/State.html?is-external=true" title="class or interface in jebl.evolution.sequences">State</a>&nbsp;stateA,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/State.html?is-external=true" title="class or interface in jebl.evolution.sequences">State</a>&nbsp;stateB)</code>
<div class="block">Same as stateA.getCanonicalStates().containsAll(stateB.getCanonicalStates())
 except that for NucleotideStates and AminoAcidStates it caches the result.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#removeGaps(java.lang.CharSequence)">removeGaps</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</code>
<div class="block">Constructs a sequence without gaps ('-') from a specified sequence that
 potentially has gaps.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#removeInvalidResidues(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean)">removeInvalidResidues</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                     boolean&nbsp;allowGaps)</code>
<div class="block">Get a sequence string identical to <code>sequence</code>
 except that any invalid residues are removed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#replaceQuestionMarksWithMaximalAmbiguitySymbol(jebl.evolution.sequences.SequenceType, java.lang.String)">replaceQuestionMarksWithMaximalAmbiguitySymbol</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence)</code>
<div class="block">get a version of a sequence string with any question marks replaces with N (for nucleotide sequences) or X (for protein sequences)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#reverseComplement(java.lang.CharSequence)">reverseComplement</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</code>
<div class="block">
 Provides a dynamic reverse complement view onto a nucletoide CharSequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#setOriginalResidueNumbering(com.biomatters.geneious.publicapi.documents.sequence.EditableSequenceDocument, int, boolean)">setOriginalResidueNumbering</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>&nbsp;document,
                           int&nbsp;startIndex,
                           boolean&nbsp;isReverse)</code>
<div class="block">set the original residue numbering of a document
 the residue index of a document will appear shifted
 if the user has "show original residue numbers" selected in the sequence view</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#toHTMLFragment(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String)">toHTMLFragment</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;additionalContent)</code>
<div class="block">Generate a HTML fragment that summarises a sequence, including the sequence string.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getForwardRegexForSequence</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getForwardRegexForSequence(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                                <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                boolean&nbsp;interpretAmbiguitiesInQuery)</pre>
<div class="block">Equivalent to calling <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)"><code>getForwardRegexForSequence(charSequence, sequenceType, interpretAmbiguities, false, true)</code></a></div>
</li>
</ul>
<a name="getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getForwardRegexForSequence</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getForwardRegexForSequence(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                                           <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                           boolean&nbsp;interpretAmbiguitiesInQuery,
                                           boolean&nbsp;interpretAmbiguitiesInTarget)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)"><code>getForwardRegexForSequence(CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)</code></a> instead</i></div>
<div class="block">Equivalent to <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)"><code>getForwardRegexForSequence(...,true)</code></a></div>
</li>
</ul>
<a name="getForwardRegexForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getForwardRegexForSequence</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getForwardRegexForSequence(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                                <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                boolean&nbsp;interpretAmbiguitiesInQuery,
                                boolean&nbsp;interpretAmbiguitiesInTarget,
                                boolean&nbsp;allowExtraGapsInTarget)</pre>
<div class="block">Given a nucleotide or amino acid sequence, returns a regular
 expression that matches forward occurrences of this sequence in a larger
 sequence, i.e. a String s such that Pattern.compile(s, Pattern.CASE_INSENSITIVE)
 will find all case insensitive forward matches of sequenceString in a larger
 sequence.
 The regular expression returned will also match sequences with gaps inserted at any point within the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>querySequence</code> - The nucleotide or amino acid sequence to search for.</dd><dd><code>sequenceType</code> - The type of the sequence</dd><dd><code>interpretAmbiguitiesInQuery</code> - If true, then an ambiguous character
        (e.g. R for nucleotides) in querySequence will match the
         corresponding canonical states (A and G) in the target.</dd><dd><code>interpretAmbiguitiesInTarget</code> - If true, then an ambiguous character
        (e.g. R for nucleotides) in the sequence being searched within will match the
         corresponding canonical states (A and G) in the querySequence.</dd><dd><code>allowExtraGapsInTarget</code> - If true, then additional gaps will be allowed in the sequence
        being search within</dd>
<dt><span class="strong">Returns:</span></dt><dd>a regular expression that matches forward occurrences of
         this search string in a larger sequence string or null if any of the characters
         in the sequence string are not valid residues for sequenceType</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
<a name="getForwardRegexPatternForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getForwardRegexPatternForSequence</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/regex/Pattern.html?is-external=true" title="class or interface in java.util.regex">Pattern</a>&nbsp;getForwardRegexPatternForSequence(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                                        <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                        boolean&nbsp;interpretAmbiguitiesInQuery)</pre>
<div class="block">Equivalent to calling <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getForwardRegexPatternForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)"><code>getForwardRegexPatternForSequence(charSequence, sequenceType, interpretAmbiguities, false)</code></a></div>
</li>
</ul>
<a name="getForwardRegexPatternForSequence(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getForwardRegexPatternForSequence</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/regex/Pattern.html?is-external=true" title="class or interface in java.util.regex">Pattern</a>&nbsp;getForwardRegexPatternForSequence(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;querySequence,
                                        <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                        boolean&nbsp;interpretAmbiguitiesInQuery,
                                        boolean&nbsp;interpretAmbiguitiesInTarget)</pre>
<div class="block">Given a nucleotide or amino acid sequence string, returns a regular
 expression pattern that matches forward occurrences of this search string in a larger
 sequence string,</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>querySequence</code> - The nucleotide or amino acid sequence to search for.</dd><dd><code>sequenceType</code> - The type of the sequence</dd><dd><code>interpretAmbiguitiesInQuery</code> - If true, then an ambiguous character
        (e.g. R for nucleotides) in sequenceString will match the
         corresponding canonical states (A and G) in the target.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a regular expression that matches forward occurrences of
         this search string in a larger sequence string or null if any of the characters
         in the sequence string are not valid residues for sequenceType</dd></dl>
</li>
</ul>
<a name="isStateAssignableFrom(jebl.evolution.sequences.State, jebl.evolution.sequences.State)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isStateAssignableFrom</h4>
<pre>public static&nbsp;boolean&nbsp;isStateAssignableFrom(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/State.html?is-external=true" title="class or interface in jebl.evolution.sequences">State</a>&nbsp;stateA,
                            <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/State.html?is-external=true" title="class or interface in jebl.evolution.sequences">State</a>&nbsp;stateB)</pre>
<div class="block">Same as stateA.getCanonicalStates().containsAll(stateB.getCanonicalStates())
 except that for NucleotideStates and AminoAcidStates it caches the result.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>stateA</code> - A state (e.g. a NucleotideState or AminoAcidState)</dd><dd><code>stateB</code> - A state of the same type as stateB</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if stateA.getCanonicalStates().containsAll(stateB.getCanonicalStates))</dd></dl>
</li>
</ul>
<a name="createSequenceDocument(jebl.evolution.sequences.SequenceType, java.lang.String, java.lang.String, java.lang.CharSequence, java.util.Date)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createSequenceDocument</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;createSequenceDocument(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description,
                                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceString,
                                             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;creationDate)</pre>
<div class="block">Creates a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultNucleotideSequence</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultAminoAcidSequence</code></a>
 depending on sequenceType.
 See the documentation of these classes' constructors for the semantics of the parameters.</div>
</li>
</ul>
<a name="setOriginalResidueNumbering(com.biomatters.geneious.publicapi.documents.sequence.EditableSequenceDocument, int, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOriginalResidueNumbering</h4>
<pre>public static&nbsp;void&nbsp;setOriginalResidueNumbering(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>&nbsp;document,
                               int&nbsp;startIndex,
                               boolean&nbsp;isReverse)</pre>
<div class="block">set the original residue numbering of a document
 the residue index of a document will appear shifted
 if the user has "show original residue numbers" selected in the sequence view</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - document to set the residue numbering for</dd><dd><code>startIndex</code> - start index for the residue numbering.  The first original residue is residue 1.</dd><dd><code>isReverse</code> - true if the residue numbering should count down from startIndex, false if it should count up.</dd></dl>
</li>
</ul>
<a name="containsInvalidResidues(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>containsInvalidResidues</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;containsInvalidResidues(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                             boolean&nbsp;allowGaps,
                             boolean&nbsp;fastIncompleteCheck)</pre>
<div class="block">Checks if a sequence contain invalid sequence residues.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceDocument</code> - the sequence to check for validity.</dd><dd><code>allowGaps</code> - true if the sequence is allowed to contain gaps</dd><dd><code>fastIncompleteCheck</code> - This parameter is ignored. It was added when Java 5 was widely used which is 10 times slower than Java 6. Checking enormous sequences is slow (a 2GB sequence takes about 20 seconds in Java 5, 2 seconds in Java 6). Set this parameter to true to check only the first and last 1,000,000 residues which catches almost all invalid cases and is much faster on enormous sequences.</dd>
<dt><span class="strong">Returns:</span></dt><dd>null if the sequence residues are all valid, or if the sequence contains invalid residues a message describing the first invalid residue is returned.</dd></dl>
</li>
</ul>
<a name="getSequenceType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceType</h4>
<pre>public static&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;getSequenceType(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</pre>
<div class="block">Gets a jebl library SequenceType that is equivalent to a Geneious alphabet.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alphabet</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>sequence type that is equivalent to this alphabet</dd></dl>
</li>
</ul>
<a name="getAlphabet(jebl.evolution.sequences.SequenceType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlphabet</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;getAlphabet(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType)</pre>
<div class="block">Gets a Geneious alphabet type that is equivalent to a jebl library SequenceType.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceType</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>alphabet that is equivalent to this sequence tyqpe</dd></dl>
</li>
</ul>
<a name="containsInvalidResidues(java.lang.CharSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>containsInvalidResidues</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;containsInvalidResidues(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceResidues,
                             <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                             boolean&nbsp;allowGaps,
                             boolean&nbsp;fastIncompleteCheck)</pre>
<div class="block">Checks if a sequence contain invalid sequence residues.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceResidues</code> - sequence residues to check for validity.</dd><dd><code>alphabet</code> - the alphabet of residues expected to be in sequenceResidues</dd><dd><code>allowGaps</code> - true if the sequence is allowed to contain gaps</dd><dd><code>fastIncompleteCheck</code> - This parameter is ignored. It was added when Java 5 was widely used which is 10 times slower than Java 6. Checking enormous sequences is slow (a 2GB sequence takes about 20 seconds in Java 5, 2 seconds in Java 6). Set this parameter to true to check only the first and last 1,000,000 residues which catches almost all invalid cases and is much faster on enormous sequences.</dd>
<dt><span class="strong">Returns:</span></dt><dd>null if the sequence residues are all valid, or if the sequence contains invalid residues a message describing the first invalid residue is returned.</dd></dl>
</li>
</ul>
<a name="containsInvalidResidues(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>containsInvalidResidues</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;containsInvalidResidues(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceResidues,
                             <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                             boolean&nbsp;allowGaps,
                             boolean&nbsp;fastIncompleteCheck)</pre>
<div class="block">Checks if a sequence contain invalid sequence residues.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceResidues</code> - sequence residues to check for validity.</dd><dd><code>sequenceType</code> - the type of residues expected to be in sequenceResidues</dd><dd><code>allowGaps</code> - true if the sequence is allowed to contain gaps</dd><dd><code>fastIncompleteCheck</code> - This parameter is ignored. It was added when Java 5 was widely used which is 10 times slower than Java 6. Checking enormous sequences is slow (a 2GB sequence takes about 20 seconds in Java 5, 2 seconds in Java 6). Set this parameter to true to check only the first and last 1,000,000 residues which catches almost all invalid cases and is much faster on enormous sequences.</dd>
<dt><span class="strong">Returns:</span></dt><dd>null if the sequence residues are all valid, or if the sequence contains invalid residues a message describing the first invalid residue is returned.</dd></dl>
</li>
</ul>
<a name="removeInvalidResidues(java.lang.CharSequence, jebl.evolution.sequences.SequenceType, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeInvalidResidues</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;removeInvalidResidues(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                                 <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                 boolean&nbsp;allowGaps)</pre>
<div class="block">Get a sequence string identical to <code>sequence</code>
 except that any invalid residues are removed. Gaps are only removed if allowGaps is false.
 All valid characters remain unchanged (they maintain their original case and there are no U->T replacements for nucleotides.)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - a string of residues that may or may not be valid residues</dd><dd><code>sequenceType</code> - the type of residues in sequence</dd><dd><code>allowGaps</code> - if this is true, then gaps are not removed.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a sequence string identical to <code>sequence</code>
 except that any invalid residues are removed. If there are no invalid residues, <code>sequence</code> is returned.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/OutOfMemoryError.html?is-external=true" title="class or interface in java.lang">OutOfMemoryError</a></code> - if a sequence comtains invalid residues and a valid version of the sequence cannot fit in memory</dd></dl>
</li>
</ul>
<a name="getValidSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getValidSequence</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;getValidSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                            boolean&nbsp;allowGaps)</pre>
<div class="block">Replace any invalid bases/residues in the given sequence document with ambiguity symbols.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceDocument</code> - sequence document to replace the invalid bases in</dd><dd><code>allowGaps</code> - whether gaps are allowed (if false they will be replaced with ambiguity symbols)</dd>
<dt><span class="strong">Returns:</span></dt><dd>the version of the sequence string with the invalid bases/residues replaced with ambiguity symbols</dd></dl>
</li>
</ul>
<a name="getValidSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getValidSequence</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;getValidSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                            boolean&nbsp;allowGaps,
                            boolean&nbsp;replaceWithGaps)</pre>
<div class="block">Replace any invalid bases/residues in the given sequence document with ambiguity symbols or gaps.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceDocument</code> - sequence document to replace the invalid bases in</dd><dd><code>allowGaps</code> - whether gaps are allowed (if false they will be replaced with ambiguity symbols)</dd><dd><code>replaceWithGaps</code> - whether invalid bases/residues should be replaced with gaps - should only be done if sequence is in an alignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>the version of the sequence string with the invalid bases/residues replaced with ambiguity symbols</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if allowGaps is false but replaceWithGaps is true</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.20 (Geneious 5.2.0)</dd></dl>
</li>
</ul>
<a name="asRna(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asRna</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;asRna(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence)</pre>
<div class="block">Views an underlying (nucleotide) CharSequence as RNA by dynamically
 translating 'T's to 'U's and 't's to 'u's

 It is guaranteed that if charSequence instanceof SequenceCharSequence, the returned
 value will also be instanceof SequenceCharSequence (but it may not support log-time modifications).</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideCharSequence</code> - A nucleotide sequence which may already have some RNA residues; it is
        not guaranteed that it is checked whether the sequence contains invalid residues. Must not be null.</dd>
<dt><span class="strong">Returns:</span></dt><dd>A CharSequence with the same sequence of characters as charSequence, except
         that 'T's are replaced with 'U's and 't's are replaced with 'u's</dd></dl>
</li>
</ul>
<a name="asDna(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asDna</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;asDna(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence)</pre>
<div class="block">Views an underlying (nucleotide) CharSequence as DNA by dynamically
 translating 'U's to 'T's and 'u's to 't's

 It is guaranteed that if charSequence instanceof SequenceCharSequence, the returned
 value will also be instanceof SequenceCharSequence (but it may not support log-time modifications).</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideCharSequence</code> - A nucleotide sequence which may already have some DNA residues; it is
        not guaranteed that it is checked whether the sequence contains invalid residues. Must not be null.</dd>
<dt><span class="strong">Returns:</span></dt><dd>A CharSequence with the same sequence of characters as charSequence, except
         that 'U's are replaced with 'T's and 'u's are replaced with 't's</dd></dl>
</li>
</ul>
<a name="asTranslation(java.lang.CharSequence, jebl.evolution.sequences.GeneticCode)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asTranslation</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;asTranslation(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence,
                                    <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a>&nbsp;geneticCode)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asTranslation(java.lang.CharSequence, jebl.evolution.sequences.GeneticCode, boolean)"><code>asTranslation(CharSequence, jebl.evolution.sequences.GeneticCode, boolean)</code></a> instead.</i></div>
</li>
</ul>
<a name="asTranslation(java.lang.CharSequence, jebl.evolution.sequences.GeneticCode, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asTranslation</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;asTranslation(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;nucleotideCharSequence,
                         <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a>&nbsp;geneticCode,
                         boolean&nbsp;translateFirstCodonUsingFirstCodonTable)</pre>
<div class="block">Views an underlying (nucleotide) CharSequence as its translation. If
 the CharSequence is not a multiple of 3, the extra 1 or 2 characters are ignored.
 The translated sequence will have length nucleotideCharSequence.length()/3.
 If the nucleotide sequence contains unknown nucleotide characters, these are treated as unknown states
 and the corresponding translated site will also be the unknown state (?) unless
 the nucleotide base would not affect the translation (e.g. the 3rd base in some triplets).

 The concrete type of the return value is not guaranteed. The specified
 charSequence must not change after it was passed to this method, but
 it is not guaranteed that violations of this contract will be detected.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideCharSequence</code> - A nucleotide sequence which may be dna, rna or a mixture. Must not be null and must be immutable. Must not contain gaps.</dd><dd><code>geneticCode</code> - the genetic code to use for the translation. Must not be null.</dd><dd><code>translateFirstCodonUsingFirstCodonTable</code> - each genetic code specifies a set of codons which get translated as M if they are the first codon even though they normally wouldn't translate as an M when occurring elsewhere a coding region. If this parameter is true the first codon will be translated using this alternative translation table for the genetic code.</dd>
<dt><span class="strong">Returns:</span></dt><dd>A CharSequence which is a translation of nucleotideCharSequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if nucleotideCharSequence contains gaps.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if nucleotideCharSequence or geneticCode is null.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.41 (Geneious 5.4.1)</dd></dl>
</li>
</ul>
<a name="reverseComplement(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplement</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;reverseComplement(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</pre>
<div class="block"><p>
 Provides a dynamic reverse complement view onto a nucletoide CharSequence.
 For performance, it is not guaranteed whether the charSequence will be checked
 for invalid residues. If an invalid nucleotide CharSequence is passed in,
 arbitrary nondeterministic behaviour may occur at any later time, such as
 e.g. unchecked exceptions thrown from <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#charAt(int)" title="class or interface in java.lang"><code>CharSequence.charAt(int)</code></a>.
 </p>

 <p>
 It is guaranteed that if charSequence instanceof SequenceCharSequence, the returned
 value will also be instanceof SequenceCharSequence (but it may not support log-time modifications).
 </p>

 <p>
 <b>Attention</b>: Unlike <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Utils.html?is-external=true#reverseComplement(java.lang.String)" title="class or interface in jebl.evolution.sequences"><code>Utils.reverseComplement(String)</code></a>,
 this method preserves case and doesn't remove gaps. To remove gaps and convert the sequence
 to upper case, use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#removeGaps(java.lang.CharSequence)"><code>removeGaps(CharSequence)</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/utilities/CharSequenceUtilities.html#asUpperCase(java.lang.CharSequence)"><code>CharSequenceUtilities.asUpperCase(CharSequence)</code></a>.
 </p></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>charSequence</code> - The charSequence for which to construct a reverse complement view.</dd>
<dt><span class="strong">Returns:</span></dt><dd>A reverse complement view onto charSequence, with case and gaps preserved.</dd></dl>
</li>
</ul>
<a name="isPredominantlyRna(java.lang.CharSequence, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isPredominantlyRna</h4>
<pre>public static&nbsp;boolean&nbsp;isPredominantlyRna(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence,
                         int&nbsp;maximumNonGapsToLookAt)</pre>
<div class="block">Checks whether a sequence is predominantly RNA (rather than DNA).
 Same as <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Utils.html?is-external=true#isPredominantlyRNA(java.lang.CharSequence, int)" title="class or interface in jebl.evolution.sequences"><code>Utils.isPredominantlyRNA(CharSequence, int)</code></a>, but more efficient on <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceCharSequence</code></a>s.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>charSequence</code> - A charSequence that may contain DNA or RNA characters.</dd><dd><code>maximumNonGapsToLookAt</code> - Maximum number of non-gap residues to look at before making a decision; Pass in Integer.MAX_VALUE to look at all residues</dd>
<dt><span class="strong">Returns:</span></dt><dd>true of the non-gap residues of charSequence are predominantly RNA</dd><dt><span class="strong">See Also:</span></dt><dd><code>#isRna(CharSequence</code></dd></dl>
</li>
</ul>
<a name="isRna(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isRna</h4>
<pre>public static&nbsp;boolean&nbsp;isRna(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</pre>
<div class="block">Checks whether a sequence is RNA (rather than DNA) based on whether the sequence contains either a T/t or a U/u first.
 If it contains neither T/t nor U/u, this method returns false.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>charSequence</code> - A charSequence that may contain DNA or RNA characters.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this charSequence is RNA (rather than DNA)</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#isPredominantlyRna(java.lang.CharSequence, int)"><code>isPredominantlyRna(CharSequence, int)</code></a></dd></dl>
</li>
</ul>
<a name="removeGaps(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeGaps</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;removeGaps(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</pre>
<div class="block">Constructs a sequence without gaps ('-') from a specified sequence that
 potentially has gaps. If the specified sequence does contain gaps, then a
 gapless copy is returned. Otherwise, the original charSequence is returned.

 It is guaranteed that

    CharSequenceUtilities.equals(removeGaps(cs), cs.toString().replace("-", ""))

 for any CharSequence cs that fulfills its contract.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>charSequence</code> - A nucleotide or amino acid sequence, potentially with gaps ('-')</dd>
<dt><span class="strong">Returns:</span></dt><dd>A CharSequence that contains the same sequence of characters but without the gaps ('-').
         Returns charSequence if charSequence doesn't contain any gaps.</dd></dl>
</li>
</ul>
<a name="getLeadingGapsLength(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLeadingGapsLength</h4>
<pre>public static&nbsp;int&nbsp;getLeadingGapsLength(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</pre>
<div class="block">Returns the start index of the non-gap regions in the specified charSequence,
 i.e. the length of the longset prefix of charSequence that contains only '-' characters.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>charSequence</code> - A CharSequence that may contain some leading gap characters '-'</dd>
<dt><span class="strong">Returns:</span></dt><dd>The length of the longest prefix of charSequence that contains only '-' characters.</dd></dl>
</li>
</ul>
<a name="getTrailingGapsLength(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrailingGapsLength</h4>
<pre>public static&nbsp;int&nbsp;getTrailingGapsLength(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</pre>
<div class="block">Get the number of trailing gap ('-') characters in the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>charSequence</code> - A CharSequence that may have trailing gap characters.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the number of trailing gap ('-') characters in the sequence or 0 if the sequence is entirely gaps.</dd></dl>
</li>
</ul>
<a name="getTrailingGapsStartIndex(java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrailingGapsStartIndex</h4>
<pre>public static&nbsp;int&nbsp;getTrailingGapsStartIndex(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</pre>
<div class="block">Returns the end index of the non-gap regions in the specified charSequence.
 This is identical to charSequence.length() minus the length of the longest
 suffix of charSequence that consists only of '-', <i>except</i> when
 charSequence consists only of '-', in which case this method returns
 charSequence.length() because there are no non-gap regions.
 In other words, in a sequence that consists only of gaps, all gaps are
 considered leading rather than trailing gaps, i.e. the non-gap region
 is considered to start just beyond the end of the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>charSequence</code> - A CharSequence that may contain some leading gap characters '-'</dd>
<dt><span class="strong">Returns:</span></dt><dd>1+the index of the last nongap character in charSequence, or charSequence.length() if charSequence consists only of gaps</dd></dl>
</li>
</ul>
<a name="getAlphabet(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlphabet</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;getAlphabet(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Get the Alphabet of a sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - a SequenceDocument to get the alphabet for.</dd>
<dt><span class="strong">Returns:</span></dt><dd>Alphabet of sequence</dd></dl>
</li>
</ul>
<a name="getSequenceType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceType</h4>
<pre>public static&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;getSequenceType(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Get the (jebl) sequence type.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - a SequenceDocument to get the sequence type of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>type of sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if sequence is not either a NucleotideSequenceDocument or a AminoAcidSequenceDocument.</dd></dl>
</li>
</ul>
<a name="getSequenceType(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceType</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&gt;&nbsp;getSequenceType(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</pre>
<div class="block">Examines a document and determines what the (jebl) sequence type (or types) of the document is (or are), and returns it (or them).<br/><br/>

 Always returns a List&lt;SequenceType&gt; of size 0, 1 or 2.
 <ul>
     <li>Size 0: when the given document was a type that could have either <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true#AMINO_ACID" title="class or interface in jebl.evolution.sequences"><code>SequenceType.AMINO_ACID</code></a> or <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true#NUCLEOTIDE" title="class or interface in jebl.evolution.sequences"><code>SequenceType.NUCLEOTIDE</code></a> or both, and that document
     has no sequences at all, for example an empty <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a></li>
     <li>Size 1: when the given document just contains a single sequence or is of a type where the SequenceType is always known, e,g a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> or a
     <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>.</li>
     <li>Size 2: when the given document has sequences of both types, e.g. a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a> with sequences of both types.</li>
 </ul></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - the document to determine the SequenceType of</dd>
<dt><span class="strong">Returns:</span></dt><dd>a list containing the sequence type or types of the given document.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if the given document type wasn't a valid type to determine the SequenceType of.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
<a name="toHTMLFragment(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toHTMLFragment</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toHTMLFragment(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;additionalContent)</pre>
<div class="block">Generate a HTML fragment that summarises a sequence, including the sequence string.
 If the sequence is longer than a certain threshold X, then only the first X residues are shown.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - a SequenceDocument</dd><dd><code>additionalContent</code> - additional content to include</dd>
<dt><span class="strong">Returns:</span></dt><dd>the html formatted summary</dd></dl>
</li>
</ul>
<a name="asJeblSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblSequence</h4>
<pre>public static&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&nbsp;asJeblSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Convert from a Geneious sequence to a jebl sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - a Geneious sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequence as a jebl sequence.</dd></dl>
</li>
</ul>
<a name="asJeblSequences(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblSequences</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;asJeblSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences)</pre>
<div class="block">Convert a set of Geneious sequences to jebl sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - Geneious sequences</dd>
<dt><span class="strong">Returns:</span></dt><dd>the Geneious sequences as jebl sequences</dd></dl>
</li>
</ul>
<a name="asJeblSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblSequences</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;asJeblSequences(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>...&nbsp;sequences)</pre>
<div class="block">Convert a set of Geneious sequences to jebl sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - Geneious sequences</dd>
<dt><span class="strong">Returns:</span></dt><dd>the Geneious sequences as jebl sequences</dd></dl>
</li>
</ul>
<a name="asJeblAlignment(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblAlignment</h4>
<pre>public static&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a>&nbsp;asJeblAlignment(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences)</pre>
<div class="block">Convert a list of (aligned) Geneious sequences to a jebl alignmnent</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - aligned Geneious sequences</dd>
<dt><span class="strong">Returns:</span></dt><dd>the Geneious sequences as a jebl alignmnent</dd></dl>
</li>
</ul>
<a name="createSequenceCopy(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createSequenceCopy</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;createSequenceCopy(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;original)</pre>
<div class="block">Creates a copy of the original sequence if necessary. If the sequence is an
 immutable sequence (<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a>)
 then it is not copied and is just returned from this method.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>original</code> - the original sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence document or the original sequence if the original sequence is immutable.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createSequenceCopyEditable(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>createSequenceCopyEditable(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)</code></a></dd></dl>
</li>
</ul>
<a name="createSequenceCopyEditable(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createSequenceCopyEditable</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;createSequenceCopyEditable(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;original)</pre>
<div class="block">Creates a copy of the original sequence that is editable.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>original</code> - the original sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence document</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createSequenceCopy(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>createSequenceCopy(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceAnnotationsIncludingImmutableSequencesTrims(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAnnotationsIncludingImmutableSequencesTrims</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getSequenceAnnotationsIncludingImmutableSequencesTrims(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Gets all the annotations on the given sequence. Additionally if it is an <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a> with <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getLeadingTrimLength()"><code>ImmutableSequence.getLeadingTrimLength()</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getTrailingTrimLength()"><code>ImmutableSequence.getTrailingTrimLength()</code></a>>0 then annotations are created to represent these trims.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to get annotations from</dd>
<dt><span class="strong">Returns:</span></dt><dd>the annotations from the sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.52 (Geneious 5.5.2)</dd></dl>
</li>
</ul>
<a name="asJeblSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblSequence</h4>
<pre>public static&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&nbsp;asJeblSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
                      <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)
                               throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Convert from a Geneious sequence to a jebl sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referencedSequence</code> - original referenced sequence to copy additional fields from.  May be null.</dd><dd><code>sequence</code> - a Geneious sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequence as a jebl sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - when the referenced sequence cannot be loaded</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="asJeblSequence(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblSequence</h4>
<pre>public static&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&nbsp;asJeblSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;referenceDocument,
                      <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#asJeblSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>asJeblSequence(SequenceAlignmentDocument.ReferencedSequence, SequenceDocument)</code></a></i></div>
<div class="block">Convert from a Geneious sequence to a jebl sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceDocument</code> - original AnnotatedPluginDocument to copy additional fields from.  May be null.</dd><dd><code>sequence</code> - a Geneious sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequence as a jebl sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.43 (Geneious 5.4.3)</dd></dl>
</li>
</ul>
<a name="replaceQuestionMarksWithMaximalAmbiguitySymbol(jebl.evolution.sequences.SequenceType, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>replaceQuestionMarksWithMaximalAmbiguitySymbol</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;replaceQuestionMarksWithMaximalAmbiguitySymbol(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence)</pre>
<div class="block">get a version of a sequence string with any question marks replaces with N (for nucleotide sequences) or X (for protein sequences)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceType</code> - sequence type of sequence</dd><dd><code>sequence</code> - sequence string</dd>
<dt><span class="strong">Returns:</span></dt><dd>version of sequence with any question marks replaces with N (for nucleotide sequences) or X (for protein sequences)</dd></dl>
</li>
</ul>
<a name="getMaximalAmbiguitySymbol(jebl.evolution.sequences.SequenceType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMaximalAmbiguitySymbol</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getMaximalAmbiguitySymbol(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType)</pre>
<div class="block">get the code for the state in this sequence type which represents a base/residue that is completely unknown</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceType</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getAnnotationsOfType(java.util.List, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationsOfType</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getAnnotationsOfType(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                                            <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<div class="block">Get all annotations in list matching the given type</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotations</code> - annotations</dd><dd><code>type</code> - type of annotations to get</dd>
<dt><span class="strong">Returns:</span></dt><dd>all annotations in document matching the given type</dd></dl>
</li>
</ul>
<a name="getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationsOfType</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getAnnotationsOfType(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;document,
                                            <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, boolean)"><code>getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, String, boolean)</code></a> instead.</i></div>
<div class="block">Get all annotations in document matching the given type</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - document to get annotations form</dd><dd><code>type</code> - type of annotations to get</dd>
<dt><span class="strong">Returns:</span></dt><dd>all annotations in document matching the given type</dd></dl>
</li>
</ul>
<a name="getAnnotationsOfType(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationsOfType</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getAnnotationsOfType(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;document,
                                            <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type,
                                            boolean&nbsp;returnAnnotationsInTracks)</pre>
<div class="block">Get all annotations in document matching the given type.<p/>
 <b>WARNING:</b> this list may not include all SequenceAnnotations represented as annotations in the sequence viewer.
 One such case is with trim annotations, which should be found using
 <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAnnotationsIncludingImmutableSequencesTrims(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>getSequenceAnnotationsIncludingImmutableSequencesTrims(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)</code></a>.<br/><br/>

 This may also return annotations that are not visible in the sequence viewer, such as <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION"><code>SequenceAnnotation.TYPE_EXTRACTED_REGION</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - document to get annotations form</dd><dd><code>type</code> - type of annotations to get</dd><dd><code>returnAnnotationsInTracks</code> - true iff we want annotations from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a> as well as those annotated directly on a document</dd>
<dt><span class="strong">Returns:</span></dt><dd>all annotations in document matching the given type</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAndTrackAnnotations</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getSequenceAndTrackAnnotations(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">A convenience method to get all annotations on the sequence and all annotations on all <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a> on this sequence.
 Most code should instead manually load tracks on demand since they may be too large to fit into memory. To get tracks on a sequence use
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html#getTrackManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceTrack.getTrackManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)</code></a> followed by
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html#getTracks()"><code>SequenceTrack.Manager.getTracks()</code></a>.
 <br><br>
 Iterating over the returned value may throw a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a> whose
 cause is an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a> if there is insufficient memory available to load the annotations.
 When running <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperations</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerators</code></a>
 core Geneious will automatic catch such exceptions and display a nice message to the user.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to get annotations for</dd>
<dt><span class="strong">Returns:</span></dt><dd>iterator containing the annotations. Will not return null.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="createSequenceCopyAdjustedForGapInsertion(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createSequenceCopyAdjustedForGapInsertion</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;createSequenceCopyAdjustedForGapInsertion(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceDocument,
                                                         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;gappedSequenceCharacters)</pre>
<div class="block">Creates a copy of the given sequence with annotations, sequence residues, and chromatogram values adjusted to account for gap insertion.
 Note, the returned copy does not create gapped versions of <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTracks</code></a>. Tracks are instead automatically propagated from referenced documents in alignments.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceDocument</code> - a sequence to insert gaps into. If the sequence alreayd contains gaps, the gaps are removed first</dd><dd><code>gappedSequenceCharacters</code> - the sequence characters to appear in the new gapped sequence. The positions of gaps in this character sequence determine how annotations and chromatograms are adjusted.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a copy of sequenceDocument adjusted for gap insertion.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)"><code>SequenceExtractionUtilities.removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)</code></a></dd></dl>
</li>
</ul>
<a name="getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequences</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequences(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;documents,
                                            <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                                            <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                                     throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">get all the sequences out of a set of AnnotatedPluginDocuments that may wrap SequenceDocuments, SequenceListDocuments or SequenceAlignmentDocuments.
 For large sequence lists (<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>) and genome sized sequences (those longer than <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD"><code>SequenceDocument.GENOME_SEQUENCE_THRESHOLD</code></a>) in other sequence lists, these are only loaded into memory on demand
 to ensure this method doesn't use excessive memory. If this method is potentially called on thousands of documents, then <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequencesWithoutImmediateLoading(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)"><code>getSequencesWithoutImmediateLoading</code></a>
 should be considered instead.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>documents</code> - documents to get the sequences out of</dd><dd><code>alphabet</code> - alphabet the sequences need to be to be included</dd><dd><code>progressListener</code> - for notifying the caller about progress of this method and for cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>all the sequences. Sequences are ordered by the AnnotatedPluginDocument they are in, and then by their index in that document.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if there is a problem getting the PluginDocument out of an AnnotatedPluginDocument or if the progress listener cancels the request.</dd></dl>
</li>
</ul>
<a name="getSequences(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequences</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents,
                                            <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                                            <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                                     throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">get all the sequences out of a set of AnnotatedPluginDocuments that may wrap SequenceDocuments, SequenceListDocuments or SequenceAlignmentDocuments.
 For large sequence lists (<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>) and genome sized sequences (those longer than <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD"><code>SequenceDocument.GENOME_SEQUENCE_THRESHOLD</code></a>) in other sequence lists, these are only loaded into memory on demand
 to ensure this method doesn't use excessive memory. If this method is potentially called on thousands of documents, then <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequencesWithoutImmediateLoading(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)"><code>getSequencesWithoutImmediateLoading</code></a>
 should be considered instead.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>documents</code> - documents to get the sequences out of</dd><dd><code>alphabet</code> - alphabet the sequences need to be to be included</dd><dd><code>progressListener</code> - for notifying the caller about progress of this method and for cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>all the sequences</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if there is a problem getting the PluginDocument out of an AnnotatedPluginDocument or if the progress listener cancels the request.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getSequencesWithoutImmediateLoading(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesWithoutImmediateLoading</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequencesWithoutImmediateLoading(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;documents,
                                                                         <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)
                                                                                  throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Like <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)"><code>getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)</code></a>
 but doesn't require each plugin document to be in memory as long as this Collection is around. The trade-off is that the sequences can only be accessed sequentially
 (hence the Collection return type of this method). Also the Collection does not support removal.<br/><br/>
 <p/>
 Since this Collection doesn't store the sequences immediately, DocumentOperationExceptions may be thrown down the line. Such a situation may be caught by surrounding the given iteration with
 <code>try {... } catch (<a href="../../../../../com/biomatters/geneious/publicapi/plugin/RuntimeDocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>RuntimeDocumentOperationException</code></a> e)</code> and then handling the exception from there.
 <p/>
 Note that using <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)"><code>getSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, jebl.util.ProgressListener)</code></a>
 is preferable to using getSequencesWithoutImmediateLoading when dealing with under a thousand documents.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>documents</code> - documents to get the sequences out of</dd><dd><code>alphabet</code> - alphabet the sequences need to be to be included</dd>
<dt><span class="strong">Returns:</span></dt><dd>all the sequences whose iterator may throw a RuntimeDocumentOperationException</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if one or more of the documents has more than Integer.MAX_VALUE sequences.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/RuntimeDocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>RuntimeDocumentOperationException</code></a></dd></dl>
</li>
</ul>
<a name="getOriginalIndex(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOriginalIndex</h4>
<pre>public static&nbsp;int&nbsp;getOriginalIndex(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                   int&nbsp;index)</pre>
<div class="block">Gets the original numbering of the given index if it is covered by a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION"><code>SequenceAnnotation.TYPE_EXTRACTED_REGION</code></a> annotation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence this index belongs to.</dd><dd><code>index</code> - the index to get the original numbering for.</dd>
<dt><span class="strong">Returns:</span></dt><dd>'translated' index or the original index if no other numbering can be found.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd></dl>
</li>
</ul>
<a name="getNumberOfSequences(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public static&nbsp;long&nbsp;getNumberOfSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</pre>
<div class="block">Gets the total number of nucleotide or amino acid sequences contained in the given documents which may be individual sequences, sequence lists, or alignments/contigs.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>documents</code> - the documents to get the number of sequences in</dd><dd><code>alphabet</code> - the alphabet (nucleotide or amino acid) of the sequences to count.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the total number of nucleotide sequences or amino acid contained in the given documents</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getNumberOfSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public static&nbsp;long&nbsp;getNumberOfSequences(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</pre>
<div class="block">Gets the total number of nucleotide or amino acid sequences contained in the given document which may be an individual sequence, sequence list, or alignment/contig.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - the document to get the number of sequences in</dd><dd><code>alphabet</code> - the alphabet (nucleotide or amino acid) of the sequences to count.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the total number of nucleotide or amino acid sequences contained in the given document</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="generateConsensusSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>generateConsensusSequence</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;generateConsensusSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
                                                    <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#generateConsensus(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)"><code>generateConsensus(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)</code></a> instead</i></div>
<div class="block">Generates a consensus sequence for an alignment using default consensus settings.
 Note that the returned sequence may contain gaps. If it is to be used as a stand-alone sequence,
 then <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceExtractionUtilities.removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)</code></a>
 should be used.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - the alignment to generate the consensus sequence for</dd><dd><code>progressListener</code> - for reporting progress can cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a sequence equal in length to the alignment. The sequence may contain gaps. May return null if progressListener requests this get cancelled.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd></dl>
</li>
</ul>
<a name="generateConsensus(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>generateConsensus</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;generateConsensus(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
                                 <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                          throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Generates a consensus sequence for an alignment using default consensus settings.
 Note that the returned sequence may contain gaps. If it is to be used as a stand-alone sequence,
 then <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceExtractionUtilities.removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)</code></a>
 should be used.
 <p/>
 To generate consensus sequences with non-default options, use PluginUtilities.getDocumentOperation("Generate_Consensus"). Note that this operation generates an sequence with gaps removed by default.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - the alignment to generate the consensus sequence for</dd><dd><code>progressListener</code> - for reporting progress can cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a sequence equal in length to the alignment. The sequence may contain gaps. Will not return null</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if the consensus can't be generated because there is insufficient free memory.</dd>
<dd><code>DocumentOperationException.Canceled</code> - if the progressListener requests the consensus generation be cancelled.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
<a name="getBlastAlignmentText(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBlastAlignmentText</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getBlastAlignmentText(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
                           boolean&nbsp;geneiousFriendly)</pre>
<div class="block">Formats the given alignment in BLAST text format</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - alignment to format</dd><dd><code>geneiousFriendly</code> - whether to format the alignment in an html-formatted "Geneious friendly" way
                         that is useful generally for alignments and not just for BLAST output</dd>
<dt><span class="strong">Returns:</span></dt><dd>alignment represented in BLAST text format</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="alignmentFromJeblSequences(java.lang.String, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>alignmentFromJeblSequences</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a>&nbsp;alignmentFromJeblSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                                                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;jeblSequences)</pre>
<div class="block">Converts the given alignment of Jebl sequences into a DefaultAlignmentDocument</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - name for alignment</dd><dd><code>jeblSequences</code> - aligned jebl sequences</dd>
<dt><span class="strong">Returns:</span></dt><dd>a DefaultAlignmentDocument representing the given alignment.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="createNewDocumentsByTransformingSequences(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener, java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>createNewDocumentsByTransformingSequences</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;createNewDocumentsByTransformingSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;sourceDocuments,
                                                                      <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a>&nbsp;transformer,
                                                                      <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener,
                                                                      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newSequenceOrDocumentNamePrefix)
                                                                               throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Transforms the sequence(s) in each input document and returns a new document corresponding to each input document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sourceDocuments</code> - the source documents containing sequences to transform. These may be SequenceDocuments or SequenceListDocuments or SequenceAlignmentDocuments</dd><dd><code>transformer</code> - the transformer for transforming each sequence</dd><dd><code>progressListener</code> - for reporting progress and canceling</dd><dd><code>newSequenceOrDocumentNamePrefix</code> - an optional prefix to assign to the name of each newly generated document. May be an empty String to leave names unchanged.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the new documents</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if documents can't be loaded, or if the input documents are not SequenceDocuments or SequenceListDocuments or SequenceAlignmentDocuments</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.701 (Geneious 7.0.1)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceUtilities.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/utilities/ProgressInputStream.ProgressInterruptedIOException.html" title="class in com.biomatters.geneious.publicapi.utilities"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/utilities/StandardIcons.html" title="enum in com.biomatters.geneious.publicapi.utilities"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html" target="_top">Frames</a></li>
<li><a href="SequenceUtilities.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
