<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:47 NZDT 2016 -->
<title>AssemblerInput.Properties (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AssemblerInput.Properties (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AssemblerInput.Properties.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.DataType.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html" target="_top">Frames</a></li>
<li><a href="AssemblerInput.Properties.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.plugin</div>
<h2 title="Class AssemblerInput.Properties" class="title">Class AssemblerInput.Properties</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.plugin.AssemblerInput.Properties</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Enclosing class:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput</a></dd>
</dl>
<hr>
<br>
<pre>public static final class <span class="strong">AssemblerInput.Properties</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Some properties that briefly summarise what the input will look like. This allows the assembler to provide options
 that take into account that data without needing to actually load all the data first.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#AssemblerInput.Properties(long)">AssemblerInput.Properties</a></strong>(long&nbsp;numberOfSequences)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#addChangeListener(org.virion.jam.util.SimpleListener)">addChangeListener</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/org/virion/jam/util/SimpleListener.html?is-external=true" title="class or interface in org.virion.jam.util">SimpleListener</a>&nbsp;listener)</code>
<div class="block">Adds a listener to be notified when the values of any properties change.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getMaximumReferenceSequenceLength()">getMaximumReferenceSequenceLength</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getMaximumSequenceLength()">getMaximumSequenceLength</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getMinimumSequenceLength()">getMinimumSequenceLength</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getNumberOfNucleotidesInAllSequences()">getNumberOfNucleotidesInAllSequences</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getNumberOfPairedSequences()">getNumberOfPairedSequences</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getNumberOfReferenceSequences()">getNumberOfReferenceSequences</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getNumberOfSequencesWithQuality()">getNumberOfSequencesWithQuality</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getReferenceSequences()">getReferenceSequences</a></strong>()</code>
<div class="block">Returns the reference sequences currently selected by the user in the options.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getReferenceSequencesLength()">getReferenceSequencesLength</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isAnyReferenceSequenceFromAnUndissolvedContig()">isAnyReferenceSequenceFromAnUndissolvedContig</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isSingleContigMappingToReferenceSequenceFromSeparateDocument()">isSingleContigMappingToReferenceSequenceFromSeparateDocument</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isSingleInputContig()">isSingleInputContig</a></strong>()</code>
<div class="block">returns true if there is just a single contig being provided as input to the assembler.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setAnyReferenceSequenceFromAnUndissolvedContig(boolean)">setAnyReferenceSequenceFromAnUndissolvedContig</a></strong>(boolean&nbsp;anyReferenceSequenceFromAnUndissolvedContig)</code>
<div class="block">Sets if there are any reference sequences which are from an undissolved contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setLengthOfReferenceSequences(long)">setLengthOfReferenceSequences</a></strong>(long&nbsp;lengthOfReferenceSequences)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setLengthOfReferenceSequences(long, long)"><code>setLengthOfReferenceSequences(long, long)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setLengthOfReferenceSequences(long, long)">setLengthOfReferenceSequences</a></strong>(long&nbsp;totalLengthOfReferenceSequences,
                             long&nbsp;maxReferenceSequenceLength)</code>
<div class="block">Sets the total length of all reference sequences or 0 if there are no reference sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setMaximumSequenceLength(long)">setMaximumSequenceLength</a></strong>(long&nbsp;maximumSequenceLength)</code>
<div class="block">Sets the maximum length of all sequences, excluding any reference sequences</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setMinimumSequenceLength(long)">setMinimumSequenceLength</a></strong>(long&nbsp;minimumSequenceLength)</code>
<div class="block">Sets the minimum length of all sequences, excluding any reference sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setNumberOfNucleotidesInAllSequences(long)">setNumberOfNucleotidesInAllSequences</a></strong>(long&nbsp;numberOfNucleotidesInAllSequences)</code>
<div class="block">Sets the number of bases in total in all the input sequences, excluding any reference sequences</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setNumberOfPairedSequences(long)">setNumberOfPairedSequences</a></strong>(long&nbsp;numberOfPairedSequences)</code>
<div class="block">Sets the number of sequences which are paired with another sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setNumberOfReferenceSequences(long)">setNumberOfReferenceSequences</a></strong>(long&nbsp;numberOfReferenceSequences)</code>
<div class="block">Sets the number of reference sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setNumberOfSequences(long)">setNumberOfSequences</a></strong>(long&nbsp;numberOfSequences)</code>
<div class="block">Sets numberOfSequences the number of sequences in the data, excluding any reference sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setNumberOfSequencesWithQuality(long)">setNumberOfSequencesWithQuality</a></strong>(long&nbsp;numberOfSequencesWithQuality)</code>
<div class="block">Sets the number of sequences (excluding any reference sequences) which have quality (i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setProperties(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Properties)">setProperties</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.Properties</a>&nbsp;propertiesToCopy)</code>
<div class="block">Assigns values to all properties by copying them from the given properties.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setReferenceSequences(java.util.List)">setReferenceSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;&nbsp;referenceSequences)</code>
<div class="block">Sets reference sequences currently selected by the user in the options.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setSingleContigMappingToReferenceSequenceFromSeparateDocument(boolean)">setSingleContigMappingToReferenceSequenceFromSeparateDocument</a></strong>(boolean&nbsp;singleContigMappingToReferenceSequenceFromSeparateDocument)</code>
<div class="block">Sets the result of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isSingleContigMappingToReferenceSequenceFromSeparateDocument()"><code>isSingleContigMappingToReferenceSequenceFromSeparateDocument()</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setSingleInputContig(boolean)">setSingleInputContig</a></strong>(boolean&nbsp;inputIsSingleContig)</code>
<div class="block">Sets if there is just a single contig being provided as input to the assembler.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="AssemblerInput.Properties(long)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AssemblerInput.Properties</h4>
<pre>public&nbsp;AssemblerInput.Properties(long&nbsp;numberOfSequences)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfSequences</code> - the number of sequences in the data, excluding any reference sequences</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfSequences()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences in the data, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="setNumberOfSequences(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNumberOfSequences</h4>
<pre>public&nbsp;void&nbsp;setNumberOfSequences(long&nbsp;numberOfSequences)</pre>
<div class="block">Sets numberOfSequences the number of sequences in the data, excluding any reference sequences</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfSequences</code> - the number of sequences in the data, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="getNumberOfNucleotidesInAllSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfNucleotidesInAllSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfNucleotidesInAllSequences()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of bases in total in all the input sequences, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="setNumberOfNucleotidesInAllSequences(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNumberOfNucleotidesInAllSequences</h4>
<pre>public&nbsp;void&nbsp;setNumberOfNucleotidesInAllSequences(long&nbsp;numberOfNucleotidesInAllSequences)</pre>
<div class="block">Sets the number of bases in total in all the input sequences, excluding any reference sequences</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfNucleotidesInAllSequences</code> - the number of bases in total in all the input sequences, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="getMinimumSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMinimumSequenceLength</h4>
<pre>public&nbsp;long&nbsp;getMinimumSequenceLength()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the minimum length of all sequences, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="setMinimumSequenceLength(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMinimumSequenceLength</h4>
<pre>public&nbsp;void&nbsp;setMinimumSequenceLength(long&nbsp;minimumSequenceLength)</pre>
<div class="block">Sets the minimum length of all sequences, excluding any reference sequences</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumSequenceLength</code> - the minimum length of all sequences, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="getMaximumSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMaximumSequenceLength</h4>
<pre>public&nbsp;long&nbsp;getMaximumSequenceLength()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum length of all sequences, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="setMaximumSequenceLength(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMaximumSequenceLength</h4>
<pre>public&nbsp;void&nbsp;setMaximumSequenceLength(long&nbsp;maximumSequenceLength)</pre>
<div class="block">Sets the maximum length of all sequences, excluding any reference sequences</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>maximumSequenceLength</code> - the maximum length of all sequences, excluding any reference sequences</dd></dl>
</li>
</ul>
<a name="getNumberOfPairedSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfPairedSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfPairedSequences()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences which are paired with another sequence. Since each sequence is counted, this returns twice the number of pairs.</dd></dl>
</li>
</ul>
<a name="setNumberOfPairedSequences(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNumberOfPairedSequences</h4>
<pre>public&nbsp;void&nbsp;setNumberOfPairedSequences(long&nbsp;numberOfPairedSequences)</pre>
<div class="block">Sets the number of sequences which are paired with another sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfPairedSequences</code> - the number of sequences which are paired with another sequence. Since each sequence is counted, this should be twice the number of pairs.</dd></dl>
</li>
</ul>
<a name="getNumberOfSequencesWithQuality()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequencesWithQuality</h4>
<pre>public&nbsp;long&nbsp;getNumberOfSequencesWithQuality()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences (excluding any reference sequences) which have quality (i.e. return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()"><code>NucleotideGraph.hasSequenceQualities()</code></a></dd></dl>
</li>
</ul>
<a name="setNumberOfSequencesWithQuality(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNumberOfSequencesWithQuality</h4>
<pre>public&nbsp;void&nbsp;setNumberOfSequencesWithQuality(long&nbsp;numberOfSequencesWithQuality)</pre>
<div class="block">Sets the number of sequences (excluding any reference sequences) which have quality (i.e. return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()"><code>NucleotideGraph.hasSequenceQualities()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfSequencesWithQuality</code> - the number of sequences (excluding any reference sequences) which have quality</dd></dl>
</li>
</ul>
<a name="getNumberOfReferenceSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfReferenceSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfReferenceSequences()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of reference sequences (0 for de novo assembly)</dd></dl>
</li>
</ul>
<a name="setNumberOfReferenceSequences(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNumberOfReferenceSequences</h4>
<pre>public&nbsp;void&nbsp;setNumberOfReferenceSequences(long&nbsp;numberOfReferenceSequences)</pre>
<div class="block">Sets the number of reference sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfReferenceSequences</code> - the number of reference sequences (0 for de novo assembly)</dd></dl>
</li>
</ul>
<a name="setReferenceSequences(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferenceSequences</h4>
<pre>public&nbsp;void&nbsp;setReferenceSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;&nbsp;referenceSequences)</pre>
<div class="block">Sets reference sequences currently selected by the user in the options.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequences</code> - the reference sequences currently selected by the user in the options. Must not be null but may be an empty list.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.51 (Geneious 5.5.1)</dd></dl>
</li>
</ul>
<a name="getReferenceSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;&nbsp;getReferenceSequences()</pre>
<div class="block">Returns the reference sequences currently selected by the user in the options. Will not be null but may be an empty list (for de novo assemblers). Be careful using this method. It may take time to load
 the actual sequences (via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html#getSequenceDocument(jebl.util.ProgressListener)"><code>AssemblerInput.ReferenceSequence.getSequenceDocument(jebl.util.ProgressListener)</code></a> so this should not be done from the swing thread.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the reference sequences currently selected by the user in the options.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.51 (Geneious 5.5.1)</dd></dl>
</li>
</ul>
<a name="getReferenceSequencesLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceSequencesLength</h4>
<pre>public&nbsp;long&nbsp;getReferenceSequencesLength()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the total length of all reference sequences or 0 if there are no reference sequences.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getMaximumReferenceSequenceLength()"><code>getMaximumReferenceSequenceLength()</code></a></dd></dl>
</li>
</ul>
<a name="setLengthOfReferenceSequences(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setLengthOfReferenceSequences</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;void&nbsp;setLengthOfReferenceSequences(long&nbsp;lengthOfReferenceSequences)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#setLengthOfReferenceSequences(long, long)"><code>setLengthOfReferenceSequences(long, long)</code></a> instead</i></div>
<div class="block">Sets the total length of all reference sequences or 0 if there are no reference sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>lengthOfReferenceSequences</code> - the total length of all reference sequences or 0 if there are no reference sequences.</dd></dl>
</li>
</ul>
<a name="setLengthOfReferenceSequences(long, long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setLengthOfReferenceSequences</h4>
<pre>public&nbsp;void&nbsp;setLengthOfReferenceSequences(long&nbsp;totalLengthOfReferenceSequences,
                                 long&nbsp;maxReferenceSequenceLength)</pre>
<div class="block">Sets the total length of all reference sequences or 0 if there are no reference sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>totalLengthOfReferenceSequences</code> - the total length of all reference sequences or 0 if there are no reference sequences.</dd><dd><code>maxReferenceSequenceLength</code> - the maximum length of any of the reference sequences or 0 if there are no reference sequences.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getMaximumReferenceSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMaximumReferenceSequenceLength</h4>
<pre>public&nbsp;long&nbsp;getMaximumReferenceSequenceLength()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum length of any of the reference sequences or 0 if there are no reference sequences.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#getReferenceSequencesLength()"><code>getReferenceSequencesLength()</code></a></dd></dl>
</li>
</ul>
<a name="setSingleInputContig(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSingleInputContig</h4>
<pre>public&nbsp;void&nbsp;setSingleInputContig(boolean&nbsp;inputIsSingleContig)</pre>
<div class="block">Sets if there is just a single contig being provided as input to the assembler. See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isSingleInputContig()"><code>isSingleInputContig()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>inputIsSingleContig</code> - true if there is just a single contig being provided as input to the assembler.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isSingleInputContig()"><code>()</code></a></dd></dl>
</li>
</ul>
<a name="isSingleInputContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSingleInputContig</h4>
<pre>public&nbsp;boolean&nbsp;isSingleInputContig()</pre>
<div class="block">returns true if there is just a single contig being provided as input to the assembler. Assemblers which return false from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#supportsSingleInputContig()"><code>Assembler.supportsSingleInputContig()</code></a> need not handle this case.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if there is just a single contig being provided as input to the assembler.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd></dl>
</li>
</ul>
<a name="setAnyReferenceSequenceFromAnUndissolvedContig(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnyReferenceSequenceFromAnUndissolvedContig</h4>
<pre>public&nbsp;void&nbsp;setAnyReferenceSequenceFromAnUndissolvedContig(boolean&nbsp;anyReferenceSequenceFromAnUndissolvedContig)</pre>
<div class="block">Sets if there are any reference sequences which are from an undissolved contig.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>anyReferenceSequenceFromAnUndissolvedContig</code> - true if there are any reference sequences which are from an undissolved contig. In other words at some point one of the assembly reads will be a contig which returns a non-negative value from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getIndexOfAssemblyReferenceSequenceForContig()"><code>AssemblerInput.Read.getIndexOfAssemblyReferenceSequenceForContig()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isAnyReferenceSequenceFromAnUndissolvedContig()"><code>isAnyReferenceSequenceFromAnUndissolvedContig()</code></a></dd></dl>
</li>
</ul>
<a name="isAnyReferenceSequenceFromAnUndissolvedContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isAnyReferenceSequenceFromAnUndissolvedContig</h4>
<pre>public&nbsp;boolean&nbsp;isAnyReferenceSequenceFromAnUndissolvedContig()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if there are any reference sequences which are from an undissolved contig. In other words at some point one of the assembly reads will be a contig which returns a non-negative value from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getIndexOfAssemblyReferenceSequenceForContig()"><code>AssemblerInput.Read.getIndexOfAssemblyReferenceSequenceForContig()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="addChangeListener(org.virion.jam.util.SimpleListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addChangeListener</h4>
<pre>public&nbsp;void&nbsp;addChangeListener(<a href="http://jebl.sourceforge.net/doc/api/org/virion/jam/util/SimpleListener.html?is-external=true" title="class or interface in org.virion.jam.util">SimpleListener</a>&nbsp;listener)</pre>
<div class="block">Adds a listener to be notified when the values of any properties change.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>listener</code> - the listener to be notified.</dd></dl>
</li>
</ul>
<a name="setProperties(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Properties)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setProperties</h4>
<pre>public&nbsp;void&nbsp;setProperties(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.Properties</a>&nbsp;propertiesToCopy)</pre>
<div class="block">Assigns values to all properties by copying them from the given properties. All change listeners (added using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#addChangeListener(org.virion.jam.util.SimpleListener)"><code>addChangeListener(org.virion.jam.util.SimpleListener)</code></a>
 are fired by this method.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>propertiesToCopy</code> - the properties to copy values from.</dd></dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="setSingleContigMappingToReferenceSequenceFromSeparateDocument(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSingleContigMappingToReferenceSequenceFromSeparateDocument</h4>
<pre>public&nbsp;void&nbsp;setSingleContigMappingToReferenceSequenceFromSeparateDocument(boolean&nbsp;singleContigMappingToReferenceSequenceFromSeparateDocument)</pre>
<div class="block">Sets the result of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html#isSingleContigMappingToReferenceSequenceFromSeparateDocument()"><code>isSingleContigMappingToReferenceSequenceFromSeparateDocument()</code></a></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.710 (Geneious 7.1.0)</dd></dl>
</li>
</ul>
<a name="isSingleContigMappingToReferenceSequenceFromSeparateDocument()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>isSingleContigMappingToReferenceSequenceFromSeparateDocument</h4>
<pre>public&nbsp;boolean&nbsp;isSingleContigMappingToReferenceSequenceFromSeparateDocument()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if the user is not disolving input contigs and they are mapping just a single contig to the reference sequence(s).</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.710 (Geneious 7.1.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AssemblerInput.Properties.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.DataType.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html" target="_top">Frames</a></li>
<li><a href="AssemblerInput.Properties.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
