<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:47 NZDT 2016 -->
<title>AssemblerInput (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AssemblerInput (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AssemblerInput.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.ReferenceSequenceSupport.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.DataType.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/AssemblerInput.html" target="_top">Frames</a></li>
<li><a href="AssemblerInput.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.plugin</div>
<h2 title="Class AssemblerInput" class="title">Class AssemblerInput</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.plugin.AssemblerInput</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">AssemblerInput</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Represents the input to an assembly algorithm. Geneious will instantiate an instance of this class
 and pass it to an assembler via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#assemble(com.biomatters.geneious.publicapi.plugin.Options, com.biomatters.geneious.publicapi.plugin.AssemblerInput, jebl.util.ProgressListener, com.biomatters.geneious.publicapi.plugin.Assembler.Callback)"><code>Assembler.assemble(Options, AssemblerInput, jebl.util.ProgressListener, com.biomatters.geneious.publicapi.plugin.Assembler.Callback)</code></a>.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1.0)</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.DataType.html" title="enum in com.biomatters.geneious.publicapi.plugin">AssemblerInput.DataType</a></strong></code>
<div class="block">The sequencing technology that a sequencing read was generated from.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.Properties</a></strong></code>
<div class="block">Some properties that briefly summarise what the input will look like.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.Read</a></strong></code>
<div class="block">Represents a single read or input contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Reads.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.Reads</a></strong></code>
<div class="block">Provides the ability to iterate over all reads to be assembled.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a></strong></code>
<div class="block">Represents a reference sequence in an assembly.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#AssemblerInput(java.util.List, java.util.List, boolean)">AssemblerInput</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;annotatedPluginDocuments,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;&nbsp;referenceSequences,
              boolean&nbsp;extractSequencesFromContigs)</code>
<div class="block">Constructs some assembly input from documents that need not be loaded into memory.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#AssemblerInput(java.util.List, java.util.List, boolean, double)">AssemblerInput</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;annotatedPluginDocuments,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;&nbsp;referenceSequences,
              boolean&nbsp;extractSequencesFromContigs,
              double&nbsp;fractionOfDataToUse)</code>
<div class="block">Constructs some assembly input from documents that need not be loaded into memory.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#AssemblerInput(java.util.List, java.util.List, java.util.List)">AssemblerInput</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;referenceSequences,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;sequences,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&gt;&nbsp;referencedDocuments)</code>
<div class="block">Constructs some assembly input from sequences that are all in memory.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#AssemblerInput(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceListDocument)">AssemblerInput</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;referenceSequences,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListDocument</a>&nbsp;sequenceListDocument)</code>
<div class="block">Constructs some assembly input from sequences that are all in memory.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getMaximumNumberOfContigsToGenerate()">getMaximumNumberOfContigsToGenerate</a></strong>()</code>
<div class="block">For performance reasons, the user may limit the algorithm to generating this many contigs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getNumberOfNucleotideSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">getNumberOfNucleotideSequences</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</code>
<div class="block">Gets the total number of nucleotide sequences contained in the given document which may be an individual sequence, sequence list, or alignment/contig.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getNumberOfNucleotideSequences(java.util.List)">getNumberOfNucleotideSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents)</code>
<div class="block">Gets the total number of nucleotide sequences contained in the given documents which may be individual sequences, sequence lists, or alignments/contigs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getNumberOfReadSequences()">getNumberOfReadSequences</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getNumberOfReferenceSequences()">getNumberOfReferenceSequences</a></strong>()</code>
<div class="block">Get the total number of reference sequences being assembled to.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getNumberOfSequencesToUseFromDocument(long, double)">getNumberOfSequencesToUseFromDocument</a></strong>(long&nbsp;numberOfSequences,
                                     double&nbsp;fractionOfDataToUse)</code>
<div class="block">When assembling using only a fraction of the data from each document, this is used to return the exact number of sequences used from each document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Reads.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.Reads</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReads()">getReads</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReferenceSequence(int, jebl.util.ProgressListener)">getReferenceSequence</a></strong>(int&nbsp;referenceSequenceIndex,
                    <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Gets one of the reference sequences for this assembly.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReferenceSequenceReference(int)">getReferenceSequenceReference</a></strong>(int&nbsp;referenceSequenceIndex)</code>
<div class="block">Gets a reference to the document that the reference sequence came from.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getSampleName()">getSampleName</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#isGenerateContigs()">isGenerateContigs</a></strong>()</code>
<div class="block">Returns whether or not contigs should be produced by the assembly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#isSaveUnusedReads()">isSaveUnusedReads</a></strong>()</code>
<div class="block">Returns whether or not unused reads should be saved by the assembly.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#isSaveUsedReads()">isSaveUsedReads</a></strong>()</code>
<div class="block">Returns whether or not used reads should be saved by the assembly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#isSaveUsedReadsIncludeMates()">isSaveUsedReadsIncludeMates</a></strong>()</code>
<div class="block">Returns whether or not the mates of used reads (even if that mate doesn't map) should be saved by the assembly.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#isSetReferencedDocumentsOnContig()">isSetReferencedDocumentsOnContig</a></strong>()</code>
<div class="block">Returns true if the assembly implementation should set referenced documents/sequences (see <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)"><code>SequenceAlignmentDocument.getReferencedSequence(int)</code></a>)
 for the reads on generated contigs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#setAllowDuplicateReferenceSequenceNames(boolean)">setAllowDuplicateReferenceSequenceNames</a></strong>(boolean&nbsp;allowDuplicateReferenceSequenceNames)</code>
<div class="block">Sets whether <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReferenceSequence(int, jebl.util.ProgressListener)"><code>getReferenceSequence(int, jebl.util.ProgressListener)</code></a> throws a <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperationException</code></a> if there are duplicate reference sequence names</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#setGenerateContigs(boolean)">setGenerateContigs</a></strong>(boolean&nbsp;saveContigs)</code>
<div class="block">Sets whether or not contigs should be saved by the assembly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#setMaximumNumberOfContigsToGenerate(int)">setMaximumNumberOfContigsToGenerate</a></strong>(int&nbsp;maximumContigs)</code>
<div class="block">For performance reasons, the user may limit the algorithm to generating this many contigs.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#setSampleName(java.lang.String)">setSampleName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sampleName)</code>
<div class="block">Sets a name describing the sequencing reads used as input</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#setSaveUnusedReads(boolean)">setSaveUnusedReads</a></strong>(boolean&nbsp;saveUnusedReads)</code>
<div class="block">Sets whether or not unused reads should be saved by the assembly.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#setSaveUsedReads(boolean)">setSaveUsedReads</a></strong>(boolean&nbsp;saveUsedReads)</code>
<div class="block">Sets whether or not used reads should be saved by the assembly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#setSaveUsedReadsIncludeMates(boolean)">setSaveUsedReadsIncludeMates</a></strong>(boolean&nbsp;saveUsedReadsIncludeMates)</code>
<div class="block">Sets whether or not the mates of used reads (even if that mate doesn't map) should be saved by the assembly.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="AssemblerInput(java.util.List, java.util.List, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AssemblerInput</h4>
<pre>public&nbsp;AssemblerInput(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;annotatedPluginDocuments,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;&nbsp;referenceSequences,
              boolean&nbsp;extractSequencesFromContigs)</pre>
<div class="block">Constructs some assembly input from documents that need not be loaded into memory.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotatedPluginDocuments</code> - the documents containing the reference sequences and reads to assembly</dd><dd><code>referenceSequences</code> - the reference sequences. This may be empty but must not be null.</dd><dd><code>extractSequencesFromContigs</code> - true to extract sequences from any contigs and provide them as stand-alone sequences rather than as part of a contig</dd></dl>
</li>
</ul>
<a name="AssemblerInput(java.util.List, java.util.List, boolean, double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AssemblerInput</h4>
<pre>public&nbsp;AssemblerInput(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;annotatedPluginDocuments,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.ReferenceSequence.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.ReferenceSequence</a>&gt;&nbsp;referenceSequences,
              boolean&nbsp;extractSequencesFromContigs,
              double&nbsp;fractionOfDataToUse)</pre>
<div class="block">Constructs some assembly input from documents that need not be loaded into memory.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotatedPluginDocuments</code> - the documents containing the reference sequences and reads to assembly</dd><dd><code>referenceSequences</code> - the reference sequences. This may be empty but must not be null.</dd><dd><code>extractSequencesFromContigs</code> - true to extract sequences from any contigs and provide them as stand-alone sequences rather than as part of a contig</dd><dd><code>fractionOfDataToUse</code> - the fraction of the input data to provide to the assembler. Must be between 0 and 1. If any reference sequences are specified, this must be 1.0</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="AssemblerInput(java.util.List, java.util.List, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AssemblerInput</h4>
<pre>public&nbsp;AssemblerInput(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;referenceSequences,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;sequences,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&gt;&nbsp;referencedDocuments)</pre>
<div class="block">Constructs some assembly input from sequences that are all in memory. Generally this constructor should only be used for test cases.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequences</code> - the reference sequences. This may be empty but must not be null.</dd><dd><code>sequences</code> - the sequences to assemble. This may (but does not need to) contain the reference sequences too.</dd><dd><code>referencedDocuments</code> - an optional parallel list (equal in length to sequences) contaiing the referenced documents for each sequence. This may be null.</dd></dl>
</li>
</ul>
<a name="AssemblerInput(java.util.List, com.biomatters.geneious.publicapi.documents.sequence.SequenceListDocument)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AssemblerInput</h4>
<pre>public&nbsp;AssemblerInput(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;referenceSequences,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListDocument</a>&nbsp;sequenceListDocument)</pre>
<div class="block">Constructs some assembly input from sequences that are all in memory. Generally this constructor should only be used for test cases.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequences</code> - the reference sequences. This may be empty but must not be null.</dd><dd><code>sequenceListDocument</code> - a sequence list containg the sequences to assemble. This must not include any reference sequences.</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="setAllowDuplicateReferenceSequenceNames(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAllowDuplicateReferenceSequenceNames</h4>
<pre>public&nbsp;void&nbsp;setAllowDuplicateReferenceSequenceNames(boolean&nbsp;allowDuplicateReferenceSequenceNames)</pre>
<div class="block">Sets whether <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReferenceSequence(int, jebl.util.ProgressListener)"><code>getReferenceSequence(int, jebl.util.ProgressListener)</code></a> throws a <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperationException</code></a> if there are duplicate reference sequence names</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>allowDuplicateReferenceSequenceNames</code> - false if <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReferenceSequence(int, jebl.util.ProgressListener)"><code>getReferenceSequence(int, jebl.util.ProgressListener)</code></a> throws a <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperationException</code></a> if there are duplicate reference sequence names</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.710 (Geneious 7.1.0)</dd></dl>
</li>
</ul>
<a name="setSampleName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSampleName</h4>
<pre>public&nbsp;void&nbsp;setSampleName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sampleName)</pre>
<div class="block">Sets a name describing the sequencing reads used as input</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sampleName</code> - the sample name, or empty String for no sample name</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd></dl>
</li>
</ul>
<a name="getSampleName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSampleName</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSampleName()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>a name describing the sequencing reads used as input or empty String for no name. Will not be null.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd></dl>
</li>
</ul>
<a name="getNumberOfNucleotideSequences(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfNucleotideSequences</h4>
<pre>public static&nbsp;long&nbsp;getNumberOfNucleotideSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents)</pre>
<div class="block">Gets the total number of nucleotide sequences contained in the given documents which may be individual sequences, sequence lists, or alignments/contigs.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>documents</code> - the documents to get the number of sequences in</dd>
<dt><span class="strong">Returns:</span></dt><dd>the total number of nucleotide sequences contained in the given documents</dd></dl>
</li>
</ul>
<a name="getNumberOfNucleotideSequences(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfNucleotideSequences</h4>
<pre>public static&nbsp;long&nbsp;getNumberOfNucleotideSequences(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</pre>
<div class="block">Gets the total number of nucleotide sequences contained in the given document which may be an individual sequence, sequence list, or alignment/contig.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - the document to get the number of sequences in</dd>
<dt><span class="strong">Returns:</span></dt><dd>the total number of nucleotide sequences contained in the given document</dd></dl>
</li>
</ul>
<a name="getNumberOfSequencesToUseFromDocument(long, double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequencesToUseFromDocument</h4>
<pre>public static&nbsp;long&nbsp;getNumberOfSequencesToUseFromDocument(long&nbsp;numberOfSequences,
                                         double&nbsp;fractionOfDataToUse)</pre>
<div class="block">When assembling using only a fraction of the data from each document, this is used to return the exact number of sequences used from each document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfSequences</code> - the total number of sequences in a single document.</dd><dd><code>fractionOfDataToUse</code> - the fraction of sequences to use from each document. Must be between 0 and 1. At least 1 sequence will be used from each document.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the number of sequences to use between 1 and numberOfSequences</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getReads()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReads</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Reads.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput.Reads</a>&nbsp;getReads()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>returns an object that allows you to iterate once over the reads, loading them into memory as required.</dd></dl>
</li>
</ul>
<a name="getNumberOfReadSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfReadSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfReadSequences()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the total number of reads, where a paired read counts as 2 reads for the purposes of this method.</dd></dl>
</li>
</ul>
<a name="getNumberOfReferenceSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfReferenceSequences</h4>
<pre>public&nbsp;int&nbsp;getNumberOfReferenceSequences()</pre>
<div class="block">Get the total number of reference sequences being assembled to.<p>
 This will always be 0 if the algorithm returned <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.ReferenceSequenceSupport.html#NoReferenceSequence"><code>Assembler.ReferenceSequenceSupport.NoReferenceSequence</code></a> from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#getReferenceSequenceSupport()"><code>Assembler.getReferenceSequenceSupport()</code></a>.<p>
 It will always be 1 if the algorithm returned <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.ReferenceSequenceSupport.html#SingleReferenceSequence"><code>Assembler.ReferenceSequenceSupport.SingleReferenceSequence</code></a> from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#getReferenceSequenceSupport()"><code>Assembler.getReferenceSequenceSupport()</code></a>.<p>
 It will always be greater than or equal to 1 if the algorithm returned <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.ReferenceSequenceSupport.html#MultipleReferenceSequences"><code>Assembler.ReferenceSequenceSupport.MultipleReferenceSequences</code></a> from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#getReferenceSequenceSupport()"><code>Assembler.getReferenceSequenceSupport()</code></a>.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the total number of reference sequences being assembled to.</dd></dl>
</li>
</ul>
<a name="getReferenceSequence(int, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceSequence</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;getReferenceSequence(int&nbsp;referenceSequenceIndex,
                                              <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                                throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Gets one of the reference sequences for this assembly. De novo assemblers must not call this method.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequenceIndex</code> - the index of the reference sequence to get between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getNumberOfReferenceSequences()"><code>getNumberOfReferenceSequences()</code></a>-1 inclusive.</dd><dd><code>progressListener</code> - a progress listener to report progress to while the reference sequence is loaded from disk.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the reference sequence. Will not be null</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if the reference sequence cannot be loaded</dd></dl>
</li>
</ul>
<a name="getReferenceSequenceReference(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceSequenceReference</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getReferenceSequenceReference(int&nbsp;referenceSequenceIndex)
                                                                           throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Gets a reference to the document that the reference sequence came from.
 The referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in
 the alignment/contig viewer.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequenceIndex</code> - the index of the reference sequence to get between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getNumberOfReferenceSequences()"><code>getNumberOfReferenceSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a reference to the document that the reference sequence came from or null if there is no reference sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if the referenced document cannot be obtained because of a failure to load the reference sequence.</dd></dl>
</li>
</ul>
<a name="isSetReferencedDocumentsOnContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSetReferencedDocumentsOnContig</h4>
<pre>public&nbsp;boolean&nbsp;isSetReferencedDocumentsOnContig()</pre>
<div class="block">Returns true if the assembly implementation should set referenced documents/sequences (see <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)"><code>SequenceAlignmentDocument.getReferencedSequence(int)</code></a>)
 for the reads on generated contigs. Note that a referened sequence is not the same thing as a reference sequence (see <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>SequenceAlignmentDocument.getContigReferenceSequenceIndex()</code></a>))
 A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.
 <p/>
 No matter what this method returns, assembly implementations should still set the a referenced document for the reference sequence.
 <p/>
 For performance reasons this method will return false on big data sets (those with more than 10,000 sequences by default)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if the assembly implementation should set referenced documents (which are not the same thing as a reference sequence) on generated contigs.</dd></dl>
</li>
</ul>
<a name="isSaveUnusedReads()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSaveUnusedReads</h4>
<pre>public&nbsp;boolean&nbsp;isSaveUnusedReads()</pre>
<div class="block">Returns whether or not unused reads should be saved by the assembly. Geneious will only set this
 to true if <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#providesUnusedReads()"><code>Assembler.providesUnusedReads()</code></a> returned true and the user checked the relevant checkbox.
 An assembler doesn't need to respect this setting. It can always provide unused reads using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.Callback.html#addUnusedReads(com.biomatters.geneious.publicapi.documents.sequence.SequenceListDocument, jebl.util.ProgressListener)"><code>Assembler.Callback.addUnusedReads(com.biomatters.geneious.publicapi.documents.sequence.SequenceListDocument, jebl.util.ProgressListener)</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.Callback.html#addUnusedRead(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Read, jebl.util.ProgressListener)"><code>Assembler.Callback.addUnusedRead(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Read, jebl.util.ProgressListener)</code></a> and the callback
 will just ignore it if it doesn't want unused reads.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>whether or not unused reads should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#providesUnusedReads()"><code>Assembler.providesUnusedReads()</code></a></dd></dl>
</li>
</ul>
<a name="setSaveUnusedReads(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSaveUnusedReads</h4>
<pre>public&nbsp;void&nbsp;setSaveUnusedReads(boolean&nbsp;saveUnusedReads)</pre>
<div class="block">Sets whether or not unused reads should be saved by the assembly.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>saveUnusedReads</code> - true if unused reads should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="isSaveUsedReads()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSaveUsedReads</h4>
<pre>public&nbsp;boolean&nbsp;isSaveUsedReads()</pre>
<div class="block">Returns whether or not used reads should be saved by the assembly. Geneious will only set this
 to true if <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#providesUsedReads()"><code>Assembler.providesUsedReads()</code></a> returned true and the user checked the relevant checkbox.
 An assembler doesn't need to respect this setting. It can always provide used reads using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.Callback.html#addUsedRead(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Read, jebl.util.ProgressListener)"><code>Assembler.Callback.addUsedRead(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Read, jebl.util.ProgressListener)</code></a> and the callback
 will just ignore it if it doesn't want used reads.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>whether or not used reads should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#providesUsedReads()"><code>Assembler.providesUsedReads()</code></a></dd></dl>
</li>
</ul>
<a name="setSaveUsedReads(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSaveUsedReads</h4>
<pre>public&nbsp;void&nbsp;setSaveUsedReads(boolean&nbsp;saveUsedReads)</pre>
<div class="block">Sets whether or not used reads should be saved by the assembly.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>saveUsedReads</code> - true if used reads should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="isSaveUsedReadsIncludeMates()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSaveUsedReadsIncludeMates</h4>
<pre>public&nbsp;boolean&nbsp;isSaveUsedReadsIncludeMates()</pre>
<div class="block">Returns whether or not the mates of used reads (even if that mate doesn't map) should be saved by the assembly. Geneious will only set this
 to true if <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#providesUsedReadsIncludeMates()"><code>Assembler.providesUsedReadsIncludeMates()</code></a> returned true and the user checked the relevant checkbox.
 An assembler doesn't need to respect this setting. It can always provide used reads using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.Callback.html#addUsedRead(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Read, jebl.util.ProgressListener)"><code>Assembler.Callback.addUsedRead(com.biomatters.geneious.publicapi.plugin.AssemblerInput.Read, jebl.util.ProgressListener)</code></a> and the callback
 will just ignore it if it doesn't want used reads.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>whether or not used reads should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#providesUsedReads()"><code>Assembler.providesUsedReads()</code></a></dd></dl>
</li>
</ul>
<a name="setGenerateContigs(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenerateContigs</h4>
<pre>public&nbsp;void&nbsp;setGenerateContigs(boolean&nbsp;saveContigs)</pre>
<div class="block">Sets whether or not contigs should be saved by the assembly.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>saveContigs</code> - true if contigs should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="isGenerateContigs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isGenerateContigs</h4>
<pre>public&nbsp;boolean&nbsp;isGenerateContigs()</pre>
<div class="block">Returns whether or not contigs should be produced by the assembly. Geneious will only set this
 to true if <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#getContigOutputSupport()"><code>Assembler.getContigOutputSupport()</code></a> did not return ConsensusOnly and the user checked the relevant checkbox (or they choose to save a generated consensus)
 An assembler doesn't need to respect this setting. It can always provide contigs using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.Callback.html#addContigDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, boolean, jebl.util.ProgressListener)"><code>Assembler.Callback.addContigDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, boolean, jebl.util.ProgressListener)</code></a> and the callback
 will just ignore it if it doesn't want contigs.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>whether or not contigs should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#getContigOutputSupport()"><code>)</code></a></dd></dl>
</li>
</ul>
<a name="setSaveUsedReadsIncludeMates(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSaveUsedReadsIncludeMates</h4>
<pre>public&nbsp;void&nbsp;setSaveUsedReadsIncludeMates(boolean&nbsp;saveUsedReadsIncludeMates)</pre>
<div class="block">Sets whether or not the mates of used reads (even if that mate doesn't map) should be saved by the assembly.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>saveUsedReadsIncludeMates</code> - true if used reads and their mates should be saved by the assembly.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="setMaximumNumberOfContigsToGenerate(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMaximumNumberOfContigsToGenerate</h4>
<pre>public&nbsp;void&nbsp;setMaximumNumberOfContigsToGenerate(int&nbsp;maximumContigs)</pre>
<div class="block">For performance reasons, the user may limit the algorithm to generating this many contigs. The implementation is recommended to only return
 the biggest contigs in this case.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>maximumContigs</code> - the maximum number of contigs the algorithm should return.</dd></dl>
</li>
</ul>
<a name="getMaximumNumberOfContigsToGenerate()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getMaximumNumberOfContigsToGenerate</h4>
<pre>public&nbsp;int&nbsp;getMaximumNumberOfContigsToGenerate()</pre>
<div class="block">For performance reasons, the user may limit the algorithm to generating this many contigs. The implementation is recommended to only return
 the biggest contigs in this case.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum number of contigs the algorithm should return.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AssemblerInput.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.ReferenceSequenceSupport.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.DataType.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/AssemblerInput.html" target="_top">Frames</a></li>
<li><a href="AssemblerInput.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
