<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:47 NZDT 2016 -->
<title>AssemblerInput.Read (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AssemblerInput.Read (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AssemblerInput.Read.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Reads.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" target="_top">Frames</a></li>
<li><a href="AssemblerInput.Read.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.plugin</div>
<h2 title="Class AssemblerInput.Read" class="title">Class AssemblerInput.Read</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.plugin.AssemblerInput.Read</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Enclosing class:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html" title="class in com.biomatters.geneious.publicapi.plugin">AssemblerInput</a></dd>
</dl>
<hr>
<br>
<pre>public static class <span class="strong">AssemblerInput.Read</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Represents a single read or input contig. A read may either be 1 unpaired sequence, or 2 paired sequences.
 These are supplied to the assembly algorithm via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReads()"><code>getReads()</code></a>.<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Reads.html#getNextReadPair()"><code>getNextReadPair()</code></a>
 <p/>
 Since different sequencing technologies orientate paired data in different relative orientations, paired data can be obtained in 2 different ways:
 <ol>
 <li>Normalized such that the 2nd read is expected to appear in the same orientation as the first, but to the right. In other words, Forward/Forward orientation (e.g. SOLiD data style)<br>
 The reads and expected distance in this case should be obtained using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalized()"><code>getMateNormalized()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceNormalized()"><code>getExpectedMateDistanceNormalized()</code></a><br>
 Alternatively use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalizedReversed()"><code>getMateNormalizedReversed()</code></a> to get the reads as forward/reverse (e.g. Illumina data style)
 </li>
 <li>
 Unnormalized using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()"><code>getMate()</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceForRead()"><code>getExpectedMateDistanceForRead()</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceForMate()"><code>getExpectedMateDistanceForMate()</code></a>. For a full discussion of
 what data means in this case, see <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a>
 </li>
 </ol>
 It is recommended that most implementations use the normalized system.
 See <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a> for a full discussion of what unnormalized expected distances mean.
 <p/>
 If the assembly algorithm specified that it supports input contigs (see <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#supportsInputContigs()"><code>Assembler.supportsInputContigs()</code></a>
 then instead of a read representing a sequence or paired of sequences, it may be a contig, in which case it will
 return null from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a> and instead return a non-null value from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getContig()"><code>getContig()</code></a>.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int)">AssemblerInput.Read</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;referencedAnnotatedDocument,
                   int&nbsp;indexInReferencedDocument)</code>
<div class="block">Constructs a new read consisting of a single unpaired sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, int, int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int)">AssemblerInput.Read</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;mate,
                   int&nbsp;expectedMateDistanceForRead,
                   int&nbsp;expectedMateDistanceForMate,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;referencedAnnotatedDocument,
                   int&nbsp;indexOfReadInReferencedAnnotatedDocument,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;mateReferencedAnnotatedDocument,
                   int&nbsp;indexOfMateInReferencedAnnotatedDocument)</code>
<div class="block">Constructs a read consisting of a pair of sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">AssemblerInput.Read</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;mate,
                   int&nbsp;expectedMateDistanceForRead,
                   int&nbsp;expectedMateDistanceForMate,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;mateReferencedSequence)</code>
<div class="block">Constructs a read consisting of a pair of sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">AssemblerInput.Read</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedDocument)</code>
<div class="block">Constructs a new read consisting of a single unpaired sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, int)">AssemblerInput.Read</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;contig,
                   int&nbsp;indexOfAssemblyReferenceSequenceForContig)</code>
<div class="block">Constructs a new read consisting of a single contig.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getContig()">getContig</a></strong>()</code>
<div class="block">If this read represents a contig as input to the assembler, returns that contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceForMate()">getExpectedMateDistanceForMate</a></strong>()</code>
<div class="block">Get the expected distance (which may be positive or negative) to find the mate of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()"><code>getMate()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceForRead()">getExpectedMateDistanceForRead</a></strong>()</code>
<div class="block">Get the expected distance (which may be positive or negative) to find the mate of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceNormalized()">getExpectedMateDistanceNormalized</a></strong>()</code>
<div class="block">Gets the expected distance normalized to be positive.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getIndexOfAssemblyReferenceSequenceForContig()">getIndexOfAssemblyReferenceSequenceForContig</a></strong>()</code>
<div class="block">If this read represents a contig (i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()">getMate</a></strong>()</code>
<div class="block">Gets the second read in the pair or null if the read in unpaired.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalized()">getMateNormalized</a></strong>()</code>
<div class="block">Gets the second read in the pair or null if the read is unpaired.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalizedReversed()">getMateNormalizedReversed</a></strong>()</code>
<div class="block">Gets the second read in the pair or null if the read is unpaired normalized such that
 the 2 reads are forward-reverse orientated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateReferencedDocument()">getMateReferencedDocument</a></strong>()</code>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()"><code>getMate()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateReferencedDocumentNormalized()">getMateReferencedDocumentNormalized</a></strong>()</code>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalized()"><code>getMateNormalized()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.DataType.html" title="enum in com.biomatters.geneious.publicapi.plugin">AssemblerInput.DataType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getProbableDataType()">getProbableDataType</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()">getRead</a></strong>()</code>
<div class="block">Gets the first read in the pair or the single read if the read is unpaired.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()">getReadNormalized</a></strong>()</code>
<div class="block">Gets the first read in the pair or the single read if the read is unpaired.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadReferencedDocument()">getReadReferencedDocument</a></strong>()</code>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadReferencedDocumentNormalized()">getReadReferencedDocumentNormalized</a></strong>()</code>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a>.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AssemblerInput.Read</h4>
<pre>public&nbsp;AssemblerInput.Read(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;contig,
                   int&nbsp;indexOfAssemblyReferenceSequenceForContig)</pre>
<div class="block">Constructs a new read consisting of a single contig.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>contig</code> - the contig.</dd><dd><code>indexOfAssemblyReferenceSequenceForContig</code> - the value to be returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getIndexOfAssemblyReferenceSequenceForContig()"><code>getIndexOfAssemblyReferenceSequenceForContig()</code></a></dd></dl>
</li>
</ul>
<a name="AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AssemblerInput.Read</h4>
<pre>public&nbsp;AssemblerInput.Read(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedDocument)</pre>
<div class="block">Constructs a new read consisting of a single unpaired sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>read</code> - the sequence</dd><dd><code>referencedDocument</code> - an optional referenced document. A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</dd></dl>
</li>
</ul>
<a name="AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AssemblerInput.Read</h4>
<pre>public&nbsp;AssemblerInput.Read(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;referencedAnnotatedDocument,
                   int&nbsp;indexInReferencedDocument)</pre>
<div class="block">Constructs a new read consisting of a single unpaired sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>read</code> - the sequence</dd><dd><code>referencedAnnotatedDocument</code> - an optional referenced document. A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</dd><dd><code>indexInReferencedDocument</code> - the index of the sequence in the referenced document</dd></dl>
</li>
</ul>
<a name="AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AssemblerInput.Read</h4>
<pre>public&nbsp;AssemblerInput.Read(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;mate,
                   int&nbsp;expectedMateDistanceForRead,
                   int&nbsp;expectedMateDistanceForMate,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;mateReferencedSequence)</pre>
<div class="block">Constructs a read consisting of a pair of sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>read</code> - the first sequence in the pair</dd><dd><code>mate</code> - the second sequence in the pair</dd><dd><code>expectedMateDistanceForRead</code> - the expected distance of the mate from read which may be positive or negative See <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a> for more information.</dd><dd><code>expectedMateDistanceForMate</code> - the expected distance of the read from mate which may be positive or negative See <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a> for more information.</dd><dd><code>referencedSequence</code> - an optional referenced document corresponding to 'read'. A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</dd><dd><code>mateReferencedSequence</code> - an optional referenced document corresponding to 'mate'. A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</dd></dl>
</li>
</ul>
<a name="AssemblerInput.Read(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument, int, int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AssemblerInput.Read</h4>
<pre>public&nbsp;AssemblerInput.Read(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;read,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;mate,
                   int&nbsp;expectedMateDistanceForRead,
                   int&nbsp;expectedMateDistanceForMate,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;referencedAnnotatedDocument,
                   int&nbsp;indexOfReadInReferencedAnnotatedDocument,
                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;mateReferencedAnnotatedDocument,
                   int&nbsp;indexOfMateInReferencedAnnotatedDocument)</pre>
<div class="block">Constructs a read consisting of a pair of sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>read</code> - the first sequence in the pair</dd><dd><code>mate</code> - the second sequence in the pair</dd><dd><code>expectedMateDistanceForRead</code> - the expected distance of the mate from read which may be positive or negative See <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a> for more information.</dd><dd><code>expectedMateDistanceForMate</code> - the expected distance of the read from mate which may be positive or negative See <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a> for more information.</dd><dd><code>referencedAnnotatedDocument</code> - an optional referenced document for 'read'. A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</dd><dd><code>indexOfReadInReferencedAnnotatedDocument</code> - the index of 'read' in the referenced document</dd><dd><code>mateReferencedAnnotatedDocument</code> - an optional referenced document for 'mate'. A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</dd><dd><code>indexOfMateInReferencedAnnotatedDocument</code> - the index of 'mate' in the referenced document</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getProbableDataType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProbableDataType</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.DataType.html" title="enum in com.biomatters.geneious.publicapi.plugin">AssemblerInput.DataType</a>&nbsp;getProbableDataType()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>The most probable type of data this read is from. In many cases, Geneious can only guess at it.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getContig</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;getContig()</pre>
<div class="block">If this read represents a contig as input to the assembler, returns that contig. If this read is a sequence or pair of sequences, this method will return null
 in which case <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a> will return a non-null value. getContig will only return non-null values if
 <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#supportsInputContigs()"><code>Assembler.supportsInputContigs()</code></a> returned true.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the input contig or null if this read is a sequence or pair of sequences.</dd></dl>
</li>
</ul>
<a name="getIndexOfAssemblyReferenceSequenceForContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIndexOfAssemblyReferenceSequenceForContig</h4>
<pre>public&nbsp;int&nbsp;getIndexOfAssemblyReferenceSequenceForContig()</pre>
<div class="block">If this read represents a contig (i.e. if <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getContig()"><code>getContig()</code></a> returned non-null) which has a reference
 sequence (i.e. getContig().<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>getContigReferenceSequenceIndex()</code></a> returns a non-negative value)
 and that reference sequence is being treated as a reference sequence for this assembly, then
 this method returns the index of the reference sequence in the assembly input.
 (that could be retrieved using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.html#getReferenceSequence(int, jebl.util.ProgressListener)"><code>AssemblerInput.getReferenceSequence(int, jebl.util.ProgressListener)</code></a>
 or -1 if this contig does not contain a reference sequence that is being treated as an input reference sequence to the assembler.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the index of the reference sequence or -1.</dd></dl>
</li>
</ul>
<a name="getRead()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRead</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;getRead()</pre>
<div class="block">Gets the first read in the pair or the single read if the read is unpaired.
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the first read in the pair or the single read if the read is unpaired. This will never return null if the <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#supportsInputContigs()"><code>Assembler.supportsInputContigs()</code></a> return false. If <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Assembler.html#supportsInputContigs()"><code>Assembler.supportsInputContigs()</code></a> returns true then getRead() may return null in which case <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getContig()"><code>getContig()</code></a> will return a non-null value</dd></dl>
</li>
</ul>
<a name="getMate()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMate</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;getMate()</pre>
<div class="block">Gets the second read in the pair or null if the read in unpaired.
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the second read in the pair or null if the read is unpaired</dd></dl>
</li>
</ul>
<a name="getReadReferencedDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReadReferencedDocument</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getReadReferencedDocument()</pre>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a>.
 A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a> or null if this read has no referenced document.</dd></dl>
</li>
</ul>
<a name="getMateReferencedDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateReferencedDocument</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getMateReferencedDocument()</pre>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()"><code>getMate()</code></a>.
 A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()"><code>getMate()</code></a> or null if this read has no referenced document.</dd></dl>
</li>
</ul>
<a name="getReadReferencedDocumentNormalized()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReadReferencedDocumentNormalized</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getReadReferencedDocumentNormalized()</pre>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a>.
 A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a> or null if this read has no referenced document.</dd></dl>
</li>
</ul>
<a name="getMateReferencedDocumentNormalized()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateReferencedDocumentNormalized</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getMateReferencedDocumentNormalized()</pre>
<div class="block">Gets the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalized()"><code>getMateNormalized()</code></a>.
 A referenced document is what Geneious goes to if you click on the little "Go to referenced document" icon to the left of the sequence names in the alignment/contig viewer.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the referenced document (not the same thing as a reference sequence) for the read returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalized()"><code>getMateNormalized()</code></a> or null if this read has no referenced document.</dd></dl>
</li>
</ul>
<a name="getExpectedMateDistanceForRead()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExpectedMateDistanceForRead</h4>
<pre>public&nbsp;int&nbsp;getExpectedMateDistanceForRead()</pre>
<div class="block">Get the expected distance (which may be positive or negative) to find the mate of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a>.
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the expected distance (which may be positive or negative) to find the mate of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getRead()"><code>getRead()</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceForMate()"><code>getExpectedMateDistanceForMate()</code></a></dd></dl>
</li>
</ul>
<a name="getExpectedMateDistanceForMate()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExpectedMateDistanceForMate</h4>
<pre>public&nbsp;int&nbsp;getExpectedMateDistanceForMate()</pre>
<div class="block">Get the expected distance (which may be positive or negative) to find the mate of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()"><code>getMate()</code></a>.
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the expected distance (which may be positive or negative) to find the mate of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMate()"><code>getMate()</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceForRead()"><code>getExpectedMateDistanceForRead()</code></a></dd></dl>
</li>
</ul>
<a name="getExpectedMateDistanceNormalized()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExpectedMateDistanceNormalized</h4>
<pre>public&nbsp;int&nbsp;getExpectedMateDistanceNormalized()</pre>
<div class="block">Gets the expected distance normalized to be positive.
 This is expected to be used in conjunction with <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalized()"><code>getMateNormalized()</code></a>
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the expected distance normalized to be positive.</dd></dl>
</li>
</ul>
<a name="getReadNormalized()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReadNormalized</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;getReadNormalized()</pre>
<div class="block">Gets the first read in the pair or the single read if the read is unpaired.
 This is expected to be used in conjunction with <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceNormalized()"><code>getExpectedMateDistanceNormalized()</code></a> and either <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalized()"><code>getMateNormalized()</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getMateNormalizedReversed()"><code>getMateNormalizedReversed()</code></a>. 
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the read normalized in direction so that its mate pair will appear to the right of it and with the same orientation</dd></dl>
</li>
</ul>
<a name="getMateNormalized()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateNormalized</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;getMateNormalized()</pre>
<div class="block">Gets the second read in the pair or null if the read is unpaired.
 This is expected to be used in conjunction with <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceNormalized()"><code>getExpectedMateDistanceNormalized()</code></a>
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the mate normalized in direction so that its mate pair will appear to the left of it and with the same orientation. Will return null if the read is unpaired.</dd></dl>
</li>
</ul>
<a name="getMateNormalizedReversed()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getMateNormalizedReversed</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;getMateNormalizedReversed()</pre>
<div class="block">Gets the second read in the pair or null if the read is unpaired normalized such that
 the 2 reads are forward-reverse orientated.
 This is expected to be used in conjunction with <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceNormalized()"><code>getExpectedMateDistanceNormalized()</code></a> where
 <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getReadNormalized()"><code>getReadNormalized()</code></a> returns the forward read, getMateNormalizedReversed() returns the reverse read and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html#getExpectedMateDistanceNormalized()"><code>getExpectedMateDistanceNormalized()</code></a> returns the distance between them.
 See <a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>AssemblerInput.Read</code></a> for a discussion of the various get methods.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the mate normalized in direction so that its mate pair will appear to the left of it and with the opposite orientation. Will return null if the read is unpaired.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AssemblerInput.Read.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Properties.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/AssemblerInput.Reads.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/AssemblerInput.Read.html" target="_top">Frames</a></li>
<li><a href="AssemblerInput.Read.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
