<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:47 NZDT 2016 -->
<title>DocumentSelectionSignature (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="DocumentSelectionSignature (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DocumentSelectionSignature.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionOption.OptionValueCreator.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" target="_top">Frames</a></li>
<li><a href="DocumentSelectionSignature.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.plugin</div>
<h2 title="Class DocumentSelectionSignature" class="title">Class DocumentSelectionSignature</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></dd>
</dl>
<hr>
<br>
<pre>public final class <span class="strong">DocumentSelectionSignature</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></pre>
<div class="block">Several plugin components (Viewers, Exporters, Operations, Actions) operate on one (or a set of) Geneious documents.
 DocumentSelectionSignature specifies which subsets of document types are valid for specific operations/viewers etc.
 For example an alignment operation may specify it requires 2 or more nucleotide sequences using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideSequences(int, int)"><code>DocumentSelectionSignature.forNucleotideSequences(2,Integer.MAX_VALUE)</code></a>.
 <p/>
 Geneious then handles correctly enabled/disabling the operation depending on what is selected and only invoking
 the operation with a valid selection of documents.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature.DocumentSelectionSignatureAtom</a></strong></code>
<div class="block">A signature is a set of Atoms.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#DocumentSelectionSignature(java.lang.Class, int, int)">DocumentSelectionSignature</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Class.html?is-external=true" title="class or interface in java.lang">Class</a>&nbsp;classOfSelected,
                          int&nbsp;from,
                          int&nbsp;to)</code>
<div class="block">convenience constructor for accepting a single class repeated a certain number of times.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#DocumentSelectionSignature(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature.DocumentSelectionSignatureAtom)">DocumentSelectionSignature</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature.DocumentSelectionSignatureAtom</a>&nbsp;a)</code>
<div class="block">convenience constructor.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#DocumentSelectionSignature(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature.DocumentSelectionSignatureAtom[])">DocumentSelectionSignature</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature.DocumentSelectionSignatureAtom</a>[]&nbsp;atoms)</code>
<div class="block">construct signature from given atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#DocumentSelectionSignature(org.jdom.Element)">DocumentSelectionSignature</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">XML deserialization constructor</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideAlignment(int, int)">forNucleotideAlignment</a></strong>(int&nbsp;minimumNumberOfSequences,
                      int&nbsp;maximumNumberOfSequences)</code>
<div class="block">Creates a selection signature that matches a single alignment that contains a specified range of nucleotide sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideAlignments(int, int)">forNucleotideAlignments</a></strong>(int&nbsp;minimumNumberOfAlignments,
                       int&nbsp;maximumNumberOfAlignments)</code>
<div class="block">Creates a selection signature that matches a number of nucleotide
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>s within the given range.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideAndProteinSequences(int, int, int, int, boolean)">forNucleotideAndProteinSequences</a></strong>(int&nbsp;minimumNumberOfNucleotideSequences,
                                int&nbsp;maximumNumberOfNucleotideSequences,
                                int&nbsp;minimumNumberOfProteinSequences,
                                int&nbsp;maximumNumberOfProteinSequences,
                                boolean&nbsp;allowSequencesInAlignments)</code>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a> (if allowSequencesInAlignments is true),
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a> 
 such that the total number of nucleotide sequences they contain is within the given range and the total number of
 protein sequences is also within the range.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideSequences(int, int)">forNucleotideSequences</a></strong>(int&nbsp;minimumNumberOfSequences,
                      int&nbsp;maximumNumberOfSequences)</code>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 such that the total number of nucleotide sequences they contain is within the given range.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideSequences(int, int, boolean)">forNucleotideSequences</a></strong>(int&nbsp;minimumNumberOfSequences,
                      int&nbsp;maximumNumberOfSequences,
                      boolean&nbsp;allowSequencesInAlignments)</code>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a> (if allowSequencesInAlignments is true)
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 such that the total number of nucleotide sequences they contain is within the given range, and there are no protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideSequencesAndAlignments(int, int, int, int)">forNucleotideSequencesAndAlignments</a></strong>(int&nbsp;minimumNumberOfSequences,
                                   int&nbsp;maximumNumberOfSequences,
                                   int&nbsp;minimumNumberOfAlignments,
                                   int&nbsp;maximumNumberOfAlignments)</code>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 such that the total number of nucleotide sequences they contain is within the given range, and there are no protein sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinAlignment(int, int)">forProteinAlignment</a></strong>(int&nbsp;minimumNumberOfSequences,
                   int&nbsp;maximumNumberOfSequences)</code>
<div class="block">Creates a selection signature that matches a single alignment that contains a specified range of protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinAlignments(int, int)">forProteinAlignments</a></strong>(int&nbsp;minimumNumberOfAlignments,
                    int&nbsp;maximumNumberOfAlignments)</code>
<div class="block">Creates a selection signature that matches a number of protein
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>s within the given range</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinSequences(int, int)">forProteinSequences</a></strong>(int&nbsp;minimumNumberOfSequences,
                   int&nbsp;maximumNumberOfSequences)</code>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>
 such that the total number of protein sequences they contain is within the given range.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinSequences(int, int, boolean)">forProteinSequences</a></strong>(int&nbsp;minimumNumberOfSequences,
                   int&nbsp;maximumNumberOfSequences,
                   boolean&nbsp;allowSequencesInAlignments)</code>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a> (if allowSequencesInAlignments is true)
  and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>
 such that the total number of protein sequences they contain is within the given range, and there are no nucleotide sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinSequencesAndAlignments(int, int, int, int)">forProteinSequencesAndAlignments</a></strong>(int&nbsp;minimumNumberOfSequences,
                                int&nbsp;maximumNumberOfSequences,
                                int&nbsp;minimumNumberOfAlignments,
                                int&nbsp;maximumNumberOfAlignments)</code>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>
 such that the total number of protein sequences they contain is within the given range, and there are no nucleotide sequences,
 plus a number of protein <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>s within the given range</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getAlignmentSequenceType()">getAlignmentSequenceType</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getNucleotideSequenceMaximum()">getNucleotideSequenceMaximum</a></strong>()</code>
<div class="block">get the maximum number of nucleotide sequences this signature accepts, if it is a signature created using forNucleotideSequences() or forNucleotideAndProteinSequences()</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getNucleotideSequenceMinimum()">getNucleotideSequenceMinimum</a></strong>()</code>
<div class="block">get the minimum number of nucleotide sequences this signature accepts, if it is a signature created using forNucleotideSequences() or forNucleotideAndProteinSequences()</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getProteinSequenceMaximum()">getProteinSequenceMaximum</a></strong>()</code>
<div class="block">get the maximum number of protein sequences this signature accepts, if it is a signature created using forProteinSequences() or forNucleotideAndProteinSequences()</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getProteinSequenceMinimum()">getProteinSequenceMinimum</a></strong>()</code>
<div class="block">get the minimum number of protein sequences this signature accepts, if it is a signature created using forProteinSequences() or forNucleotideAndProteinSequences()</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getSequencesInAlignmentMaximum()">getSequencesInAlignmentMaximum</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getSequencesInAlignmentMinimum()">getSequencesInAlignmentMinimum</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature.DocumentSelectionSignatureAtom</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getSignature()">getSignature</a></strong>()</code>
<div class="block">Get the signature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#isAllowSequencesInNucleotideAlignments()">isAllowSequencesInNucleotideAlignments</a></strong>()</code>
<div class="block">get whether a signature created forNucleotideSequences() or forNucleotideAndProteinSequences() allows sequences in alignments as well as from stand-alone sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#isAllowSequencesInProteinAlignments()">isAllowSequencesInProteinAlignments</a></strong>()</code>
<div class="block">get whether a signature created using forProteinSequences() or forNucleotideAndProteinSequences() allows sequences in alignments as well as from stand-alone sequences</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#isMoreGeneral(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[])">isMoreGeneral</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;potentiallyMoreGeneralSignatures,
             <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;potentiallyMoreSpecificSignatures)</code>
<div class="block">Returns true if all possible documents accepted by potentiallyMoreSpecificSignatures will also be accepted by potentiallyMoreGeneralSignatures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#isNucleotideSequenceSignature()">isNucleotideSequenceSignature</a></strong>()</code>
<div class="block">get whether this signature was created using forNucleotideSequences() or forNucleotideAndProteinSequences()</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#isProteinSequenceSignature()">isProteinSequenceSignature</a></strong>()</code>
<div class="block">get whether this signature was created using forProteinSequences() or forNucleotideAndProteinSequences()</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#matches(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], java.lang.Class[])">matches</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;signatures,
       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Class.html?is-external=true" title="class or interface in java.lang">Class</a>[]&nbsp;classes)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#matches(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], java.util.List)"><code>matches(DocumentSelectionSignature[], java.util.List)</code></a> instead.</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#matches(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], java.util.List)">matches</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;signatures,
       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#matches(java.util.List)">matches</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents)</code>
<div class="block">Check whether this selection signature matches these documents.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="DocumentSelectionSignature(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DocumentSelectionSignature</h4>
<pre>public&nbsp;DocumentSelectionSignature(org.jdom.Element&nbsp;element)
                           throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">XML deserialization constructor</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - an element returned from a previous call to <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#toXML()"><code>toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="DocumentSelectionSignature(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature.DocumentSelectionSignatureAtom[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DocumentSelectionSignature</h4>
<pre>public&nbsp;DocumentSelectionSignature(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature.DocumentSelectionSignatureAtom</a>[]&nbsp;atoms)</pre>
<div class="block">construct signature from given atoms. The signature only matches a set of class if the requirements of all
 atoms are met and if every class in the set matches at least 1 of the atoms.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>atoms</code> - signature parts.</dd></dl>
</li>
</ul>
<a name="DocumentSelectionSignature(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature.DocumentSelectionSignatureAtom)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DocumentSelectionSignature</h4>
<pre>public&nbsp;DocumentSelectionSignature(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature.DocumentSelectionSignatureAtom</a>&nbsp;a)</pre>
<div class="block">convenience constructor.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a</code> - single selection atom.</dd></dl>
</li>
</ul>
<a name="DocumentSelectionSignature(java.lang.Class, int, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DocumentSelectionSignature</h4>
<pre>public&nbsp;DocumentSelectionSignature(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Class.html?is-external=true" title="class or interface in java.lang">Class</a>&nbsp;classOfSelected,
                          int&nbsp;from,
                          int&nbsp;to)</pre>
<div class="block">convenience constructor for accepting a single class repeated a certain number of times.
 (say a set of 2 sequences or more - for alignment)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>classOfSelected</code> - class of selected documents</dd><dd><code>from</code> - lower bound of number of documents (from 0 to Integer.MAX_VALUE)</dd><dd><code>to</code> - upper bound of number of documents (from 0 to Integer.MAX_VALUE, must be greater than or equal to <code>from</code>)</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;element)
             throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - representation from a previous call to <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="forNucleotideSequences(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forNucleotideSequences</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forNucleotideSequences(int&nbsp;minimumNumberOfSequences,
                                                int&nbsp;maximumNumberOfSequences)</pre>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 such that the total number of nucleotide sequences they contain is within the given range. If an operation uses this selection signature, it must
 be prepared to accept <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocuments</code></a> as input (i.e. Geneious won't split the sequence list into stand-alone sequences for input to the operation).</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of nucleotide sequences to match.</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of nucleotide sequences to match.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of nucleotide sequences.</dd></dl>
</li>
</ul>
<a name="forNucleotideSequences(int, int, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forNucleotideSequences</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forNucleotideSequences(int&nbsp;minimumNumberOfSequences,
                                                int&nbsp;maximumNumberOfSequences,
                                                boolean&nbsp;allowSequencesInAlignments)</pre>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a> (if allowSequencesInAlignments is true)
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 such that the total number of nucleotide sequences they contain is within the given range, and there are no protein sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of nucleotide sequences to match.</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of nucleotide sequences to match.</dd><dd><code>allowSequencesInAlignments</code> - whether to allow sequences in alignments as well as stand-alone sequences. If you want a signature that just matches an alignment, use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideAlignment(int, int)"><code>forNucleotideAlignment(int, int)</code></a> instead.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of nucleotide sequences.</dd></dl>
</li>
</ul>
<a name="isNucleotideSequenceSignature()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isNucleotideSequenceSignature</h4>
<pre>public&nbsp;boolean&nbsp;isNucleotideSequenceSignature()</pre>
<div class="block">get whether this signature was created using forNucleotideSequences() or forNucleotideAndProteinSequences()</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>whether this signature was created using forNucleotideSequences() or forNucleotideAndProteinSequences()</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="isProteinSequenceSignature()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isProteinSequenceSignature</h4>
<pre>public&nbsp;boolean&nbsp;isProteinSequenceSignature()</pre>
<div class="block">get whether this signature was created using forProteinSequences() or forNucleotideAndProteinSequences()</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>whether this signature was created using forProteinSequences() or forNucleotideAndProteinSequences()</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getNucleotideSequenceMinimum()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNucleotideSequenceMinimum</h4>
<pre>public&nbsp;int&nbsp;getNucleotideSequenceMinimum()</pre>
<div class="block">get the minimum number of nucleotide sequences this signature accepts, if it is a signature created using forNucleotideSequences() or forNucleotideAndProteinSequences()</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the minimum number of nucleotide sequences this signature accepts, if it is a signature created using forNucleotideSequences() or forNucleotideAndProteinSequences()</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if the signature was not created using forNucleotideSequences() or forNucleotideAndProteinSequences();</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getNucleotideSequenceMaximum()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNucleotideSequenceMaximum</h4>
<pre>public&nbsp;int&nbsp;getNucleotideSequenceMaximum()</pre>
<div class="block">get the maximum number of nucleotide sequences this signature accepts, if it is a signature created using forNucleotideSequences() or forNucleotideAndProteinSequences()</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum number of nucleotide sequences this signature accepts, if it is a signature created using forNucleotideSequences() or forNucleotideAndProteinSequences()</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if the signature was not created using forNucleotideSequences() or forNucleotideAndProteinSequences();</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getProteinSequenceMinimum()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequenceMinimum</h4>
<pre>public&nbsp;int&nbsp;getProteinSequenceMinimum()</pre>
<div class="block">get the minimum number of protein sequences this signature accepts, if it is a signature created using forProteinSequences() or forNucleotideAndProteinSequences()</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the minimum number of nucleotide sequences this signature accepts, if it is a signature created using forProteinSequences() or forNucleotideAndProteinSequences()</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if the signature was not created using forProteinSequences() or forNucleotideAndProteinSequences();</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getProteinSequenceMaximum()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequenceMaximum</h4>
<pre>public&nbsp;int&nbsp;getProteinSequenceMaximum()</pre>
<div class="block">get the maximum number of protein sequences this signature accepts, if it is a signature created using forProteinSequences() or forNucleotideAndProteinSequences()</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum number of nucleotide sequences this signature accepts, if it is a signature created using forProteinSequences() or forNucleotideAndProteinSequences()</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if the signature was not created using forProteinSequences() or forNucleotideAndProteinSequences();</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="isAllowSequencesInNucleotideAlignments()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isAllowSequencesInNucleotideAlignments</h4>
<pre>public&nbsp;boolean&nbsp;isAllowSequencesInNucleotideAlignments()</pre>
<div class="block">get whether a signature created forNucleotideSequences() or forNucleotideAndProteinSequences() allows sequences in alignments as well as from stand-alone sequences</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if a signature created using forNucleotideSequences() or forNucleotideAndProteinSequences() allows sequences in alignments as well as from stand-alone sequences</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if the signature was not created using forNucleotideSequences() or forNucleotideAndProteinSequences();</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="isAllowSequencesInProteinAlignments()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isAllowSequencesInProteinAlignments</h4>
<pre>public&nbsp;boolean&nbsp;isAllowSequencesInProteinAlignments()</pre>
<div class="block">get whether a signature created using forProteinSequences() or forNucleotideAndProteinSequences() allows sequences in alignments as well as from stand-alone sequences</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if a signature created using forProteinSequences() or forNucleotideAndProteinSequences() allows sequences in alignments as well as from stand-alone sequences</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if the signature was not created using forProteinSequences() or forNucleotideAndProteinSequences();</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="forNucleotideSequencesAndAlignments(int, int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forNucleotideSequencesAndAlignments</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forNucleotideSequencesAndAlignments(int&nbsp;minimumNumberOfSequences,
                                                             int&nbsp;maximumNumberOfSequences,
                                                             int&nbsp;minimumNumberOfAlignments,
                                                             int&nbsp;maximumNumberOfAlignments)</pre>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 such that the total number of nucleotide sequences they contain is within the given range, and there are no protein sequences.
 plus a number of nucleotide <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>s within the given range</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of nucleotide sequences to match.</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of nucleotide sequences to match.</dd><dd><code>minimumNumberOfAlignments</code> - the minimum (inclusive) number of nucleotide alignments to match.</dd><dd><code>maximumNumberOfAlignments</code> - the maximum (inclusive) number of nucleotide alignments to match.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of nucleotide sequences and alignments.</dd></dl>
</li>
</ul>
<a name="forNucleotideAlignments(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forNucleotideAlignments</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forNucleotideAlignments(int&nbsp;minimumNumberOfAlignments,
                                                 int&nbsp;maximumNumberOfAlignments)</pre>
<div class="block">Creates a selection signature that matches a number of nucleotide
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>s within the given range. Note that if this is being used on a <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerator</code></a> then the generator must implement
 <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html#generateOnAlignments(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], jebl.util.ProgressListener, com.biomatters.geneious.publicapi.plugin.Options)"><code>SequenceAnnotationGenerator.generateOnAlignments</code></a>
 instead of the usual <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html#generate(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.plugin.SequenceAnnotationGenerator.SelectionRange, jebl.util.ProgressListener, com.biomatters.geneious.publicapi.plugin.Options)"><code>SequenceAnnotationGenerator.generate</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfAlignments</code> - the minimum (inclusive) number of nucleotide alignments to match.</dd><dd><code>maximumNumberOfAlignments</code> - the maximum (inclusive) number of nucleotide alignments to match.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of nucleotide alignments.</dd></dl>
</li>
</ul>
<a name="forProteinSequences(int, int, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forProteinSequences</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forProteinSequences(int&nbsp;minimumNumberOfSequences,
                                             int&nbsp;maximumNumberOfSequences,
                                             boolean&nbsp;allowSequencesInAlignments)</pre>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a> (if allowSequencesInAlignments is true)
  and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>
 such that the total number of protein sequences they contain is within the given range, and there are no nucleotide sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of protein sequences to match.</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of protein sequences to match.</dd><dd><code>allowSequencesInAlignments</code> - whether to allow sequences in alignments as well as stand-alone sequences. If you want a signature that just matches an alignment, use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinAlignment(int, int)"><code>forProteinAlignment(int, int)</code></a> instead</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of protein sequences.</dd></dl>
</li>
</ul>
<a name="forProteinSequences(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forProteinSequences</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forProteinSequences(int&nbsp;minimumNumberOfSequences,
                                             int&nbsp;maximumNumberOfSequences)</pre>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>
 such that the total number of protein sequences they contain is within the given range. If an operation uses this selection signature, it must
 be prepared to accept <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocuments</code></a> as input (i.e. Geneious won't split the sequence list into stand-alone sequences for input to the operation).</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of protein sequences to match.</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of protein sequences to match.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of protein sequences.</dd></dl>
</li>
</ul>
<a name="forProteinSequencesAndAlignments(int, int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forProteinSequencesAndAlignments</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forProteinSequencesAndAlignments(int&nbsp;minimumNumberOfSequences,
                                                          int&nbsp;maximumNumberOfSequences,
                                                          int&nbsp;minimumNumberOfAlignments,
                                                          int&nbsp;maximumNumberOfAlignments)</pre>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>
 such that the total number of protein sequences they contain is within the given range, and there are no nucleotide sequences,
 plus a number of protein <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>s within the given range</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of protein sequences to match.</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of protein sequences to match.</dd><dd><code>minimumNumberOfAlignments</code> - the minimum (inclusive) number of protein alignments to match.</dd><dd><code>maximumNumberOfAlignments</code> - the maximum (inclusive) number of protein alignments to match.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of protein sequences and alignments.</dd></dl>
</li>
</ul>
<a name="forProteinAlignments(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forProteinAlignments</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forProteinAlignments(int&nbsp;minimumNumberOfAlignments,
                                              int&nbsp;maximumNumberOfAlignments)</pre>
<div class="block">Creates a selection signature that matches a number of protein
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>s within the given range</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfAlignments</code> - the minimum (inclusive) number of protein alignments to match.</dd><dd><code>maximumNumberOfAlignments</code> - the maximum (inclusive) number of protein alignments to match.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of proteine alignments.</dd></dl>
</li>
</ul>
<a name="forNucleotideAndProteinSequences(int, int, int, int, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forNucleotideAndProteinSequences</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forNucleotideAndProteinSequences(int&nbsp;minimumNumberOfNucleotideSequences,
                                                          int&nbsp;maximumNumberOfNucleotideSequences,
                                                          int&nbsp;minimumNumberOfProteinSequences,
                                                          int&nbsp;maximumNumberOfProteinSequences,
                                                          boolean&nbsp;allowSequencesInAlignments)</pre>
<div class="block">Creates a selection signature that matches any combination of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>,
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a> (if allowSequencesInAlignments is true),
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a> 
 such that the total number of nucleotide sequences they contain is within the given range and the total number of
 protein sequences is also within the range.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfNucleotideSequences</code> - the minimum (inclusive) number of nucleotide sequences to match. May be 0.</dd><dd><code>maximumNumberOfNucleotideSequences</code> - the maximum (inclusive) number of nucleotide sequences to match.</dd><dd><code>minimumNumberOfProteinSequences</code> - the minimum (inclusive) number of protein sequences to match. May be 0.</dd><dd><code>maximumNumberOfProteinSequences</code> - the maximum (inclusive) number of protein sequences to match.</dd><dd><code>allowSequencesInAlignments</code> - whether to allow sequences in alignments
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>s that are not of the right type</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches the specified number of nucleotide and protein sequences.</dd></dl>
</li>
</ul>
<a name="forNucleotideAlignment(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forNucleotideAlignment</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forNucleotideAlignment(int&nbsp;minimumNumberOfSequences,
                                                int&nbsp;maximumNumberOfSequences)</pre>
<div class="block">Creates a selection signature that matches a single alignment that contains a specified range of nucleotide sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of sequences the alignment must contain</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of sequences the alignment must contain</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches a single alignment that contains a specified range of nucleotide sequences.</dd></dl>
</li>
</ul>
<a name="forProteinAlignment(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forProteinAlignment</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>&nbsp;forProteinAlignment(int&nbsp;minimumNumberOfSequences,
                                             int&nbsp;maximumNumberOfSequences)</pre>
<div class="block">Creates a selection signature that matches a single alignment that contains a specified range of protein sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumNumberOfSequences</code> - the minimum (inclusive) number of sequences the alignment must contain</dd><dd><code>maximumNumberOfSequences</code> - the maximum (inclusive) number of sequences the alignment must contain</dd>
<dt><span class="strong">Returns:</span></dt><dd>a selection signature that matches a single alignment that contains a specified range of protein sequences.</dd></dl>
</li>
</ul>
<a name="getSignature()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSignature</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature.DocumentSelectionSignatureAtom</a>[]&nbsp;getSignature()</pre>
<div class="block">Get the signature.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the signatures</dd></dl>
</li>
</ul>
<a name="matches(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>matches</h4>
<pre>public&nbsp;boolean&nbsp;matches(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents)</pre>
<div class="block">Check whether this selection signature matches these documents.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>documents</code> - the documents to check against. null is equivalent to an empty document list.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this selection signature matches these documents.</dd></dl>
</li>
</ul>
<a name="matches(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>matches</h4>
<pre>public static&nbsp;boolean&nbsp;matches(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;signatures,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>signatures</code> - the signatures to be matched</dd><dd><code>documents</code> - the documents to be matched. null is equivalent to an empty document list.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if at least one of the document selection signatures match the given documents</dd></dl>
</li>
</ul>
<a name="isMoreGeneral(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isMoreGeneral</h4>
<pre>public static&nbsp;boolean&nbsp;isMoreGeneral(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;potentiallyMoreGeneralSignatures,
                    <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;potentiallyMoreSpecificSignatures)</pre>
<div class="block">Returns true if all possible documents accepted by potentiallyMoreSpecificSignatures will also be accepted by potentiallyMoreGeneralSignatures.
 <p/>
 Note that this method does not cover all possible scenarios so it may return false in some cases even  where the documents accepted by
 potentiallyMoreSpecificSignatures are also accepted by potentiallyMoreGeneralSignatures.
 For example cases where signatures contain more than 1 atom are not covered. And generic handling of sequence lists comparing with specific SequenceDocument classes are not covered.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>potentiallyMoreGeneralSignatures</code> - the signatures to see if they are more general than the other ones</dd><dd><code>potentiallyMoreSpecificSignatures</code> - the signatures to see if they are more specific than the other ones</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if all possible documents accepted by potentiallyMoreSpecificSignatures will also be accepted by potentiallyMoreGeneralSignatures.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="matches(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], java.lang.Class[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>matches</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;boolean&nbsp;matches(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentSelectionSignature</a>[]&nbsp;signatures,
                         <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Class.html?is-external=true" title="class or interface in java.lang">Class</a>[]&nbsp;classes)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>Use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#matches(com.biomatters.geneious.publicapi.plugin.DocumentSelectionSignature[], java.util.List)"><code>matches(DocumentSelectionSignature[], java.util.List)</code></a> instead.</i></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>signatures</code> - the signatures to be matched</dd><dd><code>classes</code> - the classes to be matched</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if at least one of the document selection signatures match the given classes</dd></dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="getAlignmentSequenceType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlignmentSequenceType</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;getAlignmentSequenceType()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the type of alignment this signature for if it was created using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideAlignment(int, int)"><code>forNucleotideAlignment(int, int)</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinAlignment(int, int)"><code>forProteinAlignment(int, int)</code></a> or null if it was created using neither of these signatures</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
<a name="getSequencesInAlignmentMinimum()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesInAlignmentMinimum</h4>
<pre>public&nbsp;int&nbsp;getSequencesInAlignmentMinimum()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the minimum number of sequences in an alignment if this signature was created using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideAlignment(int, int)"><code>forNucleotideAlignment(int, int)</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinAlignment(int, int)"><code>forProteinAlignment(int, int)</code></a>.
 You can determine if this signature was created using either of these via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getAlignmentSequenceType()"><code>getAlignmentSequenceType()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
<a name="getSequencesInAlignmentMaximum()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getSequencesInAlignmentMaximum</h4>
<pre>public&nbsp;int&nbsp;getSequencesInAlignmentMaximum()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum number of sequences in an alignment if this signature was created using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forNucleotideAlignment(int, int)"><code>forNucleotideAlignment(int, int)</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#forProteinAlignment(int, int)"><code>forProteinAlignment(int, int)</code></a>.
 You can determine if this signature was created using either of these via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html#getAlignmentSequenceType()"><code>getAlignmentSequenceType()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DocumentSelectionSignature.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionOption.OptionValueCreator.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.DocumentSelectionSignatureAtom.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/DocumentSelectionSignature.html" target="_top">Frames</a></li>
<li><a href="DocumentSelectionSignature.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
