<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:48 NZDT 2016 -->
<title>SequenceGraph.SequencePropertyRetriever (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceGraph.SequencePropertyRetriever (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceGraph.SequencePropertyRetriever.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.Location.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraphFactory.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html" target="_top">Frames</a></li>
<li><a href="SequenceGraph.SequencePropertyRetriever.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.plugin</div>
<h2 title="Class SequenceGraph.SequencePropertyRetriever" class="title">Class SequenceGraph.SequencePropertyRetriever</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.plugin.SequenceGraph.SequencePropertyRetriever</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceViewerExtension.PropertyRetrieverAndEditor.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceViewerExtension.PropertyRetrieverAndEditor</a></dd>
</dl>
<dl>
<dt>Enclosing class:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceGraph</a></dd>
</dl>
<hr>
<br>
<pre>public abstract static class <span class="strong">SequenceGraph.SequencePropertyRetriever</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></pre>
<div class="block">Provides additional information about the sequence(s) (e.g. the sequence name) for which the graph is being drawn.
 <p/>
 A property retriever can be obtained by graph plugins via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.html#getSequencePropertyRetriever()"><code>SequenceGraph.getSequencePropertyRetriever</code></a>
 (the sequence viewer sets it as a gettable property immediately after creating the graph).
 <p/>
 A property retriever is supplied to sequence viewer extension plugins when they are created via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceViewerExtension.Factory.html#createSequenceViewerExtension(com.biomatters.geneious.publicapi.plugin.DocumentViewer, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], com.biomatters.geneious.publicapi.plugin.SequenceViewerExtension.PropertyRetrieverAndEditor)"><code>SequenceViewerExtension.Factory.createSequenceViewerExtension</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceViewerExtension.StatisticsFactory.html#createStatistics(com.biomatters.geneious.publicapi.plugin.SequenceViewerExtension.PropertyRetrieverAndEditor, jebl.util.ProgressListener)"><code>SequenceViewerExtension.StatisticsFactory.createStatistics</code></a>.
 <p/>
 Plugins that wish to write test cases can instantiate a property retriever using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#forAlignment(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument)"><code>forAlignment</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceViewerExtension.PropertyRetrieverAndEditor.html#forSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)"><code>forSequences</code></a>.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#SequenceGraph.SequencePropertyRetriever()">SequenceGraph.SequencePropertyRetriever</a></strong>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceGraph.SequencePropertyRetriever</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#forAlignment(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument)">forAlignment</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment)</code>
<div class="block">Creates a SequencePropertyRetriever for this alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getAlphabet()">getAlphabet</a></strong>()</code>
<div class="block">Get the alphabet of the sequence(s)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getCircularAlignmentLength()">getCircularAlignmentLength</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" title="class in com.biomatters.geneious.publicapi.implementations">EndGapsManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getEndGapsManager()">getEndGapsManager</a></strong>()</code>
<div class="block">If this graph is on an alignment, gets an end gaps manager for quickly finding sequences that intersect a given base number, ignoring end gaps.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></strong>()</code>
<div class="block">An immutable view of the paired reads managed by this list or alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getLeadingTrimLength(int)">getLeadingTrimLength</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Get the length (if any) of the leading trimmed region of this sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getMappingQuality(int)">getMappingQuality</a></strong>(int&nbsp;sequenceIndex)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getMateIndex(int)">getMateIndex</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getMaximumSequenceLength()">getMaximumSequenceLength</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedNucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">ExtendedNucleotideGraph</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getNucleotideGraph(int)">getNucleotideGraph</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Get the nucleotide graph (chromatogram) for the sequence at the specified index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getNumberOfMappedLocations(int)">getNumberOfMappedLocations</a></strong>(int&nbsp;sequenceIndex)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>
<div class="block">Get the number of sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getReferenceSequenceIndex()">getReferenceSequenceIndex</a></strong>()</code>
<div class="block">Get the index of the reference sequence in this alignment (if any)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getSequenceCharSequence(int)">getSequenceCharSequence</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Get the SequenceCharSequence for the sequence at the specified index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getSequenceListSummary()">getSequenceListSummary</a></strong>()</code>
<div class="block">Gets a summary of all the sequences in a sequence list, or all the non-reference sequences in an alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getSequenceName(int)">getSequenceName</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the name of the specified sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#getTrailingTrimLength(int)">getTrailingTrimLength</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Get the length (if any) of the traling trimmed region of this sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#isCircularSequence(int)">isCircularSequence</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Returns true if this sequence is circular</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#isReferenceSequnece(int)">isReferenceSequnece</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Get whether or not the specified sequence is a reference sequence in an alignment or contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#isReversedSequence(int)">isReversedSequence</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Get the direction of the specified sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html#isTrimmed(int, int)">isTrimmed</a></strong>(int&nbsp;sequenceIndex,
         int&nbsp;residueIndex)</code>
<div class="block">Get whether or not the specified position in the specified sequence is trimmed or not</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceGraph.SequencePropertyRetriever()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceGraph.SequencePropertyRetriever</h4>
<pre>public&nbsp;SequenceGraph.SequencePropertyRetriever()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="isReversedSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReversedSequence</h4>
<pre>public&nbsp;boolean&nbsp;isReversedSequence(int&nbsp;sequenceIndex)</pre>
<div class="block">Get the direction of the specified sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence to get the name of. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if the sequence is in the reversed direction (i.e. its name ends with " (reversed)")</dd></dl>
</li>
</ul>
<a name="getSequenceName(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceName</h4>
<pre>public abstract&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSequenceName(int&nbsp;sequenceIndex)</pre>
<div class="block">Gets the name of the specified sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence to get the name of. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the name of the specified sequence.</dd></dl>
</li>
</ul>
<a name="isReferenceSequnece(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReferenceSequnece</h4>
<pre>public abstract&nbsp;boolean&nbsp;isReferenceSequnece(int&nbsp;sequenceIndex)</pre>
<div class="block">Get whether or not the specified sequence is a reference sequence in an alignment or contig.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if the specified sequence is a reference sequence in an alignment or contig.</dd></dl>
</li>
</ul>
<a name="getReferenceSequenceIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceSequenceIndex</h4>
<pre>public abstract&nbsp;int&nbsp;getReferenceSequenceIndex()</pre>
<div class="block">Get the index of the reference sequence in this alignment (if any)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the index of the reference sequence or -1 if this is not an alignment or if this alignment has no reference sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="isTrimmed(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isTrimmed</h4>
<pre>public abstract&nbsp;boolean&nbsp;isTrimmed(int&nbsp;sequenceIndex,
                int&nbsp;residueIndex)</pre>
<div class="block">Get whether or not the specified position in the specified sequence is trimmed or not</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd><dd><code>residueIndex</code> - the 0-based index of the residue in the sequence.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if specified position in the specified sequence is trimmed</dd></dl>
</li>
</ul>
<a name="getLeadingTrimLength(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLeadingTrimLength</h4>
<pre>public abstract&nbsp;int&nbsp;getLeadingTrimLength(int&nbsp;sequenceIndex)</pre>
<div class="block">Get the length (if any) of the leading trimmed region of this sequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the length of the trimmed region at the start of this sequence (excluding end gaps), or 0 if there is no trimmed region</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="getTrailingTrimLength(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrailingTrimLength</h4>
<pre>public abstract&nbsp;int&nbsp;getTrailingTrimLength(int&nbsp;sequenceIndex)</pre>
<div class="block">Get the length (if any) of the traling trimmed region of this sequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the length of the trimmed region at the end of this sequence (excluding end gaps), or 0 if there is no trimmed region</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="getNucleotideGraph(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNucleotideGraph</h4>
<pre>public abstract&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedNucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">ExtendedNucleotideGraph</a>&nbsp;getNucleotideGraph(int&nbsp;sequenceIndex)</pre>
<div class="block">Get the nucleotide graph (chromatogram) for the sequence at the specified index.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>an ExtendedNucleotideGraph for the sequence at index <code>sequenceIndex</code> or null if the sequence does not have a nucleotide graph</dd></dl>
</li>
</ul>
<a name="getAlphabet()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlphabet</h4>
<pre>public abstract&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;getAlphabet()</pre>
<div class="block">Get the alphabet of the sequence(s)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the SequenceDocument.Alphabet giving the alphabet of the sequence(s)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd></dl>
</li>
</ul>
<a name="getEndGapsManager()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getEndGapsManager</h4>
<pre>public abstract&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" title="class in com.biomatters.geneious.publicapi.implementations">EndGapsManager</a>&nbsp;getEndGapsManager()</pre>
<div class="block">If this graph is on an alignment, gets an end gaps manager for quickly finding sequences that intersect a given base number, ignoring end gaps.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>If this graph is on an alignment an end gaps manager or null if this graph is for an unaligned sequence.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="getSequenceCharSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceCharSequence</h4>
<pre>public abstract&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&nbsp;getSequenceCharSequence(int&nbsp;sequenceIndex)</pre>
<div class="block">Get the SequenceCharSequence for the sequence at the specified index.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a SequenceCharSequence for the sequence at index <code>sequenceIndex</code></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public&nbsp;int&nbsp;getNumberOfSequences()</pre>
<div class="block">Get the number of sequences.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences in the alignment, or 1 for sequence based graphs.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="forAlignment(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forAlignment</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceGraph.SequencePropertyRetriever</a>&nbsp;forAlignment(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment)</pre>
<div class="block">Creates a SequencePropertyRetriever for this alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - the alignment to create a properties retriever for.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a newly created properties retriever.</dd></dl>
</li>
</ul>
<a name="getSequenceListSummary()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceListSummary</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a>&nbsp;getSequenceListSummary()</pre>
<div class="block">Gets a summary of all the sequences in a sequence list, or all the non-reference sequences in an alignment.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a summary or null if no summary is available.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getMaximumSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMaximumSequenceLength</h4>
<pre>public&nbsp;int&nbsp;getMaximumSequenceLength()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum length of all the sequences in a sequence list or alignment.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="isCircularSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isCircularSequence</h4>
<pre>public&nbsp;boolean&nbsp;isCircularSequence(int&nbsp;sequenceIndex)</pre>
<div class="block">Returns true if this sequence is circular</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this sequence is circular (see <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a>)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getCircularAlignmentLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCircularAlignmentLength</h4>
<pre>public&nbsp;int&nbsp;getCircularAlignmentLength()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the circular length of the alignment or 0 if this is not a circular alignment as defined by <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()"><code>SequenceAlignmentDocument.getCircularLength()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getNumberOfMappedLocations(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfMappedLocations</h4>
<pre>public&nbsp;int&nbsp;getNumberOfMappedLocations(int&nbsp;sequenceIndex)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the number of locations this sequence is mapped to against the reference sequence. See <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#NUMBER_OF_MAPPED_LOCATIONS"><code>DocumentField.NUMBER_OF_MAPPED_LOCATIONS</code></a>. For sequences without this field, 1 will be returned.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getMappingQuality(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMappingQuality</h4>
<pre>public&nbsp;int&nbsp;getMappingQuality(int&nbsp;sequenceIndex)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the 0-based index of the sequence. This is only applicable on alignment based graphs. This must be 0 on sequence based graphs.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the quality of a mapped read. -1 for no quality or between 0 and 254 inclusive. Equivalent to <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#MAPPING_QUALITY"><code>DocumentField.MAPPING_QUALITY</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd></dl>
</li>
</ul>
<a name="isFreeEndGapAlignment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isFreeEndGapAlignment</h4>
<pre>public&nbsp;boolean&nbsp;isFreeEndGapAlignment()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this is an alignment with free end gaps as defined by <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isFreeEndGapAlignment()"><code>SequenceAlignmentDocument.isFreeEndGapAlignment()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getMateExpectedDistance(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateExpectedDistance</h4>
<pre>public&nbsp;int&nbsp;getMateExpectedDistance(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate. This is defined
 as the length of the 2 reads plus the distance between them.
 For example if the 2 reads both have length 30 with a 40 bp gap between them, then the
 expected mate distance is 100.
 If the mate is expected to appear after this sequence with this sequence in its current orientation, then
 the expected distance will be positive. If the mate is expected to appear before this sequence with this sequence in its current orientation, then
 the expected distance will be negative.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence for which to get the mate index of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the expected mate distance or 0 if this read is not paired. Note that a read may still have a mate distance!=0 but its mate may not be present in this document (in which case <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a> will be -1.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a></dd></dl>
</li>
</ul>
<a name="getMateIndex(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateIndex</h4>
<pre>public&nbsp;int&nbsp;getMateIndex(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">getMateIndex</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence to get the mate index of</dd>
<dt><span class="strong">Returns:</span></dt><dd>an index of the mate of the sequence, or -1 if the sequence has no mate.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>PairedReads.getMateExpectedDistance(int)</code></a></dd></dl>
</li>
</ul>
<a name="getImmutablePairedReadManager()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getImmutablePairedReadManager</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;getImmutablePairedReadManager()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">PairedReads</a></code></strong></div>
<div class="block">An immutable view of the paired reads managed by this list or alignment. Even if the list or alignment
 that implements this method is later modified, the
 returned PairedReadManager will not change.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>An immutable view of the paired reads managed by the list or alignment that implements this method or null if this class doesn't provide an immutable view of its paired reads.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceGraph.SequencePropertyRetriever.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.Location.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraphFactory.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html" target="_top">Frames</a></li>
<li><a href="SequenceGraph.SequencePropertyRetriever.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
