<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:48 NZDT 2016 -->
<title>SequenceSelection.SelectionInterval (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceSelection.SelectionInterval (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceSelection.SelectionInterval.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" target="_top">Frames</a></li>
<li><a href="SequenceSelection.SelectionInterval.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.plugin</div>
<h2 title="Class SequenceSelection.SelectionInterval" class="title">Class SequenceSelection.SelectionInterval</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.plugin.SequenceSelection.SelectionInterval</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></dd>
</dl>
<dl>
<dt>Enclosing class:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection</a></dd>
</dl>
<hr>
<br>
<pre>public static final class <span class="strong">SequenceSelection.SelectionInterval</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">A contiguous selection of residues in a sequence.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#SequenceSelection.SelectionInterval(org.jdom.Element)">SequenceSelection.SelectionInterval</a></strong>(org.jdom.Element&nbsp;element)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.utilities.Interval, com.biomatters.geneious.publicapi.utilities.Interval, jebl.evolution.sequences.SequenceType, com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceDocumentType)">SequenceSelection.SelectionInterval</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;residueInterval,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;sequenceInterval,
                                   <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceDocumentType</a>&nbsp;sequenceDocumentType)</code>
<div class="block">Constructs a rectangular selection (i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.utilities.Interval, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">SequenceSelection.SelectionInterval</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;residueInterval,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <code>#SelectionInterval(SequenceIndex, com.biomatters.geneious.publicapi.utilities.Interval)</code> instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#SequenceSelection.SelectionInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">SequenceSelection.SelectionInterval</a></strong>(int&nbsp;fromResidue,
                                   int&nbsp;toResidue,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <code>#SelectionInterval(SequenceIndex, int, int)</code> instead</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.utilities.Interval)">SequenceSelection.SelectionInterval</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;sequenceIndex,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;residueInterval)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, int, int)">SequenceSelection.SelectionInterval</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;sequenceIndex,
                                   int&nbsp;fromResidue,
                                   int&nbsp;toResidue)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#equals(java.lang.Object)">equals</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;o)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getFromResidue()">getFromResidue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getMaxResidue()">getMaxResidue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getMinResidue()">getMinResidue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getResidueInterval()">getResidueInterval</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequence()">getSequence</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceIndex()"><code>getSequenceIndex()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html#getSequence(java.util.List)"><code>SequenceSelection.SequenceIndex.getSequence(java.util.List)</code></a> instead.</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceDocumentType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceDocumentType()">getSequenceDocumentType</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceIndex()">getSequenceIndex</a></strong>()</code>
<div class="block">Returns the sequence index if this is a selection for a single sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceIndices()">getSequenceIndices</a></strong>()</code>
<div class="block">Gets all selected sequence indices.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequencesRange()">getSequencesRange</a></strong>()</code>
<div class="block">Gets the range of sequences selected</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceType()">getSequenceType</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getToResidue()">getToResidue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#hashCode()">hashCode</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#toXML()">toXML</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceSelection.SelectionInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection.SelectionInterval</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;SequenceSelection.SelectionInterval(int&nbsp;fromResidue,
                                              int&nbsp;toResidue,
                                              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <code>#SelectionInterval(SequenceIndex, int, int)</code> instead</i></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fromResidue</code> - 0-offset residue index where the interval starts, inclusive</dd><dd><code>toResidue</code> - residue 0-offset index where the interval finishes, exclusive - may be less than fromResidue for reverse direction</dd><dd><code>sequence</code> - the sequence in which the residues are selected, must not be null</dd></dl>
</li>
</ul>
<a name="SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection.SelectionInterval</h4>
<pre>public&nbsp;SequenceSelection.SelectionInterval(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;sequenceIndex,
                                   int&nbsp;fromResidue,
                                   int&nbsp;toResidue)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the sequence in which the residues are selected.</dd><dd><code>fromResidue</code> - 0-offset residue index where the interval starts, inclusive</dd><dd><code>toResidue</code> - residue 0-offset index where the interval finishes, exclusive - may be less than fromResidue for reverse direction. May be Integer.MAX_VALUE to represent selecting the entire sequence.
 If <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html#getMaxExclusive()"><code>Interval.getMaxExclusive()</code></a> is greater than the sequence length, if a single sequence is selected which is circular then the selected is treated as wrapped back to the start of the sequence.
 in other cases the selection is truncated to the sequence length.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.utilities.Interval, com.biomatters.geneious.publicapi.utilities.Interval, jebl.evolution.sequences.SequenceType, com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceDocumentType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection.SelectionInterval</h4>
<pre>public&nbsp;SequenceSelection.SelectionInterval(<a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;residueInterval,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;sequenceInterval,
                                   <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;sequenceType,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceDocumentType</a>&nbsp;sequenceDocumentType)</pre>
<div class="block">Constructs a rectangular selection (i.e. a range of sequences and a range of residues)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueInterval</code> - the range of residues.  <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html#getMaxExclusive()"><code>Interval.getMaxExclusive()</code></a> may be Integer.MAX_VALUE to represent selecting the entire sequence.
 If <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html#getMaxExclusive()"><code>Interval.getMaxExclusive()</code></a> is greater than the sequence length, if a single sequence is selected which is circular then the selected is treated as wrapped back to the start of the sequence.
 in other cases the selection is truncated to the sequence length.</dd><dd><code>sequenceInterval</code> - the range of sequences</dd><dd><code>sequenceType</code> - the type of sequence (nucleotide or amino acid) selected</dd><dd><code>sequenceDocumentType</code> - the type of documents (aligned or unaligned) the sequences are selected in.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.utilities.Interval, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection.SelectionInterval</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;SequenceSelection.SelectionInterval(<a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;residueInterval,
                                              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <code>#SelectionInterval(SequenceIndex, com.biomatters.geneious.publicapi.utilities.Interval)</code> instead</i></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueInterval</code> - the residues selected for the given sequence.  <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html#getMaxExclusive()"><code>Interval.getMaxExclusive()</code></a> may be Integer.MAX_VALUE to represent selecting the entire sequence</dd><dd><code>sequence</code> - the sequence in which the residues are selected, must not be null</dd></dl>
</li>
</ul>
<a name="SequenceSelection.SelectionInterval(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.utilities.Interval)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection.SelectionInterval</h4>
<pre>public&nbsp;SequenceSelection.SelectionInterval(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;sequenceIndex,
                                   <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;residueInterval)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the sequence in which the residues are selected. May be -1 for the consensus sequence (if it exists)</dd><dd><code>residueInterval</code> - the residues selected for the given sequence.   <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html#getMaxExclusive()"><code>Interval.getMaxExclusive()</code></a> may be Integer.MAX_VALUE to represent selecting the entire sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="SequenceSelection.SelectionInterval(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceSelection.SelectionInterval</h4>
<pre>public&nbsp;SequenceSelection.SelectionInterval(org.jdom.Element&nbsp;element)
                                    throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - an element returned from a previous call to <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#toXML()"><code>toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if XML does not conform to our unpublished specification</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getFromResidue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFromResidue</h4>
<pre>public&nbsp;int&nbsp;getFromResidue()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>residue index where the interval starts, inclusive</dd></dl>
</li>
</ul>
<a name="getToResidue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getToResidue</h4>
<pre>public&nbsp;int&nbsp;getToResidue()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>residue index where the interval finishes, exclusive - may be less than fromResidue for reverse direction</dd></dl>
</li>
</ul>
<a name="getSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;getSequence()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceIndex()"><code>getSequenceIndex()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html#getSequence(java.util.List)"><code>SequenceSelection.SequenceIndex.getSequence(java.util.List)</code></a> instead.</i></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>sequence the sequence in which the residues are selected</dd></dl>
</li>
</ul>
<a name="getSequenceIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceIndex</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;getSequenceIndex()</pre>
<div class="block">Returns the sequence index if this is a selection for a single sequence.  This method should only be called on SelectionIntervals returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals()"><code>SequenceSelection.getIntervals(true)</code></a>.
 For SelectionIntervals returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals(boolean)"><code>getIntervals(false)</code></a> use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequencesRange()"><code>getSequencesRange()</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceIndices()"><code>getSequenceIndices()</code></a> instead.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sequence index</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - If this is a selection for multiple sequences (created via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals(boolean)"><code>getIntervals(false)</code></a>)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getSequenceIndices()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceIndices</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&gt;&nbsp;getSequenceIndices()</pre>
<div class="block">Gets all selected sequence indices.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>all selected sequence indices.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getSequencesRange()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesRange</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;getSequencesRange()</pre>
<div class="block">Gets the range of sequences selected</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the range of sequences selected</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getSequenceType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceType</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;getSequenceType()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the type (nucleotide or amino acid) of sequences selected</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getSequenceDocumentType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceDocumentType</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceDocumentType</a>&nbsp;getSequenceDocumentType()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the document type (aligned or unaliged) of sequences selected</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getMinResidue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMinResidue</h4>
<pre>public&nbsp;int&nbsp;getMinResidue()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the minimum residue index in the interval, inclusive</dd></dl>
</li>
</ul>
<a name="getMaxResidue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMaxResidue</h4>
<pre>public&nbsp;int&nbsp;getMaxResidue()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum residue index in the interval, exclusive (never less than minResidue). This may be Integer.MAX_VALUE to represent selecting the entire sequence</dd></dl>
</li>
</ul>
<a name="getResidueInterval()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getResidueInterval</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;getResidueInterval()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the interval of selected residues. This interval may be [0,Integer.MAX_VALUE) to represent selecting the entire sequence</dd></dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="equals(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre>public&nbsp;boolean&nbsp;equals(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;o)</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="hashCode()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hashCode</h4>
<pre>public&nbsp;int&nbsp;hashCode()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>an element that can be used with <code>#SelectionInterval(org.jdom.Element)</code></dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceSelection.SelectionInterval.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" target="_top">Frames</a></li>
<li><a href="SequenceSelection.SelectionInterval.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
