<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:48 NZDT 2016 -->
<title>SequenceSelection (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceSelection (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceSelection.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraphFactory.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" target="_top">Frames</a></li>
<li><a href="SequenceSelection.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.plugin</div>
<h2 title="Class SequenceSelection" class="title">Class SequenceSelection</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.plugin.SequenceSelection</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></dd>
</dl>
<hr>
<br>
<pre>public final class <span class="strong">SequenceSelection</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">Specifies a selection of regions in a sequence, list of sequences, alignment or other sequence containing document.
 <p/>
 A SequenceSelection is provided to <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperation.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentOperations</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceAnnotationGenerator.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceAnnotationGenerators</code></a> to allow them to optionally make their behaviour depend on
 what the user has currently selected in the sequence/alignment viewer. <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentViewer.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>DocumentViewers</code></a> are also informed of the current selection in other viewers via <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentViewerMessageHandler.html#setSequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection, java.lang.String)"><code>DocumentViewerMessageHandler.setSequenceSelection</code></a>
 <p/>
 Each SequenceSelection consists of one or more <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SelectionIntervals</code></a>. A SelectionInterval
 has a range of selected residues/bases (see <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getResidueInterval()"><code>SelectionInterval.getResidueInterval()</code></a>)
 and an associated <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>SequenceSelection.SequenceIndex</code></a>.
 <p/>
  The SequenceIndex indicates which sequence in the set of sequences is selected.
 The index is relative to the first sequence in the first document. So for example if 2 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocuments</code></a> are
 selected, and the first sequence list has 5 sequences, then a SequenceIndex of 6 corresponds to the second sequence in the second list.
 <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html#getSequence(java.util.List)"><code>SequenceIndex.getSequence</code></a> automatically applies this logic.

 <p/>
 For a slightly more complicated example, consider a set of 2 documents, the first of which is a SequenceDocument and the second is a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>CombinedAlignmentAndSequenceDocument</code></a>.
 <p/>
 Let [a,b] denote a SequenceIndex with <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html#getSequenceIndex()"><code>SequenceSelection.SequenceIndex.getSequenceIndex()</code></a>==a and <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html#getSequenceDocumentType()"><code>SequenceSelection.SequenceIndex.getSequenceDocumentType()</code></a>==b)
 Then:<br>
 [0,SequenceDocumentType.Sequence] refers to the sequence contained in the first document, the SequenceDocument<br>
 [1,SequenceDocumentType.Sequence] refers to the un-aligned sequence from the CombinedAlignmentAndSequenceDocument<br>
 [0,SequenceDocumentType.Alignment] refers to the first sequence in the alignment in the CombinedAlignmentAndSequenceDocument<br>
 [1,SequenceDocumentType.Alignment] refers to the second sequence in the alignment in the CombinedAlignmentAndSequenceDocument<br>
 [2,SequenceDocumentType.Alignment] refers to the third sequence in the alignment in the CombinedAlignmentAndSequenceDocument<br></div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a></strong></code>
<div class="block">A contiguous selection of residues in a sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceDocumentType</a></strong></code>
<div class="block">Used for identifying the type of view the selection applies for identifying which list the index applies to in a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>CombinedAlignmentAndSequenceDocument</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a></strong></code>
<div class="block">Represents the index of the selected sequence and the type (so that we can identify which list the index applies to in <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocuments</code></a>.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#SequenceSelection(org.jdom.Element)">SequenceSelection</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Construct a SequenceSelection from the results of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#toXML()"><code>toXML()</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#SequenceSelection(java.util.List)">SequenceSelection</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;&nbsp;intervals)</code>
<div class="block">Create a SequenceSelection</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#SequenceSelection(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">SequenceSelection</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)"><code>SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SelectionInterval...)">SequenceSelection</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>...&nbsp;intervals)</code>
<div class="block">Create a SequenceSelection</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">SequenceSelection</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;sequenceIndex,
                 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>
<div class="block">Construct for a selected SequenceAnnotation on the specified sequence.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#equals(java.lang.Object)">equals</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;o)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#extractIntervals()">extractIntervals</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;&nbsp;</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getAnnotation()">getAnnotation</a></strong>()</code>
<div class="block">Get the selected annotation if there is one</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getAnnotationSequenceIndex()">getAnnotationSequenceIndex</a></strong>()</code>
<div class="block">Get the sequence index of the selected annotation if there is an annotation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals()">getIntervals</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals(boolean)"><code>getIntervals(boolean)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals(boolean)">getIntervals</a></strong>(boolean&nbsp;splitMultiSequenceIntervalsIntoSingleSelectionIntervals)</code>
<div class="block">Gets the intervals selected</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getNonZeroLengthIntervals()">getNonZeroLengthIntervals</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getSelectedSequenceCount()">getSelectedSequenceCount</a></strong>()</code>
<div class="block">Returns the number of sequences which have any residues selected (0 length selections are included).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getSelectedSequenceIndices()">getSelectedSequenceIndices</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getSelectedSequences()">getSelectedSequences</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getSelectedSequenceIndices()"><code>getSelectedSequenceIndices()</code></a></i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getTotalSelection(java.util.List, jebl.evolution.sequences.SequenceType, com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceDocumentType)">getTotalSelection</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents,
                 <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;type,
                 <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceDocumentType</a>&nbsp;sequenceDocumentType)</code>
<div class="block">For a List of AnnotatedPluginDocument, creates a selection that represents a "Select all" on each of the documents.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#hashCode()">hashCode</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#withoutZeroLengthIntervals()">withoutZeroLengthIntervals</a></strong>()</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceSelection(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection</h4>
<pre>public&nbsp;SequenceSelection(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;&nbsp;intervals)</pre>
<div class="block">Create a SequenceSelection</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>intervals</code> - the intervals</dd></dl>
</li>
</ul>
<a name="SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SelectionInterval...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection</h4>
<pre>public&nbsp;SequenceSelection(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>...&nbsp;intervals)</pre>
<div class="block">Create a SequenceSelection</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>intervals</code> - the intervals</dd></dl>
</li>
</ul>
<a name="SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection</h4>
<pre>public&nbsp;SequenceSelection(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;sequenceIndex,
                 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<div class="block">Construct for a selected SequenceAnnotation on the specified sequence. The intervals returned by <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals()"><code>getIntervals()</code></a>
 will correspond to the intervals of this annotation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the sequence</dd><dd><code>annotation</code> - the selected annotation, must not be null</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="SequenceSelection(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceSelection</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;SequenceSelection(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                            <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)"><code>SequenceSelection(com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceIndex, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)</code></a> instead</i></div>
<div class="block">Construct for a selected SequenceAnnotation on the specified sequence. The intervals returned by <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals()"><code>getIntervals()</code></a>
 will correspond to the intervals of this annotation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence in which the annotation is selected, must not be null</dd><dd><code>annotation</code> - the selected annotation, must not be null</dd></dl>
</li>
</ul>
<a name="SequenceSelection(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceSelection</h4>
<pre>public&nbsp;SequenceSelection(org.jdom.Element&nbsp;element)
                  throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Construct a SequenceSelection from the results of <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#toXML()"><code>toXML()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - an element returned from <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#toXML()"><code>toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if XML does not conform to our unpublished specification</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="withoutZeroLengthIntervals()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>withoutZeroLengthIntervals</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection</a>&nbsp;withoutZeroLengthIntervals()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>an exact copy of this SequenceSelection but with any zero length SelectionIntervals removed.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="getAnnotation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotation</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;getAnnotation()</pre>
<div class="block">Get the selected annotation if there is one</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the selected annotation or <code>null</code> if none</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getAnnotationSequenceIndex()"><code>getAnnotationSequenceIndex()</code></a></dd></dl>
</li>
</ul>
<a name="getAnnotationSequenceIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationSequenceIndex</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&nbsp;getAnnotationSequenceIndex()</pre>
<div class="block">Get the sequence index of the selected annotation if there is an annotation</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sequence index of the selected annotation or <code>null</code> if no annotation</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - If this SequenceSelection was deserialized from XML Created in Geneious 5.5 or earlier and the annotation had no intervals.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getAnnotation()"><code>getAnnotation()</code></a></dd></dl>
</li>
</ul>
<a name="getIntervals()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIntervals</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;&nbsp;getIntervals()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals(boolean)"><code>getIntervals(boolean)</code></a> instead</i></div>
<div class="block">Equivalent to <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getIntervals(boolean)"><code>getIntervals(true)</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the intervals</dd></dl>
</li>
</ul>
<a name="getIntervals(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIntervals</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;&nbsp;getIntervals(boolean&nbsp;splitMultiSequenceIntervalsIntoSingleSelectionIntervals)</pre>
<div class="block">Gets the intervals selected</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>splitMultiSequenceIntervalsIntoSingleSelectionIntervals</code> - true to split rectangular SelectionIntervals (those that cover more than 1 sequence) into 1 SelectionInterval per sequence. When this is true , <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html#getSequenceIndex()"><code>SequenceSelection.SelectionInterval.getSequenceIndex()</code></a> may be called on the returned intervals.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the intervals selected, possibly including rectangular selections if splitMultiSequenceIntervalsIntoSingleSelectionIntervals is false (a range of sequences and a range of residues). Note that this may return empty intervals, for example to denote a cursor position. Using
 <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getNonZeroLengthIntervals()"><code>getNonZeroLengthIntervals()</code></a> will return a list guaranteed to correspond exactly to the nonempty selections that a user has made.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getNonZeroLengthIntervals()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNonZeroLengthIntervals</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SelectionInterval</a>&gt;&nbsp;getNonZeroLengthIntervals()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the (potentially ranged) intervals in this selection. Normally this will be a single element list unless the selection is
 discontiguous. The List of SelectionInterval that this method returns does not include any with zero length.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getSelectedSequenceCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSelectedSequenceCount</h4>
<pre>public&nbsp;int&nbsp;getSelectedSequenceCount()</pre>
<div class="block">Returns the number of sequences which have any residues selected (0 length selections are included). If the same sequence occurs in multiple SelectionIntervals it will be counted once.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences which have any residues selected (0 length selections are included)</dd></dl>
</li>
</ul>
<a name="getSelectedSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSelectedSequences</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSelectedSequences()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html#getSelectedSequenceIndices()"><code>getSelectedSequenceIndices()</code></a></i></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>set of sequences which have any residues selected (0 length selections are included)</dd></dl>
</li>
</ul>
<a name="getSelectedSequenceIndices()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSelectedSequenceIndices</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceIndex.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceIndex</a>&gt;&nbsp;getSelectedSequenceIndices()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>set of sequence indices which have any residues selected (0 length selections are included)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="extractIntervals()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractIntervals</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&gt;&nbsp;extractIntervals()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i></i></div>
<div class="block">Extract the selected intervals as individual sequence documents. If multiple intervals are selected in a single
 sequence document then each of the intervals is extracted separately.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the selected intervals as individual sequence documents</dd></dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="equals(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre>public&nbsp;boolean&nbsp;equals(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;o)</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="hashCode()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hashCode</h4>
<pre>public&nbsp;int&nbsp;hashCode()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="getTotalSelection(java.util.List, jebl.evolution.sequences.SequenceType, com.biomatters.geneious.publicapi.plugin.SequenceSelection.SequenceDocumentType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTotalSelection</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceSelection</a>&nbsp;getTotalSelection(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;documents,
                                  <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;type,
                                  <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html" title="enum in com.biomatters.geneious.publicapi.plugin">SequenceSelection.SequenceDocumentType</a>&nbsp;sequenceDocumentType)</pre>
<div class="block">For a List of AnnotatedPluginDocument, creates a selection that represents a "Select all" on each of the documents. This depends on the SequenceType and SequenceDocumentType passed in
 which determine whether to use the nucleotide or amino acid sequences from a List, and whether to use the stand-alone sequences or the aligned sequences from a document (which is ambiguous
 in the case of a CombinedAlignmentAndSequenceDocument). If <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html#Sequence"><code>SequenceSelection.SequenceDocumentType.Sequence</code></a> is provided for sequenceDocumentType, then the sequences
 from SequenceListDocument, SequenceDocument and the stand-alone sequence from a CombinedAlignmentAndSequenceDocument will be included, or else ignored. If <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html#Alignment"><code>SequenceSelection.SequenceDocumentType.Alignment</code></a>
 is provided, then the aligned sequences from a SequenceAlignmentDocument will be included, or else ignored.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>documents</code> - the documents to create a total selection for</dd><dd><code>type</code> - the type of sequences (nucleotide or amino-acid) to select in each document</dd><dd><code>sequenceDocumentType</code> - whether to use the stand-alone sequences or the aligned sequences in each document. If <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html#Sequence"><code>SequenceSelection.SequenceDocumentType.Sequence</code></a> is provided, then the sequences
 from SequenceListDocument, SequenceDocument and the stand-alone sequence from a CombinedAlignmentAndSequenceDocument will be included, or else ignored. If <a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SequenceDocumentType.html#Alignment"><code>SequenceSelection.SequenceDocumentType.Alignment</code></a>
 is provided, then the aligned sequences from a SequenceAlignmentDocument will be included, or else ignored.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the SequenceSelection representing the total selection.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;element)
             throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - representation from a previous call to <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceSelection.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraphFactory.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/plugin/SequenceSelection.SelectionInterval.html" title="class in com.biomatters.geneious.publicapi.plugin"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/plugin/SequenceSelection.html" target="_top">Frames</a></li>
<li><a href="SequenceSelection.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
