<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>DefaultSequenceListDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="DefaultSequenceListDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultSequenceListDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultNucleotideGraph.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultSequenceListDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Class DefaultSequenceListDocument" class="title">Class DefaultSequenceListDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">com.biomatters.geneious.publicapi.documents.AbstractPluginDocument</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">ExtendedEditableSequenceListDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializableWithProgress</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">DefaultSequenceListDocument</span>
extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializableWithProgress</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">ExtendedEditableSequenceListDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">A default implementation of a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a>.

 The constructors provided for this class allow for the quick creation of DefaultSequenceListDocuments that fit entirely into memory. For constructing larger
 SequenceListDocuments, see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a></div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#DefaultSequenceListDocument()">DefaultSequenceListDocument</a></strong>()</code>
<div class="block">XML deserialization constructor.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#addAminoAcidSequence(com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">addAminoAcidSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Adds the given amino acid sequence to the end of the list of amino acid sequences in this document
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#addAminoAcidSequence(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">addAminoAcidSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</code>
<div class="block">Adds an amino acid sequence to this list that is wrapped inside an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#addNucleotideSequence(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">addNucleotideSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</code>
<div class="block">Adds a nucleotide sequence to this list that is wrapped inside an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#addNucleotideSequence(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">addNucleotideSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Adds the given nucleotide sequence to the end of the list of nucleotide sequences in this document
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener)">createNewDocumentByTransformingSequences</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                                        <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a>&nbsp;transformer,
                                        <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener, boolean)"><code>createNewDocumentByTransformingSequences(alphabet, transformer, progressListener, true)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener, boolean)">createNewDocumentByTransformingSequences</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                                        <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a>&nbsp;transformer,
                                        <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener,
                                        boolean&nbsp;allowMultipleThreads)</code>
<div class="block">Creates a new DefaultSequenceListDocument (whose sequences will be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>) by applying a transformation to each of the sequences in this list.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forAminoAcidSequences(java.util.List)">forAminoAcidSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;aminoAcidSequences)</code>
<div class="block">Construct a new DefaultSequenceListDocument.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forBothSequenceTypes(java.util.List, jebl.util.ProgressListener)">forBothSequenceTypes</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences,
                    <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Construct a new DefaultSequenceListDocument.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forBothSequenceTypes(java.util.List, java.util.List)">forBothSequenceTypes</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;nucleotideSequences,
                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;aminoAcidSequences)</code>
<div class="block">Construct a new DefaultSequenceListDocument.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forBothSequenceTypes(java.util.List, java.util.List, jebl.util.ProgressListener)">forBothSequenceTypes</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;nucleotideSequences,
                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;aminoAcidSequences,
                    <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Construct a new DefaultSequenceListDocument.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forBothSequenceTypes(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)">forBothSequenceTypes</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>...&nbsp;sequences)</code>
<div class="block">Construct a new DefaultSequenceListDocument.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forNucleotideSequences(java.util.List)">forNucleotideSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;nucleotideSequences)</code>
<div class="block">Construct a new DefaultSequenceListDocument.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;root)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#fromXML(org.jdom.Element, jebl.util.ProgressListener)">fromXML</a></strong>(org.jdom.Element&nbsp;root,
       <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Deserializes this document from XML</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getAminoAcidSequences()">getAminoAcidSequences</a></strong>()</code>
<div class="block">Gets the amino acid sequences in this document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getAnnotatedPluginDocumentsForAminoAcidSequences()">getAnnotatedPluginDocumentsForAminoAcidSequences</a></strong>()</code>
<div class="block">When a SequneceList is created for some existing sequences wrapped in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>,
 then the original AnnotatedPluginDocuments are stored in the sequence list.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getAnnotatedPluginDocumentsForNucleotideSequences()">getAnnotatedPluginDocumentsForNucleotideSequences</a></strong>()</code>
<div class="block">When a SequneceList is created for some existing sequences wrapped in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>,
 then the original AnnotatedPluginDocuments are stored in the sequence list.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getDescription()">getDescription</a></strong>()</code>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></strong>()</code>
<div class="block">An immutable view of the paired reads managed by this list or alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getMateIndex(int)">getMateIndex</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getName()">getName</a></strong>()</code>
<div class="block">A short name (preferably one or two words) for the document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getNucleotideSequences()">getNucleotideSequences</a></strong>()</code>
<div class="block">Gets the nucleotide sequences in this document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getOrderingRevisionNumber()">getOrderingRevisionNumber</a></strong>()</code>
<div class="block">Return a revision number that is incremented each time sequences are reordered or removed from this sequence list.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getSizeRequiredToLoadIntoMemory()">getSizeRequiredToLoadIntoMemory</a></strong>()</code>
<div class="block">Return an estimate of the number of bytes required to load this document into memory.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#incrementOrderingRevisionNumber()">incrementOrderingRevisionNumber</a></strong>()</code>
<div class="block">Increments the value returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getOrderingRevisionNumber()"><code>getOrderingRevisionNumber()</code></a> to indicate that alignments that reference sequences by index in this list should discard their reference.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#removeAminoAcidSequence(int)">removeAminoAcidSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Removes the amino acid sequence at the given index from this sequence list and shifts any subsequent sequences to the left (subtracts one from their indices).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#removeNucleotideSequence(int)">removeNucleotideSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Removes the nucleotide sequence at the given index from this sequence list and shifts any subsequent sequences to the left (subtracts one from their indices).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#reorderAminoAcidSequences(int[])">reorderAminoAcidSequences</a></strong>(int[]&nbsp;ordering)</code>
<div class="block">Adjusts the ordering of sequences in this list according to the ordering array.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#reorderNucleotideSequences(int[])">reorderNucleotideSequences</a></strong>(int[]&nbsp;ordering)</code>
<div class="block">Adjusts the ordering of sequences in this list according to the ordering array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceAminoAcidSequence(int, com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">replaceAminoAcidSequence</a></strong>(int&nbsp;index,
                        <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&nbsp;newSequence)</code>
<div class="block">Replaces the amino acid sequence at this index with a new sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)">replaceContents</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;sequenceListToCopyContentsFrom)</code>
<div class="block">Copies all the sequence data from another DefaultSequenceListDocument into this document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceNucleotideSequence(int, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">replaceNucleotideSequence</a></strong>(int&nbsp;index,
                         <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;newSequence)</code>
<div class="block">Replaces the nucleotide sequence at this index with a new sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#setMates(int, int, int)">setMates</a></strong>(int&nbsp;sequenceIndex,
        int&nbsp;mateIndex,
        int&nbsp;expectedMateDistance)</code>
<div class="block">Sets 2 sequences as mates with the given expected mate distance.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#setName(java.lang.String)">setName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Sets the name of this document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#setPairedReadsManager(com.biomatters.geneious.publicapi.implementations.PairedReadManager)">setPairedReadsManager</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;pairedReadsManager)</code>
<div class="block">Sets a paired read manager to correspond with the nucleotide sequences for this document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#setUrn(com.biomatters.geneious.publicapi.documents.URN)">setUrn</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;urn)</code>
<div class="block">Set document URN.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#toHTML()">toHTML</a></strong>()</code>
<div class="block">Provide a textual view of document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#toXML(jebl.util.ProgressListener)">toXML</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.AbstractPluginDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addDisplayableField(com.biomatters.geneious.publicapi.documents.DocumentField)">addDisplayableField</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#addReferencedDocumentUrn(com.biomatters.geneious.publicapi.documents.URN)">addReferencedDocumentUrn</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#clearReferencedDocumentUrns()">clearReferencedDocumentUrns</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#createFromBaseElement(org.jdom.Element)">createFromBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getBaseElement()">getBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getBaseElementString()">getBaseElementString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getFieldValueCodes()">getFieldValueCodes</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getReferencedDocumentUrns()">getReferencedDocumentUrns</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getURN()">getURN</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#hasProperties()">hasProperties</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setBaseElement(org.jdom.Element)">setBaseElement</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setCreated(java.util.Date)">setCreated</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValue(com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object)">setFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValue(java.lang.String, java.lang.Object)">setFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setFieldValues(java.util.Map)">setFieldValues</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#setURN(com.biomatters.geneious.publicapi.documents.URN)">setURN</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="DefaultSequenceListDocument()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DefaultSequenceListDocument</h4>
<pre>public&nbsp;DefaultSequenceListDocument()</pre>
<div class="block">XML deserialization constructor. To construct a new DefaultSequenceListDocument, use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forAminoAcidSequences(java.util.List)"><code>forAminoAcidSequences(java.util.List)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forNucleotideSequences(java.util.List)"><code>forNucleotideSequences(java.util.List)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#forBothSequenceTypes(java.util.List, java.util.List)"><code>forBothSequenceTypes(java.util.List, java.util.List)</code></a>.</div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="forBothSequenceTypes(java.util.List, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forBothSequenceTypes</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;forBothSequenceTypes(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;nucleotideSequences,
                                               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;aminoAcidSequences)</pre>
<div class="block">Construct a new DefaultSequenceListDocument. For constructing DefaultSequenceListDocument where all the sequences don't fit into memory, see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideSequences</code> - the nucleotide sequences that the created document will contain. This list may be empty but must not be null.</dd><dd><code>aminoAcidSequences</code> - the amino acid sequences that the created document will contain. This list may be empty but must not be null.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the newly created document</dd></dl>
</li>
</ul>
<a name="forBothSequenceTypes(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forBothSequenceTypes</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;forBothSequenceTypes(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>...&nbsp;sequences)</pre>
<div class="block">Construct a new DefaultSequenceListDocument. For constructing DefaultSequenceListDocument where all the sequences don't fit into memory, see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - all sequences, can be a mixture of nucleotide sequences and amino acid sequences.
 This method will automatically put them in the right list.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the newly created document</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.911 (Geneious 9.1.1)</dd></dl>
</li>
</ul>
<a name="forBothSequenceTypes(java.util.List, java.util.List, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forBothSequenceTypes</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;forBothSequenceTypes(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;nucleotideSequences,
                                               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;aminoAcidSequences,
                                               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block">Construct a new DefaultSequenceListDocument. For constructing DefaultSequenceListDocument where all the sequences don't fit into memory, see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideSequences</code> - the nucleotide sequences that the created document will contain. This list may be empty but must not be null.</dd><dd><code>aminoAcidSequences</code> - the amino acid sequences that the created document will contain. This list may be empty but must not be null.</dd><dd><code>progressListener</code> - for reporting progress and cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the newly created document</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.13 (Geneious 5.0.2)</dd></dl>
</li>
</ul>
<a name="forBothSequenceTypes(java.util.List, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forBothSequenceTypes</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;forBothSequenceTypes(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences,
                                               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block">Construct a new DefaultSequenceListDocument. For constructing DefaultSequenceListDocument where all the sequences don't fit into memory, see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - all sequences, can be a mixture of nucleotide sequences and amino acid sequences.
 This method will automatically put them in the right list.</dd><dd><code>progressListener</code> - for reporting progress and cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the newly created document</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.13 (Geneious 5.0.2)</dd></dl>
</li>
</ul>
<a name="forNucleotideSequences(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forNucleotideSequences</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;forNucleotideSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;nucleotideSequences)</pre>
<div class="block">Construct a new DefaultSequenceListDocument. For constructing DefaultSequenceListDocument where all the sequences don't fit into memory, see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotideSequences</code> - the nucleotide sequences that the created document will contain. This list may be empty but must not be null.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the newly created document</dd></dl>
</li>
</ul>
<a name="forAminoAcidSequences(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>forAminoAcidSequences</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;forAminoAcidSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;aminoAcidSequences)</pre>
<div class="block">Construct a new DefaultSequenceListDocument. For constructing DefaultSequenceListDocument where all the sequences don't fit into memory, see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>aminoAcidSequences</code> - the amino acid sequences that the created document will contain. This list may be empty but must not be null.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the newly created document</dd></dl>
</li>
</ul>
<a name="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getName()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">PluginDocument</a></code></strong></div>
<div class="block">A short name (preferably one or two words) for the document.
 <p/>
 If this PluginDocument is wrapped inside an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a> then this method
 will be overridden and delegate back to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getName()"><code>AnnotatedPluginDocument.getName()</code></a>. One way to look at it is
 that when you call PluginDocument.getName() you will get the current name of the document. If you are implementing PluginDocument.getName() then you are
 returning the default name for the document.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>document name. Cannot be null but may be an empty string.</dd></dl>
</li>
</ul>
<a name="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre>public&nbsp;void&nbsp;setName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block">Sets the name of this document. If the name is not set it will be given a default name such as "5,000 nucleotide sequences".</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html#setName(java.lang.String)">setName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - the name for this document</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getName()"><code>getName()</code></a></dd></dl>
</li>
</ul>
<a name="getDescription()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">PluginDocument</a></code></strong></div>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a one line simple text description of the document, or null if no description is available.</dd></dl>
</li>
</ul>
<a name="toHTML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toHTML</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toHTML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">PluginDocument</a></code></strong></div>
<div class="block">Provide a textual view of document. The contents should be at most 30K-60K characters. One reason is that this
 text is rendered using JTextPane, which is horrendously slow when handling large text. The more fundamental
 reason is that large amounts of text are generally not useful for a user. The whole idea of using a tool such as
 Geneious is to provide specialized viewers which make the large amount of bioinformatics data perceivable and
 comprehensible to a human user.
 <p/>
 Return an HTML body fragment (i.e. a fragment which sits inside a &lt;body&gt;&lt;/body&gt; tag). (Header,stylesheet and
 annotations are automatically added by GENEious code).
 <p/>
 Can optionally return an entire html tag set, as long as the first six characters are <code>&lt;html></code></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an HTML fragment of tag set, or null to have no text view.</dd></dl>
</li>
</ul>
<a name="getNucleotideSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNucleotideSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;getNucleotideSequences()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getNucleotideSequences()">SequenceListDocument</a></code></strong></div>
<div class="block">Gets the nucleotide sequences in this document. Client code should not attempt to make any changes to the returned list
 or any of the sequences in the returned list even if the sequences are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>EditableSequenceDocuments</code></a>.
 <p/>
 To add/delete/reorder sequences use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>EditableSequenceListDocument</code></a>.
 To modify individual sequences in the list, use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html#replaceNucleotideSequence(int, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)"><code>ExtendedEditableSequenceListDocument.replaceNucleotideSequence(int, NucleotideSequenceDocument)</code></a>
 
 <p/>Note that using the editing methods will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getNucleotideSequences()">getNucleotideSequences</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the nucleotide sequences contained in this document. This list may be empty but will not be null.</dd></dl>
</li>
</ul>
<a name="getAminoAcidSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAminoAcidSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;getAminoAcidSequences()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getAminoAcidSequences()">SequenceListDocument</a></code></strong></div>
<div class="block">Gets the amino acid sequences in this document. Client code should not attempt to make any changes to the returned list
 or any of the sequences in the returned list even if the sequences are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>EditableSequenceDocuments</code></a>.
 <p/>
 To add/delete/reorder sequences use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>EditableSequenceListDocument</code></a>.
 To modify individual sequences in the list, use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html#replaceAminoAcidSequence(int, com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)"><code>ExtendedEditableSequenceListDocument.replaceAminoAcidSequence(int, AminoAcidSequenceDocument)</code></a></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getAminoAcidSequences()">getAminoAcidSequences</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the amino acid sequences contained in this document. This list may be empty but will not be null.</dd></dl>
</li>
</ul>
<a name="getAnnotatedPluginDocumentsForNucleotideSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotatedPluginDocumentsForNucleotideSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;getAnnotatedPluginDocumentsForNucleotideSequences()</pre>
<div class="block">When a SequneceList is created for some existing sequences wrapped in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>,
 then the original AnnotatedPluginDocuments are stored in the sequence list.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the AnnotatedPluginDocuments for the nucleotide sequences (in which case the length of this list will be the same as the length of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getNucleotideSequences()"><code>getNucleotideSequences()</code></a> but some or all entries in the list may be null) or potentially a null list if none of the nucleotide sequences have associated AnnotatedPluginDocuments.</dd></dl>
</li>
</ul>
<a name="getAnnotatedPluginDocumentsForAminoAcidSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotatedPluginDocumentsForAminoAcidSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;getAnnotatedPluginDocumentsForAminoAcidSequences()</pre>
<div class="block">When a SequneceList is created for some existing sequences wrapped in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>,
 then the original AnnotatedPluginDocuments are stored in the sequence list.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the AnnotatedPluginDocuments for the amino acid sequences (in which case the length of this list will be the same as the length of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getAminoAcidSequences()"><code>getAminoAcidSequences()</code></a> ()} but some or all entries in the list may be null) or potentially null if none of the amino acid sequences have associated AnnotatedPluginDocuments.</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#toXML()">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="toXML(jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html#toXML(jebl.util.ProgressListener)">XMLSerializableWithProgress</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that the constructor can completely
 restore the object's representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html#toXML(jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializableWithProgress</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>progressListener</code> - for reporting progress of the serialization. The implementation may return null if the progress listener requests that it cancel.</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;root)
             throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>root</code> - representation from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;root,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
             throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Deserializes this document from XML</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>root</code> - the element returned from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#toXML()"><code>toXML()</code></a></dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if it cannot be deserialized.</dd></dl>
</li>
</ul>
<a name="setUrn(com.biomatters.geneious.publicapi.documents.URN)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setUrn</h4>
<pre>public&nbsp;void&nbsp;setUrn(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;urn)</pre>
<div class="block">Set document URN. In almost all cases there is no need to assign a URN.
 Assigning a URN to a PluginDocument is only required when creating an association between a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/SummaryDocument.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>SummaryDocument</code></a> and the full downloaded document (e.g. DocumentLists derived from WGS with multiple scaffolds)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>urn</code> - to set</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.905 (Geneious 9.0.5)</dd></dl>
</li>
</ul>
<a name="incrementOrderingRevisionNumber()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>incrementOrderingRevisionNumber</h4>
<pre>public&nbsp;void&nbsp;incrementOrderingRevisionNumber()</pre>
<div class="block">Increments the value returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getOrderingRevisionNumber()"><code>getOrderingRevisionNumber()</code></a> to indicate that alignments that reference sequences by index in this list should discard their reference.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.715 (Geneious 7.1.5)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getOrderingRevisionNumber()"><code>getOrderingRevisionNumber()</code></a></dd></dl>
</li>
</ul>
<a name="reorderNucleotideSequences(int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderNucleotideSequences</h4>
<pre>public&nbsp;void&nbsp;reorderNucleotideSequences(int[]&nbsp;ordering)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#reorderNucleotideSequences(int[])">EditableSequenceListDocument</a></code></strong></div>
<div class="block">Adjusts the ordering of sequences in this list according to the ordering array.
 If ordering[i]==j, then the sequence at index 'i' should be the sequence previously at index 'j'
 For example, if ordering = {0,1,3,4,2}
 Then:
 sequence at index 2 is moved to index 4
 sequence at index 3 is moved to index 2
 sequence at index 4 is moved to index 3
 <p/>
 This behaviour is identical to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a>
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#reorderNucleotideSequences(int[])">reorderNucleotideSequences</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>ordering</code> - Each integer between 0 and ordering.length - 1 inclusive should occur exactly once in this array.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#reorderAminoAcidSequences(int[])"><code>EditableSequenceListDocument.reorderAminoAcidSequences(int[])</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a></dd></dl>
</li>
</ul>
<a name="reorderAminoAcidSequences(int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderAminoAcidSequences</h4>
<pre>public&nbsp;void&nbsp;reorderAminoAcidSequences(int[]&nbsp;ordering)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#reorderAminoAcidSequences(int[])">EditableSequenceListDocument</a></code></strong></div>
<div class="block">Adjusts the ordering of sequences in this list according to the ordering array.
 If ordering[i]==j, then the sequence at index 'i' should be the sequence previously at index 'j'
 For example, if ordering = {0,1,3,4,2}
 Then:
 sequence at index 2 is moved to index 4
 sequence at index 3 is moved to index 2
 sequence at index 4 is moved to index 3
 <p/>
 This behaviour is identical to <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#reorderSequences(int[])"><code>DefaultAlignmentDocument.reorderSequences(int[])</code></a> 
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#reorderAminoAcidSequences(int[])">reorderAminoAcidSequences</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>ordering</code> - Each integer between 0 and ordering.length - 1 inclusive should occur exactly once in this array.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#reorderNucleotideSequences(int[])"><code>EditableSequenceListDocument.reorderNucleotideSequences(int[])</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a></dd></dl>
</li>
</ul>
<a name="removeNucleotideSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeNucleotideSequence</h4>
<pre>public&nbsp;void&nbsp;removeNucleotideSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#removeNucleotideSequence(int)">EditableSequenceListDocument</a></code></strong></div>
<div class="block">Removes the nucleotide sequence at the given index from this sequence list and shifts any subsequent sequences to the left (subtracts one from their indices).
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#removeNucleotideSequence(int)">removeNucleotideSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the nucleotide sequence to remove.</dd></dl>
</li>
</ul>
<a name="removeAminoAcidSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeAminoAcidSequence</h4>
<pre>public&nbsp;void&nbsp;removeAminoAcidSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#removeAminoAcidSequence(int)">EditableSequenceListDocument</a></code></strong></div>
<div class="block">Removes the amino acid sequence at the given index from this sequence list and shifts any subsequent sequences to the left (subtracts one from their indices).
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#removeAminoAcidSequence(int)">removeAminoAcidSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the amino acid sequence to remove.</dd></dl>
</li>
</ul>
<a name="replaceNucleotideSequence(int, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>replaceNucleotideSequence</h4>
<pre>public&nbsp;void&nbsp;replaceNucleotideSequence(int&nbsp;index,
                             <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;newSequence)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html#replaceNucleotideSequence(int, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">ExtendedEditableSequenceListDocument</a></code></strong></div>
<div class="block">Replaces the nucleotide sequence at this index with a new sequence.
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html#replaceNucleotideSequence(int, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">replaceNucleotideSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">ExtendedEditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the 0-based index of the sequence to replace</dd><dd><code>newSequence</code> - the new sequence to replace it with</dd></dl>
</li>
</ul>
<a name="replaceAminoAcidSequence(int, com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>replaceAminoAcidSequence</h4>
<pre>public&nbsp;void&nbsp;replaceAminoAcidSequence(int&nbsp;index,
                            <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&nbsp;newSequence)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html#replaceAminoAcidSequence(int, com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">ExtendedEditableSequenceListDocument</a></code></strong></div>
<div class="block">Replaces the amino acid sequence at this index with a new sequence.
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html#replaceAminoAcidSequence(int, com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">replaceAminoAcidSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ExtendedEditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">ExtendedEditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the 0-based index of the sequence to replace</dd><dd><code>newSequence</code> - the new sequence to replace it with</dd></dl>
</li>
</ul>
<a name="addNucleotideSequence(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addNucleotideSequence</h4>
<pre>public&nbsp;void&nbsp;addNucleotideSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)
                           throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Adds a nucleotide sequence to this list that is wrapped inside an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - the document containing the sequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if the sequence inside the document cannot be loaded</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - of the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>PluginDocument</code></a> inside document is not a nucleotide sequence.</dd></dl>
</li>
</ul>
<a name="addNucleotideSequence(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addNucleotideSequence</h4>
<pre>public&nbsp;void&nbsp;addNucleotideSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#addNucleotideSequence(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">EditableSequenceListDocument</a></code></strong></div>
<div class="block">Adds the given nucleotide sequence to the end of the list of nucleotide sequences in this document
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#addNucleotideSequence(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)">addNucleotideSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the nucleotide sequence to add</dd></dl>
</li>
</ul>
<a name="addAminoAcidSequence(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addAminoAcidSequence</h4>
<pre>public&nbsp;void&nbsp;addAminoAcidSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)
                          throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Adds an amino acid sequence to this list that is wrapped inside an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - the document containing the sequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if the sequence inside the document cannot be loaded</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - of the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>PluginDocument</code></a> inside document is not an amino acid sequence.</dd></dl>
</li>
</ul>
<a name="addAminoAcidSequence(com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addAminoAcidSequence</h4>
<pre>public&nbsp;void&nbsp;addAminoAcidSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#addAminoAcidSequence(com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">EditableSequenceListDocument</a></code></strong></div>
<div class="block">Adds the given amino acid sequence to the end of the list of amino acid sequences in this document
 <p/>Note that using this method will require loading the entire sequence list into memory, which
 might not be possible on large sequence lists. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>DefaultSequenceListDocument.replaceContents</code></a> for a discussion on efficiently
 handling editing large sequence lists.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#addAminoAcidSequence(com.biomatters.geneious.publicapi.documents.sequence.AminoAcidSequenceDocument)">addAminoAcidSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the amino acid sequence to add</dd></dl>
</li>
</ul>
<a name="getMateExpectedDistance(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateExpectedDistance</h4>
<pre>public&nbsp;int&nbsp;getMateExpectedDistance(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate. This is defined
 as the length of the 2 reads plus the distance between them.
 For example if the 2 reads both have length 30 with a 40 bp gap between them, then the
 expected mate distance is 100.
 If the mate is expected to appear after this sequence with this sequence in its current orientation, then
 the expected distance will be positive. If the mate is expected to appear before this sequence with this sequence in its current orientation, then
 the expected distance will be negative.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence for which to get the mate index of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the expected mate distance or 0 if this read is not paired. Note that a read may still have a mate distance!=0 but its mate may not be present in this document (in which case <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a> will be -1.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a></dd></dl>
</li>
</ul>
<a name="getMateIndex(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateIndex</h4>
<pre>public&nbsp;int&nbsp;getMateIndex(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">getMateIndex</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence to get the mate index of</dd>
<dt><span class="strong">Returns:</span></dt><dd>an index of the mate of the sequence, or -1 if the sequence has no mate.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>PairedReads.getMateExpectedDistance(int)</code></a></dd></dl>
</li>
</ul>
<a name="setMates(int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMates</h4>
<pre>public&nbsp;void&nbsp;setMates(int&nbsp;sequenceIndex,
            int&nbsp;mateIndex,
            int&nbsp;expectedMateDistance)</pre>
<div class="block">Sets 2 sequences as mates with the given expected mate distance.
 sequenceIndex will be given a mate distance of expectedMateDistance.
 If these sequences were not already paired, then mateIndex will be assigned a mate distance of -expectedMateDistance. 
 If the mate sequence is already paired with this sequence, then this method does not change the distance recorded for the mate.

 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a> for what the expected distance means.
 If all mates are set interlaced (i.e 0 is paired with 1, 2 is paired with 3 etc) then
 the implementation is more memory efficient.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the sequence for which to set the mate distance for.</dd><dd><code>mateIndex</code> - the index of the mate of sequenceIndex. May be -1 to unpair this sequence from its mate</dd><dd><code>expectedMateDistance</code> - the expected distance to its mate</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>PairedReads.getMateExpectedDistance(int)</code></a></dd></dl>
</li>
</ul>
<a name="setPairedReadsManager(com.biomatters.geneious.publicapi.implementations.PairedReadManager)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setPairedReadsManager</h4>
<pre>public&nbsp;void&nbsp;setPairedReadsManager(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;pairedReadsManager)</pre>
<div class="block">Sets a paired read manager to correspond with the nucleotide sequences for this document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>pairedReadsManager</code> - the new paired read manager, or null if the nucleotide sequences are not paired.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getImmutablePairedReadManager()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getImmutablePairedReadManager</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;getImmutablePairedReadManager()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">PairedReads</a></code></strong></div>
<div class="block">An immutable view of the paired reads managed by this list or alignment. Even if the list or alignment
 that implements this method is later modified, the
 returned PairedReadManager will not change.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>An immutable view of the paired reads managed by the list or alignment that implements this method or null if this class doesn't provide an immutable view of its paired reads.</dd></dl>
</li>
</ul>
<a name="createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createNewDocumentByTransformingSequences</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;createNewDocumentByTransformingSequences(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                                                                   <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a>&nbsp;transformer,
                                                                   <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                                                     throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener, boolean)"><code>createNewDocumentByTransformingSequences(alphabet, transformer, progressListener, true)</code></a>.</div>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createNewDocumentByTransformingSequences</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;createNewDocumentByTransformingSequences(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                                                                   <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a>&nbsp;transformer,
                                                                   <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener,
                                                                   boolean&nbsp;allowMultipleThreads)
                                                                     throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Creates a new DefaultSequenceListDocument (whose sequences will be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>) by applying a transformation to each of the sequences in this list.
 This method is useful in conjunction with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>replaceContents(DefaultSequenceListDocument)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alphabet</code> - specifies whether to transform the nucleotide or amino acid sequences in this document. If alphabet is null, both types of sequences will be transformed.</dd><dd><code>transformer</code> - the transformer</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd><dd><code>allowMultipleThreads</code> - true if the transformer can be invoked from multiple threads simultaneously to improve performance. In this case, the transformer must be prepared to accept the sequences to be transformed in any order.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new DefaultSequenceListDocument containing all the transformed sequences. Will not be null</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if the transformation cannot be completed for any reason</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.711 (Geneious 7.1.1)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)"><code>replaceContents(DefaultSequenceListDocument)</code></a></dd></dl>
</li>
</ul>
<a name="replaceContents(com.biomatters.geneious.publicapi.documents.sequence.DefaultSequenceListDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>replaceContents</h4>
<pre>public&nbsp;void&nbsp;replaceContents(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;sequenceListToCopyContentsFrom)</pre>
<div class="block">Copies all the sequence data from another DefaultSequenceListDocument into this document. The name of this document is not copied. Lists
 returned from previous calls to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getNucleotideSequences()"><code>getNucleotideSequences()</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#getAminoAcidSequences()"><code>getAminoAcidSequences()</code></a> remain valid
 until this document is saved. This method is useful in conjunction with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener)"><code>createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener)</code></a>.
 <p/>
 Sequence lists can contain a lot of sequences (e.g. over 100 million sequences). In these situations, the lists returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getNucleotideSequences()"><code>SequenceListDocument.getNucleotideSequences()</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getAminoAcidSequences()"><code>SequenceListDocument.getAminoAcidSequences()</code></a> may be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>. SequenceListOnDisks are immutable, so calling methods such as
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#addNucleotideSequence(com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)"><code>addNucleotideSequence(NucleotideSequenceDocument)</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#removeNucleotideSequence(int)"><code>removeNucleotideSequence(int)</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#replaceNucleotideSequence(int, com.biomatters.geneious.publicapi.documents.sequence.NucleotideSequenceDocument)"><code>replaceNucleotideSequence(int, NucleotideSequenceDocument)</code></a>
 on the SequenceListDocument that contains the SequenceListOnDisk will not
 be efficient and may require loading the entire list into memory, which may not be possible. Instead it is best to either create a new SequenceListDocument (using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a>),
 and/or replace the contents of a SequenceListDocument in a single step using DefaultSequenceListDocument.replaceContents.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceListToCopyContentsFrom</code> - the sequence list to copy data from</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener)"><code>createNewDocumentByTransformingSequences(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Transformer, jebl.util.ProgressListener)</code></a></dd></dl>
</li>
</ul>
<a name="getSizeRequiredToLoadIntoMemory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSizeRequiredToLoadIntoMemory</h4>
<pre>public&nbsp;long&nbsp;getSizeRequiredToLoadIntoMemory()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html#getSizeRequiredToLoadIntoMemory()">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></strong></div>
<div class="block">Return an estimate of the number of bytes required to load this document into memory. Or 0 if this document is unable to make an estimate in which case Geneious will use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getSize()"><code>AnnotatedPluginDocument.getSize()</code></a>.
 Geneious caches the value returned from this method when the document is wrapped in an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a> or when it is saved.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html#getSizeRequiredToLoadIntoMemory()">getSizeRequiredToLoadIntoMemory</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an estimate of the number of bytes required to load this document into memory. Or 0 if this document is unable to make an estimate.</dd></dl>
</li>
</ul>
<a name="getOrderingRevisionNumber()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getOrderingRevisionNumber</h4>
<pre>public&nbsp;int&nbsp;getOrderingRevisionNumber()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#getOrderingRevisionNumber()">EditableSequenceListDocument</a></code></strong></div>
<div class="block">Return a revision number that is incremented each time sequences are reordered or removed from this sequence list. Alignments or other documents
 that store references back to individual sequences in this list should record the ordering revision number at the time they begin referencing
 this document. If at a later date this ordering revision number has changed they should discard the reference or they could end up referencing the wrong sequence.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html#getOrderingRevisionNumber()">getOrderingRevisionNumber</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceListDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a revision number that is incremented each time sequences are reordered or removed from this sequence list.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#SEQUENCE_LIST_ORDERING_REVISION_NUMBER"><code>DocumentField.SEQUENCE_LIST_ORDERING_REVISION_NUMBER</code></a></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultSequenceListDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultNucleotideGraph.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultSequenceListDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
