<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>ImmutableSequence (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ImmutableSequence (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ImmutableSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" target="_top">Frames</a></li>
<li><a href="ImmutableSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Class ImmutableSequence" class="title">Class ImmutableSequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.documents.sequence.ImmutableSequence</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Comparable.html?is-external=true" title="class or interface in java.lang">Comparable</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&gt;</dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequenceImplementation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequenceImplementation</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="strong">ImmutableSequence</span>
extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></pre>
<div class="block">An implementation of a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a> that is immutable
 and has very low memory usage. Generally it is only advantageous to use ImmutableSequences when dealing with lots
 of sequences. ImmutableSequences have a very low memory usage overhead per sequence, and lots of ImmutableSequences
 constructed with an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence.Builder</code></a>
 that have names which have a common pattern will have low memory usage for storing their names.
 <p/>
 As of Geneious 5.0, this class is unable to store annotations and sequence descriptions but these may be supported in future versions.
 Data that is currently supported by this class includes
 <ul>
 <li>Sequence names</li>
 <li>Sequence bases/residues</li>
 <li>Leading/trailing gap lengths</li>
 <li>Sequence quality (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>), but not chromatogram trace data)
 </ul>
 To construct an instance of an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a>, use
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#createImmutableSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)"><code>createImmutableSequence(SequenceDocument, boolean)</code></a>
 or to achieve even lower memory usage when constructing many sequences which have a common pattern to their names, use
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence.Builder</code></a></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence.Builder</a></strong></code>
<div class="block">Provides a system for building a list of ImmutableSequence which are even more compact than normal by analyzing patterns
 in the names of sequences.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#EMPTY">EMPTY</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD">GENOME_SEQUENCE_THRESHOLD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#asDna()">asDna</a></strong>()</code>
<div class="block">Creates a new sequence that is identical to this sequence except with U replaced with T</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#asRna()">asRna</a></strong>()</code>
<div class="block">Creates a new sequence that is identical to this sequence except with T replaced with U</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#compactAminoAcidSequences(java.util.List, jebl.util.ProgressListener)">compactAminoAcidSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;sequences,
                         <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Attempts to compact the memory usage of as many sequences in the given list of sequences as possible
 by replacing them with ImmutableSequence instances.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#compactNucleotideSequences(java.util.List, jebl.util.ProgressListener)">compactNucleotideSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;sequences,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Attempts to compact the memory usage of as many sequences in the given list of sequences as possible
 by replacing them with ImmutableSequence instances.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#createCopyWithNameSuffix(java.lang.String)">createCopyWithNameSuffix</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;nameSuffix)</code>
<div class="block">Creates a new sequence identical to this once except with a different name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#createCopyWithNewName(java.lang.String)">createCopyWithNewName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName)</code>
<div class="block">Creates a new sequence identical to this once except with a different name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#createImmutableSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">createImmutableSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Creates an ImmutableSequence copy of another sequence if possible.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#createImmutableSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">createImmutableSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                       boolean&nbsp;failIfContainsNonStorableData)</code>
<div class="block">Creates an ImmutableSequence copy of another sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#delete(int, int)">delete</a></strong>(int&nbsp;deletionBegin,
      int&nbsp;deletionEnd)</code>
<div class="block">Returns a copy of this SequenceCharSequence where the characters at positions deletionBegin inclusive
 to deletionEnd exclusive have been removed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getCharAndQualityAtIgnoringEndGaps(int)">getCharAndQualityAtIgnoringEndGaps</a></strong>(int&nbsp;characterIndex)</code>
<div class="block">Provides both the sequence character and the quality in a single method call, which is almost twice
 as fast as requesting the 2 values separately.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getData(int, int, byte[], byte[], int)">getData</a></strong>(int&nbsp;internalStartIndex,
       int&nbsp;sizeOfDataToGet,
       byte[]&nbsp;characters,
       byte[]&nbsp;qualities,
       int&nbsp;startingOffsetIntoDataArrays)</code>
<div class="block">Provides sequence character and qualities for multiple positions in a single method call,
 which is much faster than requesting the values separately.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getLeadingTrimLength()">getLeadingTrimLength</a></strong>()</code>
<div class="block">Gets the number of bases and non-end gaps at the start of this sequence flagged as trimmed (equivalent to annotations of type <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>SequenceAnnotation.TYPE_TRIMMED</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getMappingQuality()">getMappingQuality</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getNumberOfIntegersOfStorage()">getNumberOfIntegersOfStorage</a></strong>()</code>
<div class="block">Returns the number of integers needed to store this document on disk and/or load it into memory</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getNumberOfMappedLocations()">getNumberOfMappedLocations</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getTrailingTrimLength()">getTrailingTrimLength</a></strong>()</code>
<div class="block">Gets the number of bases and non-end gaps at the end of this sequence flagged as trimmed (equivalent to annotations of type <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>SequenceAnnotation.TYPE_TRIMMED</code></a>)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Implements <code>XMLSerializable.OldVersionCompatible#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code>
 but note that SequenceCharSequence is guaranteed to be able to always serialize to <code>Geneious.MajorVersion#Version6_0</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#immutableSequenceFromXML(org.jdom.Element)">immutableSequenceFromXML</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Recreates this sequence from XML returned from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#toXML(boolean)"><code>toXML(boolean)</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#insert(int, java.lang.CharSequence)">insert</a></strong>(int&nbsp;index,
      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</code>
<div class="block">Creates a copy of this SequenceCharSequence with the specified CharSequence inserted
 at the specified index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#insertGaps(int, int, int[], int[])">insertGaps</a></strong>(int&nbsp;leadingGapsLength,
          int&nbsp;trailingGapsLength,
          int[]&nbsp;internalGapPositions,
          int[]&nbsp;internalGapSizes)</code>
<div class="block">Creates a new sequence identical to this one except with some gaps inserted.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#isReversed()">isReversed</a></strong>()</code>
<div class="block">Determine if this sequence is reversed (i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">readObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)</code>
<div class="block">Reads a instance of an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a> that
 was previously written using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)"><code>SequenceCharSequence.writeObject(GeneiousObjectOutputStream, jebl.util.ProgressListener)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#removeGaps()">removeGaps</a></strong>()</code>
<div class="block">Creates a new sequence identical to this once except all gaps (both end and internal) have been removed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#reverse()">reverse</a></strong>()</code>
<div class="block">Create a new sequence that is the reverse  of this sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#reverseComplement()">reverseComplement</a></strong>()</code>
<div class="block">Create a new sequence that is the reverse complement of this sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#subString(int, int)">subString</a></strong>(int&nbsp;startIndex,
         int&nbsp;endIndex)</code>
<div class="block">Returns the characters between the 2 given indices as a string</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#toXML(boolean)">toXML</a></strong>(boolean&nbsp;allowFileData)</code>
<div class="block">Serializes this SequenceCharSequence to XML into an element named
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener, boolean)">toXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener,
     boolean&nbsp;allowFileData)</code>
<div class="block">Serializes this SequenceCharSequence to XML into an element named
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#withMappingQuality(int)">withMappingQuality</a></strong>(int&nbsp;mappingQuality)</code>
<div class="block">Creates a new sequence identical to this one, except with a different value for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getMappingQuality()"><code>getMappingQuality()</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#withNumberOfMappedLocations(int)">withNumberOfMappedLocations</a></strong>(int&nbsp;numberOfMappedLocations)</code>
<div class="block">Creates a new sequence identical to this one, except with a different value for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getNumberOfMappedLocations()"><code>getNumberOfMappedLocations()</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#withTrimLengths(int, int)">withTrimLengths</a></strong>(int&nbsp;leadingTrimLength,
               int&nbsp;trailingTrimLength)</code>
<div class="block">Creates a new sequence identical to this one, except with different leading and trailing trim lengths.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Writes this sequence to a DataOutput.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
           <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Writes this sequence to a DataOutput.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#charAt(int)">charAt</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#charAtIgnoringEndGaps(int)">charAtIgnoringEndGaps</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#compareTo(com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence)">compareTo</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#contains(char)">contains</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#containsInvalidResidues(jebl.evolution.sequences.SequenceType, boolean)">containsInvalidResidues</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#count(char)">count</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#countGaps(int, int)">countGaps</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#equals(java.lang.Object)">equals</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#equalsIgnoreCase(com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence)">equalsIgnoreCase</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#fromXml(org.jdom.Element)">fromXml</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#fromXML(org.jdom.Element)">fromXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getInternalCharSequence()">getInternalCharSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getInternalSequenceLength()">getInternalSequenceLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getLeadingGapsLength()">getLeadingGapsLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getTrailingGapsLength()">getTrailingGapsLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getTrailingGapsStartIndex()">getTrailingGapsStartIndex</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getUngappedLength()">getUngappedLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#hashCode()">hashCode</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#indexOf(java.lang.CharSequence)">indexOf</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#indexOf(java.lang.CharSequence, int)">indexOf</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#isAllGaps()">isAllGaps</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#isEndGap(int)">isEndGap</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#isGap(int)">isGap</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#readObject(java.io.DataInput)">readObject</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#subSequence(int, int)">subSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#toString()">toString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#toXML()">toXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#valueOf(java.lang.CharSequence)">valueOf</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#withOnlyGaps(int)">withOnlyGaps</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#withTerminalGaps(int, java.lang.CharSequence, int)">withTerminalGaps</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)">writeObject</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">getCharSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">getSequenceLength</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">getSequenceString</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()">isCircular</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.CharSequence">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang">length</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="createImmutableSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createImmutableSequence</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;createImmutableSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                                        boolean&nbsp;failIfContainsNonStorableData)</pre>
<div class="block">Creates an ImmutableSequence copy of another sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to create a copy of</dd><dd><code>failIfContainsNonStorableData</code> - if this is true, and the sequence to copy contains data that cannot be stored in an ImmutableSequence (for example annotations)
 then this method will throw an <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang"><code>IllegalArgumentException</code></a>. If this is false and the sequence to copy contains data that cannot be stored in an ImmutableSequence then a sequence with that information missing is returned</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if failIfContainsNonStorableData is true and the seuqnece to copy contains data that cannot be stored in an ImmutableSequence (for example annotations)</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#createImmutableSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>createImmutableSequence(SequenceDocument)</code></a></dd></dl>
</li>
</ul>
<a name="createImmutableSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createImmutableSequence</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;createImmutableSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Creates an ImmutableSequence copy of another sequence if possible.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to create a copy of</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence or null if ths source sequence contains information not storeable in an ImmutableSequence (e.g. annotations)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#createImmutableSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)"><code>createImmutableSequence(SequenceDocument, boolean)</code></a></dd></dl>
</li>
</ul>
<a name="getLeadingTrimLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLeadingTrimLength</h4>
<pre>public abstract&nbsp;int&nbsp;getLeadingTrimLength()</pre>
<div class="block">Gets the number of bases and non-end gaps at the start of this sequence flagged as trimmed (equivalent to annotations of type <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>SequenceAnnotation.TYPE_TRIMMED</code></a>).</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of bases and non-end gaps at the start of this sequence flagged as trimmed</dd></dl>
</li>
</ul>
<a name="getTrailingTrimLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrailingTrimLength</h4>
<pre>public abstract&nbsp;int&nbsp;getTrailingTrimLength()</pre>
<div class="block">Gets the number of bases and non-end gaps at the end of this sequence flagged as trimmed (equivalent to annotations of type <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>SequenceAnnotation.TYPE_TRIMMED</code></a>)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of bases and non-end gaps at the end of this sequence flagged as trimmed</dd></dl>
</li>
</ul>
<a name="getNumberOfMappedLocations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfMappedLocations</h4>
<pre>public abstract&nbsp;int&nbsp;getNumberOfMappedLocations()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of locations in total that this read was mapped to. Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#NUMBER_OF_MAPPED_LOCATIONS"><code>DocumentField.NUMBER_OF_MAPPED_LOCATIONS</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getMappingQuality()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMappingQuality</h4>
<pre>public abstract&nbsp;int&nbsp;getMappingQuality()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the quality of a mapped read. -1 for no quality or between 0 and 254 inclusive. Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#MAPPING_QUALITY"><code>DocumentField.MAPPING_QUALITY</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd></dl>
</li>
</ul>
<a name="insert(int, java.lang.CharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>insert</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;insert(int&nbsp;index,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;charSequence)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#insert(int, java.lang.CharSequence)">SequenceCharSequence</a></code></strong></div>
<div class="block">Creates a copy of this SequenceCharSequence with the specified CharSequence inserted
 at the specified index. Doesn't modify this SequenceCharSequence.

 The returned SequenceCharSequence will support logarithmic time <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#insert(int, java.lang.CharSequence)"><code>insertions</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#delete(int, int)"><code>deletions</code></a>, and its length will be longer than this sequence's by
 <code>charSequence.length()</code>.
 <p/>
 In order to obey the contract of SequenceCharSequence that it is immutable,
 <b>the length of <code>charSequence</code> and the characters retured from <code>charSequence</code> must not change after calling this method.</b></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#insert(int, java.lang.CharSequence)">insert</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - The position where to insert csq. All existing characters at or to the right of this position will be moved to the right by <code>charSequence.length()</code>.</dd><dd><code>charSequence</code> - A CharSequence to insert at the specified index. <b>The length of <code>charSequence</code> and the characters retured from <code>charSequence</code> must not change after calling this method.</b></dd>
<dt><span class="strong">Returns:</span></dt><dd>A copy of this SequenceCharSequence with csq inserted at the specified index.</dd></dl>
</li>
</ul>
<a name="delete(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>delete</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;delete(int&nbsp;deletionBegin,
                       int&nbsp;deletionEnd)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#delete(int, int)">SequenceCharSequence</a></code></strong></div>
<div class="block">Returns a copy of this SequenceCharSequence where the characters at positions deletionBegin inclusive
 to deletionEnd exclusive have been removed. Doesn't modify this SequenceCharSequence.
 The returned SequenceCharSequence will support logarithmic time <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#insert(int, java.lang.CharSequence)"><code>insertions</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#delete(int, int)"><code>deletions</code></a>, and its length will be shorter than this sequence's by
 <code>deletionEnd - deletionBegin</code>.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#delete(int, int)">delete</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>deletionBegin</code> - Deletion start index, inclusive</dd><dd><code>deletionEnd</code> - Deletion end index, exclusive</dd>
<dt><span class="strong">Returns:</span></dt><dd>A copy of this SequenceCharSequence with the specified range of characters removed.</dd></dl>
</li>
</ul>
<a name="withTrimLengths(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>withTrimLengths</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;withTrimLengths(int&nbsp;leadingTrimLength,
                                int&nbsp;trailingTrimLength)</pre>
<div class="block">Creates a new sequence identical to this one, except with different leading and trailing trim lengths.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>leadingTrimLength</code> - the leading trim length as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getLeadingTrimLength()"><code>getLeadingTrimLength()</code></a></dd><dd><code>trailingTrimLength</code> - the trailing trim length as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getTrailingTrimLength()"><code>getTrailingTrimLength()</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence identical to this one, except with different leading and trailing trim lengths.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="withNumberOfMappedLocations(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>withNumberOfMappedLocations</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;withNumberOfMappedLocations(int&nbsp;numberOfMappedLocations)</pre>
<div class="block">Creates a new sequence identical to this one, except with a different value for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getNumberOfMappedLocations()"><code>getNumberOfMappedLocations()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfMappedLocations</code> - the new number of mapped locations</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence identical to this one, except with a different value for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getNumberOfMappedLocations()"><code>getNumberOfMappedLocations()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="withMappingQuality(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>withMappingQuality</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;withMappingQuality(int&nbsp;mappingQuality)</pre>
<div class="block">Creates a new sequence identical to this one, except with a different value for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getMappingQuality()"><code>getMappingQuality()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>mappingQuality</code> - the new mapping quality. -1 for no quality, or between 0 and 254 inclusive for a valid mapping quality</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence identical to this one, except with a different value for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getMappingQuality()"><code>getMappingQuality()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd></dl>
</li>
</ul>
<a name="insertGaps(int, int, int[], int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>insertGaps</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;insertGaps(int&nbsp;leadingGapsLength,
                           int&nbsp;trailingGapsLength,
                           int[]&nbsp;internalGapPositions,
                           int[]&nbsp;internalGapSizes)</pre>
<div class="block">Creates a new sequence identical to this one except with some gaps inserted.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>leadingGapsLength</code> - the number of additional leading gaps to add to this sequence</dd><dd><code>trailingGapsLength</code> - the number of additional trailing gaps to add to this sequence</dd><dd><code>internalGapPositions</code> - the positions of internal gaps to insert into this sequence. All values must be in the range 1 through to the internal sequence length-1 inclusive and indicate that a gap should be inserted before the current character at that index. This array may be null which is equivalent to an empty array.</dd><dd><code>internalGapSizes</code> - the sizes of corresponding internal gaps to insert into this sequence. This array must be equal in length to internalGapPositions. Each size must be positive. This array may be null which is equivalent to an empty array.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence identical to this one except with some gaps inserted.  If this sequence is a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>.  If this sequence is an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a></dd></dl>
</li>
</ul>
<a name="removeGaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeGaps</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;removeGaps()</pre>
<div class="block">Creates a new sequence identical to this once except all gaps (both end and internal) have been removed.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a new sequence identical to this once except all gaps (both end and internal) have been removed.  If this sequence is a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>.  If this sequence is an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a></dd></dl>
</li>
</ul>
<a name="createCopyWithNewName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createCopyWithNewName</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;createCopyWithNewName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName)</pre>
<div class="block">Creates a new sequence identical to this once except with a different name. Note that if the name
 is the same as the current name of this sequence, and the current name has been compacted according to
 a standard name compaction pattern, then the returned sequence will be the this ImmutableSequence instance.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>newName</code> - the name of the new sequence to create</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence identical to this once except with a different name. If this sequence is a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>.  If this sequence is an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd></dl>
</li>
</ul>
<a name="createCopyWithNameSuffix(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createCopyWithNameSuffix</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;createCopyWithNameSuffix(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;nameSuffix)</pre>
<div class="block">Creates a new sequence identical to this once except with a different name.
 If the current sequence has a compressed name pattern, then the returned sequence will have a
 new compressed name pattern based on the current name pattern.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nameSuffix</code> - a suffix to append to the name. If the current name ends with " (reversed)" then the suffix will be appended immediately prior to the reversed suffix.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new sequence identical to this once except with a different name.  If this sequence is a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>.  If this sequence is an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a> the returned sequence will also be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a></dd></dl>
</li>
</ul>
<a name="isReversed()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReversed</h4>
<pre>public abstract&nbsp;boolean&nbsp;isReversed()</pre>
<div class="block">Determine if this sequence is reversed (i.e. its name ends with <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>SequenceExtractionUtilities.REVERSED_NAME_SUFFIX</code></a>)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this sequence is reversed</dd></dl>
</li>
</ul>
<a name="reverseComplement()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplement</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;reverseComplement()</pre>
<div class="block">Create a new sequence that is the reverse complement of this sequence. The name of the returned sequence will have <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>SequenceExtractionUtilities.REVERSED_NAME_SUFFIX</code></a> either appended or removed</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a new sequence that is the reverse complement of this sequence. The returned sequence will be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this is not a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>. You should instead use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#reverse()"><code>reverse()</code></a></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#reverse()"><code>reverse()</code></a></dd></dl>
</li>
</ul>
<a name="reverse()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverse</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;reverse()</pre>
<div class="block">Create a new sequence that is the reverse  of this sequence.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a new sequence that is the reverse of this sequence. The returned sequence will be an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this is not a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>. You should instead use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#reverseComplement()"><code>reverseComplement()</code></a></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#reverseComplement()"><code>reverseComplement()</code></a></dd></dl>
</li>
</ul>
<a name="asDna()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asDna</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;asDna()</pre>
<div class="block">Creates a new sequence that is identical to this sequence except with U replaced with T</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a new sequence that is identical to this sequence except with U replaced with T (or this sequence if it did not already contain U). The returned sequence will be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this is not a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd></dl>
</li>
</ul>
<a name="asRna()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asRna</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;asRna()</pre>
<div class="block">Creates a new sequence that is identical to this sequence except with T replaced with U</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a new sequence that is identical to this sequence except with T replaced with U (or this sequence if it did not already contain T). The returned sequence will be a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this is not a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a>.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd></dl>
</li>
</ul>
<a name="getCharAndQualityAtIgnoringEndGaps(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCharAndQualityAtIgnoringEndGaps</h4>
<pre>public abstract&nbsp;int&nbsp;getCharAndQualityAtIgnoringEndGaps(int&nbsp;characterIndex)</pre>
<div class="block">Provides both the sequence character and the quality in a single method call, which is almost twice
 as fast as requesting the 2 values separately.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>characterIndex</code> - the index of the character to get, relative to the leadingGapsLength.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a value equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#charAtIgnoringEndGaps(int)"><code>SequenceCharSequence.charAtIgnoringEndGaps(int)</code></a>+quality*256.</dd></dl>
</li>
</ul>
<a name="getData(int, int, byte[], byte[], int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getData</h4>
<pre>public abstract&nbsp;void&nbsp;getData(int&nbsp;internalStartIndex,
           int&nbsp;sizeOfDataToGet,
           byte[]&nbsp;characters,
           byte[]&nbsp;qualities,
           int&nbsp;startingOffsetIntoDataArrays)</pre>
<div class="block">Provides sequence character and qualities for multiple positions in a single method call,
 which is much faster than requesting the values separately. This method may only be
 used to retrieve data that doesn't include end gap regions.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>internalStartIndex</code> - the index of the first character to get, relative to the leadingGapsLength. (e.g. 0 to get the first non-end-gap character)</dd><dd><code>sizeOfDataToGet</code> - the number of characters of data to get. internalStartIndex+sizeOfDataToGet must not be greater than <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getInternalSequenceLength()"><code>SequenceCharSequence.getInternalSequenceLength()</code></a></dd><dd><code>characters</code> - an array to copy the sequence characters into.</dd><dd><code>qualities</code> - an optional array to copy the sequence quality into. This may be null to not retrieve quality data. If this array is not null, and this sequence has no quality, quality values of 0 will be set. If any quality values exceed 255, they will be capped at 255.</dd><dd><code>startingOffsetIntoDataArrays</code> - the index in 'characters' and 'qualities' to start copying data into.</dd></dl>
</li>
</ul>
<a name="writeObject(java.io.DataOutput, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)">SequenceCharSequence</a></code></strong></div>
<div class="block">Writes this sequence to a DataOutput.
 The object can be reconstructed using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#readObject(java.io.DataInput)"><code>SequenceCharSequence.readObject(java.io.DataInput)</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>out</code> - the DataOutput to write to.</dd><dd><code>progressListener</code> - for reporting progress and allowing the write to be canceled.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if the write cannot be completed for any reason (including if the progress listener requests the operation be canceled)</dd></dl>
</li>
</ul>
<a name="writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out,
               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)">SequenceCharSequence</a></code></strong></div>
<div class="block">Writes this sequence to a DataOutput.
 The object can be reconstructed using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#readObject(java.io.DataInput)"><code>SequenceCharSequence.readObject(java.io.DataInput)</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput, jebl.util.ProgressListener)">writeObject</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the version number of Geneious that must be able to deserialize this.</dd><dd><code>out</code> - the DataOutput to write to.</dd><dd><code>progressListener</code> - for reporting progress and allowing the write to be canceled.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if the write cannot be completed for any reason (including if the progress listener requests the operation be canceled)</dd></dl>
</li>
</ul>
<a name="readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readObject</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;readObject(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)
                                    throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Reads a instance of an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a> that
 was previously written using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)"><code>SequenceCharSequence.writeObject(GeneiousObjectOutputStream, jebl.util.ProgressListener)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>in</code> - the stream to read from</dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence read from the stream</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if something goes wrong</dd></dl>
</li>
</ul>
<a name="compactNucleotideSequences(java.util.List, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compactNucleotideSequences</h4>
<pre>public static&nbsp;void&nbsp;compactNucleotideSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>&gt;&nbsp;sequences,
                              <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block">Attempts to compact the memory usage of as many sequences in the given list of sequences as possible
 by replacing them with ImmutableSequence instances.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - the sequences which should be compacted if possible.</dd><dd><code>progressListener</code> - a progress listener for reporting progress and providing the ability to cancel</dd></dl>
</li>
</ul>
<a name="compactAminoAcidSequences(java.util.List, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compactAminoAcidSequences</h4>
<pre>public static&nbsp;void&nbsp;compactAminoAcidSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>&gt;&nbsp;sequences,
                             <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block">Attempts to compact the memory usage of as many sequences in the given list of sequences as possible
 by replacing them with ImmutableSequence instances.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - the sequences which should be compacted if possible.</dd><dd><code>progressListener</code> - a progress listener for reporting progress and providing the ability to cancel</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">SequenceCharSequence</a></code></strong></div>
<div class="block">Implements <code>XMLSerializable.OldVersionCompatible#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code>
 but note that SequenceCharSequence is guaranteed to be able to always serialize to <code>Geneious.MajorVersion#Version6_0</code></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(boolean&nbsp;allowFileData)
                       throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#toXML(boolean)">SequenceCharSequence</a></code></strong></div>
<div class="block">Serializes this SequenceCharSequence to XML into an element named
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a>.
 Deserialization can be done via the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#fromXml(org.jdom.Element)"><code>SequenceCharSequence.fromXml(org.jdom.Element)</code></a>.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#toXML(boolean)">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>allowFileData</code> - true to allow the sequence to potentially serialize to a file if it is large enough, using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME"><code>PluginDocument.FILE_DATA_ATTRIBUTE_NAME</code></a>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>An XML Element representing this SequenceCharSequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code></dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener,
                     boolean&nbsp;allowFileData)
                       throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener, boolean)">SequenceCharSequence</a></code></strong></div>
<div class="block">Serializes this SequenceCharSequence to XML into an element named
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a>.
 Deserialization can be done via the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#fromXml(org.jdom.Element)"><code>SequenceCharSequence.fromXml(org.jdom.Element)</code></a>.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener, boolean)">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the version number of Geneious that must be able to deserialize this</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd><dd><code>allowFileData</code> - true to allow the sequence to potentially serialize to a file if it is large enough, using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME"><code>PluginDocument.FILE_DATA_ATTRIBUTE_NAME</code></a>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>An XML Element representing this SequenceCharSequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code></dd></dl>
</li>
</ul>
<a name="immutableSequenceFromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>immutableSequenceFromXML</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>&nbsp;immutableSequenceFromXML(org.jdom.Element&nbsp;element)
                                                  throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Recreates this sequence from XML returned from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#toXML(boolean)"><code>toXML(boolean)</code></a>.
 Generally this method should not be used as it is slow and memory intensive. Binary serialization
 using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream, jebl.util.ProgressListener)"><code>SequenceCharSequence.writeObject(GeneiousObjectOutputStream, jebl.util.ProgressListener)</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>readObject(GeneiousObjectInputStream)</code></a> are preferable.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - the element to deserialzide from.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the deserialized sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if anything goes wrong</dd></dl>
</li>
</ul>
<a name="subString(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>subString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;subString(int&nbsp;startIndex,
               int&nbsp;endIndex)</pre>
<div class="block">Returns the characters between the 2 given indices as a string</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>startIndex</code> - inclusive start index ( the index of the first character to get)</dd><dd><code>endIndex</code> - exclusive end index (the index after the last character to get)</dd>
<dt><span class="strong">Returns:</span></dt><dd>the characters between the 2 given indices as a string</dd></dl>
</li>
</ul>
<a name="getNumberOfIntegersOfStorage()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getNumberOfIntegersOfStorage</h4>
<pre>public abstract&nbsp;int&nbsp;getNumberOfIntegersOfStorage()</pre>
<div class="block">Returns the number of integers needed to store this document on disk and/or load it into memory</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of integers needed to store this document on disk and/or load it into memory</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ImmutableSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" target="_top">Frames</a></li>
<li><a href="ImmutableSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
