<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>PairedReads (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="PairedReads (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/PairedReads.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" target="_top">Frames</a></li>
<li><a href="PairedReads.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Interface PairedReads" class="title">Interface PairedReads</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndAminoAcidSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndAminoAcidSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndNucleotideGraphSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndNucleotideSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultPhylogenyDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/types/PhylogenyDocument.html" title="class in com.biomatters.geneious.publicapi.documents.types">PhylogenyDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/SequenceGraph.SequencePropertyRetriever.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceGraph.SequencePropertyRetriever</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/SequenceViewerExtension.PropertyRetrieverAndEditor.html" title="class in com.biomatters.geneious.publicapi.plugin">SequenceViewerExtension.PropertyRetrieverAndEditor</a></dd>
</dl>
<hr>
<br>
<pre>public interface <span class="strong">PairedReads</span></pre>
<div class="block">Provides data about whether nucleotide sequences are paired together and their expected paired distance.
 <p/>
 Different sequencing technologies have different relative orientations of reads. Geneious
 manages this by allowing each pair of a mate to specify its expected mate distance
 separately. For example, if unaligned pairs are:
 <ul>
 <li>  Forward/Forward (solid) Then the first mate of the pair will have a positive expected distance to its mate and the second mate will have a negative expected distance.</li>
 <li>  Forward/Reverse (Illumina short read kit) Then the both mates in the pair will have positive expected distances</li>
 <li>  Reverse/Forward (Illumina long read kit) Then the both mates in the pair will have negative expected distances</li>
 <li>  Reverse/Reverse (454 from NCBI) Then the first mate of the pair will have a negative expected distance to its mate and the second mate will have a positive expected distance.</li>
 </ul>
 <p/>
 The mate distance is defined as the length of the 2 reads plus the distance between them.
 <p/>
 Once sequences are placed in a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>, the expected distance now refers to where a sequence
 expects to find its mate. If a sequence is placed in the alignment without reverse-complementing, then its expected distance will remain the same as specified above. If the sequence
 is placed in an alignment reverse-complemented, then its expected distance will be negated (postive <-> negative).
 <b>So, no matter what the sequencing technology used,
 when reads are correctly orientated in the alignment, the left-hand read will have a positive expected distance and the right-hand read will have a negative expected distance.</b>
 <p/>
 For example, if aligned pairs are:
 <ul>
 <li> Forward/forward data, where the first mate has a positive expected distance and the second has a negative expected distance when unaligned, if they both align to the forward strand,
 then their expected distances remain the same. Alternatively, both could align to the reverse strand, in which case their relative order should change (first mate will be on
 the right and second mate will be on the left) so their expected distances get negated.</li>
 <li> Forward/reverse data, where both mates have a positive expected distance when unaligned , if the first aligns to the forward strand and the second aligns to the reverse strand
 then the first will have a positive expected distance and the second will get negated to become a negative expected distance. Alternatively if the first aligns to the reverse
 strand and the second aligns to the forward strand then their relative order should change and so the first mate will end up with a negative expected distance and the second
 a positive expected distance in the alignment</li>
 </ul>
 <p/>
 All <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocuments</code></a> implements this interface.
 A <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListDocument</code></a> can optionally implement this interface to provide paired read data.
 <p/>
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>getMateIndex(int)</code></a> returns the index of the mate sequence in the list (if any) and 
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a> returns the expected distance of the mate.
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()"><code>getImmutablePairedReadManager()</code></a> returns an immutable view of the paired properties and also provides additional functionality. See <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>PairedReadManager</code></a>
 <p/>
 The sequence viewer plugin displays paired reads with a little paired-read icon to the left of their name. For paired reads in an alignment, the sequence viewer
 also links them together on the same line as long as the 'Link Paired Reads' setting (under 'Advanced') is on. The sequence viewer also has a color scheme
 to color paired reads based on how close they are to their expected distance.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></strong>()</code>
<div class="block">An immutable view of the paired reads managed by this list or alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">getMateIndex</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
</td>
</tr>
</table>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getMateExpectedDistance(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateExpectedDistance</h4>
<pre>int&nbsp;getMateExpectedDistance(int&nbsp;sequenceIndex)</pre>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate. This is defined
 as the length of the 2 reads plus the distance between them.
 For example if the 2 reads both have length 30 with a 40 bp gap between them, then the
 expected mate distance is 100.
 If the mate is expected to appear after this sequence with this sequence in its current orientation, then
 the expected distance will be positive. If the mate is expected to appear before this sequence with this sequence in its current orientation, then
 the expected distance will be negative.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence for which to get the mate index of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the expected mate distance or 0 if this read is not paired. Note that a read may still have a mate distance!=0 but its mate may not be present in this document (in which case <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>getMateIndex(int)</code></a> will be -1.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>getMateIndex(int)</code></a></dd></dl>
</li>
</ul>
<a name="getMateIndex(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateIndex</h4>
<pre>int&nbsp;getMateIndex(int&nbsp;sequenceIndex)</pre>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence to get the mate index of</dd>
<dt><span class="strong">Returns:</span></dt><dd>an index of the mate of the sequence, or -1 if the sequence has no mate.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a></dd></dl>
</li>
</ul>
<a name="getImmutablePairedReadManager()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getImmutablePairedReadManager</h4>
<pre><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;getImmutablePairedReadManager()</pre>
<div class="block">An immutable view of the paired reads managed by this list or alignment. Even if the list or alignment
 that implements this method is later modified, the
 returned PairedReadManager will not change.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>An immutable view of the paired reads managed by the list or alignment that implements this method or null if this class doesn't provide an immutable view of its paired reads.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/PairedReads.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" target="_top">Frames</a></li>
<li><a href="PairedReads.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
