<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:47 NZDT 2016 -->
<title>SequenceListSummary (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceListSummary (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceListSummary.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Coverage.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" target="_top">Frames</a></li>
<li><a href="SequenceListSummary.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Class SequenceListSummary" class="title">Class SequenceListSummary</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.documents.sequence.SequenceListSummary</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></dd>
</dl>
<hr>
<br>
<pre>public final class <span class="strong">SequenceListSummary</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">Provides a summary of data in a list of sequences. For example it provides the distribution of sequence lengths, sequence qualities, and annotation types.
 <p/>
 The list returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getNucleotideSequences()"><code>SequenceListDocument.getNucleotideSequences()</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListDocument.html#getAminoAcidSequences()"><code>SequenceListDocument.getAminoAcidSequences()</code></a> may optionally implement the interface
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListSummary.Provider</code></a> to provide a summary of the information contained within it. <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSummary()"><code>DefaultAlignmentDocument.getSummary</code></a>
 also provides a summary if it stores its non-reference sequences on disk instead of in memory.
 <p/>
 This class is not thread safe while being constructed, but once <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has been called this class becomes immutable and thread safe.
 Instances of this class created using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#SequenceListSummary(org.jdom.Element)"><code>SequenceListSummary(org.jdom.Element)</code></a> are thread safe since they are immutable.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Coverage.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Coverage</a></strong></code>
<div class="block">Represents the coverage (number of sequences covering a column in an alignment)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.PairwiseSimilarity.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.PairwiseSimilarity</a></strong></code>
<div class="block">Immutable value class representing the number of identical and non-identical
 pairs of bases/residues in an alignment column.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Provider</a></strong></code>
<div class="block">An interface that lists of sequences (e.g.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#SequenceListSummary(org.jdom.Element)">SequenceListSummary</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Creates a summary from a previously serialized summary</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#SequenceListSummary(java.util.List, boolean, jebl.util.ProgressListener)">SequenceListSummary</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences,
                   boolean&nbsp;allowGaps,
                   <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">A convenience constructor which just calls <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>addSequence(SequenceDocument)</code></a> for each sequence in the list and then calls <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, int)">SequenceListSummary</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                   int&nbsp;numberOfSequencesIfGapsAllowed)</code>
<div class="block">Constructs a new summary.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, int, int)">SequenceListSummary</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                   int&nbsp;numberOfSequencesIfGapsAllowed,
                   int&nbsp;circularAlignmentLength)</code>
<div class="block">Constructs a new summary.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, int, int, boolean)">SequenceListSummary</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                   int&nbsp;numberOfSequencesIfGapsAllowed,
                   int&nbsp;circularAlignmentLength,
                   boolean&nbsp;createMultipleThreads)</code>
<div class="block">Constructs a new summary.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceListSummary, com.biomatters.geneious.publicapi.documents.sequence.SequenceListSummary)">SequenceListSummary</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a>&nbsp;summary1,
                   <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a>&nbsp;summary2)</code>
<div class="block">Constructs a new SequenceListSummary by combining 2 other summaries.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Adds a new sequence for this summary.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)">finishedAddingSequences</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Indicates that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>addSequence(SequenceDocument)</code></a> will no longer be called, and builds and sorts the summary data.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getAlignmentNumberOfColumns()">getAlignmentNumberOfColumns</a></strong>()</code>
<div class="block">If this summary is for an alignment, returns the number of columns in the alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.PairwiseSimilarity.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.PairwiseSimilarity</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getAlignmentPairwiseSimilarity(boolean)">getAlignmentPairwiseSimilarity</a></strong>(boolean&nbsp;freeEndGaps)</code>
<div class="block">Get the pairwise similarity if this summary is for an alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getAnnotationTypes()">getAnnotationTypes</a></strong>()</code>
<div class="block">Gets a set consisting of each annotation type that is present in at least 1 of the sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/BitSet.html?is-external=true" title="class or interface in java.util">BitSet</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getColumnsWithGapsOverPercentage(double)">getColumnsWithGapsOverPercentage</a></strong>(double&nbsp;percentage)</code>
<div class="block">If this summary is for an alignment gets a BitSet indicating which columns contain over the given percentage of gaps.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getCoverage(int)">getCoverage</a></strong>(int&nbsp;residueIndex)</code>
<div class="block">If this summary is for an alignment, gets the total number of sequences covering this position.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Coverage.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Coverage</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getCoverage(int, int)">getCoverage</a></strong>(int&nbsp;minResidueIndex,
           int&nbsp;maxResidueIndexExclusive)</code>
<div class="block">If this summary is for an alignment, gets an approximate coverage for the given range.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getCoverageAsDouble(int)">getCoverageAsDouble</a></strong>(int&nbsp;residueIndex)</code>
<div class="block">If this summary is for an alignment, gets the total number of sequences covering this position.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getCoverageMaximum()">getCoverageMaximum</a></strong>()</code>
<div class="block">Returns the maximum coverage in this alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getCoverageMaximumAsDouble()">getCoverageMaximumAsDouble</a></strong>()</code>
<div class="block">Returns the maximum coverage in this alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getCoverageMean()">getCoverageMean</a></strong>()</code>
<div class="block">Returns the mean coverage in this alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getCoverageStandardDeviation()">getCoverageStandardDeviation</a></strong>()</code>
<div class="block">Returns the standard deviation of coverage in this alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getFrequency(char)">getFrequency</a></strong>(char&nbsp;character)</code>
<div class="block">Gets the number of occurrences of the given character</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfAnnotations()">getNumberOfAnnotations</a></strong>()</code>
<div class="block">Get the total number of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotations</code></a> present on the sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfAnnotationsOfType(java.lang.String)">getNumberOfAnnotationsOfType</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;annotationType)</code>
<div class="block">Get the total number of annotations of the given type present on the sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfCharactersExcludingGaps()">getNumberOfCharactersExcludingGaps</a></strong>()</code>
<div class="block">returns the total number of characters (nucleotides or residues) in all the sequences, excluding gaps</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfCharactersIncludingInternalGaps()">getNumberOfCharactersIncludingInternalGaps</a></strong>()</code>
<div class="block">returns the total number of characters (nucleotides or residues) in all the sequences, including internal gaps, but excluding end gaps.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfCircularSequences()">getNumberOfCircularSequences</a></strong>()</code>
<div class="block">Returns the number of sequences in the list for which <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a> is true</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfImmutableSequenceTrimmedRegions()">getNumberOfImmutableSequenceTrimmedRegions</a></strong>()</code>
<div class="block">Returns the number of trimmed regions in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequences</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfNucleotideGraphs()">getNumberOfNucleotideGraphs</a></strong>()</code>
<div class="block">returns the number of sequences that are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>s</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfReversedSequences()">getNumberOfReversedSequences</a></strong>()</code>
<div class="block">Returns the number of sequences whose name ends with <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>SequenceExtractionUtilities.REVERSED_NAME_SUFFIX</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>
<div class="block">Returns the number of sequences which this summary is for</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfSequencesWithAnnotationsOfType(java.lang.String)">getNumberOfSequencesWithAnnotationsOfType</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;annotationType)</code>
<div class="block">Get the total number of sequences that have annotations of the given type present.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfSequencesWithQuality()">getNumberOfSequencesWithQuality</a></strong>()</code>
<div class="block">returns the number of sequences that are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>s and return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()"><code>NucleotideGraph.hasSequenceQualities()</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfTrimmedSequences()">getNumberOfTrimmedSequences</a></strong>()</code>
<div class="block">Gets the total number of sequences that either have an annotation of type <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>SequenceAnnotation.TYPE_TRIMMED</code></a> or have <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getLeadingTrimLength()"><code>ImmutableSequence.getLeadingTrimLength()</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getTrailingTrimLength()"><code>ImmutableSequence.getTrailingTrimLength()</code></a> greater than 0.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getPercentGC()">getPercentGC</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getQualitiesExcludingGaps()">getQualitiesExcludingGaps</a></strong>()</code>
<div class="block">Returns the distribution of quality values (those returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)"><code>NucleotideGraph.getSequenceQuality(int)</code></a>) across all nucleotides that are non-gaps in all sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getQualitiesExcludingGapsTrimmed()">getQualitiesExcludingGapsTrimmed</a></strong>()</code>
<div class="block">Returns the distribution of quality values (those returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)"><code>NucleotideGraph.getSequenceQuality(int)</code></a>) across all non-gap non-trimmed nucleotides in all sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getReferenceSequenceBases()">getReferenceSequenceBases</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getReferenceSequenceBasesCovered()">getReferenceSequenceBasesCovered</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getSequenceLengthsExcludingGaps()">getSequenceLengthsExcludingGaps</a></strong>()</code>
<div class="block">Gets the distribution of ungapped sequence lengths of all sequences managed by this summary.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getSequenceLengthsExcludingGapsTrimmed()">getSequenceLengthsExcludingGapsTrimmed</a></strong>()</code>
<div class="block">Gets the distribution of ungapped sequence lengths of all sequences managed by this summary after trim annotations have been applied.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Boolean.html?is-external=true" title="class or interface in java.lang">Boolean</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#hasAnyReadsMappedToMultipleLocations()">hasAnyReadsMappedToMultipleLocations</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;majorVersion,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceListSummary</h4>
<pre>public&nbsp;SequenceListSummary(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                   int&nbsp;numberOfSequencesIfGapsAllowed)
                    throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Constructs a new summary. Sequences may be added, but many of the getters will not be functional until <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has been called.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alphabet</code> - specifies whether this summary is for nucleotide or amino acid sequences</dd><dd><code>numberOfSequencesIfGapsAllowed</code> - if this is not negative then gaps will be allowed, and this must be the number of sequences that will be added.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we can't write data to disk.</dd></dl>
</li>
</ul>
<a name="SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceListSummary</h4>
<pre>public&nbsp;SequenceListSummary(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                   int&nbsp;numberOfSequencesIfGapsAllowed,
                   int&nbsp;circularAlignmentLength)
                    throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Constructs a new summary. Sequences may be added, but many of the getters will not be functional until <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has been called.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alphabet</code> - specifies whether this summary is for nucleotide or amino acid sequences</dd><dd><code>numberOfSequencesIfGapsAllowed</code> - if this is not negative then gaps will be allowed, and this must be the number of sequences that will be added.</dd><dd><code>circularAlignmentLength</code> - the circular length of the alignment (if gaps allowed) as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()"><code>SequenceAlignmentDocument.getCircularLength()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we can't write data to disk.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, int, int, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceListSummary</h4>
<pre>public&nbsp;SequenceListSummary(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                   int&nbsp;numberOfSequencesIfGapsAllowed,
                   int&nbsp;circularAlignmentLength,
                   boolean&nbsp;createMultipleThreads)
                    throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Constructs a new summary. Sequences may be added, but many of the getters will not be functional until <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has been called.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alphabet</code> - specifies whether this summary is for nucleotide or amino acid sequences</dd><dd><code>numberOfSequencesIfGapsAllowed</code> - if this is not negative then gaps will be allowed, and this must be the number of sequences that will be added.</dd><dd><code>circularAlignmentLength</code> - the circular length of the alignment (if gaps allowed) as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()"><code>SequenceAlignmentDocument.getCircularLength()</code></a></dd><dd><code>createMultipleThreads</code> - true to allow creating multiple threads when building the summary to improve performance</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we can't write data to disk.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.800 (Geneious 8.0.0)</dd></dl>
</li>
</ul>
<a name="SequenceListSummary(com.biomatters.geneious.publicapi.documents.sequence.SequenceListSummary, com.biomatters.geneious.publicapi.documents.sequence.SequenceListSummary)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceListSummary</h4>
<pre>public&nbsp;SequenceListSummary(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a>&nbsp;summary1,
                   <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a>&nbsp;summary2)
                    throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Constructs a new SequenceListSummary by combining 2 other summaries. Only summaries which don't support gaps may be combined with this method.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>summary1</code> - the first summary to combine</dd><dd><code>summary2</code> - the second summary to combine</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there is a problem writing to disk</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if the summaries are different sequence types (DNA vs Protein) or if one allows gaps while the other doesn't</dd></dl>
</li>
</ul>
<a name="SequenceListSummary(java.util.List, boolean, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceListSummary</h4>
<pre>public&nbsp;SequenceListSummary(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;sequences,
                   boolean&nbsp;allowGaps,
                   <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                    throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">A convenience constructor which just calls <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>addSequence(SequenceDocument)</code></a> for each sequence in the list and then calls <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - the sequences to create a summary of</dd><dd><code>allowGaps</code> - true if gaps are allowed in the sequences</dd><dd><code>progressListener</code> - for reporting progress and cancelling.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we can't write the summary to disk</dd></dl>
</li>
</ul>
<a name="SequenceListSummary(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceListSummary</h4>
<pre>public&nbsp;SequenceListSummary(org.jdom.Element&nbsp;element)
                    throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Creates a summary from a previously serialized summary</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - an element returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#toXML()"><code>toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if we can't deserialize it</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;majorVersion,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>majorVersion</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;element)
             throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - representation from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequence</h4>
<pre>public&nbsp;void&nbsp;addSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Adds a new sequence for this summary. <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> must be called after all sequences have been added.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to add to the summary</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we can't write the summary to disk</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has already been called</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if sequence doesn't contain valid nucleotide or amino acid characters or contains gaps when not allowed (as specified by the constructor)</dd></dl>
</li>
</ul>
<a name="finishedAddingSequences(jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>finishedAddingSequences</h4>
<pre>public&nbsp;void&nbsp;finishedAddingSequences(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                             throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Indicates that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>addSequence(SequenceDocument)</code></a> will no longer be called, and builds and sorts the summary data. Once this method
 has finished, this class becomes immutable and thread safe.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we can't write the summary to disk</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this method is called more than once</dd></dl>
</li>
</ul>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfSequences()</pre>
<div class="block">Returns the number of sequences which this summary is for</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences which this summary is for</dd></dl>
</li>
</ul>
<a name="getNumberOfCharactersIncludingInternalGaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfCharactersIncludingInternalGaps</h4>
<pre>public&nbsp;long&nbsp;getNumberOfCharactersIncludingInternalGaps()</pre>
<div class="block">returns the total number of characters (nucleotides or residues) in all the sequences, including internal gaps, but excluding end gaps.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the total number of characters (nucleotides or residues) in all the sequences, including internal gaps, but excluding end gaps.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfCharactersExcludingGaps()"><code>getNumberOfCharactersExcludingGaps()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getFrequency(char)"><code>getFrequency(char)</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfCharactersExcludingGaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfCharactersExcludingGaps</h4>
<pre>public&nbsp;long&nbsp;getNumberOfCharactersExcludingGaps()</pre>
<div class="block">returns the total number of characters (nucleotides or residues) in all the sequences, excluding gaps</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the total number of characters (nucleotides or residues) in all the sequences, excluding gaps</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfCharactersIncludingInternalGaps()"><code>getNumberOfCharactersIncludingInternalGaps()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getFrequency(char)"><code>getFrequency(char)</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfImmutableSequenceTrimmedRegions()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfImmutableSequenceTrimmedRegions</h4>
<pre>public&nbsp;long&nbsp;getNumberOfImmutableSequenceTrimmedRegions()</pre>
<div class="block">Returns the number of trimmed regions in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequences</code></a>.
 (i.e. the number of sequences that returned a non-zero value from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getLeadingTrimLength()"><code>ImmutableSequence.getLeadingTrimLength()</code></a> plus the
 number of sequences that returned a non-zero value from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getTrailingTrimLength()"><code>ImmutableSequence.getTrailingTrimLength()</code></a>)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of trimmed regions in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequences</code></a>.</dd></dl>
</li>
</ul>
<a name="getFrequency(char)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFrequency</h4>
<pre>public&nbsp;long&nbsp;getFrequency(char&nbsp;character)</pre>
<div class="block">Gets the number of occurrences of the given character</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>character</code> - the character to get the frequency of. The character is case sensitive. If this character is '-' then this counts internal gaps only - end gaps are not counted.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the number of occurrences of this character in all sequences.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfCharactersExcludingGaps()"><code>getNumberOfCharactersExcludingGaps()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfCharactersIncludingInternalGaps()"><code>getNumberOfCharactersIncludingInternalGaps()</code></a></dd></dl>
</li>
</ul>
<a name="getPercentGC()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPercentGC</h4>
<pre>public&nbsp;double&nbsp;getPercentGC()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the percentage of A, C, G, T, U, S (G or C), W (A or T) nucleotides that are either G, C, or S. Returns 0, if no nucleotides are any of these.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is for amino acid sequences</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd></dl>
</li>
</ul>
<a name="getNumberOfSequencesWithQuality()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequencesWithQuality</h4>
<pre>public&nbsp;long&nbsp;getNumberOfSequencesWithQuality()</pre>
<div class="block">returns the number of sequences that are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>s and return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()"><code>NucleotideGraph.hasSequenceQualities()</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences that are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>s and return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#hasSequenceQualities()"><code>NucleotideGraph.hasSequenceQualities()</code></a></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfNucleotideGraphs()"><code>getNumberOfNucleotideGraphs()</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfNucleotideGraphs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfNucleotideGraphs</h4>
<pre>public&nbsp;long&nbsp;getNumberOfNucleotideGraphs()</pre>
<div class="block">returns the number of sequences that are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>s</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences that are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>s</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfSequencesWithQuality()"><code>getNumberOfSequencesWithQuality()</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfAnnotations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfAnnotations</h4>
<pre>public&nbsp;long&nbsp;getNumberOfAnnotations()</pre>
<div class="block">Get the total number of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotations</code></a> present on the sequences.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the total number of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotations</code></a> present on the sequences.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfAnnotationsOfType(java.lang.String)"><code>getNumberOfAnnotationsOfType(String)</code></a></dd></dl>
</li>
</ul>
<a name="getAnnotationTypes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationTypes</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getAnnotationTypes()</pre>
<div class="block">Gets a set consisting of each annotation type that is present in at least 1 of the sequences.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a set consisting of each annotation type that is present in at least 1 of the sequences.</dd></dl>
</li>
</ul>
<a name="getNumberOfAnnotationsOfType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfAnnotationsOfType</h4>
<pre>public&nbsp;long&nbsp;getNumberOfAnnotationsOfType(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;annotationType)</pre>
<div class="block">Get the total number of annotations of the given type present on the sequences.
 If a sequence has 2 annotations of the same type, that counts as 2 towards the value returned from this method. To make it count as one, use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfSequencesWithAnnotationsOfType(java.lang.String)"><code>getNumberOfSequencesWithAnnotationsOfType(String)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotationType</code> - the type of annotation (e.g. <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CDS"><code>SequenceAnnotation.TYPE_CDS</code></a>)</dd>
<dt><span class="strong">Returns:</span></dt><dd>the number of annotations of this type</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfAnnotations()"><code>getNumberOfAnnotations()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfSequencesWithAnnotationsOfType(java.lang.String)"><code>getNumberOfSequencesWithAnnotationsOfType(String)</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfSequencesWithAnnotationsOfType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequencesWithAnnotationsOfType</h4>
<pre>public&nbsp;long&nbsp;getNumberOfSequencesWithAnnotationsOfType(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;annotationType)</pre>
<div class="block">Get the total number of sequences that have annotations of the given type present.
 If a sequence has 2 annotations of the same type, that counts as 1 towards the value returned from this method. To make it count as 2, use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfAnnotationsOfType(java.lang.String)"><code>getNumberOfAnnotationsOfType(String)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotationType</code> - the type of annotation (e.g. <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CDS"><code>SequenceAnnotation.TYPE_CDS</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>the number of annotations of this type</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfAnnotations()"><code>getNumberOfAnnotations()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getNumberOfAnnotationsOfType(java.lang.String)"><code>getNumberOfAnnotationsOfType(String)</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfTrimmedSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfTrimmedSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfTrimmedSequences()</pre>
<div class="block">Gets the total number of sequences that either have an annotation of type <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>SequenceAnnotation.TYPE_TRIMMED</code></a> or have <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getLeadingTrimLength()"><code>ImmutableSequence.getLeadingTrimLength()</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getTrailingTrimLength()"><code>ImmutableSequence.getTrailingTrimLength()</code></a> greater than 0.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the total number of sequences that are trimmed.</dd></dl>
</li>
</ul>
<a name="getSequenceLengthsExcludingGaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLengthsExcludingGaps</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a>&nbsp;getSequenceLengthsExcludingGaps()</pre>
<div class="block">Gets the distribution of ungapped sequence lengths of all sequences managed by this summary.
 For example if a sequence is "--GATT--ACA----" then its ungapped length is 7.
 <p/><p/>
 This method can not be called until <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has been called.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the distribution of ungapped sequence lengths of all sequences managed by this summary</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has not yet been called.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getSequenceLengthsExcludingGapsTrimmed()"><code>getSequenceLengthsExcludingGapsTrimmed()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceLengthsExcludingGapsTrimmed()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLengthsExcludingGapsTrimmed</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a>&nbsp;getSequenceLengthsExcludingGapsTrimmed()</pre>
<div class="block">Gets the distribution of ungapped sequence lengths of all sequences managed by this summary after trim annotations have been applied.
 If there are no trim annotations present, the lengths are the same as those returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getSequenceLengthsExcludingGaps()"><code>getSequenceLengthsExcludingGaps()</code></a>.
 For example if a sequence is "--GATT--ACA----" and it has a trim annotation covering the "GA" then its ungapped trimmed length is 5.
 <p/><p/>
 This method can not be called until <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has been called.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the distribution of ungapped sequence lengths of all sequences managed by this summary after trim annotations have been applied.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has not yet been called.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getSequenceLengthsExcludingGaps()"><code>getSequenceLengthsExcludingGaps()</code></a></dd></dl>
</li>
</ul>
<a name="getQualitiesExcludingGaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getQualitiesExcludingGaps</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a>&nbsp;getQualitiesExcludingGaps()</pre>
<div class="block">Returns the distribution of quality values (those returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)"><code>NucleotideGraph.getSequenceQuality(int)</code></a>) across all nucleotides that are non-gaps in all sequences.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the distribution of quality values across all non-gap nucleotides in all sequences.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has not yet been called.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getQualitiesExcludingGapsTrimmed()"><code>getQualitiesExcludingGapsTrimmed()</code></a></dd></dl>
</li>
</ul>
<a name="getQualitiesExcludingGapsTrimmed()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getQualitiesExcludingGapsTrimmed</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ValueAndFrequencyDistribution.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ValueAndFrequencyDistribution</a>&nbsp;getQualitiesExcludingGapsTrimmed()</pre>
<div class="block">Returns the distribution of quality values (those returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html#getSequenceQuality(int)"><code>NucleotideGraph.getSequenceQuality(int)</code></a>) across all non-gap non-trimmed nucleotides in all sequences.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the distribution of quality values across all non-gap non-trimmed nucleotides in all sequences.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#finishedAddingSequences(jebl.util.ProgressListener)"><code>finishedAddingSequences(jebl.util.ProgressListener)</code></a> has not yet been called.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getQualitiesExcludingGaps()"><code>getQualitiesExcludingGaps()</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfCircularSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfCircularSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfCircularSequences()</pre>
<div class="block">Returns the number of sequences in the list for which <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a> is true</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences in the list for which <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a> is true</dd></dl>
</li>
</ul>
<a name="getCoverage(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCoverage</h4>
<pre>public&nbsp;int&nbsp;getCoverage(int&nbsp;residueIndex)</pre>
<div class="block">If this summary is for an alignment, gets the total number of sequences covering this position.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the 0-based index of the position to get the coverage for</dd>
<dt><span class="strong">Returns:</span></dt><dd>the coverage</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd></dl>
</li>
</ul>
<a name="getCoverageAsDouble(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCoverageAsDouble</h4>
<pre>public&nbsp;double&nbsp;getCoverageAsDouble(int&nbsp;residueIndex)</pre>
<div class="block">If this summary is for an alignment, gets the total number of sequences covering this position.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the 0-based index of the position to get the coverage for</dd>
<dt><span class="strong">Returns:</span></dt><dd>the coverage</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getCoverage(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCoverage</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Coverage.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Coverage</a>&nbsp;getCoverage(int&nbsp;minResidueIndex,
                                       int&nbsp;maxResidueIndexExclusive)</pre>
<div class="block">If this summary is for an alignment, gets an approximate coverage for the given range. The approximation is that
 for performance reasons it may return a coverage for a slightly larger range of values than those provided.
 For example if residues 70 through 90 are requested, the calling code may return values for the range 64 to 96.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minResidueIndex</code> - the first 0-based index of the position to get the coverage for</dd><dd><code>maxResidueIndexExclusive</code> - the final 0-based index (exclusive) of the position to get the coverage for</dd>
<dt><span class="strong">Returns:</span></dt><dd>an approximation of the coverage over this range</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd></dl>
</li>
</ul>
<a name="getColumnsWithGapsOverPercentage(double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getColumnsWithGapsOverPercentage</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/BitSet.html?is-external=true" title="class or interface in java.util">BitSet</a>&nbsp;getColumnsWithGapsOverPercentage(double&nbsp;percentage)</pre>
<div class="block">If this summary is for an alignment gets a BitSet indicating which columns contain over the given percentage of gaps.
 This method may return null if this information is not cached with this summary. For circular alignments, the BitSet
 is only equal in length to circular alignment length.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>percentage</code> - the percentage</dd>
<dt><span class="strong">Returns:</span></dt><dd>a BitSet where <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/BitSet.html?is-external=true#get(int)" title="class or interface in java.util"><code>BitSet.get(int)</code></a> is true for each column that contains over the given percentage of gaps, or null if this information is unavailable.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
<a name="getCoverageMaximum()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCoverageMaximum</h4>
<pre>public&nbsp;int&nbsp;getCoverageMaximum()</pre>
<div class="block">Returns the maximum coverage in this alignment.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum coverage in this alignment</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd></dl>
</li>
</ul>
<a name="getCoverageMaximumAsDouble()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCoverageMaximumAsDouble</h4>
<pre>public&nbsp;double&nbsp;getCoverageMaximumAsDouble()</pre>
<div class="block">Returns the maximum coverage in this alignment.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the maximum coverage in this alignment</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getCoverageMean()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCoverageMean</h4>
<pre>public&nbsp;double&nbsp;getCoverageMean()</pre>
<div class="block">Returns the mean coverage in this alignment.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the mean coverage in this alignment</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd></dl>
</li>
</ul>
<a name="getCoverageStandardDeviation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCoverageStandardDeviation</h4>
<pre>public&nbsp;double&nbsp;getCoverageStandardDeviation()</pre>
<div class="block">Returns the standard deviation of coverage in this alignment.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the standard deviation of coverage in this alignment.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd></dl>
</li>
</ul>
<a name="getAlignmentPairwiseSimilarity(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlignmentPairwiseSimilarity</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.PairwiseSimilarity.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.PairwiseSimilarity</a>&nbsp;getAlignmentPairwiseSimilarity(boolean&nbsp;freeEndGaps)</pre>
<div class="block">Get the pairwise similarity if this summary is for an alignment</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>freeEndGaps</code> - true end gaps should contribute towards the similarity</dd>
<dt><span class="strong">Returns:</span></dt><dd>the pairwise similarity</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd></dl>
</li>
</ul>
<a name="getAlignmentNumberOfColumns()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlignmentNumberOfColumns</h4>
<pre>public&nbsp;int&nbsp;getAlignmentNumberOfColumns()</pre>
<div class="block">If this summary is for an alignment, returns the number of columns in the alignment</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>If this summary is for an alignment, the number of columns in the alignment</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment.</dd></dl>
</li>
</ul>
<a name="getNumberOfReversedSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfReversedSequences</h4>
<pre>public&nbsp;long&nbsp;getNumberOfReversedSequences()</pre>
<div class="block">Returns the number of sequences whose name ends with <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>SequenceExtractionUtilities.REVERSED_NAME_SUFFIX</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences whose name ends with <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>SequenceExtractionUtilities.REVERSED_NAME_SUFFIX</code></a></dd></dl>
</li>
</ul>
<a name="getReferenceSequenceBases()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceSequenceBases</h4>
<pre>public&nbsp;int&nbsp;getReferenceSequenceBases()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of non-gap bases/residues in the reference sequence for an alignment/contig</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getReferenceSequenceBasesCovered()"><code>getReferenceSequenceBasesCovered()</code></a></dd></dl>
</li>
</ul>
<a name="getReferenceSequenceBasesCovered()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceSequenceBasesCovered</h4>
<pre>public&nbsp;int&nbsp;getReferenceSequenceBasesCovered()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of non-gap bases/residues in the reference sequence that have at least 1 read covering them in an alignment/contig</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this summary is not for an alignment</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html#getReferenceSequenceBases()"><code>getReferenceSequenceBases()</code></a></dd></dl>
</li>
</ul>
<a name="hasAnyReadsMappedToMultipleLocations()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>hasAnyReadsMappedToMultipleLocations</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Boolean.html?is-external=true" title="class or interface in java.lang">Boolean</a>&nbsp;hasAnyReadsMappedToMultipleLocations()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>null if this summary was created prior to Geneious 6.0. True if this summary is for an alignment and there is at least 1 read in the alignment mapped to more than 1 location as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#NUMBER_OF_MAPPED_LOCATIONS"><code>DocumentField.NUMBER_OF_MAPPED_LOCATIONS</code></a>, false if not.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceListSummary.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Coverage.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" target="_top">Frames</a></li>
<li><a href="SequenceListSummary.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
