<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:47 NZDT 2016 -->
<title>SequenceListOnDisk.Builder (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceListOnDisk.Builder (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceListOnDisk.Builder.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" target="_top">Frames</a></li>
<li><a href="SequenceListOnDisk.Builder.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Class SequenceListOnDisk.Builder" class="title">Class SequenceListOnDisk.Builder&lt;T extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk.Builder&lt;T&gt;</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl><dt><span class="strong">Type Parameters:</span></dt><dd><code>T</code> - the type of sequence (<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a>)</dd></dl>
<dl>
<dt>Enclosing class:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="type parameter in SequenceListOnDisk">T</a> extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</dd>
</dl>
<hr>
<br>
<pre>public static class <span class="strong">SequenceListOnDisk.Builder&lt;T extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Used for building a SequenceListOnDisk.
 <p/>
 The simplest way to use this builder is to call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence</code></a> for each sequence to be added,
 then call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument</code></a> when done.

 However, to improve performance when using sequence compression (see tryCompressingSequences parameter in constructor) it is recommended a first pass be made of the sequences, providing their names to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addNameOfSequence(java.lang.String)"><code>addNameOfSequence</code></a>
 before making the 2nd pass using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence</code></a>
 <p/>
 This class is not thread safe.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#SequenceListOnDisk.Builder(boolean, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean)">SequenceListOnDisk.Builder</a></strong>(boolean&nbsp;tryCompressingSequences,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                          boolean&nbsp;allowGaps)</code>
<div class="block">Create a new builder</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#SequenceListOnDisk.Builder(boolean, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean, jebl.util.ProgressListener, int)">SequenceListOnDisk.Builder</a></strong>(boolean&nbsp;tryCompressingSequences,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                          boolean&nbsp;allowGaps,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListenerForSortingAlignment,
                          int&nbsp;maximumNumberOfSequencesThatWillBeAdded)</code>
<div class="block">Create a new builder</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#SequenceListOnDisk.Builder(boolean, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean, jebl.util.ProgressListener, int, boolean)">SequenceListOnDisk.Builder</a></strong>(boolean&nbsp;tryCompressingSequences,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                          boolean&nbsp;allowGaps,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListenerForSortingAlignment,
                          int&nbsp;maximumNumberOfSequencesThatWillBeAdded,
                          boolean&nbsp;createMultipleThreads)</code>
<div class="block">Create a new builder</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addAlignmentReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)">addAlignmentReferenceSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;referenceSequence,
                             <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Adds a reference sequence when building an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addNameOfSequence(java.lang.String)">addNameOfSequence</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName)</code>
<div class="block">Adds the name of a sequence that will be later added using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence(SequenceDocument, jebl.util.ProgressListener)</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)">addSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Adds a new unpaired sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequenceWithMate(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, int, int, jebl.util.ProgressListener)">addSequenceWithMate</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                   <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;mateSequence,
                   int&nbsp;expectedDistance1,
                   int&nbsp;expectedDistance2,
                   <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Similar to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence</code></a> but this version adds 2 sequences which are paired</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedContigSizeForCreatingContigsOnDisk()">getMinimumSuggestedContigSizeForCreatingContigsOnDisk</a></strong>()</code>
<div class="block">Returns a suggested minimum number of sequences that should be in a SequenceListOnDisk.Builder when creating one.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()">getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk</a></strong>()</code>
<div class="block">Returns a suggested minimum sequence length for the reference sequence in a SequenceListOnDisk.Builder when creating one.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>
<div class="block">Returns the number of sequences added so far to this builder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="type parameter in SequenceListOnDisk.Builder">T</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getSequenceList()">getSequenceList</a></strong>()</code>
<div class="block">Gets the result returned from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#setCircularAlignmentLength(int)">setCircularAlignmentLength</a></strong>(int&nbsp;circularAlignmentLength)</code>
<div class="block">Sets the alignment being built as circular as specified by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()"><code>SequenceAlignmentDocument.getCircularLength()</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#shouldCreateContigOnDisk(int, int)">shouldCreateContigOnDisk</a></strong>(int&nbsp;sequenceCount,
                        int&nbsp;referenceLength)</code>
<div class="block">Returns true if the sequence count and/or reference length of this contig are sufficient for it to be recommended to create a contig on disk.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#shouldCreateContigOnDisk(int, int, long)">shouldCreateContigOnDisk</a></strong>(int&nbsp;sequenceCount,
                        int&nbsp;referenceLength,
                        long&nbsp;totalLengthOfMappedReads)</code>
<div class="block">Returns true if the sequence count and/or reference length and/or total length of all the mapped reads in this contig are sufficient for it to be recommended to create a contig on disk.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)">toAlignmentDocument</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Creates an Alignment document from this builder.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="type parameter in SequenceListOnDisk.Builder">T</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)">toSequenceList</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Creates a SequenceListOnDisk from this builder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)">toSequenceListDocument</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Creates a new SequenceListDocument from this builder.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceListOnDisk.Builder(boolean, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceListOnDisk.Builder</h4>
<pre>public&nbsp;SequenceListOnDisk.Builder(boolean&nbsp;tryCompressingSequences,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                          boolean&nbsp;allowGaps)
                           throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Create a new builder</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>tryCompressingSequences</code> - true if we should analyze sequences names for compressible patterns and convert to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a> where possible. When using compression, for optimal performance, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addNameOfSequence(java.lang.String)"><code>addNameOfSequence(String)</code></a> should be called on a first pass for all sequences before calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence(SequenceDocument, jebl.util.ProgressListener)</code></a></dd><dd><code>allowGaps</code> - true if gaps are allowed in the sequences. This should be false when constructing sequences lists documents (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a>)
 and true when constructing alignments (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument</code></a>)
 When allowGaps is true, sequences will be sorted according to the number of leading gaps (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html#getLeadingGapsLength()"><code>SequenceCharSequence.getLeadingGapsLength</code></a>) they have.</dd><dd><code>alphabet</code> - the alphabet (nucleotide or amino acid) of the sequences to be added. If the calling code chooses to use generics on this builder, it is up to it to ensure the builder type matches this parameter. The generics are useful for use with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList(jebl.util.ProgressListener)</code></a>, but for code that will only be using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument(jebl.util.ProgressListener)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument(jebl.util.ProgressListener)</code></a> on this builder, they can safely not use generics.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to a temporary file on disk.</dd></dl>
</li>
</ul>
<a name="SequenceListOnDisk.Builder(boolean, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean, jebl.util.ProgressListener, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceListOnDisk.Builder</h4>
<pre>public&nbsp;SequenceListOnDisk.Builder(boolean&nbsp;tryCompressingSequences,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                          boolean&nbsp;allowGaps,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListenerForSortingAlignment,
                          int&nbsp;maximumNumberOfSequencesThatWillBeAdded)
                           throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Create a new builder</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>tryCompressingSequences</code> - true if we should analyze sequences names for compressible patterns and convert to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a> where possible. When using compression, for optimal performance, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addNameOfSequence(java.lang.String)"><code>addNameOfSequence(String)</code></a> should be called on a first pass for all sequences before calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence(SequenceDocument, jebl.util.ProgressListener)</code></a></dd><dd><code>allowGaps</code> - true if gaps are allowed in the sequences. This should be false when constructing sequences lists documents (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a>)
 and true when constructing alignments (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument</code></a>)
 When allowGaps is true, sequences will be sorted according to the number of leading gaps they have.</dd><dd><code>alphabet</code> - the alphabet (nucleotide or amino acid) of the sequences to be added. If the calling code chooses to use generics on this builder, it is up to it to ensure the builder type matches this parameter. The generics are useful for use with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList(jebl.util.ProgressListener)</code></a>, but for code that will only be using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument(jebl.util.ProgressListener)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument(jebl.util.ProgressListener)</code></a> on this builder, they can safely not use generics.</dd><dd><code>progressListenerForSortingAlignment</code> - when creating alignments, sorting and saving may need to be report progress prior to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument(jebl.util.ProgressListener)</code></a> is called. This progress listener will continue to have progress reported to it after returning from this method.</dd><dd><code>maximumNumberOfSequencesThatWillBeAdded</code> - an upper bound number of sequences that will be added to the alignment builder, or -1 if unknown. Progress reporting will be inaccurate if -1 is provided.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to a temporary file on disk.</dd></dl>
</li>
</ul>
<a name="SequenceListOnDisk.Builder(boolean, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet, boolean, jebl.util.ProgressListener, int, boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceListOnDisk.Builder</h4>
<pre>public&nbsp;SequenceListOnDisk.Builder(boolean&nbsp;tryCompressingSequences,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet,
                          boolean&nbsp;allowGaps,
                          <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListenerForSortingAlignment,
                          int&nbsp;maximumNumberOfSequencesThatWillBeAdded,
                          boolean&nbsp;createMultipleThreads)
                           throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Create a new builder</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>tryCompressingSequences</code> - true if we should analyze sequences names for compressible patterns and convert to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a> where possible. When using compression, for optimal performance, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addNameOfSequence(java.lang.String)"><code>addNameOfSequence(String)</code></a> should be called on a first pass for all sequences before calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence(SequenceDocument, jebl.util.ProgressListener)</code></a></dd><dd><code>allowGaps</code> - true if gaps are allowed in the sequences. This should be false when constructing sequences lists documents (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a>)
 and true when constructing alignments (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument</code></a>)
 When allowGaps is true, sequences will be sorted according to the number of leading gaps they have.</dd><dd><code>alphabet</code> - the alphabet (nucleotide or amino acid) of the sequences to be added. If the calling code chooses to use generics on this builder, it is up to it to ensure the builder type matches this parameter. The generics are useful for use with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList(jebl.util.ProgressListener)</code></a>, but for code that will only be using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument(jebl.util.ProgressListener)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument(jebl.util.ProgressListener)</code></a> on this builder, they can safely not use generics.</dd><dd><code>progressListenerForSortingAlignment</code> - when creating alignments, sorting and saving may need to be report progress prior to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument(jebl.util.ProgressListener)</code></a> is called. This progress listener will continue to have progress reported to it after returning from this method.</dd><dd><code>maximumNumberOfSequencesThatWillBeAdded</code> - an upper bound number of sequences that will be added to the alignment builder, or -1 if unknown. Progress reporting will be inaccurate if -1 is provided.</dd><dd><code>createMultipleThreads</code> - true to create multiple threads to improve performance when necessary. Normally this should be true except in cases where code is creating many SequenceListOnDisk.Builders at once in which case you might run out of system threads.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to a temporary file on disk.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.800 (Geneious 8.0.0)</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getMinimumSuggestedContigSizeForCreatingContigsOnDisk()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMinimumSuggestedContigSizeForCreatingContigsOnDisk</h4>
<pre>public static&nbsp;int&nbsp;getMinimumSuggestedContigSizeForCreatingContigsOnDisk()</pre>
<div class="block">Returns a suggested minimum number of sequences that should be in a SequenceListOnDisk.Builder when creating one. There is a bit of overhead for each contig creating on disk versions so it is better
 to create contigs with a small number of sequences and a short reference using the standard <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocument</code></a> constructors. For SequenceLists, the overhead is small so creating a SequenceListOnDisk
 for even a list of 2 sequences is fine.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a suggested minimum number of sequences that should be in a contig created from a SequenceListOnDisk.Builder when creating one.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()"><code>getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#shouldCreateContigOnDisk(int, int)"><code>shouldCreateContigOnDisk(int, int)</code></a></dd></dl>
</li>
</ul>
<a name="getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk</h4>
<pre>public static&nbsp;int&nbsp;getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()</pre>
<div class="block">Returns a suggested minimum sequence length for the reference sequence in a SequenceListOnDisk.Builder when creating one. There is a bit of overhead for each contig creating on disk versions so it is better
 to create contigs with a small number of sequences and a short reference using the standard <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocument</code></a> constructors.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a suggested minimum sequence length for the reference sequence in a contig created from a SequenceListOnDisk.Builder when creating one.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedContigSizeForCreatingContigsOnDisk()"><code>getMinimumSuggestedContigSizeForCreatingContigsOnDisk()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#shouldCreateContigOnDisk(int, int)"><code>shouldCreateContigOnDisk(int, int)</code></a></dd></dl>
</li>
</ul>
<a name="shouldCreateContigOnDisk(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>shouldCreateContigOnDisk</h4>
<pre>public static&nbsp;boolean&nbsp;shouldCreateContigOnDisk(int&nbsp;sequenceCount,
                               int&nbsp;referenceLength)</pre>
<div class="block">Returns true if the sequence count and/or reference length of this contig are sufficient for it to be recommended to create a contig on disk.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceCount</code> - number of sequences in the contig</dd><dd><code>referenceLength</code> - length of the reference sequence or -1 if no reference.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if the sequence count and/or reference length of this contig are sufficient for it to be recommended to create a contig on disk.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()"><code>getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedContigSizeForCreatingContigsOnDisk()"><code>getMinimumSuggestedContigSizeForCreatingContigsOnDisk()</code></a></dd></dl>
</li>
</ul>
<a name="shouldCreateContigOnDisk(int, int, long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>shouldCreateContigOnDisk</h4>
<pre>public static&nbsp;boolean&nbsp;shouldCreateContigOnDisk(int&nbsp;sequenceCount,
                               int&nbsp;referenceLength,
                               long&nbsp;totalLengthOfMappedReads)</pre>
<div class="block">Returns true if the sequence count and/or reference length and/or total length of all the mapped reads in this contig are sufficient for it to be recommended to create a contig on disk.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceCount</code> - number of sequences in the contig</dd><dd><code>referenceLength</code> - length of the reference sequence or -1 if no reference.</dd><dd><code>totalLengthOfMappedReads</code> - total length of all mapped reads or -1 if unknown</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if the sequence count and/or reference length and/or total length of all the mapped reads in this contig are sufficient for it to be recommended to create a contig on disk.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()"><code>getMinimumSuggestedReferenceLengthForCreatingContigsOnDisk()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#getMinimumSuggestedContigSizeForCreatingContigsOnDisk()"><code>getMinimumSuggestedContigSizeForCreatingContigsOnDisk()</code></a></dd></dl>
</li>
</ul>
<a name="addNameOfSequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addNameOfSequence</h4>
<pre>public&nbsp;boolean&nbsp;addNameOfSequence(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName)</pre>
<div class="block">Adds the name of a sequence that will be later added using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence(SequenceDocument, jebl.util.ProgressListener)</code></a>. It is optional to make
 a first pass of the sequences providing their names, but doing so will improve performance if tryCompressingSequences==true was passed to the builder constructor.
 This method should not be called for the name of the reference sequence that will be passed to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addAlignmentReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addAlignmentReferenceSequence</code></a> (if any).</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceName</code> - the name of the sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>false if a compressible pattern in the sequence names could not be found, and further calls to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addNameOfSequence(java.lang.String)"><code>addNameOfSequence(String)</code></a> will be ignored. The calling code could skip straight to the 2nd pass in this case.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence</code></a> has already been called</dd></dl>
</li>
</ul>
<a name="setCircularAlignmentLength(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCircularAlignmentLength</h4>
<pre>public&nbsp;void&nbsp;setCircularAlignmentLength(int&nbsp;circularAlignmentLength)</pre>
<div class="block">Sets the alignment being built as circular as specified by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()"><code>SequenceAlignmentDocument.getCircularLength()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>circularAlignmentLength</code> - the circular alignment length or 0 for not circular (<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()"><code>SequenceAlignmentDocument.getCircularLength()</code></a>)</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this is called after adding any sequences or if allowGaps==true wasn't passed to the constructor.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="addSequenceWithMate(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, int, int, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequenceWithMate</h4>
<pre>public&nbsp;void&nbsp;addSequenceWithMate(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                       <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;mateSequence,
                       int&nbsp;expectedDistance1,
                       int&nbsp;expectedDistance2,
                       <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                         throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Similar to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence</code></a> but this version adds 2 sequences which are paired</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to add</dd><dd><code>mateSequence</code> - the mate sequence to add</dd><dd><code>expectedDistance1</code> - the expected distance from the first sequence to its mate. Must not be 0.</dd><dd><code>expectedDistance2</code> - the expected distance from the second sequence to its mate. This must be equal to either expectedDistance1 or -1*expectedDistance1, depending on the relative orientation of the pairs. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a> for expected distance meanings.</dd><dd><code>progressListener</code> - for reporting progress if this is a large sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to disk or if the progressListener indicates we should cancel.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList(jebl.util.ProgressListener)</code></a> has already been called on this builder or if the sequence is not a NucleotideSequenceDocument or AminoAcidSequenceDocument</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if the sequence contains gaps when not allowed, or if it contains invalid nucleotide or amino acid characters or if invalid expected distances are provided.</dd></dl>
</li>
</ul>
<a name="addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequence</h4>
<pre>public&nbsp;void&nbsp;addSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
               <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                 throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Adds a new unpaired sequence. <p/>
 If this builder is for a sequence list, then paired sequences can added either by using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequenceWithMate(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, int, int, jebl.util.ProgressListener)"><code>addSequenceWithMate(SequenceDocument, SequenceDocument, int, int, jebl.util.ProgressListener)</code></a> or can be be paired later by using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html#setPairedReadsManager(com.biomatters.geneious.publicapi.implementations.PairedReadManager)"><code>DefaultSequenceListDocument.setPairedReadsManager</code></a> on the result of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a>
 <p/>
 If this list is being built for use with an alignment, since sequences will get sorted by leading gaps, paired sequences need to be added using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequenceWithMate(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, int, int, jebl.util.ProgressListener)"><code>addSequenceWithMate</code></a>.
 <p/>
 This method should not be called to add the reference sequence to a contig.
 Instead <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addAlignmentReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addAlignmentReferenceSequence(SequenceDocument, jebl.util.ProgressListener addAlignmentReferenceSequence)</code></a> should be called prior to adding any sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to add</dd><dd><code>progressListener</code> - for reporting progress if this is a large sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to disk or if the progressListener indicates we should cancel.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList(jebl.util.ProgressListener)</code></a> has already been called on this builder or if the sequence is not a NucleotideSequenceDocument or AminoAcidSequenceDocument</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if the sequence contains gaps when not allowed, or if it contains invalid nucleotide or amino acid characters.</dd></dl>
</li>
</ul>
<a name="addAlignmentReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addAlignmentReferenceSequence</h4>
<pre>public&nbsp;void&nbsp;addAlignmentReferenceSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;referenceSequence,
                                 <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                   throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Adds a reference sequence when building an alignment. Only a single reference sequence may be added, and it must be
 added prior to other sequences via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequence</code> - the reference sequence</dd><dd><code>progressListener</code> - for reporting progress if this is a large sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to disk or if the progressListener indicates we should cancel.</dd></dl>
</li>
</ul>
<a name="toSequenceList(jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toSequenceList</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="type parameter in SequenceListOnDisk.Builder">T</a>&gt;&nbsp;toSequenceList(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                                              throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Creates a SequenceListOnDisk from this builder. In most situations you probably want to use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument</code></a> instead.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>progressListener</code> - for reporting progress and cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a SequenceListOnDisk from this builder.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to disk or if the progressListener indicates we should cancel.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this method or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument(jebl.util.ProgressListener)</code></a> has already been called or if no sequences have been added</dd></dl>
</li>
</ul>
<a name="getSequenceList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceList</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="type parameter in SequenceListOnDisk.Builder">T</a>&gt;&nbsp;getSequenceList()</pre>
<div class="block">Gets the result returned from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList</code></a>. In most situations you probably want to use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument</code></a> instead.
 <p/>
 This method need not be used unless you are constructing your own alignment or sequence list implementation rather than using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a SequenceListOnDisk from this builder.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if at least one of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceListDocument(jebl.util.ProgressListener)"><code>toSequenceListDocument(jebl.util.ProgressListener)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList(jebl.util.ProgressListener)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toAlignmentDocument(jebl.util.ProgressListener)"><code>toAlignmentDocument(jebl.util.ProgressListener)</code></a> have not been called yet</dd></dl>
</li>
</ul>
<a name="toSequenceListDocument(jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toSequenceListDocument</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/DefaultSequenceListDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">DefaultSequenceListDocument</a>&nbsp;toSequenceListDocument(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                                   throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Creates a new SequenceListDocument from this builder.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>progressListener</code> - for reporting progress and cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a DefaultSequenceListDocument from this builder whose sequences are not loaded into memory.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to disk or if the progressListener indicates we should cancel.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this method or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#toSequenceList(jebl.util.ProgressListener)"><code>toSequenceList(jebl.util.ProgressListener)</code></a> has already been called
 or if <code>allowGaps</code> is true for this builder.</dd></dl>
</li>
</ul>
<a name="toAlignmentDocument(jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toAlignmentDocument</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a>&nbsp;toAlignmentDocument(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                             throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Creates an Alignment document from this builder. This method may only be called if allowGaps==true was passed to the constructor <code>Builder#Builder(boolean, SequenceDocument.Alphabet, boolean)</code>.
 For constructing reference sequence alignments, use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addAlignmentReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addAlignmentReferenceSequence</code></a> prior to adding
 sequences using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, jebl.util.ProgressListener)"><code>addSequence</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>progressListener</code> - for reporting progress and cancelling.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a DefaultAlignmentDocument from this builder whose sequences are not loaded into memory.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if we can't write to disk or if the progressListener indicates we should cancel.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if <code>allowGaps</code> is false for this builder</dd></dl>
</li>
</ul>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public&nbsp;int&nbsp;getNumberOfSequences()</pre>
<div class="block">Returns the number of sequences added so far to this builder.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences added so far to this builder.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceListOnDisk.Builder.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" target="_top">Frames</a></li>
<li><a href="SequenceListOnDisk.Builder.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
