<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>SequenceAlignmentDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceAlignmentDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceAlignmentDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" target="_top">Frames</a></li>
<li><a href="SequenceAlignmentDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Class SequenceAlignmentDocument" class="title">Class SequenceAlignmentDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="strong">SequenceAlignmentDocument</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></pre>
<div class="block">An alignment or contig of sequences. The only difference between an alignment or contig (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isContig()"><code>isContig()</code></a>) is how they are displayed to the user. At the code level
 they are identical. Alignments and contigs may have a reference sequence (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>getContigReferenceSequenceIndex()</code></a>)
 <p/>
 Unless the alignment is editable (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a>), the only methods that need to be implemented are <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)"><code>getSequence(int)</code></a>.
 <p/>
 To construct an alignment, it is recommended you use a <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocument</code></a>
 or to construct alignments that are too large to fit in memory, use a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.Builder.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.Builder</code></a>
 <p/></div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></strong></code>
<div class="block">Represents a reference to the original document that a sequence in this alignment was aligned from.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#SequenceAlignmentDocument()">SequenceAlignmentDocument</a></strong>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Add a sequence to the end of the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignment(boolean)">asJeblAlignment</a></strong>(boolean&nbsp;silentlyStripDuplicates)</code>
<div class="block">Get alignment as a Jebl alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignmentIfMemoryAllows(boolean)">asJeblAlignmentIfMemoryAllows</a></strong>(boolean&nbsp;silentlyStripDuplicates)</code>
<div class="block">This method calls #asJeblAlignment(boolean) and re-throws any OutOfMemoryError as a DocumentOperationException</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a></strong>()</code>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()">canSetAnnotations</a></strong>()</code>
<div class="block">Returns true if this alignment can have its annotations on sequences (including the consensus sequence) edited alone via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>setAnnotations(int, java.util.List, boolean)</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>setAnnotationsOnConsensus(java.util.List)</code></a>
 Note that some alignment implementations may return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a> and allow updating their annotations via
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a> even if they return false from this method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()">canSetAnnotationsOnConsensus</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">canSetFieldValue</a></strong>(int&nbsp;sequenceIndex,
                <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field)</code>
<div class="block">Returns true if this alignment supports updating field values using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()">canSetSequenceNames</a></strong>()</code>
<div class="block">Specifies whether this alignment supports changing the sequence names.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()">getAnnotationsOnConsensus</a></strong>()</code>
<div class="block">Gets annotations that are associated with the consensus sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()">getCircularLength</a></strong>()</code>
<div class="block">If this alignment is circular, returns the gapped length at which it wraps back to the start or 0 if this alignment is not circular.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()">getContigReferenceSequenceIndex</a></strong>()</code>
<div class="block">Get the index (if any) of the reference sequence in a contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></strong>()</code>
<div class="block">An immutable view of the paired reads managed by this list or alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateIndex(int)">getMateIndex</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>
<div class="block">Number of sequences in this alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRealignedName(java.lang.String)">getRealignedName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Given the name of an existing alignment, obtain a name that should
 be used for a realignment of this alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)">getReferencedDocument</a></strong>(int&nbsp;sequenceIndexInAlignment)</code>
<div class="block">Get an original sequence document that this alignment was made from.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocuments()">getReferencedDocuments</a></strong>()</code>
<div class="block">Return the results of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a> for each sequence in the
 alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)">getReferencedSequence</a></strong>(int&nbsp;sequenceIndexInAlignment)</code>
<div class="block">Get an original sequence document that this alignment was made from.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequences()">getReferencedSequences</a></strong>()</code>
<div class="block">Return the results of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)"><code>getReferencedSequence(int)</code></a> for each sequence in the
 alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRefinedName(java.lang.String)">getRefinedName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Given the name of an existing alignment, obtain a name that should
 be used for a refinement of this alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)">getSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Gets one row from the alignment as a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceAsJeblSequence(int)">getSequenceAsJeblSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Gets a sequence as a jebl sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()">getSequences</a></strong>()</code>
<div class="block">Get the alignment as a list of sequences (Convenience function).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()">getSequencesInImmutableList</a></strong>()</code>
<div class="block">Gets an immutable list (whose elements will not change after returning from this method).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceType()">getSequenceType</a></strong>()</code>
<div class="block">Returns the sequence type of the first sequence in this alignment (which should be the same as all other sequences in the alignment)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isContig()">isContig</a></strong>()</code>
<div class="block">Returns false by default.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()">isEditable</a></strong>()</code>
<div class="block">true if this alignment is editable.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a></strong>()</code>
<div class="block">True if this alignment was built using free end gaps.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isQueryCentricAlignment()">isQueryCentricAlignment</a></strong>()</code>
<div class="block">Returns true if this alignment is a query centric alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isReferencedDocumentReversed(int)">isReferencedDocumentReversed</a></strong>(int&nbsp;index)</code>
<div class="block">Sequence direction.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)">removeSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Removes the specified sequence from the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])">reorderSequences</a></strong>(int[]&nbsp;ordering)</code>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)">setAnnotations</a></strong>(int&nbsp;index,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
              boolean&nbsp;allowModifyingReferencedDocuments)</code>
<div class="block">Updates just the annotations on a sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)">setAnnotationsOnConsensus</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets the annotations on the consensus sequence that will be returned from a later call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>getAnnotationsOnConsensus()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">setFieldValue</a></strong>(int&nbsp;sequenceIndex,
             <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field,
             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;fieldValue,
             boolean&nbsp;allowModifyingOriginals)</code>
<div class="block">Sets the value of a displayable field on one of the sequences in the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setMates(int, int, int)">setMates</a></strong>(int&nbsp;sequenceIndex,
        int&nbsp;sequenceIndexOfMate,
        int&nbsp;expectedMateDistance)</code>
<div class="block">Records the 2nd sequence as being paired with the first sequence with the given distance.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">setReferencedDocument</a></strong>(int&nbsp;sequenceIndexInAlignment,
                     <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</code>
<div class="block">Sets the sequence at the given sequenceIndexInAlignment in this alignment to refer to the given document as if it is the original source.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">setReferencedSequence</a></strong>(int&nbsp;sequenceIndexInAlignment,
                     <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence)</code>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)"><code>setReferencedSequence(sequenceIndexInAlignment, referencedSequence, -1</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">setReferencedSequence</a></strong>(int&nbsp;sequenceIndexInAlignment,
                     <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
                     int&nbsp;referencedDocumentRevisionNumberSequenceWasCopiedFrom)</code>
<div class="block">Sets the sequence at the given index in this alignment to refer to the given document as if it is the original source.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a></strong>(int&nbsp;index,
               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
               boolean&nbsp;allowModifyingReferencedDocuments)</code>
<div class="block">change the name of a sequence within the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#toHTML()">toHTML</a></strong>()</code>
<div class="block">Provide a textual view of document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></strong>(int&nbsp;index,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
              <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
              boolean&nbsp;allowModifyingReferencedDocuments)</code>
<div class="block">Set new residues and related data for one sequence.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceAlignmentDocument()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceAlignmentDocument</h4>
<pre>public&nbsp;SequenceAlignmentDocument()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public abstract&nbsp;int&nbsp;getNumberOfSequences()</pre>
<div class="block">Number of sequences in this alignment</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>number of sequences</dd></dl>
</li>
</ul>
<a name="getSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre>public abstract&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;getSequence(int&nbsp;index)</pre>
<div class="block">Gets one row from the alignment as a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>. As this sequence
 is an alignment row, it will normally contain gaps.

 If implementation stores the URN of the original sequence document, see 
 the note in <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence at the specified index.</dd></dl>
</li>
</ul>
<a name="getSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequences()</pre>
<div class="block">Get the alignment as a list of sequences (Convenience function).

 The default implementation returns an unmodifiable list view backed by
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)"><code>getSequence(int)</code></a>.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a list of sequences</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()"><code>getSequencesInImmutableList()</code></a></dd></dl>
</li>
</ul>
<a name="getSequencesInImmutableList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesInImmutableList</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequencesInImmutableList()</pre>
<div class="block">Gets an immutable list (whose elements will not change after returning from this method). However the contents of those sequences may change unless the sequences are instances of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a>.
 Despite that, this method is still useful since if sequences are deleted, added or replaced in this alignment, the returned list will not change.

 The default implementation of this method constructs a new list using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()"><code>getSequences()</code></a> and returns it.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>an immutable list (whose elements will not change after returning from this method). However the contents of those sequences may change unless the sequences are instances of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a>.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()"><code>getSequences()</code></a></dd></dl>
</li>
</ul>
<a name="getReferencedDocument(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedDocument</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;getReferencedDocument(int&nbsp;sequenceIndexInAlignment)</pre>
<div class="block">Get an original sequence document that this alignment was made from.

 This may be null if the alignment was not generated from
 sequence documents, or if the alignment has been user modified, so the reference
 to the original sequence document would not be relevant.

 Note that when an alignment implements this by storing the sequence URN and loading it
 on request it needs to check that the seqeunce residues has not been changed by a later edit
 operation. In that case the seqeunce document is no longer valid for that alignment and a null should be returned.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the index of the sequence in the alignment to get the original document for. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the original document or null if this alignment does not a reference an original document for this sequence index. The default implementation of this method returns null</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)"><code>getReferencedSequence(int)</code></a></dd></dl>
</li>
</ul>
<a name="getReferencedSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedSequence</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getReferencedSequence(int&nbsp;sequenceIndexInAlignment)</pre>
<div class="block">Get an original sequence document that this alignment was made from.

 This may be null if the alignment was not generated from
 sequence documents, or if the alignment has been user modified, so the reference
 to the original sequence document would not be relevant.

 Note that when an alignment implements this by storing the sequence URN and loading it
 on request it needs to check that the seqeunce residues has not been changed by a later edit
 operation. In that case the seqeunce document is no longer valid for that alignment and a null should be returned.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the index of the sequence in the alignment to get the original document for. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the original document or null if this alignment does not a reference an original document for this sequence index. The default implementation of this method forwards to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a></dd></dl>
</li>
</ul>
<a name="getReferencedDocuments()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedDocuments</h4>
<pre>public final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&gt;&nbsp;getReferencedDocuments()</pre>
<div class="block">Return the results of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a> for each sequence in the
 alignment. (Convenience function)

 This list will be equal length to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> with null elements
 for sequences with no referenced document. There may be a mixture of referenced and
 unreferenced documents in an alignment.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the list of referenced documents</dd></dl>
</li>
</ul>
<a name="getReferencedSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedSequences</h4>
<pre>public final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&gt;&nbsp;getReferencedSequences()</pre>
<div class="block">Return the results of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)"><code>getReferencedSequence(int)</code></a> for each sequence in the
 alignment. (Convenience method)

 This list will be equal length to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> with null elements
 for sequences with no referenced document. There may be a mixture of referenced and
 unreferenced documents in an alignment.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the list of referenced documents</dd></dl>
</li>
</ul>
<a name="isReferencedDocumentReversed(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReferencedDocumentReversed</h4>
<pre>public&nbsp;boolean&nbsp;isReferencedDocumentReversed(int&nbsp;index)</pre>
<div class="block">Sequence direction.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a> is not null and it is in the reverse direction of its sequence string in the alignment.</dd></dl>
</li>
</ul>
<a name="isEditable()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEditable</h4>
<pre>public&nbsp;boolean&nbsp;isEditable()</pre>
<div class="block">true if this alignment is editable.
 if it is, it must implement <br>
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>reorderSequences(int[])</code></a>, <br>
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)"><code>removeSequence(int)</code></a> , <br>
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a>, <br>
 and may want to implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)"><code>setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)</code></a> <br><br>

 An implementation may also want to return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()"><code>canSetAnnotations()</code></a> to allow updating just the annotations via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>setAnnotations(int, java.util.List, boolean)</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this alignment is editable.</dd></dl>
</li>
</ul>
<a name="canSetAnnotations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetAnnotations</h4>
<pre>public&nbsp;boolean&nbsp;canSetAnnotations()</pre>
<div class="block">Returns true if this alignment can have its annotations on sequences (including the consensus sequence) edited alone via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>setAnnotations(int, java.util.List, boolean)</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>setAnnotationsOnConsensus(java.util.List)</code></a>
 Note that some alignment implementations may return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a> and allow updating their annotations via
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a> even if they return false from this method.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this alignment can have its annotations edited alone via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>setAnnotations(int, java.util.List, boolean)</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>setAnnotations(int, java.util.List, boolean)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>setAnnotationsOnConsensus(java.util.List)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>canSetAnnotationsOnConsensus()</code></a></dd></dl>
</li>
</ul>
<a name="reorderSequences(int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderSequences</h4>
<pre>public&nbsp;void&nbsp;reorderSequences(int[]&nbsp;ordering)</pre>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.
 If ordering[i]==j, then the sequence at index 'i' should be the sequence previously at index 'j'
 For example, if ordering = {0,1,3,4,2}
 Then:
 sequence at index 2 is moved to index 4
 sequence at index 3 is moved to index 2
 sequence at index 4 is moved to index 3

 This function may only be called on alignments that return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>ordering</code> - Each integer between 0 and ordering.length - 1 inclusive should occur exactly once in this array.</dd></dl>
</li>
</ul>
<a name="removeSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeSequence</h4>
<pre>public&nbsp;void&nbsp;removeSequence(int&nbsp;index)</pre>
<div class="block">Removes the specified sequence from the alignment. All other sequences should
 maintain their original ordering.

 This function may only be called on alignments that return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be removed. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> - 1</dd></dl>
</li>
</ul>
<a name="canAddAndRemoveSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canAddAndRemoveSequences</h4>
<pre>public&nbsp;boolean&nbsp;canAddAndRemoveSequences()</pre>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if adding and removing sequences is supported, false otherwise.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>addSequence(SequenceDocument)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)"><code>removeSequence(int)</code></a></dd></dl>
</li>
</ul>
<a name="addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequence</h4>
<pre>public&nbsp;void&nbsp;addSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Add a sequence to the end of the alignment. <code>sequence</code> should be the
 gapped (aligned) sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - sequence to add</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>reorderSequences(int[])</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)"><code>setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)</code></a></dd></dl>
</li>
</ul>
<a name="updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>updateSequence</h4>
<pre>public&nbsp;void&nbsp;updateSequence(int&nbsp;index,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                  <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
                  boolean&nbsp;allowModifyingReferencedDocuments)</pre>
<div class="block">Set new residues and related data for one sequence.

 This function may only be called on alignments that return true from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a>
 (otherwise, the method should throw <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang"><code>UnsupportedOperationException</code></a>).</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be updated. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> - 1</dd><dd><code>sequence</code> - the new residues</dd><dd><code>annotations</code> - a list of the annotations</dd><dd><code>graph</code> - Must be null if the original sequence at the specified index is not a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>.
 Otherwise (i.e. if the sequence was a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>), specifies the new
 graph to use for that sequence.</dd><dd><code>allowModifyingReferencedDocuments</code> - if the change being made to the sequence would affect the unaligned version of it
 (for example, the sequence residues are changing rather than just the gaps)
 and the alignment stores a reference an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be modified.
 If this parameter is not true and the change would have otherwise modified the original SequenceDocument,
 then any reference to the original document will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></code> - if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a> is false.</dd></dl>
</li>
</ul>
<a name="setAnnotations(int, java.util.List, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotations</h4>
<pre>public&nbsp;void&nbsp;setAnnotations(int&nbsp;index,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                  boolean&nbsp;allowModifyingReferencedDocuments)</pre>
<div class="block">Updates just the annotations on a sequence. This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()"><code>canSetAnnotations()</code></a> returns true<p/>
 This is used for updating non-track annotations only. Track annotations are handled via a different API. To update or add tracks check if
 a sequence in the alignment implements <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the 0-based index of the sequence to set annotations on, or -1 to set annotations on the consensus sequence.</dd><dd><code>annotations</code> - the new annotations for this sequence, which will replace any existing annotations.</dd><dd><code>allowModifyingReferencedDocuments</code> - if this is true  the original referenced sequence document will be modified to use these annotations
 If this parameter is not true and then any reference to the original document will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd></dl>
</li>
</ul>
<a name="setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedDocument</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedDocument(int&nbsp;sequenceIndexInAlignment,
                            <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</pre>
<div class="block">Sets the sequence at the given sequenceIndexInAlignment in this alignment to refer to the given document as if it is the original source.
 This document must contain a sequence consisting of residues matching the aligned sequence stored at <code> sequenceIndexInAlignment </code>,
 with gap residues removed.
 This method may be used when the user chooses to extract a sequence from an alignment, so the alignment
 should now be updated to reference the extracted sequence.
 Any alignment implementations may choose to not implement this method, in which case of the default implementation is used which always returns false.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the sequenceIndexInAlignment of the sequence referred to. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> - 1</dd><dd><code>document</code> - the document to reference</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this document was successfully set as a referenced document. Note that some implementations may allow setting the reference and return true,
 but then discard the reference next time it is requested if the sequence from the referenced document does not correspond to sequence in the alignment.
 Call <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a> to check if it really worked.</dd></dl>
</li>
</ul>
<a name="setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedSequence</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedSequence(int&nbsp;sequenceIndexInAlignment,
                            <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
                            int&nbsp;referencedDocumentRevisionNumberSequenceWasCopiedFrom)</pre>
<div class="block">Sets the sequence at the given index in this alignment to refer to the given document as if it is the original source.
 This document must contain a sequence consisting of residues matching the aligned sequence stored at <code> sequenceIndexInAlignment </code>,
 with gap residues removed.
 This method may be used when the user chooses to extract a sequence from an alignment, so the alignment
 should now be updated to reference the extracted sequence.
 Any alignment implementations may choose to not implement this method, in which case of the default implementation is used which always returns false.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the sequenceIndexInAlignment of the sequence referred to. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> - 1</dd><dd><code>referencedSequence</code> - the document to reference</dd><dd><code>referencedDocumentRevisionNumberSequenceWasCopiedFrom</code> - normally an alignment document stores the most recent revision number it updated its local cache of the sequence from. In order to improve performance, if it is known what revision number (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getRevisionNumber()"><code>AnnotatedPluginDocument.getRevisionNumber()</code></a> the current alignment sequence was created from then it can be supplied here. Use -1 if unsure.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this document was successfully set as a referenced document</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedSequence</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedSequence(int&nbsp;sequenceIndexInAlignment,
                            <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence)</pre>
<div class="block">Equivalent to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)"><code>setReferencedSequence(sequenceIndexInAlignment, referencedSequence, -1</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the sequenceIndexInAlignment of the sequence referred to. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> - 1</dd><dd><code>referencedSequence</code> - the document to reference</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this document was successfully set as a referenced document</dd></dl>
</li>
</ul>
<a name="asJeblAlignmentIfMemoryAllows(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblAlignmentIfMemoryAllows</h4>
<pre>public final&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a>&nbsp;asJeblAlignmentIfMemoryAllows(boolean&nbsp;silentlyStripDuplicates)
                                              throws <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">This method calls #asJeblAlignment(boolean) and re-throws any OutOfMemoryError as a DocumentOperationException</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>silentlyStripDuplicates</code> - Unlike Geneious alignments, Jebl alignments cannot contain sequences
 with the same names. If this parameter is true, duplicate sequences will be absent from the returned alignment.
 if this parameter is false, sequences will be given unique names by appending "1", "2", etc .</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequences as a Jebl alignment</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if there is not enough memory to create the Jebl alignment</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.65 (Geneious 5.6.5)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignment(boolean)"><code>asJeblAlignment(boolean)</code></a></dd></dl>
</li>
</ul>
<a name="asJeblAlignment(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asJeblAlignment</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a>&nbsp;asJeblAlignment(boolean&nbsp;silentlyStripDuplicates)</pre>
<div class="block">Get alignment as a Jebl alignment. Throws an OutOfMemoryError if there is not enough memory to do so.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>silentlyStripDuplicates</code> - Unlike Geneious alignments, Jebl alignments cannot contain sequences
 with the same names. If this parameter is true, duplicate sequences will be absent from the returned alignment.
 if this parameter is false, sequences will be given unique names by appending "1", "2", etc .

 This function is implemented for all alignments. Sub-classes need not implement it, but may wish to
 if they internally store their alignment in jebl format already.</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequences as a Jebl alignment</dd></dl>
</li>
</ul>
<a name="getSequenceAsJeblSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAsJeblSequence</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&nbsp;getSequenceAsJeblSequence(int&nbsp;index)</pre>
<div class="block">Gets a sequence as a jebl sequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to get. between 0 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a> - 1</dd>
<dt><span class="strong">Returns:</span></dt><dd>sequence as a jebl sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang">RuntimeException</a></code> - wrapping a DocumentOperationException if the referenced sequence of the alignment cannot be loaded.
 This function is implemented for all alignments by converting the result of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)"><code>getSequence(int)</code></a>.
 Sub-classes need not implement it, but may wish to
 if they internally store their sequences in jebl format already.</dd></dl>
</li>
</ul>
<a name="toHTML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toHTML</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toHTML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">PluginDocument</a></code></strong></div>
<div class="block">Provide a textual view of document. The contents should be at most 30K-60K characters. One reason is that this
 text is rendered using JTextPane, which is horrendously slow when handling large text. The more fundamental
 reason is that large amounts of text are generally not useful for a user. The whole idea of using a tool such as
 Geneious is to provide specialized viewers which make the large amount of bioinformatics data perceivable and
 comprehensible to a human user.
 <p/>
 Return an HTML body fragment (i.e. a fragment which sits inside a &lt;body&gt;&lt;/body&gt; tag). (Header,stylesheet and
 annotations are automatically added by GENEious code).
 <p/>
 Can optionally return an entire html tag set, as long as the first six characters are <code>&lt;html></code></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an HTML fragment of tag set, or null to have no text view.</dd></dl>
</li>
</ul>
<a name="isContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isContig</h4>
<pre>public&nbsp;boolean&nbsp;isContig()</pre>
<div class="block">Returns false by default. If true then this alignment may should be
 treated as a contig (may affect e.g. the way consensus sequence
 are calculated).
 <p/>
 If this is overridden to return true by classes other than
 <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocument</code></a>
 then the document will not be given the contig icon (to set non default icons for
 a document see <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/DocumentType.html#OVERRIDE_DOCUMENT_TYPE_KEY"><code>DocumentType.OVERRIDE_DOCUMENT_TYPE_KEY</code></a>.
 <p/>
 Use <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)"><code>DefaultAlignmentDocument.setContig(boolean)</code></a>
 to create a contig.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this alignment should be treated as a contig</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)"><code>DefaultAlignmentDocument.setContig(boolean)</code></a></dd></dl>
</li>
</ul>
<a name="canSetSequenceNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetSequenceNames</h4>
<pre>public&nbsp;boolean&nbsp;canSetSequenceNames()</pre>
<div class="block">Specifies whether this alignment supports changing the sequence names.
 If this method returns true, then this alignment must implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)"><code>setSequenceName(int, String, boolean)</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this alignment supports changing the sequence names.</dd></dl>
</li>
</ul>
<a name="setSequenceName(int, java.lang.String, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceName</h4>
<pre>public&nbsp;void&nbsp;setSequenceName(int&nbsp;index,
                   <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
                   boolean&nbsp;allowModifyingReferencedDocuments)</pre>
<div class="block">change the name of a sequence within the alignment.
 This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()"><code>canSetSequenceNames()</code></a> returns true.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - index of the sequence name to change</dd><dd><code>newName</code> - the new name for the sequence.</dd><dd><code>allowModifyingReferencedDocuments</code> - if the alignment stores a reference to an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be renamed too.
 Otherwise,  any existing reference to the original SequenceDocument will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="isFreeEndGapAlignment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isFreeEndGapAlignment</h4>
<pre>public&nbsp;boolean&nbsp;isFreeEndGapAlignment()</pre>
<div class="block">True if this alignment was built using free end gaps. This means gaps at either end of the sequences are
 not penalized.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this alignment was built using free end gaps. The default implementation returns true.</dd></dl>
</li>
</ul>
<a name="getRefinedName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRefinedName</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getRefinedName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block">Given the name of an existing alignment, obtain a name that should
 be used for a refinement of this alignment.
 Appends a " - refined" suffix to the name of an alignment if it is not already present.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - the name of the original alignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>the name for the refined alignment.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRealignedName(java.lang.String)"><code>getRealignedName(String)</code></a></dd></dl>
</li>
</ul>
<a name="getRealignedName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRealignedName</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getRealignedName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block">Given the name of an existing alignment, obtain a name that should
 be used for a realignment of this alignment.
 Appends a " - realigned" suffix to the name of an alignment if it is not already present.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - the name of the original alignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>the name for the realigned alignment.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRefinedName(java.lang.String)"><code>getRefinedName(String)</code></a></dd></dl>
</li>
</ul>
<a name="getContigReferenceSequenceIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getContigReferenceSequenceIndex</h4>
<pre>public&nbsp;int&nbsp;getContigReferenceSequenceIndex()</pre>
<div class="block">Get the index (if any) of the reference sequence in a contig. Note - despite the name, this method can also be used to identify
 the query sequence in a query centric alignment.
 <p/>
 The default implementation returns -1.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the index (in the range [0,<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a>-1]) or -1 to if this contig has no reference sequence.</dd></dl>
</li>
</ul>
<a name="isQueryCentricAlignment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isQueryCentricAlignment</h4>
<pre>public&nbsp;boolean&nbsp;isQueryCentricAlignment()</pre>
<div class="block">Returns true if this alignment is a query centric alignment. A query centric alignment must also return a non-negative number
 from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>getContigReferenceSequenceIndex()</code></a>.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this alignment is a query centric alignment.</dd></dl>
</li>
</ul>
<a name="getMateExpectedDistance(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateExpectedDistance</h4>
<pre>public&nbsp;int&nbsp;getMateExpectedDistance(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate. This is defined
 as the length of the 2 reads plus the distance between them.
 For example if the 2 reads both have length 30 with a 40 bp gap between them, then the
 expected mate distance is 100.
 If the mate is expected to appear after this sequence with this sequence in its current orientation, then
 the expected distance will be positive. If the mate is expected to appear before this sequence with this sequence in its current orientation, then
 the expected distance will be negative.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence for which to get the mate index of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the expected mate distance or 0 if this read is not paired. Note that a read may still have a mate distance!=0 but its mate may not be present in this document (in which case <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a> will be -1.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a></dd></dl>
</li>
</ul>
<a name="getMateIndex(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateIndex</h4>
<pre>public&nbsp;int&nbsp;getMateIndex(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">getMateIndex</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence to get the mate index of</dd>
<dt><span class="strong">Returns:</span></dt><dd>an index of the mate of the sequence, or -1 if the sequence has no mate.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>PairedReads.getMateExpectedDistance(int)</code></a></dd></dl>
</li>
</ul>
<a name="getImmutablePairedReadManager()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getImmutablePairedReadManager</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;getImmutablePairedReadManager()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">PairedReads</a></code></strong></div>
<div class="block">An immutable view of the paired reads managed by this list or alignment. Even if the list or alignment
 that implements this method is later modified, the
 returned PairedReadManager will not change.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>An immutable view of the paired reads managed by the list or alignment that implements this method or null if this class doesn't provide an immutable view of its paired reads.</dd></dl>
</li>
</ul>
<a name="setMates(int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMates</h4>
<pre>public&nbsp;void&nbsp;setMates(int&nbsp;sequenceIndex,
            int&nbsp;sequenceIndexOfMate,
            int&nbsp;expectedMateDistance)</pre>
<div class="block">Records the 2nd sequence as being paired with the first sequence with the given distance.
 <p/>
 If the mate sequence is not yet paired with this sequence then this method also sets the
 expected mate's distance to be -expectedMateDistance.
 If the mate sequence is already paired with this sequence, then this method does not change the distance recorded for the mate.
 This is so that if a contig is created where the 2 mates haven been mapped in opposite directions (due to an inversion from the reference sequence), the
 expected distance can be correctly set for each of them since both values should be either positive or negative in this case. (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a>)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the sequence to set the mate for.</dd><dd><code>sequenceIndexOfMate</code> - the index of its mate, or -1 if its mate is not in this contig.</dd><dd><code>expectedMateDistance</code> - the expected mate distance. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if this sequence already has a different mate or if its mate is paired to a different sequence.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></code> - if this alignment does not support setting mates. All editable alignment implementations (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>isEditable()</code></a> for which <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getImmutablePairedReadManager()"><code>getImmutablePairedReadManager()</code></a>.<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#hasAnyPairedReads()"><code>hasAnyPairedReads()</code></a> returns true must not throw a UnsupportedOperationException.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="canSetAnnotationsOnConsensus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetAnnotationsOnConsensus</h4>
<pre>public&nbsp;boolean&nbsp;canSetAnnotationsOnConsensus()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this alignment implementation supports calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>setAnnotationsOnConsensus(java.util.List)</code></a>.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>setAnnotationsOnConsensus(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="getAnnotationsOnConsensus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationsOnConsensus</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getAnnotationsOnConsensus()</pre>
<div class="block">Gets annotations that are associated with the consensus sequence.
 Note - this doesn't include consensus annotations in tracks. Consensus tracks can be obtained by checking if this alignment
 implements <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>annotations that are associated with the consensus sequence. Never null but may be empty</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="setAnnotationsOnConsensus(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotationsOnConsensus</h4>
<pre>public&nbsp;void&nbsp;setAnnotationsOnConsensus(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block">Sets the annotations on the consensus sequence that will be returned from a later call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>getAnnotationsOnConsensus()</code></a>.
 It is only legal to call this method if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>canSetAnnotationsOnConsensus()</code></a> returns true.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotations</code> - the annotations to store on the consensus. Must not be null. This replaces any existing annotations on the consensus.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>getAnnotationsOnConsensus()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>canSetAnnotationsOnConsensus()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceType</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;getSequenceType()</pre>
<div class="block">Returns the sequence type of the first sequence in this alignment (which should be the same as all other sequences in the alignment)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sequence type of the first sequence in this alignment (which should be the same as all other sequences in the alignment)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getCircularLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCircularLength</h4>
<pre>public&nbsp;int&nbsp;getCircularLength()</pre>
<div class="block">If this alignment is circular, returns the gapped length at which it wraps back to the start or 0 if this alignment is not circular.
 For example, if an alignment looks like<br/><br/>
 <code>
 RefSeq: -GATTACA----<br/>
 Read 1: ----TACAAGA-<br/>
 Read 2: -----ACA-GAT<br/>
 </code><br/>
 this could have a circular length of 8 to indicate the reads extending past the last base of the refSeq and considered to wrap back to the start.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>if this alignment is circular, returns the gapped length at which it wraps back to the start or 0 if this alignment is not circular</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetFieldValue</h4>
<pre>public&nbsp;boolean&nbsp;canSetFieldValue(int&nbsp;sequenceIndex,
                       <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field)</pre>
<div class="block">Returns true if this alignment supports updating field values using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of a sequence in the alignment</dd><dd><code>field</code> - the field to update</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment supports updating field values using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></dd></dl>
</li>
</ul>
<a name="setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setFieldValue</h4>
<pre>public&nbsp;void&nbsp;setFieldValue(int&nbsp;sequenceIndex,
                 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field,
                 <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;fieldValue,
                 boolean&nbsp;allowModifyingOriginals)</pre>
<div class="block">Sets the value of a displayable field on one of the sequences in the alignment. This method may only be called if <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)"><code>canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)</code></a> returns true.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of a sequence in the alignment to update the field value on</dd><dd><code>field</code> - the field to update</dd><dd><code>fieldValue</code> - the new value for the field.</dd><dd><code>allowModifyingOriginals</code> - true if this field change may be propagated to a referenced sequence (if any). If this is false the sequence will be disconnected from its referenced document.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)"><code>canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)</code></a></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceAlignmentDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" target="_top">Frames</a></li>
<li><a href="SequenceAlignmentDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
