<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>SequenceAnnotation (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceAnnotation (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceAnnotation.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationGroup.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" target="_top">Frames</a></li>
<li><a href="SequenceAnnotation.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Class SequenceAnnotation" class="title">Class SequenceAnnotation</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></dd>
</dl>
<hr>
<br>
<pre>public final class <span class="strong">SequenceAnnotation</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">Provides information on a subset of the sequence. SequenceAnnotations are known as features in genbank format.
 <p/>
 Each annotation has a name (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getName()"><code>getName()</code></a>) and a type (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a>) along with a set of name-value pairs (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getQualifiers()"><code>getQualifiers()</code></a>
 and the intervals of the sequence which the annotation covers (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getIntervals()"><code>getIntervals()</code></a>
 <p/>
 Annotations are obtained via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()"><code>SequenceDocument.getSequenceAnnotations</code></a>
 or from a track via <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html#getSequenceAnnotations(jebl.util.ProgressListener)"><code>SequenceTrack.getSequenceAnnotations()</code></a></div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#HIDDEN_TYPE_FOR_ACTIVELY_LINKED_PARENTS">HIDDEN_TYPE_FOR_ACTIVELY_LINKED_PARENTS</a></strong></code>
<div class="block">An annotation type for annotations added to documents which are actively linked to their descendants (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/OperationRecordDocument.html#hasAnyActiveLinks()"><code>OperationRecordDocument.hasAnyActiveLinks()</code></a>).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#INVISIBLE_ANNOTATION_TYPE_PREFIX">INVISIBLE_ANNOTATION_TYPE_PREFIX</a></strong></code>
<div class="block">Annotations whose type (<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a>) has this prefix are not visible in the sequence viewer.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#KEY_ANNOTATION_GROUP">KEY_ANNOTATION_GROUP</a></strong></code>
<div class="block"><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#getName()"><code>Name</code></a>
 of a qualifier whose value identifies a group that this annotation belongs to
 on a single sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#KEY_CREATED_BY">KEY_CREATED_BY</a></strong></code>
<div class="block">A special key used to identify the origin of this annotation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#KEY_MODIFIED_BY">KEY_MODIFIED_BY</a></strong></code>
<div class="block">A special key to identify the last modifier of this annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CDS">TYPE_CDS</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CHAIN">TYPE_CHAIN</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_COIL">TYPE_COIL</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_COILED_COIL">TYPE_COILED_COIL</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CONCATENATED_SEQUENCE">TYPE_CONCATENATED_SEQUENCE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CONFLICT">TYPE_CONFLICT</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CRISPR">TYPE_CRISPR</a></strong></code>
<div class="block">Candidate CRISPR site</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_DISULFIDE_BOND">TYPE_DISULFIDE_BOND</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_DNA_PROBE">TYPE_DNA_PROBE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_DOMAIN">TYPE_DOMAIN</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_DELETION">TYPE_EDITING_HISTORY_DELETION</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_INSERTION">TYPE_EDITING_HISTORY_INSERTION</a></strong></code>
<div class="block">A directionless annotation type to record deletions, insertions and replacements of bases in a sequence compared
 to the original imported sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_REPLACEMENT">TYPE_EDITING_HISTORY_REPLACEMENT</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXCLUDED_REGION">TYPE_EXCLUDED_REGION</a></strong></code>
<div class="block">Annotations of this type are length 0 annotations between 2 sites, and represent a region
 of the sequence that has been excluded for some reason, usually for example when viewing
 a query centric alignment with gaps hidden in the reference sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXON">TYPE_EXON</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXPRESSION_DIFFERENCE">TYPE_EXPRESSION_DIFFERENCE</a></strong></code>
<div class="block">A type used to represent differential expression levels (for example when dealing with RNA sequencing data)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXPRESSION_LEVEL">TYPE_EXPRESSION_LEVEL</a></strong></code>
<div class="block">A type used to represent expression levels (for example when dealing with RNA sequencing data)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION">TYPE_EXTRACTED_REGION</a></strong></code>
<div class="block">This is an invisible annotation that contains a mapping from the original sequence indices to the current sequence indices.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_GENE">TYPE_GENE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_GIBSON_PRIMER_EXTENSION">TYPE_GIBSON_PRIMER_EXTENSION</a></strong></code>
<div class="block">A type to mark bases that have to be added as a primer extension (instead of primer binding site) during the gibson assembly operation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_HELIX">TYPE_HELIX</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_HIGH_COVERAGE">TYPE_HIGH_COVERAGE</a></strong></code>
<div class="block">Annotation types created by High/Low coverage finder when it finds high coverage regions</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_INSERTED_SEQUENCE">TYPE_INSERTED_SEQUENCE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_INTRON">TYPE_INTRON</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_JUNCTION">TYPE_JUNCTION</a></strong></code>
<div class="block">The annotation type created by reference mapping and SAM/BAM import when there are large deletions or introns</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_LIGATION">TYPE_LIGATION</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_LONG_TERMINAL_REPEAT">TYPE_LONG_TERMINAL_REPEAT</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_LOW_COVERAGE">TYPE_LOW_COVERAGE</a></strong></code>
<div class="block">Annotation types created by High/Low coverage finder when it finds low coverage regions</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_LOW_QUALITY">TYPE_LOW_QUALITY</a></strong></code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>replaced with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>TYPE_TRIMMED</code></a>.</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_MATURE_PEPTIDE">TYPE_MATURE_PEPTIDE</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_MISC_DIFFERENCE">TYPE_MISC_DIFFERENCE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_MISC_FEATURE">TYPE_MISC_FEATURE</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_MISC_RNA">TYPE_MISC_RNA</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_MOTIF">TYPE_MOTIF</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_MRNA">TYPE_MRNA</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_NCRNA">TYPE_NCRNA</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_OPTIMIZED_CODON">TYPE_OPTIMIZED_CODON</a></strong></code>
<div class="block">A type used to represent a codon that has been replaced wih an optimized equivalent</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_ORF">TYPE_ORF</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_ORIGIN">TYPE_ORIGIN</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_ORIGIN_OF_REPLICATION">TYPE_ORIGIN_OF_REPLICATION</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_OVERHANG">TYPE_OVERHANG</a></strong></code>
<div class="block">A terminal region of a sequence annotated with an annotation of this type is an overhang on one of the strands.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PEPTIDE">TYPE_PEPTIDE</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_POLYA_SIGNAL">TYPE_POLYA_SIGNAL</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_POLYA_SITE">TYPE_POLYA_SITE</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_POLYLINKER">TYPE_POLYLINKER</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_POLYMORPHISM">TYPE_POLYMORPHISM</a></strong></code>
<div class="block">Used for any type of variant annotation, for example SNPs or longer variants.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PRECURSOR_RNA">TYPE_PRECURSOR_RNA</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PRIMER_BIND">TYPE_PRIMER_BIND</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PRIMER_BIND_REVERSE">TYPE_PRIMER_BIND_REVERSE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PROMOTER">TYPE_PROMOTER</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PROTEIN_BIND">TYPE_PROTEIN_BIND</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_REGULATORY">TYPE_REGULATORY</a></strong></code>
<div class="block">From Genbank definition: "Any region of sequence that functions in the regulation of transcription or translation"</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_REPEAT_REGION">TYPE_REPEAT_REGION</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_REPEAT_UNIT">TYPE_REPEAT_UNIT</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_RESTRICTION_SITE">TYPE_RESTRICTION_SITE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_RRNA">TYPE_RRNA</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_SEARCH_HIT">TYPE_SEARCH_HIT</a></strong></code>
<div class="block">The type of annotation added to a sequence search result to indicate the region that matched the query sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_SIGNAL_PEPTIDE">TYPE_SIGNAL_PEPTIDE</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_SITE">TYPE_SITE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_SOURCE">TYPE_SOURCE</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_SOURCE_SEQUENCE">TYPE_SOURCE_SEQUENCE</a></strong></code>
<div class="block">The type of annotations created via 'Paste as Active Link'</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_STATISTICS">TYPE_STATISTICS</a></strong></code>
<div class="block">A type used to represent statistical data</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_STRAND">TYPE_STRAND</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TERMINATOR">TYPE_TERMINATOR</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TMRNA">TYPE_TMRNA</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRANSLATED_REGION">TYPE_TRANSLATED_REGION</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRANSMEMBRANE">TYPE_TRANSMEMBRANE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED">TYPE_TRIMMED</a></strong></code>
<div class="block">Regions of a sequence covered by trim annotations are ignored for many operations, including assembly and consensus sequence calculation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRNA">TYPE_TRNA</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TURN">TYPE_TURN</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_UTR_3">TYPE_UTR_3</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_UTR_5">TYPE_UTR_5</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_VECTOR_MODERATE">TYPE_VECTOR_MODERATE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_VECTOR_SEGMENT_OF_SUSPECT_ORIGIN">TYPE_VECTOR_SEGMENT_OF_SUSPECT_ORIGIN</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_VECTOR_STRONG">TYPE_VECTOR_STRONG</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_VECTOR_WEAK">TYPE_VECTOR_WEAK</a></strong></code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#SequenceAnnotation(org.jdom.Element)">SequenceAnnotation</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">XML deserialization constructor.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#SequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">SequenceAnnotation</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>
<div class="block">creates a copy of the given annotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#SequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation, int)">SequenceAnnotation</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation,
                  int&nbsp;offset)</code>
<div class="block">creates a copy of the given annotation offset by a certain amount</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#SequenceAnnotation(java.lang.String, java.lang.String)">SequenceAnnotation</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>
<div class="block">Constructs a new SequenceAnnotation with the specified name and type.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#SequenceAnnotation(java.lang.String, java.lang.String, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval...)">SequenceAnnotation</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type,
                  <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>...&nbsp;intervals)</code>
<div class="block">Constructs a new SequenceAnnotation with the specified name and type.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addInterval(int, int)">addInterval</a></strong>(int&nbsp;from,
           int&nbsp;to)</code>
<div class="block">Add one left to right interval to this annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval.Direction)">addInterval</a></strong>(int&nbsp;minIndex,
           int&nbsp;maxIndex,
           <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a>&nbsp;direction)</code>
<div class="block">Add one interval to this annotation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addInterval(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">addInterval</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;interval)</code>
<div class="block">add a new interval to this annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addIntervals(java.util.Collection)">addIntervals</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;intervalsToAdd)</code>
<div class="block">adds a collection of intervals to this annotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifier(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationQualifier)">addQualifier</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&nbsp;qualifier)</code>
<div class="block">add a new <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>qualifier</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifier(java.lang.String, java.lang.String)">addQualifier</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
            <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;value)</code>
<div class="block">Add a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>qualifier</code></a> with the specified name-value mapping.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifiers(java.util.Collection)">addQualifiers</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&gt;&nbsp;qualifiersToAdd)</code>
<div class="block">Adds all the given qualifiers to this annotation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#copyOf(java.util.List)">copyOf</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;sequenceAnnotationsToCopy)</code>
<div class="block">Creates a deep copy of a list of SequenceAnnotations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#copyWithoutIntervals()">copyWithoutIntervals</a></strong>()</code>
<div class="block">Creates a copy of this annotation, without any intervals.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#createTrimAnnotation(int, int)">createTrimAnnotation</a></strong>(int&nbsp;minimumIndexInclusive,
                    int&nbsp;maximumIndexInclusive)</code>
<div class="block">Creates a new <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>TYPE_TRIMMED</code></a> annotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#equals(java.lang.Object)">equals</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;obj)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getAnnotationAdjustedForDeletion(int, int, boolean, int)">getAnnotationAdjustedForDeletion</a></strong>(int&nbsp;deletionStart,
                                int&nbsp;deletionLength,
                                boolean&nbsp;isDeletionAllGaps,
                                int&nbsp;newSequenceLengthIfCircular)</code>
<div class="block">Handles a deletion from the sequence which this annotation is for, by either returning a new annotation adjusted to take the deletion into account, or this
 annotation if it doesn't change, or null if the annotation should be deleted (for example the user deletes nucleotides from a primer annotation)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getAnnotationAdjustedForInsertion(int, int, boolean, int)">getAnnotationAdjustedForInsertion</a></strong>(int&nbsp;indexToInsertAt,
                                 int&nbsp;lengthOfInsertion,
                                 boolean&nbsp;isInsertionAllGaps,
                                 int&nbsp;newSequenceLengthIfCircular)</code>
<div class="block">Handles an insertion into the sequence which this annotation is for, by either returning a new annotation adjusted to take the insertion into account, or this
 annotation if it doesn't change, or null if the annotation should be deleted (for example the user inserts nucleotides into a primer annotation)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getAnnotationAdjustedForReplacement(int, int, int)">getAnnotationAdjustedForReplacement</a></strong>(int&nbsp;replacementPosition,
                                   int&nbsp;replacementLength,
                                   int&nbsp;sequenceLengthIfCircular)</code>
<div class="block">Handles a non-gap replacement in the sequence which this annotation is for, by either returning this
 annotation if it doesn't change, or null if the annotation should be deleted (for example the user replaces nucleotides in a primer annotation)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getCodonStart()">getCodonStart</a></strong>()</code>
<div class="block">The value of the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#NCBI_CODON_START"><code>NCBI_CODON_START</code></a> qualifier indicates the offset at which the first complete codon of a coding feature can be found,
 relative to the first base of that feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getInterval()">getInterval</a></strong>()</code>
<div class="block">If this annotation has exactly 1 interval gets that interval or throws an <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang"><code>IllegalStateException</code></a> if it doesn't have exactly 1 interval.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getInterval(int)">getInterval</a></strong>(int&nbsp;intervalIndex)</code>
<div class="block">Returns one of the intervals from this annotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getIntervals()">getIntervals</a></strong>()</code>
<div class="block">Gets all the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>intervals</code></a> this annotation contains.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getName()">getName</a></strong>()</code>
<div class="block">The name of the annotation which should be a short human readable text description.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getNumberOfIntervals()">getNumberOfIntervals</a></strong>()</code>
<div class="block">Returns the number of intervals this annotation has</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getQualifiers()">getQualifiers</a></strong>()</code>
<div class="block">A list of qualifiers for the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getQualifierValue(java.lang.String)">getQualifierValue</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Get the value of the first qualifier in the list of the given name or "" if one is not found.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getTranslation(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">getTranslation</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document,
              <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
              <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>
<div class="block">Returns the translation (if applicable) for the given annotation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()">getType</a></strong>()</code>
<div class="block">Type of the annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getVersionSupportStatic(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupportStatic</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">A static implementation of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code></a>
 that can be used in place of that method since all SequenceAnnotation instances have the same version support.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#hashCode()">hashCode</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#isEditingAnnotationType(java.lang.String)">isEditingAnnotationType</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#isInvisibleAnnotationType(java.lang.String)">isInvisibleAnnotationType</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>
<div class="block">Returns true if annotations of this type are not visible to the user.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#isReadOnlyAnnotationType(java.lang.String)">isReadOnlyAnnotationType</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#readObject(java.io.DataInput)">readObject</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataInput.html?is-external=true" title="class or interface in java.io">DataInput</a>&nbsp;in)</code>
<div class="block">Reads data previously created using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#writeObject(java.io.DataOutput)"><code>writeObject(java.io.DataOutput)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">readObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)</code>
<div class="block">Reads data previously created using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)"><code>writeObject(GeneiousObjectOutputStream)</code></a>
 It is a good idea to use a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>GeneiousObjectInputStream</code></a> when deserializing
 lots of annotations.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#removeInterval(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">removeInterval</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;interval)</code>
<div class="block">Removes the given interval from this annotation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#removeQualifier(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationQualifier)">removeQualifier</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&nbsp;qualifierToRemove)</code>
<div class="block">Removes the given qualifier from this annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#removeQualifier(java.lang.String)">removeQualifier</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Deletes all existing qualifiers with the given name</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#reverse(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">reverse</a></strong>(int&nbsp;sequenceLength,
       <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</code>
<div class="block">Creates a new annotation that is either the reverse or the reverse complements this annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setIntervals(java.util.List)">setIntervals</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;intervals)</code>
<div class="block">set the intervals on this annotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setName(java.lang.String)">setName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setQualifier(java.lang.String, java.lang.String)">setQualifier</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
            <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;value)</code>
<div class="block">Deletes all existing qualifiers with the given name, and adds a new qualifier with the given value.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setQualifiers(java.util.List)">setQualifiers</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&gt;&nbsp;qualifiers)</code>
<div class="block">Set a new list of qualifiers on this SequenceAnnotation, removing any
 existing qualifiers added via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifier(java.lang.String, java.lang.String)"><code>addQualifier(String, String)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifier(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationQualifier)"><code>addQualifier(SequenceAnnotationQualifier)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifiers(java.util.Collection)"><code>addQualifiers(java.util.Collection)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setQualifiers(java.util.List)"><code>setQualifiers(java.util.List)</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setType(java.lang.String)">setType</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>
<div class="block">Set the type of this annotation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#toString()">toString</a></strong>()</code>
<div class="block">Returns a human readable string with information about this annotation;
 the exact format of the string is not guaranteed and should not be relied
 on, because it may change at any time.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;majorVersion,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serialization of annotation - to.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#writeObject(java.io.DataOutput)">writeObject</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out)</code>
<div class="block">Writes this annotation to a DataOutput</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput)">writeObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
           <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out)</code>
<div class="block">Writes this annotation to a DataOutput in a form that can potentially be read by old versions of Geneious</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)">writeObject</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;out)</code>
<div class="block">Writes this annotation to an output stream as defined by <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io"><code>Serializable</code></a>.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="INVISIBLE_ANNOTATION_TYPE_PREFIX">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INVISIBLE_ANNOTATION_TYPE_PREFIX</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> INVISIBLE_ANNOTATION_TYPE_PREFIX</pre>
<div class="block">Annotations whose type (<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a>) has this prefix are not visible in the sequence viewer.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#isInvisibleAnnotationType(java.lang.String)"><code>isInvisibleAnnotationType(String)</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.INVISIBLE_ANNOTATION_TYPE_PREFIX">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="HIDDEN_TYPE_FOR_ACTIVELY_LINKED_PARENTS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HIDDEN_TYPE_FOR_ACTIVELY_LINKED_PARENTS</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> HIDDEN_TYPE_FOR_ACTIVELY_LINKED_PARENTS</pre>
<div class="block">An annotation type for annotations added to documents which are actively linked to their descendants (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/OperationRecordDocument.html#hasAnyActiveLinks()"><code>OperationRecordDocument.hasAnyActiveLinks()</code></a>).
 When a document containing an annotation of this type is regenerated because its parent has changed, then despite all other data on the document being discarded, if the region covered by this annotation
 can be unambiguously identified on the new version of the document, then this annotation will be copied to the new document covering that region.
 <p/>
 Annotations of this type will be hidden from the user interface. They are intended for programatic use only.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#HIDDEN_QUALIFIER_FOR_ACTIVELY_LINKED_PARENTS"><code>SequenceAnnotationQualifier.HIDDEN_QUALIFIER_FOR_ACTIVELY_LINKED_PARENTS</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.HIDDEN_TYPE_FOR_ACTIVELY_LINKED_PARENTS">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_ORF">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_ORF</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_ORF</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_ORF">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_CDS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_CDS</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_CDS</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_CDS">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_GENE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_GENE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_GENE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_GENE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EXTRACTED_REGION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EXTRACTED_REGION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EXTRACTED_REGION</pre>
<div class="block">This is an invisible annotation that contains a mapping from the original sequence indices to the current sequence indices.
 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#EXTRACTED_REGION_INTERVALS_NAME"><code>SequenceAnnotationQualifier.EXTRACTED_REGION_INTERVALS_NAME</code></a> for the name of the qualifier that contains the mapping.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EXTRACTED_REGION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_TRANSLATED_REGION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_TRANSLATED_REGION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_TRANSLATED_REGION</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_TRANSLATED_REGION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_ORIGIN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_ORIGIN</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_ORIGIN</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_ORIGIN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_UTR_5">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_UTR_5</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_UTR_5</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_UTR_5">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_UTR_3">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_UTR_3</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_UTR_3</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_UTR_3">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_PRIMER_BIND">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_PRIMER_BIND</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_PRIMER_BIND</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_PRIMER_BIND">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_PRIMER_BIND_REVERSE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_PRIMER_BIND_REVERSE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_PRIMER_BIND_REVERSE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_PRIMER_BIND_REVERSE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_PEPTIDE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_PEPTIDE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_PEPTIDE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_PEPTIDE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_MISC_FEATURE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_MISC_FEATURE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_MISC_FEATURE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_MISC_FEATURE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_SOURCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_SOURCE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_SOURCE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_SOURCE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EXON">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EXON</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EXON</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EXON">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_INTRON">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_INTRON</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_INTRON</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_INTRON">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_DNA_PROBE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_DNA_PROBE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_DNA_PROBE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_DNA_PROBE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_RESTRICTION_SITE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_RESTRICTION_SITE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_RESTRICTION_SITE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_RESTRICTION_SITE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_MOTIF">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_MOTIF</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_MOTIF</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_MOTIF">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_REGULATORY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_REGULATORY</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_REGULATORY</pre>
<div class="block">From Genbank definition: "Any region of sequence that functions in the regulation of transcription or translation"</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_REGULATORY">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_CRISPR">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_CRISPR</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_CRISPR</pre>
<div class="block">Candidate CRISPR site</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.800 (Geneious 8.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_CRISPR">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_SEARCH_HIT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_SEARCH_HIT</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_SEARCH_HIT</pre>
<div class="block">The type of annotation added to a sequence search result to indicate the region that matched the query sequence</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.703 (Geneious 7.0.3)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_SEARCH_HIT">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_TRIMMED">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_TRIMMED</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_TRIMMED</pre>
<div class="block">Regions of a sequence covered by trim annotations are ignored for many operations, including assembly and consensus sequence calculation. <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getTrailingTrimLength()"><code>ImmutableSequence.getTrailingTrimLength()</code></a> and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html#getLeadingTrimLength()"><code>ImmutableSequence.getLeadingTrimLength()</code></a> may also be used as an alternative method
 for indicating trimmed regions. Trim annotations should be created using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#createTrimAnnotation(int, int)"><code>createTrimAnnotation(int, int)</code></a></div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_TRIMMED">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_HELIX">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_HELIX</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_HELIX</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_HELIX">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_STRAND">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_STRAND</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_STRAND</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_STRAND">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_TURN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_TURN</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_TURN</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_TURN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_COIL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_COIL</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_COIL</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_COIL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_VECTOR_SEGMENT_OF_SUSPECT_ORIGIN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_VECTOR_SEGMENT_OF_SUSPECT_ORIGIN</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_VECTOR_SEGMENT_OF_SUSPECT_ORIGIN</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_VECTOR_SEGMENT_OF_SUSPECT_ORIGIN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_VECTOR_WEAK">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_VECTOR_WEAK</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_VECTOR_WEAK</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_VECTOR_WEAK">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_VECTOR_MODERATE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_VECTOR_MODERATE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_VECTOR_MODERATE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_VECTOR_MODERATE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_VECTOR_STRONG">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_VECTOR_STRONG</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_VECTOR_STRONG</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_VECTOR_STRONG">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_POLYLINKER">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_POLYLINKER</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_POLYLINKER</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_POLYLINKER">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_DISULFIDE_BOND">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_DISULFIDE_BOND</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_DISULFIDE_BOND</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_DISULFIDE_BOND">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_ORIGIN_OF_REPLICATION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_ORIGIN_OF_REPLICATION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_ORIGIN_OF_REPLICATION</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_ORIGIN_OF_REPLICATION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_TERMINATOR">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_TERMINATOR</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_TERMINATOR</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_TERMINATOR">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_SIGNAL_PEPTIDE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_SIGNAL_PEPTIDE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_SIGNAL_PEPTIDE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_SIGNAL_PEPTIDE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_MATURE_PEPTIDE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_MATURE_PEPTIDE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_MATURE_PEPTIDE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_MATURE_PEPTIDE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_PROMOTER">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_PROMOTER</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_PROMOTER</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_PROMOTER">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_SITE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_SITE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_SITE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_SITE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_LONG_TERMINAL_REPEAT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_LONG_TERMINAL_REPEAT</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_LONG_TERMINAL_REPEAT</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_LONG_TERMINAL_REPEAT">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_REPEAT_REGION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_REPEAT_REGION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_REPEAT_REGION</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_REPEAT_REGION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_REPEAT_UNIT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_REPEAT_UNIT</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_REPEAT_UNIT</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_REPEAT_UNIT">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_MISC_DIFFERENCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_MISC_DIFFERENCE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_MISC_DIFFERENCE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_MISC_DIFFERENCE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_CONFLICT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_CONFLICT</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_CONFLICT</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_CONFLICT">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_POLYA_SIGNAL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_POLYA_SIGNAL</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_POLYA_SIGNAL</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_POLYA_SIGNAL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_POLYA_SITE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_POLYA_SITE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_POLYA_SITE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_POLYA_SITE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_PROTEIN_BIND">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_PROTEIN_BIND</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_PROTEIN_BIND</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_PROTEIN_BIND">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_DOMAIN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_DOMAIN</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_DOMAIN</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_DOMAIN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_TRANSMEMBRANE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_TRANSMEMBRANE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_TRANSMEMBRANE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_TRANSMEMBRANE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_COILED_COIL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_COILED_COIL</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_COILED_COIL</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_COILED_COIL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_POLYMORPHISM">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_POLYMORPHISM</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_POLYMORPHISM</pre>
<div class="block">Used for any type of variant annotation, for example SNPs or longer variants. All polymorphism annotations should be given a name which indicates the nucleotides in the variant. They may provide qualifiers
 for <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#VARIANT_CHANGE"><code>SequenceAnnotationQualifier.VARIANT_CHANGE</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#VARIANT_NUCLEOTIDES"><code>SequenceAnnotationQualifier.VARIANT_NUCLEOTIDES</code></a>,
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#VARIANT_REFERENCE_NUCLEOTIDES"><code>SequenceAnnotationQualifier.VARIANT_REFERENCE_NUCLEOTIDES</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#VARIANT_AMINO_ACIDS"><code>SequenceAnnotationQualifier.VARIANT_AMINO_ACIDS</code></a></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.615 (Geneious 6.1.5)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_POLYMORPHISM">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_CHAIN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_CHAIN</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_CHAIN</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_CHAIN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_MRNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_MRNA</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_MRNA</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_MRNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_TMRNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_TMRNA</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_TMRNA</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_TMRNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_TRNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_TRNA</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_TRNA</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_TRNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_RRNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_RRNA</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_RRNA</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_RRNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_MISC_RNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_MISC_RNA</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_MISC_RNA</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_MISC_RNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_NCRNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_NCRNA</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_NCRNA</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_NCRNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_PRECURSOR_RNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_PRECURSOR_RNA</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_PRECURSOR_RNA</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_PRECURSOR_RNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_CONCATENATED_SEQUENCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_CONCATENATED_SEQUENCE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_CONCATENATED_SEQUENCE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_CONCATENATED_SEQUENCE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_INSERTED_SEQUENCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_INSERTED_SEQUENCE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_INSERTED_SEQUENCE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_INSERTED_SEQUENCE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_SOURCE_SEQUENCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_SOURCE_SEQUENCE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_SOURCE_SEQUENCE</pre>
<div class="block">The type of annotations created via 'Paste as Active Link'</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.910 (Geneious 9.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_SOURCE_SEQUENCE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EXCLUDED_REGION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EXCLUDED_REGION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EXCLUDED_REGION</pre>
<div class="block">Annotations of this type are length 0 annotations between 2 sites, and represent a region
 of the sequence that has been excluded for some reason, usually for example when viewing
 a query centric alignment with gaps hidden in the reference sequence.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EXCLUDED_REGION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_OVERHANG">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_OVERHANG</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_OVERHANG</pre>
<div class="block">A terminal region of a sequence annotated with an annotation of this type is an overhang on one of the strands.
 If the annotation is directed from left to right
 (<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#leftToRight"><code>SequenceAnnotationInterval.Direction.leftToRight</code></a>),
 the overhang is on the forward strand. If the annotation is directed right to left
 (<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#rightToLeft"><code>SequenceAnnotationInterval.Direction.rightToLeft</code></a>),
  it is an overhang on the reverse strand.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_OVERHANG">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_LIGATION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_LIGATION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_LIGATION</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_LIGATION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EDITING_HISTORY_INSERTION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EDITING_HISTORY_INSERTION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EDITING_HISTORY_INSERTION</pre>
<div class="block">A directionless annotation type to record deletions, insertions and replacements of bases in a sequence compared
 to the original imported sequence.
 The bases that the annotation covers will always be bases the user has inserted into the sequence.
 And the qualifier <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#EDITING_HISTORY_ORIGINAL_BASES"><code>SequenceAnnotationQualifier.EDITING_HISTORY_ORIGINAL_BASES</code></a> within the annotation records the bases that
 originally existed there. For example:
 <pre>
  1   2   3   4   5   6   7   8   9
  G   A   T   T   A   C   A   G   G
         [ ]       [C]       [  T  ]
 </pre>
 Here we have 3 annotations. The first covers only the 3rd base and therefore represents
 and insertion of T at that point. The second annotation represents a deletion of a C between the 5th and 6th bases.
 The final annotation represents a replacement of a T with GG at the end of the sequence.
 This works well for the implementation but isn't that nice to look at so the
 text displayed to the user might be something like:
 "Inserted T", for the first annotaion, "Deleted C" for the second, and
 "Replaced T with GG" for the third.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_REPLACEMENT"><code>TYPE_EDITING_HISTORY_REPLACEMENT</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_DELETION"><code>TYPE_EDITING_HISTORY_DELETION</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EDITING_HISTORY_INSERTION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EDITING_HISTORY_REPLACEMENT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EDITING_HISTORY_REPLACEMENT</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EDITING_HISTORY_REPLACEMENT</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_INSERTION"><code>TYPE_EDITING_HISTORY_INSERTION</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EDITING_HISTORY_REPLACEMENT">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EDITING_HISTORY_DELETION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EDITING_HISTORY_DELETION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EDITING_HISTORY_DELETION</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_INSERTION"><code>TYPE_EDITING_HISTORY_INSERTION</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EDITING_HISTORY_DELETION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_GIBSON_PRIMER_EXTENSION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_GIBSON_PRIMER_EXTENSION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_GIBSON_PRIMER_EXTENSION</pre>
<div class="block">A type to mark bases that have to be added as a primer extension (instead of primer binding site) during the gibson assembly operation.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_GIBSON_PRIMER_EXTENSION">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_LOW_QUALITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_LOW_QUALITY</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_LOW_QUALITY</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>replaced with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>TYPE_TRIMMED</code></a>.</i></div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_LOW_QUALITY">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_STATISTICS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_STATISTICS</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_STATISTICS</pre>
<div class="block">A type used to represent statistical data</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_STATISTICS">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EXPRESSION_LEVEL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EXPRESSION_LEVEL</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EXPRESSION_LEVEL</pre>
<div class="block">A type used to represent expression levels (for example when dealing with RNA sequencing data)</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EXPRESSION_LEVEL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_EXPRESSION_DIFFERENCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_EXPRESSION_DIFFERENCE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_EXPRESSION_DIFFERENCE</pre>
<div class="block">A type used to represent differential expression levels (for example when dealing with RNA sequencing data)</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.801 (Geneious 8.0.1)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_EXPRESSION_DIFFERENCE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_OPTIMIZED_CODON">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_OPTIMIZED_CODON</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_OPTIMIZED_CODON</pre>
<div class="block">A type used to represent a codon that has been replaced wih an optimized equivalent</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_OPTIMIZED_CODON">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="KEY_CREATED_BY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>KEY_CREATED_BY</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_CREATED_BY</pre>
<div class="block">A special key used to identify the origin of this annotation.
 eg. "ORF finder" if this annotation was generated by the ORF finder plugin
 "User" if this annotation was created by the user
 Annotations from databases such as NCBI should not be given a created tag.
 <p/>
 To add a created by tag it should be added to the qualifiers of the annotation
 with this as the name</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.KEY_CREATED_BY">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="KEY_MODIFIED_BY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>KEY_MODIFIED_BY</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_MODIFIED_BY</pre>
<div class="block">A special key to identify the last modifier of this annotation.
 eg. "User" if this annotation was modifed by the user.
 Annotations this tag if they have been modified after creation (do not
 give modified by tag to a new annotation).
 <p/>
 To add a modified by tag it should be added to the qualifiers of the annotation
 with this as the name</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.KEY_MODIFIED_BY">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="KEY_ANNOTATION_GROUP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>KEY_ANNOTATION_GROUP</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_ANNOTATION_GROUP</pre>
<div class="block"><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#getName()"><code>Name</code></a>
 of a qualifier whose value identifies a group that this annotation belongs to
 on a single sequence.  In the sequence viewer, if the user puts the mouse over
 one of these annotations then all others of the same group will be highlighted.
 (eg) Primer pairs are associated in this way.  This value should be unique on a
 per sequence basis and groups cannot span several sequences.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.KEY_ANNOTATION_GROUP">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_LOW_COVERAGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_LOW_COVERAGE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_LOW_COVERAGE</pre>
<div class="block">Annotation types created by High/Low coverage finder when it finds low coverage regions</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.902 (Geneious 9.0.2)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_HIGH_COVERAGE"><code>TYPE_HIGH_COVERAGE</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_LOW_COVERAGE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_HIGH_COVERAGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_HIGH_COVERAGE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_HIGH_COVERAGE</pre>
<div class="block">Annotation types created by High/Low coverage finder when it finds high coverage regions</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.902 (Geneious 9.0.2)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_LOW_COVERAGE"><code>TYPE_LOW_COVERAGE</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_HIGH_COVERAGE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="TYPE_JUNCTION">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>TYPE_JUNCTION</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> TYPE_JUNCTION</pre>
<div class="block">The annotation type created by reference mapping and SAM/BAM import when there are large deletions or introns</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.911 (Geneious 9.1.1)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#JUNCTION_SOURCE"><code>SequenceAnnotationQualifier.JUNCTION_SOURCE</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#JUNCTION_DESTINATION"><code>SequenceAnnotationQualifier.JUNCTION_DESTINATION</code></a>, 
<a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation.TYPE_JUNCTION">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceAnnotation(java.lang.String, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotation</h4>
<pre>public&nbsp;SequenceAnnotation(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<div class="block">Constructs a new SequenceAnnotation with the specified name and type.
 The type can be e.g. one of the TYPE_* constants in this class, but may be
 any arbitrary non-null value.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - Human readable name for the annotation; must not be null. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getName()"><code>getName()</code></a>.</dd><dd><code>type</code> - Identifier for the type of the annotation, e.g. <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CDS"><code>TYPE_CDS</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_ORF"><code>TYPE_ORF</code></a>. May be
                    any arbitrary String, but must not be null. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a>.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if name or type is null</dd></dl>
</li>
</ul>
<a name="SequenceAnnotation(java.lang.String, java.lang.String, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotation</h4>
<pre>public&nbsp;SequenceAnnotation(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type,
                  <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>...&nbsp;intervals)</pre>
<div class="block">Constructs a new SequenceAnnotation with the specified name and type.
 The type can be e.g. one of the TYPE_* constants in this class, but may be
 any arbitrary non-null value.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - Human readable name for the annotation; must not be null. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getName()"><code>getName()</code></a>.</dd><dd><code>type</code> - Identifier for the type of the annotation, e.g. <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_CDS"><code>TYPE_CDS</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_ORF"><code>TYPE_ORF</code></a>. May be
                    any arbitrary String, but must not be null. See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a>.</dd><dd><code>intervals</code> - 0 or more intervals to add to this annotation.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if name or type is null</dd></dl>
</li>
</ul>
<a name="SequenceAnnotation(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotation</h4>
<pre>public&nbsp;SequenceAnnotation(org.jdom.Element&nbsp;element)
                   throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">XML deserialization constructor.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - an element returned from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#toXML()"><code>toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if unable to deserialize the annotation</dd></dl>
</li>
</ul>
<a name="SequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotation</h4>
<pre>public&nbsp;SequenceAnnotation(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<div class="block">creates a copy of the given annotation</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotation</code> - SequenceAnnotation of which to create a copy</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if annotation is null</dd></dl>
</li>
</ul>
<a name="SequenceAnnotation(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceAnnotation</h4>
<pre>public&nbsp;SequenceAnnotation(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation,
                  int&nbsp;offset)</pre>
<div class="block">creates a copy of the given annotation offset by a certain amount</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotation</code> - SequenceAnnotation of which to create a copy</dd><dd><code>offset</code> - amount to offset by</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if annotation is null</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getIntervals()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIntervals</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;getIntervals()</pre>
<div class="block">Gets all the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>intervals</code></a> this annotation contains.
 <p/>
 If the annotation is for the entire sequence,
 this method must return an interval specifying the entire sequence range.
 <p/>
 For a CDS annotation the order of the intervals indicates the order in which the translation proceeds.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>all annotation intervals</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getNumberOfIntervals()"><code>getNumberOfIntervals()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getInterval(int)"><code>getInterval(int)</code></a></dd></dl>
</li>
</ul>
<a name="getNumberOfIntervals()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfIntervals</h4>
<pre>public&nbsp;int&nbsp;getNumberOfIntervals()</pre>
<div class="block">Returns the number of intervals this annotation has</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of intervals this annotation has</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getInterval(int)"><code>getInterval(int)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getIntervals()"><code>getIntervals()</code></a></dd></dl>
</li>
</ul>
<a name="getInterval(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInterval</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;getInterval(int&nbsp;intervalIndex)</pre>
<div class="block">Returns one of the intervals from this annotation</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>intervalIndex</code> - the 0-based index of the interval to get. Must be in the range [0,<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getNumberOfIntervals()"><code>getNumberOfIntervals()</code></a>)</dd>
<dt><span class="strong">Returns:</span></dt><dd>one of the intervals from this annotation</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.810 (Geneious 8.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getNumberOfIntervals()"><code>getNumberOfIntervals()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getIntervals()"><code>getIntervals()</code></a></dd></dl>
</li>
</ul>
<a name="getInterval()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInterval</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;getInterval()</pre>
<div class="block">If this annotation has exactly 1 interval gets that interval or throws an <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang"><code>IllegalStateException</code></a> if it doesn't have exactly 1 interval.
 <p/>
 Annotations are generally made of several intervals.
 However, certain types of annotations (such as <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_PRIMER_BIND"><code>TYPE_PRIMER_BIND</code></a>) never have more than one interval, and in such cases this method can be called to avoid
 needing to call .getIntervals().get(0).</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sole interval of this annotation</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this annotation does not have exactly 1 interval</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getIntervals()"><code>getIntervals()</code></a></dd></dl>
</li>
</ul>
<a name="getQualifiers()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getQualifiers</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&gt;&nbsp;getQualifiers()</pre>
<div class="block">A list of qualifiers for the sequence. If there are no qualifiers,
 this method must return an empty list (e.g. <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collections.html?is-external=true#emptyList()" title="class or interface in java.util"><code>Collections.emptyList()</code></a>).</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the qualifiers</dd></dl>
</li>
</ul>
<a name="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getName()</pre>
<div class="block">The name of the annotation which should be a short human readable text description.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>name of the annotation</dd></dl>
</li>
</ul>
<a name="getType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getType</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getType()</pre>
<div class="block">Type of the annotation. types are non structured - for example "CDS" or "gene".
 These can be any of the strings defined at SequenceAnnotation.TYPE_*
 or any custom string.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the type</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setType(java.lang.String)"><code>setType(String)</code></a></dd></dl>
</li>
</ul>
<a name="setType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setType</h4>
<pre>public&nbsp;void&nbsp;setType(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<div class="block">Set the type of this annotation. Types are non structured - for example "CDS" or "gene".
 These can be any of the strings defined at SequenceAnnotation.TYPE_* or any custom string.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>type</code> - the new type of this annotation</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a></dd></dl>
</li>
</ul>
<a name="isInvisibleAnnotationType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isInvisibleAnnotationType</h4>
<pre>public static&nbsp;boolean&nbsp;isInvisibleAnnotationType(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<div class="block">Returns true if annotations of this type are not visible to the user. For example <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION"><code>TYPE_EXTRACTED_REGION</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRANSLATED_REGION"><code>TYPE_TRANSLATED_REGION</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_ORIGIN"><code>TYPE_ORIGIN</code></a> or types starting with <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#INVISIBLE_ANNOTATION_TYPE_PREFIX"><code>INVISIBLE_ANNOTATION_TYPE_PREFIX</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>type</code> - an annotation type</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if annotations of this type are not visible to the user.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.605 (Geneious 6.0.5)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#INVISIBLE_ANNOTATION_TYPE_PREFIX"><code>INVISIBLE_ANNOTATION_TYPE_PREFIX</code></a></dd></dl>
</li>
</ul>
<a name="equals(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre>public final&nbsp;boolean&nbsp;equals(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;obj)</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="hashCode()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hashCode</h4>
<pre>public&nbsp;int&nbsp;hashCode()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="createTrimAnnotation(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createTrimAnnotation</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;createTrimAnnotation(int&nbsp;minimumIndexInclusive,
                                      int&nbsp;maximumIndexInclusive)</pre>
<div class="block">Creates a new <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_TRIMMED"><code>TYPE_TRIMMED</code></a> annotation</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumIndexInclusive</code> - the minimum 1-based index</dd><dd><code>maximumIndexInclusive</code> - the maximum 1-based index</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new trim annotation</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="copyOf(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>copyOf</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;copyOf(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;sequenceAnnotationsToCopy)</pre>
<div class="block">Creates a deep copy of a list of SequenceAnnotations. This is useful in cases where you obtain a list of annotations from another class and
 need to ensure the other class doesn't modify the list or any of the contained annotations after providing it to you.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceAnnotationsToCopy</code> - the annotations to copy.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a deep copy of sequenceAnnotationsToCopy</dd></dl>
</li>
</ul>
<a name="addInterval(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addInterval</h4>
<pre>public&nbsp;void&nbsp;addInterval(int&nbsp;from,
               int&nbsp;to)</pre>
<div class="block">Add one left to right interval to this annotation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>from</code> - lower limit (inclusive), in range 1 to sequence length.</dd><dd><code>to</code> - upper limit (inclusive), in range 1 to sequence length.</dd></dl>
</li>
</ul>
<a name="addInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval.Direction)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addInterval</h4>
<pre>public&nbsp;void&nbsp;addInterval(int&nbsp;minIndex,
               int&nbsp;maxIndex,
               <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a>&nbsp;direction)</pre>
<div class="block">Add one interval to this annotation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minIndex</code> - The minimum index (inclusive) spanned by this interval, in range 1 to sequence length.</dd><dd><code>maxIndex</code> - The maximum index (inclusive) spanned by this interval, in range 1 to sequence length.</dd><dd><code>direction</code> - The direction of this annotation.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationInterval.Direction</code></a></dd></dl>
</li>
</ul>
<a name="setIntervals(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setIntervals</h4>
<pre>public&nbsp;void&nbsp;setIntervals(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;intervals)</pre>
<div class="block">set the intervals on this annotation</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>intervals</code> - the intervals to set on the annotation</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.60 (Geneious 5.6.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationInterval</code></a></dd></dl>
</li>
</ul>
<a name="addInterval(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addInterval</h4>
<pre>public&nbsp;void&nbsp;addInterval(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;interval)</pre>
<div class="block">add a new interval to this annotation.
 In Geneious 5.0 and later. annotations are not required to have any intervals. If they don't have intervals they will not
 be displayed in the sequence viewer, but will be visible in the annotations viewer.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>interval</code> - an interval to copy the newly added interval from</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationInterval</code></a></dd></dl>
</li>
</ul>
<a name="addIntervals(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addIntervals</h4>
<pre>public&nbsp;void&nbsp;addIntervals(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;intervalsToAdd)</pre>
<div class="block">adds a collection of intervals to this annotation</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>intervalsToAdd</code> - the intervals to add</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationInterval</code></a></dd></dl>
</li>
</ul>
<a name="removeInterval(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeInterval</h4>
<pre>public&nbsp;void&nbsp;removeInterval(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;interval)</pre>
<div class="block">Removes the given interval from this annotation. Does nothing if the interval is not already present.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>interval</code> - the interval to remove</dd></dl>
</li>
</ul>
<a name="copyWithoutIntervals()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>copyWithoutIntervals</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;copyWithoutIntervals()</pre>
<div class="block">Creates a copy of this annotation, without any intervals. i.e. the returned annotation will have no intervals.
 The name, type and all qualifiers on this annotation will be the same on the returned annotation.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a copy of this annotation, without any intervals. i.e. the returned annotation will have no intervals.</dd></dl>
</li>
</ul>
<a name="addQualifier(java.lang.String, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addQualifier</h4>
<pre>public&nbsp;void&nbsp;addQualifier(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;value)</pre>
<div class="block">Add a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>qualifier</code></a> with the specified name-value mapping.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - name of the new SequenceAnnotationQualifier to add</dd><dd><code>value</code> - value of the new SequenceAnnotationQualifier to add</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if name or value is null</dd></dl>
</li>
</ul>
<a name="setQualifier(java.lang.String, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setQualifier</h4>
<pre>public&nbsp;void&nbsp;setQualifier(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;value)</pre>
<div class="block">Deletes all existing qualifiers with the given name, and adds a new qualifier with the given value.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - Name of the existing qualifier whose value to replace; must not be null</dd><dd><code>value</code> - Value to set on the qualifier; must not be null</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if name or value is null</dd></dl>
</li>
</ul>
<a name="removeQualifier(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeQualifier</h4>
<pre>public&nbsp;void&nbsp;removeQualifier(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block">Deletes all existing qualifiers with the given name</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - Name of the qualifier whose values to delete; must not be null</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if name is null</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getQualifierValue(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getQualifierValue</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getQualifierValue(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block">Get the value of the first qualifier in the list of the given name or "" if one is not found.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - the name of the qualifier to get the value for; must not be nll</dd>
<dt><span class="strong">Returns:</span></dt><dd>the value of the qualifier with the specified name, or "" if no such qualifier was found</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if name is null</dd></dl>
</li>
</ul>
<a name="addQualifier(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationQualifier)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addQualifier</h4>
<pre>public&nbsp;void&nbsp;addQualifier(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&nbsp;qualifier)</pre>
<div class="block">add a new <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>qualifier</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>qualifier</code> - a qualifier to copy the newly added qualifier from</dd></dl>
</li>
</ul>
<a name="addQualifiers(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addQualifiers</h4>
<pre>public&nbsp;void&nbsp;addQualifiers(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&gt;&nbsp;qualifiersToAdd)</pre>
<div class="block">Adds all the given qualifiers to this annotation</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>qualifiersToAdd</code> - the qualifiers to add to this annotation.</dd></dl>
</li>
</ul>
<a name="removeQualifier(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationQualifier)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeQualifier</h4>
<pre>public&nbsp;void&nbsp;removeQualifier(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&nbsp;qualifierToRemove)</pre>
<div class="block">Removes the given qualifier from this annotation. Does nothing if the qualifier is not already present.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>qualifierToRemove</code> - the qualifier to remove.</dd></dl>
</li>
</ul>
<a name="setQualifiers(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setQualifiers</h4>
<pre>public&nbsp;void&nbsp;setQualifiers(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationQualifier</a>&gt;&nbsp;qualifiers)</pre>
<div class="block">Set a new list of qualifiers on this SequenceAnnotation, removing any
 existing qualifiers added via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifier(java.lang.String, java.lang.String)"><code>addQualifier(String, String)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifier(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationQualifier)"><code>addQualifier(SequenceAnnotationQualifier)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#addQualifiers(java.util.Collection)"><code>addQualifiers(java.util.Collection)</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#setQualifiers(java.util.List)"><code>setQualifiers(java.util.List)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>qualifiers</code> - the new <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationQualifier</code></a>s to set; must not be null, and must not contain null entries</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if qualifiers or one of its elements is null</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="getVersionSupportStatic(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupportStatic</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupportStatic(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block">A static implementation of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code></a>
 that can be used in place of that method since all SequenceAnnotation instances have the same version support.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - the type of version number to get</dd>
<dt><span class="strong">Returns:</span></dt><dd>the version support as specified by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;majorVersion,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block">Serialization of annotation - to.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>majorVersion</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>annotation as a jdom element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;element)
             throws <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - representation from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre>public&nbsp;void&nbsp;setName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - The new name to set for this SequenceAnnotation; must not be null</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if name is null</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getName()"><code>getName()</code></a></dd></dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<div class="block">Returns a human readable string with information about this annotation;
 the exact format of the string is not guaranteed and should not be relied
 on, because it may change at any time.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>A human readable string with information about this annotation.</dd></dl>
</li>
</ul>
<a name="writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;out)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Writes this annotation to an output stream as defined by <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io"><code>Serializable</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>out</code> - the stream to write to.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we are unable to write to the stream</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>readObject(GeneiousObjectInputStream)</code></a></dd></dl>
</li>
</ul>
<a name="writeObject(java.io.DataOutput)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Writes this annotation to a DataOutput</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>out</code> - the DataOutput to write to.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we are unable to write to the DataOutput</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>readObject(GeneiousObjectInputStream)</code></a></dd></dl>
</li>
</ul>
<a name="writeObject(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, java.io.DataOutput)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeObject</h4>
<pre>public&nbsp;void&nbsp;writeObject(<a href="../../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataOutput.html?is-external=true" title="class or interface in java.io">DataOutput</a>&nbsp;out)
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Writes this annotation to a DataOutput in a form that can potentially be read by old versions of Geneious</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the version number of Geneious that must be able to deserialize this.</dd><dd><code>out</code> - the DataOutput to write to.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we are unable to write to the DataOutput</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>readObject(GeneiousObjectInputStream)</code></a></dd></dl>
</li>
</ul>
<a name="readObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readObject</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;readObject(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)
                                     throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Reads data previously created using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#writeObject(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)"><code>writeObject(GeneiousObjectOutputStream)</code></a>
 It is a good idea to use a <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>GeneiousObjectInputStream</code></a> when deserializing
 lots of annotations.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>in</code> - the stream to read from</dd>
<dt><span class="strong">Returns:</span></dt><dd>the deserialized annotation</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we are unable to read from the stream</dd></dl>
</li>
</ul>
<a name="readObject(java.io.DataInput)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readObject</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;readObject(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/DataInput.html?is-external=true" title="class or interface in java.io">DataInput</a>&nbsp;in)
                                     throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Reads data previously created using <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#writeObject(java.io.DataOutput)"><code>writeObject(java.io.DataOutput)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>in</code> - the DataInput to read from</dd>
<dt><span class="strong">Returns:</span></dt><dd>the deserialized annotation</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if we are unable to read from the DataInput</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd></dl>
</li>
</ul>
<a name="reverse(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.Alphabet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverse</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;reverse(int&nbsp;sequenceLength,
                         <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a>&nbsp;alphabet)</pre>
<div class="block">Creates a new annotation that is either the reverse or the reverse complements this annotation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceLength</code> - the length of the sequence in which the annotation is being reversed.</dd><dd><code>alphabet</code> - the alphabet of the sequence the annotation is from. This is used to determine whether or not to complement the bases in any editing annotation qualifiers (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#EDITING_HISTORY_ORIGINAL_BASES"><code>SequenceAnnotationQualifier.EDITING_HISTORY_ORIGINAL_BASES</code></a>)</dd>
<dt><span class="strong">Returns:</span></dt><dd>this annotation in the reverse direction.</dd></dl>
</li>
</ul>
<a name="getCodonStart()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCodonStart</h4>
<pre>public&nbsp;int&nbsp;getCodonStart()</pre>
<div class="block">The value of the <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#NCBI_CODON_START"><code>NCBI_CODON_START</code></a> qualifier indicates the offset at which the first complete codon of a coding feature can be found,
 relative to the first base of that feature. Valid values are 1, 2 or 3.

 If this annotation has an NCBI_CODON_START qualifier, and the value of it is between 1 and 3, return that value.  Otherwise, return -1</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the codon start of this annotation, or -1 if there is no NCBI_CODON_START qualifier or its value is not between 1 and 3</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationQualifier.html#NCBI_CODON_START"><code>SequenceAnnotationQualifier.NCBI_CODON_START</code></a></dd></dl>
</li>
</ul>
<a name="isEditingAnnotationType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEditingAnnotationType</h4>
<pre>public static&nbsp;boolean&nbsp;isEditingAnnotationType(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>type</code> - a type of annotation returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this annotation type (<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a>) is <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_DELETION"><code>TYPE_EDITING_HISTORY_DELETION</code></a> or
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_INSERTION"><code>TYPE_EDITING_HISTORY_INSERTION</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EDITING_HISTORY_REPLACEMENT"><code>TYPE_EDITING_HISTORY_REPLACEMENT</code></a>.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="isReadOnlyAnnotationType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReadOnlyAnnotationType</h4>
<pre>public static&nbsp;boolean&nbsp;isReadOnlyAnnotationType(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>type</code> - a type of annotation returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#getType()"><code>getType()</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this annotation type is read only (i.e. the user shouldn't be allowed to edit it)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getAnnotationAdjustedForInsertion(int, int, boolean, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationAdjustedForInsertion</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;getAnnotationAdjustedForInsertion(int&nbsp;indexToInsertAt,
                                                   int&nbsp;lengthOfInsertion,
                                                   boolean&nbsp;isInsertionAllGaps,
                                                   int&nbsp;newSequenceLengthIfCircular)</pre>
<div class="block">Handles an insertion into the sequence which this annotation is for, by either returning a new annotation adjusted to take the insertion into account, or this
 annotation if it doesn't change, or null if the annotation should be deleted (for example the user inserts nucleotides into a primer annotation)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>indexToInsertAt</code> - the 1-based index to insert. On a circular sequence, this must still be in the range 1 to the sequence length.</dd><dd><code>lengthOfInsertion</code> - the number of characters in the insertion</dd><dd><code>isInsertionAllGaps</code> - true if the insertion consists entirely of gaps (which means for example we don't delete primer annotations)</dd><dd><code>newSequenceLengthIfCircular</code> - if the sequence this annotation corresponds to is circular, then this should be >0 to indicate the length of a circular sequence (after the insertion)</dd>
<dt><span class="strong">Returns:</span></dt><dd>either a new annotation (if the insertion was within or prior to this annotation), or this annotation (if it is unaffected by the insertion), or null (if this annotation should be deleted)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getAnnotationAdjustedForReplacement(int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationAdjustedForReplacement</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;getAnnotationAdjustedForReplacement(int&nbsp;replacementPosition,
                                                     int&nbsp;replacementLength,
                                                     int&nbsp;sequenceLengthIfCircular)</pre>
<div class="block">Handles a non-gap replacement in the sequence which this annotation is for, by either returning this
 annotation if it doesn't change, or null if the annotation should be deleted (for example the user replaces nucleotides in a primer annotation)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>replacementPosition</code> - the 1-based index of the first character being replaced</dd><dd><code>replacementLength</code> - the length of the region being replaced</dd><dd><code>sequenceLengthIfCircular</code> - if the sequence this annotation corresponds to is circular, then this should be >0 to indicate the length of a circular sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>either this annotation (if it is unaffected by the replacement), or null (if this annotation should be deleted)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getAnnotationAdjustedForDeletion(int, int, boolean, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationAdjustedForDeletion</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;getAnnotationAdjustedForDeletion(int&nbsp;deletionStart,
                                                  int&nbsp;deletionLength,
                                                  boolean&nbsp;isDeletionAllGaps,
                                                  int&nbsp;newSequenceLengthIfCircular)</pre>
<div class="block">Handles a deletion from the sequence which this annotation is for, by either returning a new annotation adjusted to take the deletion into account, or this
 annotation if it doesn't change, or null if the annotation should be deleted (for example the user deletes nucleotides from a primer annotation)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>deletionStart</code> - the 1-based index the deletion begins at. On a circular sequence, this must still be in the range 1 to the sequence length.</dd><dd><code>deletionLength</code> - the number of characters deleted.</dd><dd><code>isDeletionAllGaps</code> - true if all the removed characters are gaps</dd><dd><code>newSequenceLengthIfCircular</code> - if the sequence this annotation corresponds to is circular, then this should be >0 to indicate the length of a circular sequence (after the deletion)</dd>
<dt><span class="strong">Returns:</span></dt><dd>either a new annotation (if the deletion was within this annotation), or this annotation (if it is unaffected by the insertion), or null (if this annotation should be deleted)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="getTranslation(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getTranslation</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getTranslation(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document,
                    <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                    <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<div class="block">Returns the translation (if applicable) for the given annotation. Usually this should only be called on CDS annotations.
 The genetic code defined by the annotations' trans_table qualifier is used if present. If that is not specified, the genetic code
 specified by the document is used. If that is not specified, the genetic code of the sequence is used. If that is not specified,
 the default genetic code (as specified from the cog section of the sequence viewer preferences) is used.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>document</code> - the document the sequence came from. Used to obtain the genetic code. May be null.</dd><dd><code>sequence</code> - the sequence the annotation to applies to</dd><dd><code>annotation</code> - the annotation to translate</dd>
<dt><span class="strong">Returns:</span></dt><dd>the translation for the given annotation or null if it cannot be translated</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.800 (Geneious 8.0.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceAnnotation.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentListDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationGroup.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" target="_top">Frames</a></li>
<li><a href="SequenceAnnotation.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
