<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>SequenceDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" target="_top">Frames</a></li>
<li><a href="SequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Interface SequenceDocument" class="title">Interface SequenceDocument</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Superinterfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></dd>
</dl>
<dl>
<dt>All Known Subinterfaces:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">AminoAcidSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ChromatogramDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">ChromatogramDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableNucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableNucleotideGraphSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">EditableSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraphSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocumentWithEditableAnnotations</a></dd>
</dl>
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndAminoAcidSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndAminoAcidSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideGraphSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndNucleotideGraphSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndNucleotideSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndNucleotideSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/CombinedAlignmentAndSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">CombinedAlignmentAndSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultAminoAcidSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultAminoAcidSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideGraphSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultNucleotideGraphSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultNucleotideSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequence</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequenceImplementation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">ImmutableSequenceImplementation</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/OligoSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">OligoSequenceDocument</a></dd>
</dl>
<hr>
<br>
<pre>public interface <span class="strong">SequenceDocument</span>
extends <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></pre>
<div class="block">Interface representing one sequence.

 Any concrete implementation of this interface <b>must</b> implement either
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/AminoAcidSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>AminoAcidSequenceDocument</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocument</code></a> to
 represent amino acid or nucleotide sequences because Geneious relies on
 these interfaces for determining the type of a sequence.
 </p>
 The most widely used implementations of a SequenceDocument are
 <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultSequenceDocument</code></a>
 and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a>
 <p/>
 Not all SequenceDocuments are editable, although most implementations also implement <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/EditableSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>EditableSequenceDocument</code></a></div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>SequenceUtilities</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>SequenceExtractionUtilities</code></a></dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Alphabet</a></strong></code>
<div class="block">Supported sequence types.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Transformer.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument.Transformer</a></strong></code>
<div class="block">Provides a method for returning a new sequence which represents a transformation made to another sequence.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#GENOME_SEQUENCE_THRESHOLD">GENOME_SEQUENCE_THRESHOLD</a></strong></code>
<div class="block">The sequence length threshold at which we decide a sequence is a genome.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()">getCharSequence</a></strong>()</code>
<div class="block">Get the sequence as a CharSequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceAnnotations()">getSequenceAnnotations</a></strong>()</code>
<div class="block">Gets sequence annotations which are directly on the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceLength()">getSequenceLength</a></strong>()</code>
<div class="block">Returns the length of the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()">getSequenceString</a></strong>()</code>
<div class="block">Get the sequence as a string.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()">isCircular</a></strong>()</code>
<div class="block">Returns true if this sequence is circular.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="GENOME_SEQUENCE_THRESHOLD">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GENOME_SEQUENCE_THRESHOLD</h4>
<pre>static final&nbsp;int GENOME_SEQUENCE_THRESHOLD</pre>
<div class="block">The sequence length threshold at which we decide a sequence is a genome. eg. sequence longer than this are serialized
 differently and have a different set of defaults/preferences in the sequence view.
 <br><br>
 At API 4.50 this is 100,000.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../constant-values.html#com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument.GENOME_SEQUENCE_THRESHOLD">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getSequenceString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceString</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSequenceString()</pre>
<div class="block">Get the sequence as a string. For nucleotide or amino acid sequences,
 each residue or ambiguity symbol is represented by one uppercase
 or lowercase character as per the IUPAC recommendations.
 <p/>
 For nucleotide sequences, any of the following are valid characters "ACGTURYMWSKBDHVN?-acgturymwskbdhvn"
 <p/>
 For amino acid sequences, any of the following are valid characters "ACDEFGHIKLMNPQRSTVWY*OUBJZX?-acdefghiklmnpqrstvwyoubjzx"

 If <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a> doesn't call this method, then one possible implementation
 for this method is the following:

 <p><code>return getCharSequence().toString();</code></p>

 <p>WARNING: It is generally a bad idea to use this method on anything but very short sequences. On large sequences (e.g. whole chromosomes) this method
 may take some time and may use significant amounts of memory. e.g. A String representation of human chr1 requires 500 MB of memory. It is preferable to use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a> instead.</p></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sequence as a string; must not return null.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLength</h4>
<pre>int&nbsp;getSequenceLength()</pre>
<div class="block">Returns the length of the sequence. This returns exactly the same as
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a>.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>length()</code></a>,
 but may be more performant on some implementations.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The length of this document's sequence.

 Note that if a class implements both this interface and <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraph</code></a>,
 the return value of this method must be consistent with the contracts from both
 interfaces.</dd></dl>
</li>
</ul>
<a name="getCharSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCharSequence</h4>
<pre><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&nbsp;getCharSequence()</pre>
<div class="block">Get the sequence as a CharSequence. Used as an alternative to
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>getSequenceString()</code></a>
 in situations where a String is unnecessary and could potentially
 use too much memory, or where we want to efficiently calculate the terminal gap lengths.

 The returned CharSequence must always return consistent
 values from any of the CharSequence methods
 (e.g. <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>CharSequence.length()</code></a> and  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#charAt(int)" title="class or interface in java.lang"><code>CharSequence.charAt(int)</code></a>.
 If a SequenceDocument is editable and changes are made to it,
 it is required that the original CharSequence obtained from
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a>
 not change. A client of the SequenceDocument is only able to obtain
 the changes by making a fresh call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>getCharSequence()</code></a>.

 If <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>getSequenceString()</code></a> doesn't call this method, then it is possible to
 implement this method as follows:

 <p><code>return SequenceCharSequence.valueOf(getSequenceString());</code></p>

 <p>However, it is recommended that a more memory efficient representation is used
 where possible; in that case, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>getSequenceString()</code></a> can be implemented
 based on this method (see its javadoc).</p>
 <p>Some implementations of this method may choose to throw a RuntimeException whose cause is an XMLSerializationException if there is
 insufficient memory to load the sequence. In most situations (e.g. from a DocumentOperation or DocumentFileImporter) catching
 this exception is unnecessary as core Geneious will have either pre-loaded the sequence or will catch and unwrap this exception itself gracefully.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sequence as a SequenceCharSequence; must not return null.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getSequenceString()"><code>getSequenceString()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceAnnotations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAnnotations</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getSequenceAnnotations()</pre>
<div class="block">Gets sequence annotations which are directly on the sequence.
 <b>This does not include annotations on <a href="../../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>sequence tracks</code></a>.</b>
 <br><br>
 To get all annotations, including those on sequence tracks, use
 <a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAndTrackAnnotations</code></a>

 This method may return an unmodifiable list and must not return the list which is used internally by the SequenceDocument to store the annotations
 (ie modifying the returned list should not modify the annotations on the SequenceDocument directly)<br/><br/>

 <b>WARNING:</b> this list may not include all SequenceAnnotations represented as annotations in the sequence viewer.
 One such case is with trim annotations, which should be found using
 <a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAnnotationsIncludingImmutableSequencesTrims(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAnnotationsIncludingImmutableSequencesTrims(SequenceDocument)</code></a>.<br/><br/>

 This may also return annotations that are not visible in the sequence viewer, such as <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html#TYPE_EXTRACTED_REGION"><code>SequenceAnnotation.TYPE_EXTRACTED_REGION</code></a>.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a list of SequenceAnnotations; must not return null, but may return an
         empty list (e.g. <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Collections.html?is-external=true#emptyList()" title="class or interface in java.util"><code>Collections.emptyList()</code></a>).</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#getSequenceAndTrackAnnotations(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceUtilities.getSequenceAndTrackAnnotations(SequenceDocument)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocumentWithEditableAnnotations.html#setAnnotations(java.util.List)"><code>SequenceDocumentWithEditableAnnotations.setAnnotations(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="isCircular()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>isCircular</h4>
<pre>boolean&nbsp;isCircular()</pre>
<div class="block">Returns true if this sequence is circular.
 <p/>
 A circular sequence is one where the end points touch forming one continuous
 sequence without ends.
 For information on how annotations are handled in circular sequences, see.
 <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getFrom()"><code>SequenceAnnotationInterval.getFrom()</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this sequence is circular.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.Alphabet.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" target="_top">Frames</a></li>
<li><a href="SequenceDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
