<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>SequenceAnnotationInterval (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceAnnotationInterval (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceAnnotationInterval.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationGroup.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" target="_top">Frames</a></li>
<li><a href="SequenceAnnotationInterval.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.documents.sequence</div>
<h2 title="Class SequenceAnnotationInterval" class="title">Class SequenceAnnotationInterval</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public final class <span class="strong">SequenceAnnotationInterval</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Immutable pair of integers with a direction, representing a directed range in the sequence annotation.
 Both ends are included in the range except in the special case of a 0 length interval between 2 bases (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isBetweenBases()"><code>isBetweenBases()</code></a>)
 <p/>
 Values are usually in range from 1 to n (sequence length) but may
 extend outside the sequence range in cases where the partial sequence it is on doesn't cover the full length
 of the annotation.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a></strong></code>
<div class="block">The direction of a sequence annotation interval.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#SequenceAnnotationInterval(com.biomatters.geneious.publicapi.utilities.Interval)">SequenceAnnotationInterval</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;interval)</code>
<div class="block">Creates a SequenceAnnotationInterval (which uses 1-based inclusive max co-ordinates) from an interval (which uses 0-based exclusive max co-ordinates)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#SequenceAnnotationInterval(int, int)">SequenceAnnotationInterval</a></strong>(int&nbsp;from,
                          int&nbsp;to)</code>
<div class="block">Backwards compatibility constructor for an interval that starts
 at index from (inclusive) and ends at index to (inclusive).</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#SequenceAnnotationInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval.Direction)">SequenceAnnotationInterval</a></strong>(int&nbsp;minIndex,
                          int&nbsp;maxIndex,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a>&nbsp;direction)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#SequenceAnnotationInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval.Direction, boolean, boolean)">SequenceAnnotationInterval</a></strong>(int&nbsp;minIndex,
                          int&nbsp;maxIndex,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a>&nbsp;direction,
                          boolean&nbsp;isTruncatedAtMinimumIndex,
                          boolean&nbsp;isTruncatedAtMaximumIndex)</code>
<div class="block">Main constructor.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#SequenceAnnotationInterval(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">SequenceAnnotationInterval</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;interval)</code>
<div class="block">Construct a SequenceAnnotationInterval that is identical to the given parameter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#SequenceAnnotationInterval(java.lang.String)">SequenceAnnotationInterval</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;text)</code>
<div class="block">constructs interval from the string representation of the form "[...]n,nnn D m,mmm[...]" where "n,nnn"
 is an integer with commas every three digits and "D" is a valid direction (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#toArrowString()"><code>SequenceAnnotationInterval.Direction.toArrowString()</code></a>).</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#adjustIntervalForGapInsertion(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)">adjustIntervalForGapInsertion</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a>&nbsp;sequenceGapInformation)</code>
<div class="block">Creates a new interval adjusted to compensate for inserting gaps into the sequence covered by this interval.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#adjustIntervalForGapRemoval(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)">adjustIntervalForGapRemoval</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a>&nbsp;sequenceGapInformation)</code>
<div class="block">Creates a new interval adjusted to compensate for removing gaps from the sequence covered by this interval.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#asInterval()">asInterval</a></strong>()</code>
<div class="block">Returns this SequenceAnnotationInterval as an Interval, which uses 0-based exclusive max co-ordinates rather than 1-based inclusive max co-ordinates used by SequenceAnnotationInterval</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#between(int, int)">between</a></strong>(int&nbsp;min,
       int&nbsp;max)</code>
<div class="block">Creates a zero-length SequenceAnnotationInterval between the given bases/residues with direction <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#none"><code>SequenceAnnotationInterval.Direction.none</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#between(int, int, int)">between</a></strong>(int&nbsp;min,
       int&nbsp;max,
       int&nbsp;gapLengthToExpandToFill)</code>
<div class="block">Creates a zero-length SequenceAnnotationInterval between the given bases/residues with direction <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#none"><code>SequenceAnnotationInterval.Direction.none</code></a> such that if it appears
 in an alignment with a gap in this region of the given size, will expand to fill that gap.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#contains(int)">contains</a></strong>(int&nbsp;index)</code>
<div class="block">Checks whether this interval includes the residue at the specified index.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#contains(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">contains</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;anotherInterval)</code>
<div class="block">Determines if this interval entirely covers another interval.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#containsOnPotentiallyCircularSequence(int, int)">containsOnPotentiallyCircularSequence</a></strong>(int&nbsp;index,
                                     int&nbsp;sequenceLengthIfCircular)</code>
<div class="block">Checks whether this interval includes the residue at the specified index in a potentially circular sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#equals(java.lang.Object)">equals</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;obj)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getDirection()">getDirection</a></strong>()</code>
<div class="block">The direction in which this annotation interval is oriented.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getFrom()">getFrom</a></strong>()</code>
<div class="block">Start position of annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getLength()">getLength</a></strong>()</code>
<div class="block">The length (number of residues and gaps) of this interval.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMaximumIndex()">getMaximumIndex</a></strong>()</code>
<div class="block">The largest sequence index covered by this interval, equivalent to Math.max(getFrom(), getTo()).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()">getMinimumIndex</a></strong>()</code>
<div class="block">The smallest sequence index covered by this interval, equivalent to Math.min(getFrom(), getTo()).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getResidues(java.lang.CharSequence, int, boolean)">getResidues</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;residues,
           int&nbsp;maximumResiduesToReturn,
           boolean&nbsp;isNucleotideSequence)</code>
<div class="block">Gets the residues covered by this interval for displaying to a user.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getReverse()">getReverse</a></strong>()</code>
<div class="block">Returns the reverse of this interval, covering the same sequence residues but having
 the opposite direction (as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#reverse()"><code>SequenceAnnotationInterval.Direction.reverse()</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getTo()">getTo</a></strong>()</code>
<div class="block">End position of annotation, usually (but not always) in the range 1 to the length of the sequence inclusive.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#hashCode()">hashCode</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWith(int, int)">intersectsWith</a></strong>(int&nbsp;minimumIndexOfOtherInterval,
              int&nbsp;maximumIndexOfOtherIntervalInclusive)</code>
<div class="block">Equivalent to calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWith(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)"><code>intersectsWith(new SequenceAnnotationInterval(minimumIndexOfOtherInterval, maximumIndexOfOtherIntervalInclusive))</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWith(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">intersectsWith</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;anotherInterval)</code>
<div class="block">Determines if this interval intersects with another interval.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWithInCircularSequence(int, int, int)">intersectsWithInCircularSequence</a></strong>(int&nbsp;minimumIndexOfOtherInterval,
                                int&nbsp;maximumIndexOfOtherIntervalInclusive,
                                int&nbsp;circularSequenceLength)</code>
<div class="block">Equivalent to calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWithInCircularSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval, int)"><code>intersectsWithInCircularSequence(new SequenceAnnotationInterval(minimumIndexOfOtherInterval, maximumIndexOfOtherIntervalInclusive),circularSequenceLength)</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWithInCircularSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval, int)">intersectsWithInCircularSequence</a></strong>(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;anotherInterval,
                                int&nbsp;circularSequenceLength)</code>
<div class="block">Determines if this interval intersects with another interval in a circular sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isBetweenBases()">isBetweenBases</a></strong>()</code>
<div class="block">Returns true if this interval represents a position between 2 bases, which is represented by the special case where <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMaximumIndex()"><code>getMaximumIndex()</code></a>==<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()"><code>getMinimumIndex()</code></a>-1.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isTruncatedAtMaximumIndex()">isTruncatedAtMaximumIndex</a></strong>()</code>
<div class="block">Checks whether the interval is incomplete and really finishes after the maximum index.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isTruncatedAtMinimumIndex()">isTruncatedAtMinimumIndex</a></strong>()</code>
<div class="block">Checks whether the interval is incomplete and really beings before the minimum index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#merge(java.util.List)">merge</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;_intervals)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>equivalent to calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#merge(java.util.List, boolean)"><code>merge(intervals, true)</code></a></i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#merge(java.util.List, boolean)">merge</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;_intervals,
     boolean&nbsp;mergeIntervalsWithDifferentDirections)</code>
<div class="block">Merges a list of intervals, i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#offsetBy(int)">offsetBy</a></strong>(int&nbsp;offsetDistance)</code>
<div class="block">Returns a new interval offset to the right by the given distance from this interval</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#reverse()">reverse</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getReverse()"><code>getReverse()</code></a> instead.</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#toString(boolean)">toString</a></strong>(boolean&nbsp;showDigitSeparator)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#toString(boolean, boolean)">toString</a></strong>(boolean&nbsp;showDigitSeparator,
        boolean&nbsp;formatNicely)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#truncate(int, int)">truncate</a></strong>(int&nbsp;minIndex,
        int&nbsp;maxIndex)</code>
<div class="block">Truncates this interval to not extend below minIndex or above maxIndex.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SequenceAnnotationInterval(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotationInterval</h4>
<pre>public&nbsp;SequenceAnnotationInterval(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;text)
                           throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NumberFormatException.html?is-external=true" title="class or interface in java.lang">NumberFormatException</a></pre>
<div class="block">constructs interval from the string representation of the form "[...]n,nnn D m,mmm[...]" where "n,nnn"
 is an integer with commas every three digits and "D" is a valid direction (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#toArrowString()"><code>SequenceAnnotationInterval.Direction.toArrowString()</code></a>).
 This is the format returned from <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#toString()"><code>toString()</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>text</code> - A value of the format returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#toString()"><code>toString()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NumberFormatException.html?is-external=true" title="class or interface in java.lang">NumberFormatException</a></code> - if the value isn't formatted correctly. This should only happen
        if text is not a value of the format returned by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#toString()"><code>toString()</code></a>.</dd></dl>
</li>
</ul>
<a name="SequenceAnnotationInterval(com.biomatters.geneious.publicapi.utilities.Interval)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotationInterval</h4>
<pre>public&nbsp;SequenceAnnotationInterval(<a href="../../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;interval)</pre>
<div class="block">Creates a SequenceAnnotationInterval (which uses 1-based inclusive max co-ordinates) from an interval (which uses 0-based exclusive max co-ordinates)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>interval</code> - the interval to create an equivalent SequenceAnnotationInterval to.</dd></dl>
</li>
</ul>
<a name="SequenceAnnotationInterval(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotationInterval</h4>
<pre>public&nbsp;SequenceAnnotationInterval(int&nbsp;from,
                          int&nbsp;to)</pre>
<div class="block">Backwards compatibility constructor for an interval that starts
 at index from (inclusive) and ends at index to (inclusive).
 The direction is leftToRight if from < end, and rightToLeft if from > end.
 For from==end, no guarantee is made what the direction of the resulting
 interval will be.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>from</code> - The start of the interval (inclusive), usually in range 1 to sequence length inclusive but it may extend outside this range in cases where the partial sequence it is on doesn't cover the full length of the annotation.</dd><dd><code>to</code> - The end of the interval (inclusive), usually in range 1 to sequence length inclusive but it may extend outside this range in cases where the partial sequence it is on doesn't cover the full length of the annotation.</dd></dl>
</li>
</ul>
<a name="SequenceAnnotationInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval.Direction)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotationInterval</h4>
<pre>public&nbsp;SequenceAnnotationInterval(int&nbsp;minIndex,
                          int&nbsp;maxIndex,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a>&nbsp;direction)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minIndex</code> - The minimum index (inclusive) spanned by this interval, usually in range 1 to sequence length inclusive but it may extend outside this range in cases where the partial sequence it is on doesn't cover the full length of the annotation.</dd><dd><code>maxIndex</code> - The maximum index (inclusive) spanned by this interval, usually in range 1 to sequence length inclusive but it may extend outside this range in cases where the partial sequence it is on doesn't cover the full length of the annotation.</dd><dd><code>direction</code> - The direction of the interval</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if maxIndex < minIndex - 1. The special case of maxIndex==minIndex-1 represents an annotation the sits between the bases maxIndex and minIndex. As of 2008-2-29, Geneious has not been checked to see if all modules handle this situation correctly, so it is generally not a good idea to use these special annotations yet.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if direction is null</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationInterval.Direction</code></a></dd></dl>
</li>
</ul>
<a name="SequenceAnnotationInterval(int, int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval.Direction, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SequenceAnnotationInterval</h4>
<pre>public&nbsp;SequenceAnnotationInterval(int&nbsp;minIndex,
                          int&nbsp;maxIndex,
                          <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a>&nbsp;direction,
                          boolean&nbsp;isTruncatedAtMinimumIndex,
                          boolean&nbsp;isTruncatedAtMaximumIndex)</pre>
<div class="block">Main constructor.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minIndex</code> - The minimum index (inclusive) spanned by this interval, indexed from 1</dd><dd><code>maxIndex</code> - The maximum index (inclusive) spanned by this interval, indexed from 1</dd><dd><code>direction</code> - The direction of the interval</dd><dd><code>isTruncatedAtMinimumIndex</code> - true to indicate that the interval is incomplete and really begins before minIndex.</dd><dd><code>isTruncatedAtMaximumIndex</code> - true to indicate that the interval is incomplete and really finishes after maxIndex.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if maxIndex < minIndex - 1. The special case of maxIndex==minIndex-1 represents an annotation the sits between the bases maxIndex and minIndex. As of 2008-2-29, Geneious has not been checked to see if all modules handle this situation correctly, so it is generally not a good idea to use these special annotations yet.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/NullPointerException.html?is-external=true" title="class or interface in java.lang">NullPointerException</a></code> - if direction is null</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationInterval.Direction</code></a></dd></dl>
</li>
</ul>
<a name="SequenceAnnotationInterval(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SequenceAnnotationInterval</h4>
<pre>public&nbsp;SequenceAnnotationInterval(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;interval)</pre>
<div class="block">Construct a SequenceAnnotationInterval that is identical to the given parameter.
 Because SequenceAnnotationInterval is an immutable class, it can be shared safely
 and it is never necessary to construct a copy of the same value. Therefore, use
 of this constructor is discouraged.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>interval</code> - interval to create a copy from</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a string representation of the form "[...]n,nnn D m,mmm[...]" where "n,nnn" is an integer with commas every three digits
        and '...' is prepended or appended if the interval beings before the minimum index or ends after the maximum index
        and D is an arrow in the format returned by Direction.toArrowString().  If this interval is between bases (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isBetweenBases()"><code>isBetweenBases()</code></a>) then the returned string will be of the form "Between 55 and 56".</dd></dl>
</li>
</ul>
<a name="toString(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString(boolean&nbsp;showDigitSeparator)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>showDigitSeparator</code> - true to separate every 3rd digit with commas</dd>
<dt><span class="strong">Returns:</span></dt><dd>a string representation of the form "[...]n,nnn D m,mmm[...]" where "n,nnn" is an integer with commas every three digits
        (if showDigitSeparator is true)
        and '...' is prepended or appended if the interval beings before the minimum index or ends after the maximum index
        and D is an arrow in the format returned by Direction.toArrowString(). If this interval is between bases (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isBetweenBases()"><code>isBetweenBases()</code></a>) then the returned string will be of the form "Between 55 and 56".</dd></dl>
</li>
</ul>
<a name="toString(boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString(boolean&nbsp;showDigitSeparator,
              boolean&nbsp;formatNicely)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>showDigitSeparator</code> - true to separate every 3rd digit with commas</dd><dd><code>formatNicely</code> - to display intervals in the reverse direction more nicely when the maximum and minimum indices are not identical. For example it would return "5->1" instead of the alternative format of "1 -< 5". Note that this nicer format is not guaranteed to be parsable by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#SequenceAnnotationInterval(java.lang.String)"><code>SequenceAnnotationInterval(String)</code></a> because formats such as "2->1" could be interpreted as an annotation between bases 1 and 2 in older versions of Geneious.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a string representation of the form "[...]n,nnn D m,mmm[...]" where "n,nnn" is an integer with commas every three digits
        (if showDigitSeparator is true)
        and '...' is prepended or appended if the interval beings before the minimum index or ends after the maximum index
        and D is an arrow in the format returned by Direction.toArrowString(). If this interval is between bases (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isBetweenBases()"><code>isBetweenBases()</code></a>) then the returned string will be of the form "Between 55 and 56".</dd></dl>
</li>
</ul>
<a name="asInterval()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>asInterval</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>&nbsp;asInterval()</pre>
<div class="block">Returns this SequenceAnnotationInterval as an Interval, which uses 0-based exclusive max co-ordinates rather than 1-based inclusive max co-ordinates used by SequenceAnnotationInterval</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>this SequenceAnnotationInterval as an Interval</dd></dl>
</li>
</ul>
<a name="between(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>between</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;between(int&nbsp;min,
                                 int&nbsp;max)</pre>
<div class="block">Creates a zero-length SequenceAnnotationInterval between the given bases/residues with direction <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#none"><code>SequenceAnnotationInterval.Direction.none</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>min</code> - base/residue this interval comes directly after</dd><dd><code>max</code> - base/residue this interval comes directly before</dd>
<dt><span class="strong">Returns:</span></dt><dd>a zero-length SequenceAnnotationInterval between the given bases/residues</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if max!=min+1 or min!=max+1</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.51 (Geneious 5.5.1)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isBetweenBases()"><code>isBetweenBases()</code></a></dd></dl>
</li>
</ul>
<a name="between(int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>between</h4>
<pre>public static&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;between(int&nbsp;min,
                                 int&nbsp;max,
                                 int&nbsp;gapLengthToExpandToFill)</pre>
<div class="block">Creates a zero-length SequenceAnnotationInterval between the given bases/residues with direction <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#none"><code>SequenceAnnotationInterval.Direction.none</code></a> such that if it appears
 in an alignment with a gap in this region of the given size, will expand to fill that gap.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>min</code> - base/residue this interval comes directly after</dd><dd><code>max</code> - base/residue this interval comes directly before</dd><dd><code>gapLengthToExpandToFill</code> - if this annotation appears in an alignment with a gap in this region of the given size, will expand to fill that gap.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a zero-length SequenceAnnotationInterval between the given bases/residues</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if max!=min+1 or min!=max+1</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.910 (Geneious 9.1.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isBetweenBases()"><code>isBetweenBases()</code></a></dd></dl>
</li>
</ul>
<a name="reverse()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverse</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;reverse()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getReverse()"><code>getReverse()</code></a> instead.</i></div>
<div class="block">Returns the reverse of this interval, covering the same sequence residues but having
 the opposite direction (as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#reverse()"><code>SequenceAnnotationInterval.Direction.reverse()</code></a>).</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the reverse of this interval</dd></dl>
</li>
</ul>
<a name="getReverse()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReverse</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;getReverse()</pre>
<div class="block">Returns the reverse of this interval, covering the same sequence residues but having
 the opposite direction (as defined by <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html#reverse()"><code>SequenceAnnotationInterval.Direction.reverse()</code></a>).</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the reverse of this interval</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="getFrom()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFrom</h4>
<pre>public&nbsp;int&nbsp;getFrom()</pre>
<div class="block">Start position of annotation. usually (but not always) in the range 1 to the length of the sequence inclusive.
 If the value is outside this range, the annotation is considered to start before (or after) the end of the sequence.
 <p/>
 In the case of circular sequences (see  <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a>),
 any multiple of the sequence length may be added (or subtracted) from the position,
 as long as at least one of <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()"><code>getMinimumIndex()</code></a>, <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMaximumIndex()"><code>getMaximumIndex()</code></a>
 is in the range [1, sequenceLength]. As this can still allow two different
 encodings of the range when it spans the end of the sequence, the recommended
 encoding is the one where <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()"><code>getMinimumIndex()</code></a> is in the range
 [1, sequenceLength].

 For example, if the sequence length of a circular sequence is 10,
 an annotation from 5 to 13 goes from residue 5 to residue 3.
 An annotation from 7 to 22 goes around the sequence 1.5 times
 starting on residue 7 and finishing on residue 2.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>start position of the annotation</dd></dl>
</li>
</ul>
<a name="getTo()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTo</h4>
<pre>public&nbsp;int&nbsp;getTo()</pre>
<div class="block">End position of annotation, usually (but not always) in the range 1 to the length of the sequence inclusive.
 If the value is outside this range, the annotation is considered to start before (or after)
 the end of the sequence.
 <p/>
 See <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getFrom()"><code>getFrom()</code></a> for how circular sequences are handled</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>end position of the annotation</dd></dl>
</li>
</ul>
<a name="getMinimumIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMinimumIndex</h4>
<pre>public&nbsp;int&nbsp;getMinimumIndex()</pre>
<div class="block">The smallest sequence index covered by this interval, equivalent to Math.min(getFrom(), getTo()).
 Usually in range 1 to sequence length.
 If the value is outside this range, the annotation is considered to start before (or after) the end of the sequence.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The smallest sequence index covered by this interval, equivalent to Math.min(getFrom(), getTo()).</dd></dl>
</li>
</ul>
<a name="getMaximumIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMaximumIndex</h4>
<pre>public&nbsp;int&nbsp;getMaximumIndex()</pre>
<div class="block">The largest sequence index covered by this interval, equivalent to Math.max(getFrom(), getTo()).
 Usually in the range 1 to sequence length.
 If the value is outside this range, the annotation is considered to start before (or after) the end of the sequence.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The largest sequence index covered by this interval, equivalent to Math.max(getFrom(), getTo()).</dd></dl>
</li>
</ul>
<a name="getDirection()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDirection</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval.Direction</a>&nbsp;getDirection()</pre>
<div class="block">The direction in which this annotation interval is oriented.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The direction in which this annotation interval is oriented.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotationInterval.Direction</code></a></dd></dl>
</li>
</ul>
<a name="isTruncatedAtMinimumIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isTruncatedAtMinimumIndex</h4>
<pre>public&nbsp;boolean&nbsp;isTruncatedAtMinimumIndex()</pre>
<div class="block">Checks whether the interval is incomplete and really beings before the minimum index.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true to indicate that the interval is incomplete and really beings before the minimum index.</dd></dl>
</li>
</ul>
<a name="isTruncatedAtMaximumIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isTruncatedAtMaximumIndex</h4>
<pre>public&nbsp;boolean&nbsp;isTruncatedAtMaximumIndex()</pre>
<div class="block">Checks whether the interval is incomplete and really finishes after the maximum index.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true to indicate that the interval is incomplete and really finishes after the maximum index.</dd></dl>
</li>
</ul>
<a name="contains(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>contains</h4>
<pre>public&nbsp;boolean&nbsp;contains(int&nbsp;index)</pre>
<div class="block">Checks whether this interval includes the residue at the specified index. If the sequence in question
 may potentially be a ciruclar sequence (see <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#isCircular()"><code>SequenceDocument.isCircular()</code></a>)
 then <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#containsOnPotentiallyCircularSequence(int, int)"><code>containsOnPotentiallyCircularSequence(int, int)</code></a> should be used instead.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - An index in a sequence (In range 1 to sequence length.)</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this interval contains the specified index. i.e. minIndex &lt;= index && index  &lt;= maxIndex</dd></dl>
</li>
</ul>
<a name="containsOnPotentiallyCircularSequence(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>containsOnPotentiallyCircularSequence</h4>
<pre>public&nbsp;boolean&nbsp;containsOnPotentiallyCircularSequence(int&nbsp;index,
                                            int&nbsp;sequenceLengthIfCircular)</pre>
<div class="block">Checks whether this interval includes the residue at the specified index in a potentially circular sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - An index in a sequence (In range 1 to sequence length.)</dd><dd><code>sequenceLengthIfCircular</code> - if the sequence this annotation corresponds to is circular, then this should be >0 to indicate the length of a circular sequence</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this interval contains the specified index. i.e. minIndex &lt;= index && index  &lt;= maxIndex</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.700 (Geneious 7.0.0)</dd></dl>
</li>
</ul>
<a name="contains(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>contains</h4>
<pre>public&nbsp;boolean&nbsp;contains(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;anotherInterval)</pre>
<div class="block">Determines if this interval entirely covers another interval.
 Two intervals with the same boundaries are considered to contain each other.
 <p>
 Intervals between two bases (the special case where <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMaximumIndex()"><code>getMaximumIndex()</code></a>==<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()"><code>getMinimumIndex()</code></a>-1)
 can only contain an identical interval. They are contained by an interval if they are within or on the edge of it.
 <p>
 This method does not handle annotations on a circular sequence (where Intervals that extend outside
 the range "1->sequence length" inclusive are considered to wrap around to
 the other end of the sequence)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>anotherInterval</code> - another interval to check containment of</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this interval entirely contains the other interval.</dd><dt><span class="strong">Since:</span></dt>
  <dd>Geneious API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="intersectsWith(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>intersectsWith</h4>
<pre>public&nbsp;boolean&nbsp;intersectsWith(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;anotherInterval)</pre>
<div class="block">Determines if this interval intersects with another interval. i.e.
 they have at least 1 residue in common. Two intervals are also
 considered to intersect if one interval entirely covers the other interval.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>anotherInterval</code> - another interval to check intersection with</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this interval intersects with the other interval.</dd></dl>
</li>
</ul>
<a name="intersectsWithInCircularSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>intersectsWithInCircularSequence</h4>
<pre>public&nbsp;boolean&nbsp;intersectsWithInCircularSequence(<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;anotherInterval,
                                       int&nbsp;circularSequenceLength)</pre>
<div class="block">Determines if this interval intersects with another interval in a circular sequence.
 i.e. they have at least 1 residue in common. Intervals that extend outside
 the range "1->sequence length" inclusive are considered to wrap around to
 the other end of the sequence.
 <p/>.
 Two intervals are also
 considered to intersect if one interval entirely covers the other interval.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>anotherInterval</code> - another interval to check intersection with</dd><dd><code>circularSequenceLength</code> - the length of the circular sequence on which both intervals are present.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this interval intersects with the other interval.</dd></dl>
</li>
</ul>
<a name="intersectsWith(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>intersectsWith</h4>
<pre>public&nbsp;boolean&nbsp;intersectsWith(int&nbsp;minimumIndexOfOtherInterval,
                     int&nbsp;maximumIndexOfOtherIntervalInclusive)</pre>
<div class="block">Equivalent to calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWith(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval)"><code>intersectsWith(new SequenceAnnotationInterval(minimumIndexOfOtherInterval, maximumIndexOfOtherIntervalInclusive))</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumIndexOfOtherInterval</code> - the 1-based minimum index of another interval to test for intersection with</dd><dd><code>maximumIndexOfOtherIntervalInclusive</code> - the 1-based maximum index (inclusive) of another interval to test for intersection with</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this interval intersects with the other interval.</dd></dl>
</li>
</ul>
<a name="intersectsWithInCircularSequence(int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>intersectsWithInCircularSequence</h4>
<pre>public&nbsp;boolean&nbsp;intersectsWithInCircularSequence(int&nbsp;minimumIndexOfOtherInterval,
                                       int&nbsp;maximumIndexOfOtherIntervalInclusive,
                                       int&nbsp;circularSequenceLength)</pre>
<div class="block">Equivalent to calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#intersectsWithInCircularSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval, int)"><code>intersectsWithInCircularSequence(new SequenceAnnotationInterval(minimumIndexOfOtherInterval, maximumIndexOfOtherIntervalInclusive),circularSequenceLength)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minimumIndexOfOtherInterval</code> - the 1-based minimum index of another interval to test for intersection with</dd><dd><code>maximumIndexOfOtherIntervalInclusive</code> - the 1-based maximum index (inclusive) of another interval to test for intersection with</dd><dd><code>circularSequenceLength</code> - the length of the circular sequence on which both intervals are present.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this interval intersects with the other interval.</dd></dl>
</li>
</ul>
<a name="isBetweenBases()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isBetweenBases</h4>
<pre>public&nbsp;boolean&nbsp;isBetweenBases()</pre>
<div class="block">Returns true if this interval represents a position between 2 bases, which is represented by the special case where <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMaximumIndex()"><code>getMaximumIndex()</code></a>==<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()"><code>getMinimumIndex()</code></a>-1. The bases it is between are getMaximumIndex() (lower of the two) and getMinimumIndex() (upper of the two), both 1-based</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this interval represents a position between 2 bases, which is represented by the special case where <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMaximumIndex()"><code>getMaximumIndex()</code></a>==<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()"><code>getMinimumIndex()</code></a>-1. The bases it is between are getMaximumIndex() (lower of the two) and getMinimumIndex() (upper of the two), both 1-based</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#between(int, int)"><code>between(int, int)</code></a></dd></dl>
</li>
</ul>
<a name="getLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLength</h4>
<pre>public&nbsp;int&nbsp;getLength()</pre>
<div class="block">The length (number of residues and gaps) of this interval.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The length of this interval, 1 + <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMaximumIndex()"><code>getMaximumIndex()</code></a> - <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#getMinimumIndex()"><code>getMinimumIndex()</code></a></dd></dl>
</li>
</ul>
<a name="merge(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>merge</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;merge(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;_intervals)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>equivalent to calling <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#merge(java.util.List, boolean)"><code>merge(intervals, true)</code></a></i></div>
</li>
</ul>
<a name="merge(java.util.List, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>merge</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;merge(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;_intervals,
                                     boolean&nbsp;mergeIntervalsWithDifferentDirections)</pre>
<div class="block">Merges a list of intervals, i.e. replaces sets of overlapping or (conditionally) directly
 adjacent intervals with a single interval. If merging intervals
 with different directions, to determine the direction of
 the resulting interval, the directions occurring in the overlapping set
 are considered in the order that they have been defined in the enum Direction.
 The first direction to have occurred is taken.
 This implies:
   - If all intervals in the overlapping set have the same direction, this direction is taken.
   - If there is a disagreement, preference is given to the direction that is the
     smallest according to Direction's natural order. The exact number of intervals
     for each direction does not matter; i.e. 10x right-to-left and 1x left-to-right yields left-to-right.
 <p/>
 Empty intervals are preserved (unless they are contained within other intervals).
 <p/>
 The returned merged intervals will not be flagged as either <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isTruncatedAtMinimumIndex()"><code>isTruncatedAtMinimumIndex()</code></a> or <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#isTruncatedAtMaximumIndex()"><code>isTruncatedAtMaximumIndex()</code></a> even
 if the original intervals were flagged in this way.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>_intervals</code> - the intervals to merge</dd><dd><code>mergeIntervalsWithDifferentDirections</code> - true to allow merging of intervals in different directions. False to merge them anyway and use one of the directions as described above.</dd>
<dt><span class="strong">Returns:</span></dt><dd>Merged set of intervals, ordered from left to right (increasing
         coordinates)</dd></dl>
</li>
</ul>
<a name="equals(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre>public&nbsp;boolean&nbsp;equals(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;obj)</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="hashCode()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hashCode</h4>
<pre>public&nbsp;int&nbsp;hashCode()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="getResidues(java.lang.CharSequence, int, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getResidues</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;getResidues(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;residues,
                       int&nbsp;maximumResiduesToReturn,
                       boolean&nbsp;isNucleotideSequence)</pre>
<div class="block">Gets the residues covered by this interval for displaying to a user.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residues</code> - the residues for the entire sequence.</dd><dd><code>maximumResiduesToReturn</code> - the maximum number of residue characters to include. Does not include the possible "..."s</dd><dd><code>isNucleotideSequence</code> - true if this is a nucleotide sequence. Used for deciding whether to reverse complement residues when a reverse annotation is encountered. For protein sequences, the residues displayed will always be in the forwards direction no matter what direction this interval has.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the residues covered by this interval, possibly truncated and with "..." appended or prepended</dd></dl>
</li>
</ul>
<a name="offsetBy(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>offsetBy</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;offsetBy(int&nbsp;offsetDistance)</pre>
<div class="block">Returns a new interval offset to the right by the given distance from this interval</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>offsetDistance</code> - the distance fo offset to the right</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new interval offset to the right by the given distance from this interval</dd></dl>
</li>
</ul>
<a name="adjustIntervalForGapInsertion(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>adjustIntervalForGapInsertion</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;adjustIntervalForGapInsertion(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a>&nbsp;sequenceGapInformation)</pre>
<div class="block">Creates a new interval adjusted to compensate for inserting gaps into the sequence covered by this interval.

 Note, if this interval was created from a previous call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#adjustIntervalForGapRemoval(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)"><code>adjustIntervalForGapRemoval(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)</code></a>
 then the original interval will be restored assuming the gaps around the ends of the interval are similar enough.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceGapInformation</code> - the gap information about the inserted gaps</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new interval adjusted to compensate for inserting gaps into the sequence covered by this interval.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#adjustIntervalForGapRemoval(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)"><code>adjustIntervalForGapRemoval(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGaps(java.lang.CharSequence, java.util.List)"><code>SequenceExtractionUtilities.adjustAnnotationsForGaps(CharSequence, java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="adjustIntervalForGapRemoval(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>adjustIntervalForGapRemoval</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;adjustIntervalForGapRemoval(<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a>&nbsp;sequenceGapInformation)</pre>
<div class="block">Creates a new interval adjusted to compensate for removing gaps from the sequence covered by this interval.
 If the end points of the interval are on or adjacent to gapped regions of the sequence, then the returned
 interval will appear to have lost information about the original position. However, this original position
 information is stored internally such that it can be recovered from a call to <a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#adjustIntervalForGapInsertion(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)"><code>adjustIntervalForGapInsertion(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceGapInformation</code> - the gap information about the removed gaps</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new interval adjusted to compensate for removing gaps from the sequence covered by this interval.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html#adjustIntervalForGapInsertion(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)"><code>adjustIntervalForGapInsertion(com.biomatters.geneious.publicapi.implementations.SequenceGapInformation)</code></a>, 
<a href="../../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGapRemoval(java.lang.CharSequence, java.util.List)"><code>SequenceExtractionUtilities.adjustAnnotationsForGapRemoval(CharSequence, java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="truncate(int, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>truncate</h4>
<pre>public&nbsp;<a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&nbsp;truncate(int&nbsp;minIndex,
                                  int&nbsp;maxIndex)</pre>
<div class="block">Truncates this interval to not extend below minIndex or above maxIndex. If this interval lies entirely outside that range null is returned.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>minIndex</code> - the 1-based minimum index</dd><dd><code>maxIndex</code> - the 1-base maximum index</dd>
<dt><span class="strong">Returns:</span></dt><dd>either this interval (if no truncation is necessary), a truncated interval, or null if this interval lies entirely outside the range.</dd><dt><span class="strong">Since:</span></dt>
  <dd>Geneious API 4.14 (Geneious 5.1)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceAnnotationInterval.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationGroup.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.Direction.html" title="enum in com.biomatters.geneious.publicapi.documents.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" target="_top">Frames</a></li>
<li><a href="SequenceAnnotationInterval.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
