<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:45 NZDT 2016 -->
<title>DefaultAlignmentDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="DefaultAlignmentDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultAlignmentDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev Class</li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultAlignmentDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations</div>
<h2 title="Class DefaultAlignmentDocument" class="title">Class DefaultAlignmentDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.DefaultAlignmentDocument</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Provider</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializableWithProgress</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/PhylogenyDocument.html" title="class in com.biomatters.geneious.publicapi.documents.types">PhylogenyDocument</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">DefaultAlignmentDocument</span>
extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>
implements <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializableWithProgress</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Provider</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a></pre>
<div class="block">An implementation of a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>.

 The constructors provided for this class allow for the quick creation of SequenceAlignmentDocuments that fit entirely into memory. For constructing larger
 SequenceAlignmentDocuments, see <code>SequenceListOnDisk.Builder</code></div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#KEY_ALIGNMENT_OPTIONS">KEY_ALIGNMENT_OPTIONS</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#KEY_ALIGNMENT_SCORE">KEY_ALIGNMENT_SCORE</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#KEY_MATCH_REGIONS">KEY_MATCH_REGIONS</a></strong></code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument()">DefaultAlignmentDocument</a></strong>()</code>
<div class="block">Constructs a new empty alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], java.lang.CharSequence[], java.lang.String, java.lang.Double)">DefaultAlignmentDocument</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;referencedDocuments,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore)</code>
<div class="block">Constructs an alignment with references to original sequence documents.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument)">DefaultAlignmentDocument</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment)</code>
<div class="block">Construct a default alignment as a deep copy of another alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], java.lang.CharSequence[], java.lang.String, java.lang.Double, java.lang.String)">DefaultAlignmentDocument</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;unalignedSequences,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;referencedDocuments,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proposedName)</code>
<div class="block">This is the constructor that should be used in most situations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], java.lang.CharSequence[], java.lang.String, java.lang.Double, java.lang.String, jebl.util.ProgressListener)">DefaultAlignmentDocument</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;unalignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proposedName,
                        <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Constructs an alignment without any references back to the original sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], java.util.List, java.lang.CharSequence[], java.lang.String, java.lang.Double, java.lang.String, jebl.util.ProgressListener)">DefaultAlignmentDocument</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;unalignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&gt;&nbsp;referencedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proposedName,
                        <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">This is the constructor that should be used in most situations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], java.lang.String, java.lang.Double, java.lang.String)">DefaultAlignmentDocument</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;alignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(java.lang.String, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)"><code>DefaultAlignmentDocument(String, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)</code></a> instead.</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(java.lang.String, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)">DefaultAlignmentDocument</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>...&nbsp;alignedSequences)</code>
<div class="block">Constructs alignment document with no references to original sequence documents.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addContigReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addContigReferenceSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;referenceSequence)</code>
<div class="block">Adds a sequence as the contig reference sequence {see @link #getContigReferenceSequenceIndex()}.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Add a sequence to the end of the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addUrnVisibleByDefaultTrack(com.biomatters.geneious.publicapi.documents.URN)">addUrnVisibleByDefaultTrack</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;urn)</code>
<div class="block">Adds a URN to this contig so that any track on this contig/alignment which returns this URN from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html#getDocumentVisibleByDefaultOn()"><code>SequenceTrack.getDocumentVisibleByDefaultOn()</code></a> will be visible by default.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#calculateScore(jebl.evolution.align.scores.Scores, java.lang.CharSequence[], double, double, boolean, boolean)">calculateScore</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/align/scores/Scores.html?is-external=true" title="class or interface in jebl.evolution.align.scores">Scores</a>&nbsp;scores,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignment,
              double&nbsp;gapOpen,
              double&nbsp;gapExtend,
              boolean&nbsp;freeEndGaps,
              boolean&nbsp;applyGapExtendCostToFirstGapResidue)</code>
<div class="block">Calculates the score for a given pairwise alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a></strong>()</code>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetAnnotations()">canSetAnnotations</a></strong>()</code>
<div class="block">Returns true if this alignment can have its annotations on sequences (including the consensus sequence) edited alone via <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a>
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>
 Note that some alignment implementations may return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a> and allow updating their annotations via
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>SequenceAlignmentDocument.updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a> even if they return false from this method.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetAnnotationsOnConsensus()">canSetAnnotationsOnConsensus</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">canSetFieldValue</a></strong>(int&nbsp;sequenceIndex,
                <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field)</code>
<div class="block">Returns true if this alignment supports updating field values using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>SequenceAlignmentDocument.setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetSequenceNames()">canSetSequenceNames</a></strong>()</code>
<div class="block">Specifies whether this alignment supports changing the sequence names.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#fromJeblAlignment(jebl.evolution.alignments.Alignment)">fromJeblAlignment</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a>&nbsp;alignment)</code>
<div class="block">Converts the given JEBL alignment into a DefaultAlignmentDocument.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;e)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#fromXML(org.jdom.Element, jebl.util.ProgressListener)">fromXML</a></strong>(org.jdom.Element&nbsp;doc,
       <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Deserializes this document with progress.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk.AlignmentData</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getAlignmentDataForSequencesNotInMemory()">getAlignmentDataForSequencesNotInMemory</a></strong>()</code>
<div class="block">If this document has its non-reference sequences stored as a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>, then this returns them along with data associated with them, or null if not.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getAnnotationsOnConsensus()">getAnnotationsOnConsensus</a></strong>()</code>
<div class="block">Gets annotations that are associated with the consensus sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getCircularLength()">getCircularLength</a></strong>()</code>
<div class="block">If this alignment is circular, returns the gapped length at which it wraps back to the start or 0 if this alignment is not circular.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getContigReferenceSequenceIndex()">getContigReferenceSequenceIndex</a></strong>()</code>
<div class="block">Get the index (if any) of the reference sequence in a contig.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getCreationDate()">getCreationDate</a></strong>()</code>
<div class="block">Get the date this document was created.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getDescription()">getDescription</a></strong>()</code>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getDescription(boolean)">getDescription</a></strong>(boolean&nbsp;includeDate)</code>
<div class="block">Short text description of alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getDisplayableFields()">getDisplayableFields</a></strong>()</code>
<div class="block">A list of columns for display in the main document panel.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getFieldValue(java.lang.String)">getFieldValue</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fieldCodeName)</code>
<div class="block">Get the value of a particular field.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></strong>()</code>
<div class="block">An immutable view of the paired reads managed by this list or alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getMateIndex(int)">getMateIndex</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getName()">getName</a></strong>()</code>
<div class="block">A short name (preferably one or two words) for the document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>
<div class="block">Number of sequences in this alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getReferencedDocument(int)">getReferencedDocument</a></strong>(int&nbsp;sequenceIndexInAlignment)</code>
<div class="block">Get an original sequence document that this alignment was made from.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getReferencedSequence(int)">getReferencedSequence</a></strong>(int&nbsp;sequenceIndexInAlignment)</code>
<div class="block">Get an original sequence document that this alignment was made from.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getScore()">getScore</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequence(int)">getSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Gets one row from the alignment as a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequences()">getSequences</a></strong>()</code>
<div class="block">Get the alignment as a list of sequences (Convenience function).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequencesInImmutableList()">getSequencesInImmutableList</a></strong>()</code>
<div class="block">Gets an immutable list (whose elements will not change after returning from this method).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequenceType()">getSequenceType</a></strong>()</code>
<div class="block">Returns the sequence type of the first sequence in this alignment (which should be the same as all other sequences in the alignment)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSizeRequiredToLoadIntoMemory()">getSizeRequiredToLoadIntoMemory</a></strong>()</code>
<div class="block">Return an estimate of the number of bytes required to load this document into memory.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSummary()">getSummary</a></strong>()</code>
<div class="block">Gets a summary of the properties of the sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getTrackManager(boolean)">getTrackManager</a></strong>(boolean&nbsp;mayReturnNullIfNoTracks)</code>
<div class="block">Gets a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager</code></a> associated with this sequence for providing sequence tracks.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getURN()">getURN</a></strong>()</code>
<div class="block">A Universal Resource Name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getUrnsForVisibleByDefaultTracks()">getUrnsForVisibleByDefaultTracks</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isContig()">isContig</a></strong>()</code>
<div class="block">Returns false by default.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isEditable()">isEditable</a></strong>()</code>
<div class="block">true if this alignment is editable.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a></strong>()</code>
<div class="block">True if this alignment was built using free end gaps.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isQueryCentricAlignment()">isQueryCentricAlignment</a></strong>()</code>
<div class="block">Returns true if this alignment is a query centric alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isReferencedDocumentReversed(int)">isReferencedDocumentReversed</a></strong>(int&nbsp;index)</code>
<div class="block">Sequence direction.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#removeSequence(int)">removeSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Removes the specified sequence from the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#reorderSequences(int[])">reorderSequences</a></strong>(int[]&nbsp;ordering)</code>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAlignmentDataForSequencesNotInMemory(com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk.AlignmentData)">setAlignmentDataForSequencesNotInMemory</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk.AlignmentData</a>&nbsp;alignmentData)</code>
<div class="block">Replaces or sets all sequences (and associated data) in this alignment that are not the reference sequence (if any - see <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>getContigReferenceSequenceIndex()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAlignmentOptions(java.lang.String)">setAlignmentOptions</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)">setAnnotations</a></strong>(int&nbsp;index,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
              boolean&nbsp;allowModifyingReferencedDocuments)</code>
<div class="block">Updates just the annotations on a sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)">setAnnotationsOnConsensus</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Sets the annotations on the consensus sequence that will be returned from a later call to <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.getAnnotationsOnConsensus()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setCircularLength(int)">setCircularLength</a></strong>(int&nbsp;circularLength)</code>
<div class="block">Sets the circular length of this alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)">setContig</a></strong>(boolean&nbsp;isContig)</code>
<div class="block">Set if this is a contig.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContigReferenceSequenceIndex(int)">setContigReferenceSequenceIndex</a></strong>(int&nbsp;referenceSequenceIndex)</code>
<div class="block">Set the index (if any) of the reference sequence in a contig.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setCreationDate(java.util.Date)">setCreationDate</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;creationDate)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">setFieldValue</a></strong>(int&nbsp;sequenceIndex,
             <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field,
             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;fieldValue,
             boolean&nbsp;allowModifyingOriginals)</code>
<div class="block">Sets the value of a displayable field on one of the sequences in the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setFreeEndGapAlignment(boolean)">setFreeEndGapAlignment</a></strong>(boolean&nbsp;freeEndGapAlignment)</code>
<div class="block">Set whether or not this alignment is considered to have free end gaps.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setMatchRegions(java.lang.String)">setMatchRegions</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;matchRegions)</code>
<div class="block">Sets information about the matching regions of a local alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setMates(int, int, int)">setMates</a></strong>(int&nbsp;sequenceIndex,
        int&nbsp;sequenceIndexOfMate,
        int&nbsp;expectedMateDistance)</code>
<div class="block">Records the 2nd sequence as being paired with the first sequence with the given distance.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setName(java.lang.String)">setName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Sets the name of this object</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setNumberOfDifferentMappedReads(int)">setNumberOfDifferentMappedReads</a></strong>(int&nbsp;numberOfDifferentMappedReads)</code>
<div class="block">Sets the number of different reads mapped to the reference sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setQueryCentricAlignment(boolean)">setQueryCentricAlignment</a></strong>(boolean&nbsp;isQueryCentricAlignment)</code>
<div class="block">Sets whether or not this alignment is a query centric alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">setReferencedDocument</a></strong>(int&nbsp;index,
                     <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</code>
<div class="block">Sets the sequence at the given sequenceIndexInAlignment in this alignment to refer to the given document as if it is the original source.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">setReferencedSequence</a></strong>(int&nbsp;sequenceIndexInAlignment,
                     <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
                     int&nbsp;referencedDocumentRevisionNumberSequenceWasCopiedFrom)</code>
<div class="block">Sets the sequence at the given index in this alignment to refer to the given document as if it is the original source.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setReferencedSequenceListDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int)">setReferencedSequenceListDocument</a></strong>(int&nbsp;sequenceIndexInAlignment,
                                 <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;sequenceListDocument,
                                 int&nbsp;sequenceIndexInSequenceListDocument)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setScore(java.lang.Double)">setScore</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;score)</code>
<div class="block">set the score of this alignment, or null if no score was calculated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a></strong>(int&nbsp;index,
               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
               boolean&nbsp;allowModifyingOriginals)</code>
<div class="block">change the name of a sequence within the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#toXML(jebl.util.ProgressListener)">toXML</a></strong>(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></strong>(int&nbsp;index,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
              boolean&nbsp;allowModifyingOriginals)</code>
<div class="block">Set new residues and related data for one sequence.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignment(boolean)">asJeblAlignment</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignmentIfMemoryAllows(boolean)">asJeblAlignmentIfMemoryAllows</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRealignedName(java.lang.String)">getRealignedName</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocuments()">getReferencedDocuments</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequences()">getReferencedSequences</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRefinedName(java.lang.String)">getRefinedName</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceAsJeblSequence(int)">getSequenceAsJeblSequence</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">setReferencedSequence</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#toHTML()">toHTML</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="KEY_ALIGNMENT_OPTIONS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>KEY_ALIGNMENT_OPTIONS</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_ALIGNMENT_OPTIONS</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.DefaultAlignmentDocument.KEY_ALIGNMENT_OPTIONS">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="KEY_ALIGNMENT_SCORE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>KEY_ALIGNMENT_SCORE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_ALIGNMENT_SCORE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.DefaultAlignmentDocument.KEY_ALIGNMENT_SCORE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="KEY_MATCH_REGIONS">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>KEY_MATCH_REGIONS</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_MATCH_REGIONS</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.DefaultAlignmentDocument.KEY_MATCH_REGIONS">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="DefaultAlignmentDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre>public&nbsp;DefaultAlignmentDocument()</pre>
<div class="block">Constructs a new empty alignment. Sequences can be added to the alignment using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)</code></a>
 <p/>
 For constructing DefaultAlignmentDocuments that are too large to fit in memory, see <code>SequenceListOnDisk.Builder</code></div>
</li>
</ul>
<a name="DefaultAlignmentDocument(java.lang.String, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre>public&nbsp;DefaultAlignmentDocument(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>...&nbsp;alignedSequences)</pre>
<div class="block">Constructs alignment document with no references to original sequence documents.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - alignment document name or null to have one generated</dd><dd><code>alignedSequences</code> - the aligned sequences, including gap residues</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3)</dd></dl>
</li>
</ul>
<a name="DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], java.lang.String, java.lang.Double, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public&nbsp;DefaultAlignmentDocument(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;alignedSequences,
                                   <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                                   <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                                   <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#DefaultAlignmentDocument(java.lang.String, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)"><code>DefaultAlignmentDocument(String, com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument...)</code></a> instead.</i></div>
<div class="block">Constructs alignment document with no references to original sequence documents.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignedSequences</code> - the aligned sequences, including gap residues</dd><dd><code>alignmentOptions</code> - text description of options used to create the alignment. Should be null because it has been replaced by document history</dd><dd><code>alignmentScore</code> - some sort of score for this alignment. May be null</dd><dd><code>name</code> - alignment document name or null to have one generated</dd></dl>
</li>
</ul>
<a name="DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], java.lang.CharSequence[], java.lang.String, java.lang.Double, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre>public&nbsp;DefaultAlignmentDocument(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;unalignedSequences,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;referencedDocuments,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proposedName)</pre>
<div class="block">This is the constructor that should be used in most situations. Constructs an alignment
 with optional references to original sequence documents.

 Both <code>unalignedSequences</code>and <code>referencedDocuments</code> are optional, but at least
 one must be supplied. If both are supplied, they
 must be the same length and for each index one of the arrays must contain a non-null value.
 If both arrays have a value at the same index, the <code>referencedDocuments</code> is used and the <code>unalignedSequences</code> entry is ignored.
 <p/>
 For constructing DefaultAlignmentDocuments that are too large to fit in memory, see <code>SequenceListOnDisk.Builder</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>unalignedSequences</code> - the original unaligned sequences. The residues must match (or be the reverse complement of) the ungapped corresponding sequence in <code>alignedSequenceStrings</code></dd><dd><code>referencedDocuments</code> - used for storing references back to the original documents so that annotations
                  or other information about the aligned sequences is available to the alignment. The residues must match (or be the reverse complement of) the ungapped corresponding sequence in <code>alignedSequenceStrings</code></dd><dd><code>alignedSequenceCharacters</code> - the aligned sequence characters, with gap characters inserted. <b>The length of the individual CharSequences and the characters retured from the CharSequences must not change after calling this method.</b></dd><dd><code>alignmentOptions</code> - a text description of options used to create the alignment. Should be null because it has been replaced by document history</dd><dd><code>alignmentScore</code> - some sort of score for this alignment. May be null</dd><dd><code>proposedName</code> - name for the resulting alignment document, or null to generate a default name</dd></dl>
</li>
</ul>
<a name="DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], java.lang.CharSequence[], java.lang.String, java.lang.Double, java.lang.String, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre>public&nbsp;DefaultAlignmentDocument(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;unalignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proposedName,
                        <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block">Constructs an alignment without any references back to the original sequences.
 <p/>
 For constructing DefaultAlignmentDocuments that are too large to fit in memory, see <code>SequenceListOnDisk.Builder</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>unalignedSequences</code> - the original unaligned sequences. The residues must match (or be the reverse complement of) the ungapped corresponding sequence in <code>alignedSequenceStrings</code></dd><dd><code>alignedSequenceCharacters</code> - the aligned sequence characters, with gap characters inserted. <b>The length of the individual CharSequences and the characters retured from the CharSequences must not change after calling this method.</b></dd><dd><code>alignmentOptions</code> - a text description of options used to create the alignment. Should be null because it has been replaced by document history</dd><dd><code>alignmentScore</code> - some sort of score for this alignment. May be null</dd><dd><code>proposedName</code> - name for the resulting alignment document, or null to generate a default name</dd><dd><code>progressListener</code> - since creating huge alignments (100,000 or more sequences) can take a while, progress will be reported to this and it can be used for canceling the creation of this alignment document, in which case the returned document will be invalid.</dd></dl>
</li>
</ul>
<a name="DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument[], java.util.List, java.lang.CharSequence[], java.lang.String, java.lang.Double, java.lang.String, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre>public&nbsp;DefaultAlignmentDocument(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>[]&nbsp;unalignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&gt;&nbsp;referencedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proposedName,
                        <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block">This is the constructor that should be used in most situations. Constructs an alignment
 with optional references to original sequence documents.

 Both <code>unalignedSequences</code>and <code>referencedDocuments</code> are optional, but at least
 one must be supplied. If both are supplied, they
 must be the same length and for each index one of the arrays must contain a non-null value.
 If both arrays have a value at the same index, the <code>referencedDocuments</code> is used and the <code>unalignedSequences</code> entry is ignored.
 <p/>
 For constructing DefaultAlignmentDocuments that are too large to fit in memory, see <code>SequenceListOnDisk.Builder</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>unalignedSequences</code> - the original unaligned sequences. The residues must match (or be the reverse complement of) the ungapped corresponding sequence in <code>alignedSequenceStrings</code></dd><dd><code>referencedSequences</code> - used for storing references back to the original documents so that annotations
                  or other information about the aligned sequences is available to the alignment. The residues must match (or be the reverse complement of) the ungapped corresponding sequence in <code>alignedSequenceStrings</code>. This may be null for no referenced sequences.</dd><dd><code>alignedSequenceCharacters</code> - the aligned sequence characters, with gap characters inserted. <b>The length of the individual CharSequences and the characters retured from the CharSequences must not change after calling this method.</b></dd><dd><code>alignmentOptions</code> - a text description of options used to create the alignment. Should be null because it has been replaced by document history</dd><dd><code>alignmentScore</code> - some sort of score for this alignment. May be null</dd><dd><code>proposedName</code> - name for the resulting alignment document, or null to generate a default name</dd><dd><code>progressListener</code> - since creating huge alignments (100,000 or more sequences) can take a while, progress will be reported to this and it can be used for canceling the creation of this alignment document, in which case the returned document will be invalid.</dd></dl>
</li>
</ul>
<a name="DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument[], java.lang.CharSequence[], java.lang.String, java.lang.Double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre>public&nbsp;DefaultAlignmentDocument(<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>[]&nbsp;referencedDocuments,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignedSequenceCharacters,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;alignmentScore)</pre>
<div class="block">Constructs an alignment with references to original sequence documents. referencedDocuments and
 alignedSequenceStrings must be the same length and contain no null entries.
 <p/>
 For constructing DefaultAlignmentDocuments that are too large to fit in memory, see <code>SequenceListOnDisk.Builder</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referencedDocuments</code> - used for storing references back to the original documents so that annotations
                  or other information about the aligned sequences is available to the alignment. Indices
                  match those in alignedSequenceStrings</dd><dd><code>alignedSequenceCharacters</code> - the aligned sequence characters, with gap characters inserted. <b>The length of the individual CharSequences and the characters retured from the CharSequences must not change after calling this method.</b></dd><dd><code>alignmentOptions</code> - a text description of options used to create the alignment. Should be null because it has been replaced by document history</dd><dd><code>alignmentScore</code> - some sort of score for this alignment. May be null</dd></dl>
</li>
</ul>
<a name="DefaultAlignmentDocument(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DefaultAlignmentDocument</h4>
<pre>public&nbsp;DefaultAlignmentDocument(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment)</pre>
<div class="block">Construct a default alignment as a deep copy of another alignment. Referenced documents will
 not be copied but they will still be referenced by the new alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - to copy</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="isReferencedDocumentReversed(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReferencedDocumentReversed</h4>
<pre>public&nbsp;boolean&nbsp;isReferencedDocumentReversed(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isReferencedDocumentReversed(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sequence direction.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isReferencedDocumentReversed(int)">isReferencedDocumentReversed</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a> is not null and it is in the reverse direction of its sequence string in the alignment.</dd></dl>
</li>
</ul>
<a name="fromJeblAlignment(jebl.evolution.alignments.Alignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromJeblAlignment</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a>&nbsp;fromJeblAlignment(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/alignments/Alignment.html?is-external=true" title="class or interface in jebl.evolution.alignments">Alignment</a>&nbsp;alignment)</pre>
<div class="block">Converts the given JEBL alignment into a DefaultAlignmentDocument.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - the alignment to convert</dd>
<dt><span class="strong">Returns:</span></dt><dd>a DefaultAlignmentDocument</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="toXML(jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html#toXML(jebl.util.ProgressListener)">XMLSerializableWithProgress</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that the constructor can completely
 restore the object's representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html#toXML(jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializableWithProgress</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>progressListener</code> - for reporting progress of the serialization. The implementation may return null if the progress listener requests that it cancel.</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;e)
             throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>e</code> - representation from a previous call to <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;doc,
           <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
             throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Deserializes this document with progress. See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>XMLSerializableWithProgress</code></a></div>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd></dl>
</li>
</ul>
<a name="getDisplayableFields()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisplayableFields</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&gt;&nbsp;getDisplayableFields()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">PluginDocument</a></code></strong></div>
<div class="block">A list of columns for display in the main document panel.
 This function only defines the columns available to display,
 not the particular values of these columns, which are returned by
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)"><code>PluginDocument.getFieldValue(String)</code></a>.
 <p/>
 Each column is specified by a Field. The fields should be obtained from a call to the appropriate
 Field.createXXXXField, where XXXX is the appropriate type.
 <p/>
 While it is not required, it is recommended that a PluginDocument
 handle returning a complete list of its displayable fields before
 the document has been properly constructed via a call to <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a>

 The returned list need not and should not be modifiable.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a list of document fields (or null to indicate no displayable fields)</dd></dl>
</li>
</ul>
<a name="getScore()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getScore</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;getScore()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the score of this alignment, or null if no score was calculated or if the alignment has been manually modified.</dd></dl>
</li>
</ul>
<a name="setScore(java.lang.Double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setScore</h4>
<pre>public&nbsp;void&nbsp;setScore(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;score)</pre>
<div class="block">set the score of this alignment, or null if no score was calculated.
 Must not be called after the alignment is modified.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>score</code> - the new score</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if this alignment has been manually modified</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd></dl>
</li>
</ul>
<a name="setNumberOfDifferentMappedReads(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNumberOfDifferentMappedReads</h4>
<pre>public&nbsp;void&nbsp;setNumberOfDifferentMappedReads(int&nbsp;numberOfDifferentMappedReads)</pre>
<div class="block">Sets the number of different reads mapped to the reference sequence. See <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#NUMBER_OF_DIFFERENT_MAPPED_READS"><code>DocumentField.NUMBER_OF_DIFFERENT_MAPPED_READS</code></a> for what this means.
 The value set on this alignment may be obtained using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getFieldValue(java.lang.String)"><code>getFieldValue</code></a>(DocumentField.NUMBER_OF_DIFFERENT_MAPPED_READS.getCode())</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfDifferentMappedReads</code> - the number of different reads mapped to the reference sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.902 (Geneious 9.0.2)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#NUMBER_OF_DIFFERENT_MAPPED_READS"><code>DocumentField.NUMBER_OF_DIFFERENT_MAPPED_READS</code></a></dd></dl>
</li>
</ul>
<a name="getFieldValue(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFieldValue</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;getFieldValue(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fieldCodeName)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">PluginDocument</a></code></strong></div>
<div class="block">Get the value of a particular field.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>fieldCodeName</code> - field code. This should be the
      <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#getCode()"><code>code</code></a> of one of the fields
      returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()"><code>PluginDocument.getDisplayableFields()</code></a>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>value for a field or null if this document does not have a field with the given field code.
          The class of the returned value must be the <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html#getValueType()"><code>DocumentField.getValueType()</code></a> (or a subclass) of the corresponding DocumentField returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()"><code>PluginDocument.getDisplayableFields()</code></a>.</dd></dl>
</li>
</ul>
<a name="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getName()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">PluginDocument</a></code></strong></div>
<div class="block">A short name (preferably one or two words) for the document.
 <p/>
 If this PluginDocument is wrapped inside an <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a> then this method
 will be overridden and delegate back to <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getName()"><code>AnnotatedPluginDocument.getName()</code></a>. One way to look at it is
 that when you call PluginDocument.getName() you will get the current name of the document. If you are implementing PluginDocument.getName() then you are
 returning the default name for the document.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>document name. Cannot be null but may be an empty string.</dd></dl>
</li>
</ul>
<a name="getURN()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getURN</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;getURN()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">PluginDocument</a></code></strong></div>
<div class="block">A Universal Resource Name.
 <p/>
 Documents from external sources should set this to a unique name inside that service. Documents generated locally
 (such as from operations) and documents that don't have a unique identifier should return null since
 Geneious handles URNs for these.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the urn, or null if this document doesn't have a unique identifier</dd></dl>
</li>
</ul>
<a name="addUrnVisibleByDefaultTrack(com.biomatters.geneious.publicapi.documents.URN)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addUrnVisibleByDefaultTrack</h4>
<pre>public&nbsp;void&nbsp;addUrnVisibleByDefaultTrack(<a href="../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&nbsp;urn)</pre>
<div class="block">Adds a URN to this contig so that any track on this contig/alignment which returns this URN from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html#getDocumentVisibleByDefaultOn()"><code>SequenceTrack.getDocumentVisibleByDefaultOn()</code></a> will be visible by default.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.902 (Geneious 9.0.2)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getUrnsForVisibleByDefaultTracks()"><code>getUrnsForVisibleByDefaultTracks()</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html#getDocumentVisibleByDefaultOn()"><code>SequenceTrack.getDocumentVisibleByDefaultOn()</code></a></dd></dl>
</li>
</ul>
<a name="getUrnsForVisibleByDefaultTracks()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getUrnsForVisibleByDefaultTracks</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/URN.html" title="class in com.biomatters.geneious.publicapi.documents">URN</a>&gt;&nbsp;getUrnsForVisibleByDefaultTracks()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>a set of URNs such that any track on this contig/alignment which returns one of these URNs from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html#getDocumentVisibleByDefaultOn()"><code>SequenceTrack.getDocumentVisibleByDefaultOn()</code></a> will be visible by default.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.902 (Geneious 9.0.2)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addUrnVisibleByDefaultTrack(com.biomatters.geneious.publicapi.documents.URN)"><code>addUrnVisibleByDefaultTrack(URN)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html#getDocumentVisibleByDefaultOn()"><code>SequenceTrack.getDocumentVisibleByDefaultOn()</code></a></dd></dl>
</li>
</ul>
<a name="getCreationDate()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCreationDate</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;getCreationDate()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">PluginDocument</a></code></strong></div>
<div class="block">Get the date this document was created. This method may return null.
 Generally clients should use <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getCreationDate()"><code>AnnotatedPluginDocument.getCreationDate()</code></a>
 instead as that will return either the date returned from
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()"><code>PluginDocument.getCreationDate()</code></a> or if that returned null,
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getCreationDate()"><code>AnnotatedPluginDocument.getCreationDate()</code></a> will return the date the
 document was first instantiated.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the date this document was created or null to use the date the document was first instantiated.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD"><code>PluginDocument.MODIFIED_DATE_FIELD</code></a></dd></dl>
</li>
</ul>
<a name="getDescription(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription(boolean&nbsp;includeDate)</pre>
<div class="block">Short text description of alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>includeDate</code> - when true, include generation time information</dd>
<dt><span class="strong">Returns:</span></dt><dd>the description</dd></dl>
</li>
</ul>
<a name="getDescription()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">PluginDocument</a></code></strong></div>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a one line simple text description of the document, or null if no description is available.</dd></dl>
</li>
</ul>
<a name="getSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;getSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Gets one row from the alignment as a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>. As this sequence
 is an alignment row, it will normally contain gaps.

 If implementation stores the URN of the original sequence document, see 
 the note in <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)">getSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence at the specified index.</dd></dl>
</li>
</ul>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public&nbsp;int&nbsp;getNumberOfSequences()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Number of sequences in this alignment</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()">getNumberOfSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>number of sequences</dd></dl>
</li>
</ul>
<a name="canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetFieldValue</h4>
<pre>public&nbsp;boolean&nbsp;canSetFieldValue(int&nbsp;sequenceIndex,
                       <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Returns true if this alignment supports updating field values using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>SequenceAlignmentDocument.setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">canSetFieldValue</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of a sequence in the alignment</dd><dd><code>field</code> - the field to update</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment supports updating field values using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>SequenceAlignmentDocument.setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)"><code>SequenceAlignmentDocument.setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, Object, boolean)</code></a></dd></dl>
</li>
</ul>
<a name="setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFieldValue</h4>
<pre>public&nbsp;void&nbsp;setFieldValue(int&nbsp;sequenceIndex,
                 <a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents">DocumentField</a>&nbsp;field,
                 <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;fieldValue,
                 boolean&nbsp;allowModifyingOriginals)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sets the value of a displayable field on one of the sequences in the alignment. This method may only be called if <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)"><code>SequenceAlignmentDocument.canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)</code></a> returns true.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">setFieldValue</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of a sequence in the alignment to update the field value on</dd><dd><code>field</code> - the field to update</dd><dd><code>fieldValue</code> - the new value for the field.</dd><dd><code>allowModifyingOriginals</code> - true if this field change may be propagated to a referenced sequence (if any). If this is false the sequence will be disconnected from its referenced document.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)"><code>SequenceAlignmentDocument.canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)</code></a></dd></dl>
</li>
</ul>
<a name="updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>updateSequence</h4>
<pre>public&nbsp;void&nbsp;updateSequence(int&nbsp;index,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                  <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
                  boolean&nbsp;allowModifyingOriginals)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Set new residues and related data for one sequence.

 This function may only be called on alignments that return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a>
 (otherwise, the method should throw <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang"><code>UnsupportedOperationException</code></a>).</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be updated. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd><dd><code>sequence</code> - the new residues</dd><dd><code>annotations</code> - a list of the annotations</dd><dd><code>graph</code> - Must be null if the original sequence at the specified index is not a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>.
 Otherwise (i.e. if the sequence was a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>), specifies the new
 graph to use for that sequence.</dd><dd><code>allowModifyingOriginals</code> - if the change being made to the sequence would affect the unaligned version of it
 (for example, the sequence residues are changing rather than just the gaps)
 and the alignment stores a reference an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be modified.
 If this parameter is not true and the change would have otherwise modified the original SequenceDocument,
 then any reference to the original document will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="setMatchRegions(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMatchRegions</h4>
<pre>public&nbsp;void&nbsp;setMatchRegions(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;matchRegions)</pre>
<div class="block">Sets information about the matching regions of a local alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>matchRegions</code> - a matching region within the alignment</dd></dl>
</li>
</ul>
<a name="calculateScore(jebl.evolution.align.scores.Scores, java.lang.CharSequence[], double, double, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>calculateScore</h4>
<pre>public static&nbsp;double&nbsp;calculateScore(<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/align/scores/Scores.html?is-external=true" title="class or interface in jebl.evolution.align.scores">Scores</a>&nbsp;scores,
                    <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>[]&nbsp;alignment,
                    double&nbsp;gapOpen,
                    double&nbsp;gapExtend,
                    boolean&nbsp;freeEndGaps,
                    boolean&nbsp;applyGapExtendCostToFirstGapResidue)</pre>
<div class="block">Calculates the score for a given pairwise alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>scores</code> - </dd><dd><code>alignment</code> - must consist of two CharSequences of equal length which represent aligned sequence characters, with gap characters inserted. <b>The length of the individual CharSequences and the characters retured from the CharSequences must not change after calling this method.</b></dd><dd><code>gapOpen</code> - </dd><dd><code>gapExtend</code> - </dd><dd><code>freeEndGaps</code> - </dd><dd><code>applyGapExtendCostToFirstGapResidue</code> - Generally there is an ambiguity in bioinformatics whether "gap opening" already includes the first gap character - in other words, whether a gap of length N has
    <pre>
    (a) a cost of gapOpen + N * gapExtend
    or
    (b) gapOpen + (N-1) * gapExtend.
    </pre>.
    <code>applyGapExtendCostToFirstGapResidue</code> should be true if using interpretation (a).</dd>
<dt><span class="strong">Returns:</span></dt><dd>the score rounded to three decimal places</dd></dl>
</li>
</ul>
<a name="isEditable()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEditable</h4>
<pre>public&nbsp;boolean&nbsp;isEditable()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">true if this alignment is editable.
 if it is, it must implement <br>
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a>, <br>
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)"><code>SequenceAlignmentDocument.removeSequence(int)</code></a> , <br>
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>SequenceAlignmentDocument.updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a>, <br>
 and may want to implement <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)"><code>SequenceAlignmentDocument.setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)</code></a> <br><br>

 An implementation may also want to return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()"><code>SequenceAlignmentDocument.canSetAnnotations()</code></a> to allow updating just the annotations via <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()">isEditable</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment is editable.</dd></dl>
</li>
</ul>
<a name="getReferencedDocument(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedDocument</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;getReferencedDocument(int&nbsp;sequenceIndexInAlignment)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get an original sequence document that this alignment was made from.

 This may be null if the alignment was not generated from
 sequence documents, or if the alignment has been user modified, so the reference
 to the original sequence document would not be relevant.

 Note that when an alignment implements this by storing the sequence URN and loading it
 on request it needs to check that the seqeunce residues has not been changed by a later edit
 operation. In that case the seqeunce document is no longer valid for that alignment and a null should be returned.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)">getReferencedDocument</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the index of the sequence in the alignment to get the original document for. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the original document or null if this alignment does not a reference an original document for this sequence index. The default implementation of this method returns null</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)"><code>SequenceAlignmentDocument.getReferencedSequence(int)</code></a></dd></dl>
</li>
</ul>
<a name="getReferencedSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferencedSequence</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;getReferencedSequence(int&nbsp;sequenceIndexInAlignment)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get an original sequence document that this alignment was made from.

 This may be null if the alignment was not generated from
 sequence documents, or if the alignment has been user modified, so the reference
 to the original sequence document would not be relevant.

 Note that when an alignment implements this by storing the sequence URN and loading it
 on request it needs to check that the seqeunce residues has not been changed by a later edit
 operation. In that case the seqeunce document is no longer valid for that alignment and a null should be returned.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequence(int)">getReferencedSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the index of the sequence in the alignment to get the original document for. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1 inclusive.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the original document or null if this alignment does not a reference an original document for this sequence index. The default implementation of this method forwards to <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a></dd></dl>
</li>
</ul>
<a name="reorderSequences(int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderSequences</h4>
<pre>public&nbsp;void&nbsp;reorderSequences(int[]&nbsp;ordering)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.
 If ordering[i]==j, then the sequence at index 'i' should be the sequence previously at index 'j'
 For example, if ordering = {0,1,3,4,2}
 Then:
 sequence at index 2 is moved to index 4
 sequence at index 3 is moved to index 2
 sequence at index 4 is moved to index 3

 This function may only be called on alignments that return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])">reorderSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>ordering</code> - Each integer between 0 and ordering.length - 1 inclusive should occur exactly once in this array.</dd></dl>
</li>
</ul>
<a name="removeSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeSequence</h4>
<pre>public&nbsp;void&nbsp;removeSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Removes the specified sequence from the alignment. All other sequences should
 maintain their original ordering.

 This function may only be called on alignments that return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)">removeSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be removed. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd></dl>
</li>
</ul>
<a name="canAddAndRemoveSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canAddAndRemoveSequences</h4>
<pre>public&nbsp;boolean&nbsp;canAddAndRemoveSequences()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canAddAndRemoveSequences()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if adding and removing sequences is supported, false otherwise.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceAlignmentDocument.addSequence(SequenceDocument)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)"><code>SequenceAlignmentDocument.removeSequence(int)</code></a></dd></dl>
</li>
</ul>
<a name="addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequence</h4>
<pre>public&nbsp;void&nbsp;addSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Add a sequence to the end of the alignment. <code>sequence</code> should be the
 gapped (aligned) sequence.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - sequence to add</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)"><code>SequenceAlignmentDocument.setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)</code></a></dd></dl>
</li>
</ul>
<a name="setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedDocument</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedDocument(int&nbsp;index,
                            <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;document)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sets the sequence at the given sequenceIndexInAlignment in this alignment to refer to the given document as if it is the original source.
 This document must contain a sequence consisting of residues matching the aligned sequence stored at <code> sequenceIndexInAlignment </code>,
 with gap residues removed.
 This method may be used when the user chooses to extract a sequence from an alignment, so the alignment
 should now be updated to reference the extracted sequence.
 Any alignment implementations may choose to not implement this method, in which case of the default implementation is used which always returns false.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">setReferencedDocument</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the sequenceIndexInAlignment of the sequence referred to. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd><dd><code>document</code> - the document to reference</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this document was successfully set as a referenced document. Note that some implementations may allow setting the reference and return true,
 but then discard the reference next time it is requested if the sequence from the referenced document does not correspond to sequence in the alignment.
 Call <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a> to check if it really worked.</dd></dl>
</li>
</ul>
<a name="setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedSequence</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedSequence(int&nbsp;sequenceIndexInAlignment,
                            <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a>&nbsp;referencedSequence,
                            int&nbsp;referencedDocumentRevisionNumberSequenceWasCopiedFrom)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sets the sequence at the given index in this alignment to refer to the given document as if it is the original source.
 This document must contain a sequence consisting of residues matching the aligned sequence stored at <code> sequenceIndexInAlignment </code>,
 with gap residues removed.
 This method may be used when the user chooses to extract a sequence from an alignment, so the alignment
 should now be updated to reference the extracted sequence.
 Any alignment implementations may choose to not implement this method, in which case of the default implementation is used which always returns false.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">setReferencedSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndexInAlignment</code> - the sequenceIndexInAlignment of the sequence referred to. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd><dd><code>referencedSequence</code> - the document to reference</dd><dd><code>referencedDocumentRevisionNumberSequenceWasCopiedFrom</code> - normally an alignment document stores the most recent revision number it updated its local cache of the sequence from. In order to improve performance, if it is known what revision number (see <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getRevisionNumber()"><code>AnnotatedPluginDocument.getRevisionNumber()</code></a> the current alignment sequence was created from then it can be supplied here. Use -1 if unsure.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this document was successfully set as a referenced document</dd></dl>
</li>
</ul>
<a name="setReferencedSequenceListDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReferencedSequenceListDocument</h4>
<pre>public&nbsp;boolean&nbsp;setReferencedSequenceListDocument(int&nbsp;sequenceIndexInAlignment,
                                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;sequenceListDocument,
                                        int&nbsp;sequenceIndexInSequenceListDocument)</pre>
</li>
</ul>
<a name="setContig(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setContig</h4>
<pre>public&nbsp;void&nbsp;setContig(boolean&nbsp;isContig)</pre>
<div class="block">Set if this is a contig.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>isContig</code> - </dd></dl>
</li>
</ul>
<a name="isContig()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isContig</h4>
<pre>public&nbsp;boolean&nbsp;isContig()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isContig()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Returns false by default. If true then this alignment may should be
 treated as a contig (may affect e.g. the way consensus sequence
 are calculated).
 <p/>
 If this is overridden to return true by classes other than
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocument</code></a>
 then the document will not be given the contig icon (to set non default icons for
 a document see <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentType.html#OVERRIDE_DOCUMENT_TYPE_KEY"><code>DocumentType.OVERRIDE_DOCUMENT_TYPE_KEY</code></a>.
 <p/>
 Use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)"><code>setContig(boolean)</code></a>
 to create a contig.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isContig()">isContig</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment should be treated as a contig</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)"><code>setContig(boolean)</code></a></dd></dl>
</li>
</ul>
<a name="isFreeEndGapAlignment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isFreeEndGapAlignment</h4>
<pre>public&nbsp;boolean&nbsp;isFreeEndGapAlignment()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isFreeEndGapAlignment()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">True if this alignment was built using free end gaps. This means gaps at either end of the sequences are
 not penalized.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment was built using free end gaps. The default implementation returns true.</dd></dl>
</li>
</ul>
<a name="setFreeEndGapAlignment(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFreeEndGapAlignment</h4>
<pre>public&nbsp;void&nbsp;setFreeEndGapAlignment(boolean&nbsp;freeEndGapAlignment)</pre>
<div class="block">Set whether or not this alignment is considered to have free end gaps.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>freeEndGapAlignment</code> - true if this</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isFreeEndGapAlignment()"><code>isFreeEndGapAlignment()</code></a></dd></dl>
</li>
</ul>
<a name="setSequenceName(int, java.lang.String, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceName</h4>
<pre>public&nbsp;void&nbsp;setSequenceName(int&nbsp;index,
                   <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
                   boolean&nbsp;allowModifyingOriginals)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">change the name of a sequence within the alignment.
 This method may only be called if <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()"><code>SequenceAlignmentDocument.canSetSequenceNames()</code></a> returns true.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - index of the sequence name to change</dd><dd><code>newName</code> - the new name for the sequence.</dd><dd><code>allowModifyingOriginals</code> - if the alignment stores a reference to an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be renamed too.
 Otherwise,  any existing reference to the original SequenceDocument will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="canSetSequenceNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetSequenceNames</h4>
<pre>public&nbsp;boolean&nbsp;canSetSequenceNames()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Specifies whether this alignment supports changing the sequence names.
 If this method returns true, then this alignment must implement <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)"><code>SequenceAlignmentDocument.setSequenceName(int, String, boolean)</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()">canSetSequenceNames</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment supports changing the sequence names.</dd></dl>
</li>
</ul>
<a name="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre>public&nbsp;void&nbsp;setName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html#setName(java.lang.String)">Renamable</a></code></strong></div>
<div class="block">Sets the name of this object</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html#setName(java.lang.String)">setName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - the new name for this object. Must not be null.</dd></dl>
</li>
</ul>
<a name="setCreationDate(java.util.Date)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCreationDate</h4>
<pre>public&nbsp;void&nbsp;setCreationDate(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Date.html?is-external=true" title="class or interface in java.util">Date</a>&nbsp;creationDate)</pre>
</li>
</ul>
<a name="setAlignmentOptions(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAlignmentOptions</h4>
<pre>public&nbsp;void&nbsp;setAlignmentOptions(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentOptions)</pre>
</li>
</ul>
<a name="setContigReferenceSequenceIndex(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setContigReferenceSequenceIndex</h4>
<pre>public&nbsp;void&nbsp;setContigReferenceSequenceIndex(int&nbsp;referenceSequenceIndex)</pre>
<div class="block">Set the index (if any) of the reference sequence in a contig.  Note - despite the name, this method can also be used to set
 the query sequence in a query centric alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequenceIndex</code> - the index (in the range [0,<a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a>-1]) or -1 to set no reference sequence</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>getContigReferenceSequenceIndex()</code></a></dd></dl>
</li>
</ul>
<a name="getContigReferenceSequenceIndex()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getContigReferenceSequenceIndex</h4>
<pre>public&nbsp;int&nbsp;getContigReferenceSequenceIndex()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get the index (if any) of the reference sequence in a contig. Note - despite the name, this method can also be used to identify
 the query sequence in a query centric alignment.
 <p/>
 The default implementation returns -1.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()">getContigReferenceSequenceIndex</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the index (in the range [0,<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a>-1]) or -1 to if this contig has no reference sequence.</dd></dl>
</li>
</ul>
<a name="isQueryCentricAlignment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isQueryCentricAlignment</h4>
<pre>public&nbsp;boolean&nbsp;isQueryCentricAlignment()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isQueryCentricAlignment()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Returns true if this alignment is a query centric alignment. A query centric alignment must also return a non-negative number
 from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>SequenceAlignmentDocument.getContigReferenceSequenceIndex()</code></a>.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isQueryCentricAlignment()">isQueryCentricAlignment</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment is a query centric alignment.</dd></dl>
</li>
</ul>
<a name="setQueryCentricAlignment(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setQueryCentricAlignment</h4>
<pre>public&nbsp;void&nbsp;setQueryCentricAlignment(boolean&nbsp;isQueryCentricAlignment)</pre>
<div class="block">Sets whether or not this alignment is a query centric alignment. The caller should
 also set the index of the query sequence using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContigReferenceSequenceIndex(int)"><code>setContigReferenceSequenceIndex(int)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>isQueryCentricAlignment</code> - true if this is a query centric alignment.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isQueryCentricAlignment()"><code>isQueryCentricAlignment()</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContigReferenceSequenceIndex(int)"><code>setContigReferenceSequenceIndex(int)</code></a></dd></dl>
</li>
</ul>
<a name="setMates(int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMates</h4>
<pre>public&nbsp;void&nbsp;setMates(int&nbsp;sequenceIndex,
            int&nbsp;sequenceIndexOfMate,
            int&nbsp;expectedMateDistance)</pre>
<div class="block">Records the 2nd sequence as being paired with the first sequence with the given distance.
 <p/>
 If the mate sequence is not yet paired with this sequence then this method also sets the
 expected mate's distance to be -expectedMateDistance.
 If the mate sequence is already paired with this sequence, then this method does not change the distance recorded for the mate.
 This is so that if a contig is created where the 2 mates haven been mapped in opposite directions (due to an inversion from the reference sequence), the
 expected distance can be correctly set for each of them since both values should be either positive or negative in this case. (see <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a>).
 <p/>
 Unlike <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setMates(int, int, int)"><code>SequenceAlignmentDocument.setMates(int, int, int)</code></a> which allows for
 an implementation to throw an <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang"><code>UnsupportedOperationException</code></a> this implementation will never throw it.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setMates(int, int, int)">setMates</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the sequence to set the mate for.</dd><dd><code>sequenceIndexOfMate</code> - the index of its mate, or -1 if its mate is not in this contig.</dd><dd><code>expectedMateDistance</code> - the expected mate distance. See <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if this sequence already has a different mate or if its mate is paired to a different sequence.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="getMateExpectedDistance(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateExpectedDistance</h4>
<pre>public&nbsp;int&nbsp;getMateExpectedDistance(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate. This is defined
 as the length of the 2 reads plus the distance between them.
 For example if the 2 reads both have length 30 with a 40 bp gap between them, then the
 expected mate distance is 100.
 If the mate is expected to appear after this sequence with this sequence in its current orientation, then
 the expected distance will be positive. If the mate is expected to appear before this sequence with this sequence in its current orientation, then
 the expected distance will be negative.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence for which to get the mate index of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the expected mate distance or 0 if this read is not paired. Note that a read may still have a mate distance!=0 but its mate may not be present in this document (in which case <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a> will be -1.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a></dd></dl>
</li>
</ul>
<a name="getMateIndex(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateIndex</h4>
<pre>public&nbsp;int&nbsp;getMateIndex(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">getMateIndex</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getMateIndex(int)">getMateIndex</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence to get the mate index of</dd>
<dt><span class="strong">Returns:</span></dt><dd>an index of the mate of the sequence, or -1 if the sequence has no mate.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>PairedReads.getMateExpectedDistance(int)</code></a></dd></dl>
</li>
</ul>
<a name="getImmutablePairedReadManager()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getImmutablePairedReadManager</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;getImmutablePairedReadManager()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">PairedReads</a></code></strong></div>
<div class="block">An immutable view of the paired reads managed by this list or alignment. Even if the list or alignment
 that implements this method is later modified, the
 returned PairedReadManager will not change.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>An immutable view of the paired reads managed by the list or alignment that implements this method or null if this class doesn't provide an immutable view of its paired reads.</dd></dl>
</li>
</ul>
<a name="canSetAnnotationsOnConsensus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetAnnotationsOnConsensus</h4>
<pre>public&nbsp;boolean&nbsp;canSetAnnotationsOnConsensus()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()">canSetAnnotationsOnConsensus</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment implementation supports calling <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="getAnnotationsOnConsensus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationsOnConsensus</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;getAnnotationsOnConsensus()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Gets annotations that are associated with the consensus sequence.
 Note - this doesn't include consensus annotations in tracks. Consensus tracks can be obtained by checking if this alignment
 implements <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()">getAnnotationsOnConsensus</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>annotations that are associated with the consensus sequence. Never null but may be empty</dd></dl>
</li>
</ul>
<a name="setAnnotationsOnConsensus(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotationsOnConsensus</h4>
<pre>public&nbsp;void&nbsp;setAnnotationsOnConsensus(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Sets the annotations on the consensus sequence that will be returned from a later call to <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.getAnnotationsOnConsensus()</code></a>.
 It is only legal to call this method if <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.canSetAnnotationsOnConsensus()</code></a> returns true.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)">setAnnotationsOnConsensus</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>annotations</code> - the annotations to store on the consensus. Must not be null. This replaces any existing annotations on the consensus.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.getAnnotationsOnConsensus()</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.canSetAnnotationsOnConsensus()</code></a></dd></dl>
</li>
</ul>
<a name="getSequencesInImmutableList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesInImmutableList</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequencesInImmutableList()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Gets an immutable list (whose elements will not change after returning from this method). However the contents of those sequences may change unless the sequences are instances of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a>.
 Despite that, this method is still useful since if sequences are deleted, added or replaced in this alignment, the returned list will not change.

 The default implementation of this method constructs a new list using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()"><code>SequenceAlignmentDocument.getSequences()</code></a> and returns it.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()">getSequencesInImmutableList</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an immutable list (whose elements will not change after returning from this method). However the contents of those sequences may change unless the sequences are instances of <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/ImmutableSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>ImmutableSequence</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()"><code>SequenceAlignmentDocument.getSequences()</code></a></dd></dl>
</li>
</ul>
<a name="canSetAnnotations()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canSetAnnotations</h4>
<pre>public&nbsp;boolean&nbsp;canSetAnnotations()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Returns true if this alignment can have its annotations on sequences (including the consensus sequence) edited alone via <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a>
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>
 Note that some alignment implementations may return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a> and allow updating their annotations via
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)"><code>SequenceAlignmentDocument.updateSequence(int, CharSequence, java.util.List, NucleotideGraph, boolean)</code></a> even if they return false from this method.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()">canSetAnnotations</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if this alignment can have its annotations edited alone via <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)"><code>SequenceAlignmentDocument.setAnnotations(int, java.util.List, boolean)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)"><code>SequenceAlignmentDocument.setAnnotationsOnConsensus(java.util.List)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotationsOnConsensus()"><code>SequenceAlignmentDocument.canSetAnnotationsOnConsensus()</code></a></dd></dl>
</li>
</ul>
<a name="setAnnotations(int, java.util.List, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotations</h4>
<pre>public&nbsp;void&nbsp;setAnnotations(int&nbsp;index,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                  boolean&nbsp;allowModifyingReferencedDocuments)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Updates just the annotations on a sequence. This method may only be called if <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetAnnotations()"><code>SequenceAlignmentDocument.canSetAnnotations()</code></a> returns true<p/>
 This is used for updating non-track annotations only. Track annotations are handled via a different API. To update or add tracks check if
 a sequence in the alignment implements <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)">setAnnotations</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the 0-based index of the sequence to set annotations on, or -1 to set annotations on the consensus sequence.</dd><dd><code>annotations</code> - the new annotations for this sequence, which will replace any existing annotations.</dd><dd><code>allowModifyingReferencedDocuments</code> - if this is true  the original referenced sequence document will be modified to use these annotations
 If this parameter is not true and then any reference to the original document will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="getSummary()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSummary</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary</a>&nbsp;getSummary()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html#getSummary()">SequenceListSummary.Provider</a></code></strong></div>
<div class="block">Gets a summary of the properties of the sequences.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html#getSummary()">getSummary</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Provider</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a summary of properties of the sequences or null if no summary is available.</dd></dl>
</li>
</ul>
<a name="setAlignmentDataForSequencesNotInMemory(com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk.AlignmentData)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAlignmentDataForSequencesNotInMemory</h4>
<pre>public&nbsp;void&nbsp;setAlignmentDataForSequencesNotInMemory(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk.AlignmentData</a>&nbsp;alignmentData)</pre>
<div class="block">Replaces or sets all sequences (and associated data) in this alignment that are not the reference sequence (if any - see <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>getContigReferenceSequenceIndex()</code></a>.
 The data is for sequences that are not loaded into memory. This method may not be used on an alignment that already has non-reference
 sequences that are loaded into memory.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignmentData</code> - the new alignmentData. Must not be null.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if there are already non-reference sequences in this alignment that are loaded into memory.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getAlignmentDataForSequencesNotInMemory()"><code>getAlignmentDataForSequencesNotInMemory()</code></a></dd></dl>
</li>
</ul>
<a name="getAlignmentDataForSequencesNotInMemory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlignmentDataForSequencesNotInMemory</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk.AlignmentData</a>&nbsp;getAlignmentDataForSequencesNotInMemory()</pre>
<div class="block">If this document has its non-reference sequences stored as a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>, then this returns them along with data associated with them, or null if not.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>If this document has its non-reference sequences stored as a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk</code></a>, then this returns them along with data associated with them, or null if its non-reference sequences are already in memory.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAlignmentDataForSequencesNotInMemory(com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk.AlignmentData)"><code>setAlignmentDataForSequencesNotInMemory(com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk.AlignmentData)</code></a></dd></dl>
</li>
</ul>
<a name="addContigReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addContigReferenceSequence</h4>
<pre>public&nbsp;void&nbsp;addContigReferenceSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;referenceSequence)</pre>
<div class="block">Adds a sequence as the contig reference sequence {see @link #getContigReferenceSequenceIndex()}. This method may only be called on contigs which do
 not currently have a reference sequence (i.e. <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getContigReferenceSequenceIndex()"><code>getContigReferenceSequenceIndex()</code></a>==-1). The sequence will be added as the first
 sequence in the alignment/contig.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>referenceSequence</code> - the new reference sequence for this contig/alignment.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;getSequences()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Get the alignment as a list of sequences (Convenience function).

 The default implementation returns an unmodifiable list view backed by
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequence(int)"><code>SequenceAlignmentDocument.getSequence(int)</code></a>.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequences()">getSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a list of sequences</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequencesInImmutableList()"><code>SequenceAlignmentDocument.getSequencesInImmutableList()</code></a></dd></dl>
</li>
</ul>
<a name="getSequenceType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceType</h4>
<pre>public&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/SequenceType.html?is-external=true" title="class or interface in jebl.evolution.sequences">SequenceType</a>&nbsp;getSequenceType()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceType()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Returns the sequence type of the first sequence in this alignment (which should be the same as all other sequences in the alignment)</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceType()">getSequenceType</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence type of the first sequence in this alignment (which should be the same as all other sequences in the alignment)</dd></dl>
</li>
</ul>
<a name="getSizeRequiredToLoadIntoMemory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSizeRequiredToLoadIntoMemory</h4>
<pre>public&nbsp;long&nbsp;getSizeRequiredToLoadIntoMemory()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html#getSizeRequiredToLoadIntoMemory()">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></strong></div>
<div class="block">Return an estimate of the number of bytes required to load this document into memory. Or 0 if this document is unable to make an estimate in which case Geneious will use <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html#getSize()"><code>AnnotatedPluginDocument.getSize()</code></a>.
 Geneious caches the value returned from this method when the document is wrapped in an <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents"><code>AnnotatedPluginDocument</code></a> or when it is saved.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html#getSizeRequiredToLoadIntoMemory()">getSizeRequiredToLoadIntoMemory</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an estimate of the number of bytes required to load this document into memory. Or 0 if this document is unable to make an estimate.</dd></dl>
</li>
</ul>
<a name="getCircularLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCircularLength</h4>
<pre>public&nbsp;int&nbsp;getCircularLength()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">If this alignment is circular, returns the gapped length at which it wraps back to the start or 0 if this alignment is not circular.
 For example, if an alignment looks like<br/><br/>
 <code>
 RefSeq: -GATTACA----<br/>
 Read 1: ----TACAAGA-<br/>
 Read 2: -----ACA-GAT<br/>
 </code><br/>
 this could have a circular length of 8 to indicate the reads extending past the last base of the refSeq and considered to wrap back to the start.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()">getCircularLength</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>if this alignment is circular, returns the gapped length at which it wraps back to the start or 0 if this alignment is not circular</dd></dl>
</li>
</ul>
<a name="setCircularLength(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCircularLength</h4>
<pre>public&nbsp;void&nbsp;setCircularLength(int&nbsp;circularLength)</pre>
<div class="block">Sets the circular length of this alignment. This is the gapped length at which the alignment is considered to wrap back to the start</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>circularLength</code> - the circular length of this alignment, or 0 if this is not a circular alignment</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.50 (Geneious 5.5.0)</dd></dl>
</li>
</ul>
<a name="getTrackManager(boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getTrackManager</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager</a>&nbsp;getTrackManager(boolean&nbsp;mayReturnNullIfNoTracks)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html#getTrackManager(boolean)">SequenceTrack.Manager.Provider</a></code></strong></div>
<div class="block">Gets a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager</code></a> associated with this sequence for providing sequence tracks. For detailed information on sequence tracks, see <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack</code></a>.
 If this method is called on an alignment, the track manager is for the consensus sequence. Individual sequences in the alignment may also implement <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence"><code>SequenceTrack.Manager.Provider</code></a> if they provide tracks.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html#getTrackManager(boolean)">getTrackManager</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>mayReturnNullIfNoTracks</code> - to save on memory usage when dealing with lots of small sequences, some implementations may not wish to construct a TrackManager unless necessary. If this parameter is true the implementation may return null rather than an empty track manager if is has no tracks</dd>
<dt><span class="strong">Returns:</span></dt><dd>a track manager or possibly null if mayReturnNullIfNoTracks is true</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultAlignmentDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev Class</li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultAlignmentDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
