<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:45 NZDT 2016 -->
<title>DefaultPhylogenyDocument (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="DefaultPhylogenyDocument (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultPhylogenyDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultSameTaxaTreesDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultPhylogenyDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations</div>
<h2 title="Class DefaultPhylogenyDocument" class="title">Class DefaultPhylogenyDocument</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">com.biomatters.geneious.publicapi.implementations.DefaultAlignmentDocument</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/PhylogenyDocument.html" title="class in com.biomatters.geneious.publicapi.documents.types">com.biomatters.geneious.publicapi.documents.types.PhylogenyDocument</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.DefaultPhylogenyDocument</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">ExportableDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/Renamable.html" title="interface in com.biomatters.geneious.publicapi.documents">Renamable</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListSummary.Provider.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceListSummary.Provider</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/types/EditableSameTaxaTreesDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types">EditableSameTaxaTreesDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/types/SameTaxaTreesDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types">SameTaxaTreesDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/types/TreeDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types">TreeDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializableWithProgress.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializableWithProgress</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/SequenceTrack.Manager.Provider.html" title="interface in com.biomatters.geneious.publicapi.implementations.sequence">SequenceTrack.Manager.Provider</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">DefaultPhylogenyDocument</span>
extends <a href="../../../../../com/biomatters/geneious/publicapi/documents/types/PhylogenyDocument.html" title="class in com.biomatters.geneious.publicapi.documents.types">PhylogenyDocument</a>
implements <a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">ExportableDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/types/EditableSameTaxaTreesDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types">EditableSameTaxaTreesDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">An implementation of a Phylogeny document.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.ReferencedSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument.ReferencedSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.ExportableDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">ExportableDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.Format.html" title="enum in com.biomatters.geneious.publicapi.documents">ExportableDocument.Format</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsAlwaysLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsAlwaysLoaded</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.ReferencedDocumentsNotLoaded.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.ReferencedDocumentsNotLoaded</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.SizeRequiredToLoadIntoMemoryProvider.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument.SizeRequiredToLoadIntoMemoryProvider</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#KEY_DOCUMENT_SIZE">KEY_DOCUMENT_SIZE</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#KEY_NEXUS_TREE_DISTANCE_MATRIX">KEY_NEXUS_TREE_DISTANCE_MATRIX</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#KEY_SEQUENCE_RESIDUES">KEY_SEQUENCE_RESIDUES</a></strong></code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.implementations.DefaultAlignmentDocument">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;com.biomatters.geneious.publicapi.implementations.<a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#KEY_ALIGNMENT_OPTIONS">KEY_ALIGNMENT_OPTIONS</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#KEY_ALIGNMENT_SCORE">KEY_ALIGNMENT_SCORE</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#KEY_MATCH_REGIONS">KEY_MATCH_REGIONS</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME">FILE_DATA_ATTRIBUTE_NAME</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MAXIMUM_HTML_LENGTH">MAXIMUM_HTML_LENGTH</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#MODIFIED_DATE_FIELD">MODIFIED_DATE_FIELD</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_com.biomatters.geneious.publicapi.documents.XMLSerializable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME">ROOT_ELEMENT_NAME</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#DefaultPhylogenyDocument()">DefaultPhylogenyDocument</a></strong>()</code>
<div class="block">XML deserialization constructor.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#DefaultPhylogenyDocument(org.jdom.Element)">DefaultPhylogenyDocument</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">XML deserialization constructor.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#DefaultPhylogenyDocument(java.util.List, java.util.List, java.lang.String, java.lang.String, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, boolean)">DefaultPhylogenyDocument</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;&nbsp;trees,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;inAlignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;summary,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;alignmentDocument,
                        boolean&nbsp;standardTranslation)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#DefaultPhylogenyDocument(java.util.List, java.util.List, java.lang.String, java.lang.String, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, boolean, boolean)">DefaultPhylogenyDocument</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;&nbsp;trees,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;inAlignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;summary,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;alignmentDocument,
                        boolean&nbsp;standardTranslation,
                        boolean&nbsp;allTreesWritable)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Add a sequence to the end of the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#areAllTreesWritable()">areAllTreesWritable</a></strong>()</code>
<div class="block">True if the tree viewer should allow the user to modify all trees in this document (if false, only the first tree will be writable)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a></strong>()</code>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#canExport(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format)">canExport</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.Format.html" title="enum in com.biomatters.geneious.publicapi.documents">ExportableDocument.Format</a>&nbsp;format)</code>
<div class="block">Check if a format is supported by document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#export(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format, java.io.OutputStream)">export</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.Format.html" title="enum in com.biomatters.geneious.publicapi.documents">ExportableDocument.Format</a>&nbsp;format,
      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;to)</code>
<div class="block">Export a document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#fromXML(org.jdom.Element)">fromXML</a></strong>(org.jdom.Element&nbsp;doc)</code>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#getDescription()">getDescription</a></strong>()</code>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#getTrees()">getTrees</a></strong>()</code>
<div class="block">Get tree set.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#removeSequence(int)">removeSequence</a></strong>(int&nbsp;index)</code>
<div class="block">Removes the specified sequence from the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#reorderSequences(int[])">reorderSequences</a></strong>(int[]&nbsp;ordering)</code>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#setNexusSource(java.lang.String, int)">setNexusSource</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fileText,
              int&nbsp;nexusBlockNumber)</code>
<div class="block">Set source of document in nexus format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a></strong>(int&nbsp;index,
               <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
               boolean&nbsp;allowModifyingOriginals)</code>
<div class="block">change the name of a sequence within the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#setTrees(java.util.List)">setTrees</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;&nbsp;newTrees)</code>
<div class="block">Set some trees.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#toHTML()">toHTML</a></strong>()</code>
<div class="block">Provide a textual view of document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#toXML()">toXML</a></strong>()</code>
<div class="block">Convert object to a JDOM element.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></strong>(int&nbsp;index,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
              boolean&nbsp;allowModifyingOriginals)</code>
<div class="block">Set new residues and related data for one sequence.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.implementations.DefaultAlignmentDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.implementations.<a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addContigReferenceSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addContigReferenceSequence</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addUrnVisibleByDefaultTrack(com.biomatters.geneious.publicapi.documents.URN)">addUrnVisibleByDefaultTrack</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#calculateScore(jebl.evolution.align.scores.Scores, java.lang.CharSequence[], double, double, boolean, boolean)">calculateScore</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetAnnotations()">canSetAnnotations</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetAnnotationsOnConsensus()">canSetAnnotationsOnConsensus</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField)">canSetFieldValue</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canSetSequenceNames()">canSetSequenceNames</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#fromJeblAlignment(jebl.evolution.alignments.Alignment)">fromJeblAlignment</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#fromXML(org.jdom.Element, jebl.util.ProgressListener)">fromXML</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getAlignmentDataForSequencesNotInMemory()">getAlignmentDataForSequencesNotInMemory</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getAnnotationsOnConsensus()">getAnnotationsOnConsensus</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getCircularLength()">getCircularLength</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getContigReferenceSequenceIndex()">getContigReferenceSequenceIndex</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getDescription(boolean)">getDescription</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getMateExpectedDistance(int)">getMateExpectedDistance</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getMateIndex(int)">getMateIndex</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getName()">getName</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getNumberOfSequences()">getNumberOfSequences</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getReferencedDocument(int)">getReferencedDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getReferencedSequence(int)">getReferencedSequence</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getScore()">getScore</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequence(int)">getSequence</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequences()">getSequences</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequencesInImmutableList()">getSequencesInImmutableList</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSequenceType()">getSequenceType</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSizeRequiredToLoadIntoMemory()">getSizeRequiredToLoadIntoMemory</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getSummary()">getSummary</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getTrackManager(boolean)">getTrackManager</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getURN()">getURN</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getUrnsForVisibleByDefaultTracks()">getUrnsForVisibleByDefaultTracks</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isContig()">isContig</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isEditable()">isEditable</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isFreeEndGapAlignment()">isFreeEndGapAlignment</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isQueryCentricAlignment()">isQueryCentricAlignment</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#isReferencedDocumentReversed(int)">isReferencedDocumentReversed</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAlignmentDataForSequencesNotInMemory(com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk.AlignmentData)">setAlignmentDataForSequencesNotInMemory</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAlignmentOptions(java.lang.String)">setAlignmentOptions</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAnnotations(int, java.util.List, boolean)">setAnnotations</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setAnnotationsOnConsensus(java.util.List)">setAnnotationsOnConsensus</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setCircularLength(int)">setCircularLength</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContig(boolean)">setContig</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setContigReferenceSequenceIndex(int)">setContigReferenceSequenceIndex</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setCreationDate(java.util.Date)">setCreationDate</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setFieldValue(int, com.biomatters.geneious.publicapi.documents.DocumentField, java.lang.Object, boolean)">setFieldValue</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setFreeEndGapAlignment(boolean)">setFreeEndGapAlignment</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setMatchRegions(java.lang.String)">setMatchRegions</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setMates(int, int, int)">setMates</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setName(java.lang.String)">setName</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setNumberOfDifferentMappedReads(int)">setNumberOfDifferentMappedReads</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setQueryCentricAlignment(boolean)">setQueryCentricAlignment</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)">setReferencedDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence, int)">setReferencedSequence</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setReferencedSequenceListDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, int)">setReferencedSequenceListDocument</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setScore(java.lang.Double)">setScore</a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#toXML(jebl.util.ProgressListener)">toXML</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.biomatters.geneious.publicapi.documents.sequence.<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignment(boolean)">asJeblAlignment</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#asJeblAlignmentIfMemoryAllows(boolean)">asJeblAlignmentIfMemoryAllows</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRealignedName(java.lang.String)">getRealignedName</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocuments()">getReferencedDocuments</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedSequences()">getReferencedSequences</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getRefinedName(java.lang.String)">getRefinedName</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getSequenceAsJeblSequence(int)">getSequenceAsJeblSequence</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedSequence(int, com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument.ReferencedSequence)">setReferencedSequence</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.biomatters.geneious.publicapi.documents.PluginDocument">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.biomatters.geneious.publicapi.documents.<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></h3>
<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getCreationDate()">getCreationDate</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDisplayableFields()">getDisplayableFields</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getFieldValue(java.lang.String)">getFieldValue</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getName()">getName</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getURN()">getURN</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="KEY_NEXUS_TREE_DISTANCE_MATRIX">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>KEY_NEXUS_TREE_DISTANCE_MATRIX</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_NEXUS_TREE_DISTANCE_MATRIX</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.DefaultPhylogenyDocument.KEY_NEXUS_TREE_DISTANCE_MATRIX">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="KEY_DOCUMENT_SIZE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>KEY_DOCUMENT_SIZE</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_DOCUMENT_SIZE</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.DefaultPhylogenyDocument.KEY_DOCUMENT_SIZE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="KEY_SEQUENCE_RESIDUES">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>KEY_SEQUENCE_RESIDUES</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> KEY_SEQUENCE_RESIDUES</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.DefaultPhylogenyDocument.KEY_SEQUENCE_RESIDUES">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="DefaultPhylogenyDocument()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultPhylogenyDocument</h4>
<pre>public&nbsp;DefaultPhylogenyDocument()</pre>
<div class="block">XML deserialization constructor. See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>XMLSerializable</code></a></div>
</li>
</ul>
<a name="DefaultPhylogenyDocument(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultPhylogenyDocument</h4>
<pre>public&nbsp;DefaultPhylogenyDocument(org.jdom.Element&nbsp;element)
                         throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">XML deserialization constructor. See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents"><code>XMLSerializable</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - the previously serialized element returned from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html#toXML()"><code>toXML()</code></a>.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if deserialization fails</dd></dl>
</li>
</ul>
<a name="DefaultPhylogenyDocument(java.util.List, java.util.List, java.lang.String, java.lang.String, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DefaultPhylogenyDocument</h4>
<pre>public&nbsp;DefaultPhylogenyDocument(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;&nbsp;trees,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;inAlignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;summary,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;alignmentDocument,
                        boolean&nbsp;standardTranslation)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>trees</code> - One or more trees whose tips are sequences of the alignment; must not be null or empty</dd><dd><code>inAlignedSequences</code> - List of aligned sequences. Can be null if alignmentDocument is given.</dd><dd><code>name</code> - document name</dd><dd><code>summary</code> - document summary</dd><dd><code>alignmentDocument</code> - Alignment document which is the origin of alignedSequences (optional, i.e. may be null)</dd><dd><code>standardTranslation</code> - when true, taxa on tree is in fact ("1", "2", ...) and so a conversion to the actual
 taxa name (from the sequences) has to be applied when appropriate.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if trees is null or empty</dd></dl>
</li>
</ul>
<a name="DefaultPhylogenyDocument(java.util.List, java.util.List, java.lang.String, java.lang.String, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument, boolean, boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DefaultPhylogenyDocument</h4>
<pre>public&nbsp;DefaultPhylogenyDocument(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;&nbsp;trees,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/Sequence.html?is-external=true" title="class or interface in jebl.evolution.sequences">Sequence</a>&gt;&nbsp;inAlignedSequences,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;summary,
                        <a href="../../../../../com/biomatters/geneious/publicapi/documents/AnnotatedPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AnnotatedPluginDocument</a>&nbsp;alignmentDocument,
                        boolean&nbsp;standardTranslation,
                        boolean&nbsp;allTreesWritable)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>trees</code> - One or more trees whose tips are sequences of the alignment; must not be null or empty</dd><dd><code>inAlignedSequences</code> - List of aligned sequences. Can be null if alignmentDocument is given.</dd><dd><code>name</code> - document name</dd><dd><code>summary</code> - document summary</dd><dd><code>alignmentDocument</code> - Alignment document which is the origin of alignedSequences (optional, i.e. may be null)</dd><dd><code>standardTranslation</code> - when true, taxa on tree is in fact ("1", "2", ...) and so a conversion to the actual
 taxa name (from the sequences) has to be applied when appropriate.</dd><dd><code>allTreesWritable</code> - True if the tree viewer should allow the user to modify all trees in this document (if false, only the first tree will be writable)</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if trees is null or empty</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="setNexusSource(java.lang.String, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNexusSource</h4>
<pre>public&nbsp;void&nbsp;setNexusSource(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fileText,
                  int&nbsp;nexusBlockNumber)</pre>
<div class="block">Set source of document in nexus format.

 Used to preserve the full content of an imported nexus file such as comments and unhandled sections.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fileText</code> - text in nexus format</dd><dd><code>nexusBlockNumber</code> - block number (or -1 for the first one)</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>org.jdom.IllegalDataException</code> - if fileText is not legal XML character data,
         as determined by <code>Verifier.checkCharacterData(String)</code>.</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalStateException.html?is-external=true" title="class or interface in java.lang">IllegalStateException</a></code> - if standardTranslationForTaxa was used.</dd></dl>
</li>
</ul>
<a name="getTrees()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrees</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;&nbsp;getTrees()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/SameTaxaTreesDocument.html#getTrees()">SameTaxaTreesDocument</a></code></strong></div>
<div class="block">Get tree set.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/SameTaxaTreesDocument.html#getTrees()">getTrees</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/SameTaxaTreesDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types">SameTaxaTreesDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the trees</dd></dl>
</li>
</ul>
<a name="areAllTreesWritable()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>areAllTreesWritable</h4>
<pre>public&nbsp;boolean&nbsp;areAllTreesWritable()</pre>
<div class="block">True if the tree viewer should allow the user to modify all trees in this document (if false, only the first tree will be writable)</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>True if the tree viewer should allow the user to modify all trees in this document (if false, only the first tree will be writable)</dd></dl>
</li>
</ul>
<a name="canExport(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canExport</h4>
<pre>public&nbsp;boolean&nbsp;canExport(<a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.Format.html" title="enum in com.biomatters.geneious.publicapi.documents">ExportableDocument.Format</a>&nbsp;format)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html#canExport(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format)">ExportableDocument</a></code></strong></div>
<div class="block">Check if a format is supported by document.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html#canExport(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format)">canExport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">ExportableDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>format</code> - export format</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if format is supported.</dd></dl>
</li>
</ul>
<a name="export(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format, java.io.OutputStream)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>export</h4>
<pre>public&nbsp;boolean&nbsp;export(<a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.Format.html" title="enum in com.biomatters.geneious.publicapi.documents">ExportableDocument.Format</a>&nbsp;format,
             <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;to)
               throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html#export(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format, java.io.OutputStream)">ExportableDocument</a></code></strong></div>
<div class="block">Export a document.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html#export(com.biomatters.geneious.publicapi.documents.ExportableDocument.Format, java.io.OutputStream)">export</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/ExportableDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">ExportableDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>format</code> - format to export</dd><dd><code>to</code> - output document to this stream.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true when file exported successfully, false otherwise.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if the OutputStream can't be written to for any reason</dd></dl>
</li>
</ul>
<a name="toXML()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">XMLSerializable</a></code></strong></div>
<div class="block">Convert object to a JDOM element. The representation should be complete so that <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)"><code>XMLSerializable.fromXML(org.jdom.Element)</code></a> can completely
 restore the objects representation.
 <p/>
 It is recommended that the returned element use <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#ROOT_ELEMENT_NAME"><code>XMLSerializable.ROOT_ELEMENT_NAME</code></a> as its name,
 in which case it must not define an attribute called "type". In this case,
 fromXML, will be called with an element whose name may differ from the element
 return from this function. This recommendation allows a more compact representation
 of the XML can be stored.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classToXML(java.lang.String, com.biomatters.geneious.publicapi.documents.XMLSerializable)"><code>XMLSerializer.classToXML(String, XMLSerializable)</code></a>
 which calls this method internally.
 <p/>
 PluginDocument implementations of this method may choose to throw an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a>,
 enclosed in a <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang"><code>RuntimeException</code></a>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#toXML()">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="setTrees(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTrees</h4>
<pre>public&nbsp;void&nbsp;setTrees(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;? extends <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/trees/Tree.html?is-external=true" title="class or interface in jebl.evolution.trees">Tree</a>&gt;&nbsp;newTrees)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/EditableSameTaxaTreesDocument.html#setTrees(java.util.List)">EditableSameTaxaTreesDocument</a></code></strong></div>
<div class="block">Set some trees.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/EditableSameTaxaTreesDocument.html#setTrees(java.util.List)">setTrees</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/types/EditableSameTaxaTreesDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types">EditableSameTaxaTreesDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>newTrees</code> - A list of new trees. A null entry means no change and the existing tree is taken.</dd></dl>
</li>
</ul>
<a name="reorderSequences(int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderSequences</h4>
<pre>public&nbsp;void&nbsp;reorderSequences(int[]&nbsp;ordering)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Adjusts the ordering of sequences in the alignment according to the ordering array.
 If ordering[i]==j, then the sequence at index 'i' should be the sequence previously at index 'j'
 For example, if ordering = {0,1,3,4,2}
 Then:
 sequence at index 2 is moved to index 4
 sequence at index 3 is moved to index 2
 sequence at index 4 is moved to index 3

 This function may only be called on alignments that return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#reorderSequences(int[])">reorderSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>ordering</code> - Each integer between 0 and ordering.length - 1 inclusive should occur exactly once in this array.</dd></dl>
</li>
</ul>
<a name="setSequenceName(int, java.lang.String, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceName</h4>
<pre>public&nbsp;void&nbsp;setSequenceName(int&nbsp;index,
                   <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName,
                   boolean&nbsp;allowModifyingOriginals)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">change the name of a sequence within the alignment.
 This method may only be called if <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canSetSequenceNames()"><code>SequenceAlignmentDocument.canSetSequenceNames()</code></a> returns true.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#setSequenceName(int, java.lang.String, boolean)">setSequenceName</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - index of the sequence name to change</dd><dd><code>newName</code> - the new name for the sequence.</dd><dd><code>allowModifyingOriginals</code> - if the alignment stores a reference to an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be renamed too.
 Otherwise,  any existing reference to the original SequenceDocument will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>updateSequence</h4>
<pre>public&nbsp;void&nbsp;updateSequence(int&nbsp;index,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequence,
                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations,
                  <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">NucleotideGraph</a>&nbsp;graph,
                  boolean&nbsp;allowModifyingOriginals)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Set new residues and related data for one sequence.

 This function may only be called on alignments that return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a>
 (otherwise, the method should throw <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang"><code>UnsupportedOperationException</code></a>).</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#updateSequence(int, java.lang.CharSequence, java.util.List, com.biomatters.geneious.publicapi.documents.sequence.NucleotideGraph, boolean)">updateSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be updated. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd><dd><code>sequence</code> - the new residues</dd><dd><code>annotations</code> - a list of the annotations</dd><dd><code>graph</code> - Must be null if the original sequence at the specified index is not a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>.
 Otherwise (i.e. if the sequence was a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraphSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphSequenceDocument</code></a>), specifies the new
 graph to use for that sequence.</dd><dd><code>allowModifyingOriginals</code> - if the change being made to the sequence would affect the unaligned version of it
 (for example, the sequence residues are changing rather than just the gaps)
 and the alignment stores a reference an editable original SequenceDocument
 (which can be determined by using <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getReferencedDocument(int)"><code>SequenceAlignmentDocument.getReferencedDocument(int)</code></a>) then if this parameter is true
 the original referenced sequence document will be modified.
 If this parameter is not true and the change would have otherwise modified the original SequenceDocument,
 then any reference to the original document will be discarded
 and a copy of the original sequence stored internally within the alignment instead.</dd></dl>
</li>
</ul>
<a name="removeSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeSequence</h4>
<pre>public&nbsp;void&nbsp;removeSequence(int&nbsp;index)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Removes the specified sequence from the alignment. All other sequences should
 maintain their original ordering.

 This function may only be called on alignments that return true from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#isEditable()"><code>SequenceAlignmentDocument.isEditable()</code></a></div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#removeSequence(int)">removeSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to be removed. between 0 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getNumberOfSequences()"><code>SequenceAlignmentDocument.getNumberOfSequences()</code></a> - 1</dd></dl>
</li>
</ul>
<a name="addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequence</h4>
<pre>public&nbsp;void&nbsp;addSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Add a sequence to the end of the alignment. <code>sequence</code> should be the
 gapped (aligned) sequence.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">addSequence</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - sequence to add</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#reorderSequences(int[])"><code>SequenceAlignmentDocument.reorderSequences(int[])</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)"><code>SequenceAlignmentDocument.setReferencedDocument(int, com.biomatters.geneious.publicapi.documents.AnnotatedPluginDocument)</code></a></dd></dl>
</li>
</ul>
<a name="canAddAndRemoveSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>canAddAndRemoveSequences</h4>
<pre>public&nbsp;boolean&nbsp;canAddAndRemoveSequences()</pre>
<div class="block"><strong>Description copied from class:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#canAddAndRemoveSequences()">SequenceAlignmentDocument</a></code></strong></div>
<div class="block">Check if this alignment supports adding and removing sequences.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#canAddAndRemoveSequences()">canAddAndRemoveSequences</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>true if adding and removing sequences is supported, false otherwise.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#addSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)"><code>SequenceAlignmentDocument.addSequence(SequenceDocument)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#removeSequence(int)"><code>SequenceAlignmentDocument.removeSequence(int)</code></a></dd></dl>
</li>
</ul>
<a name="fromXML(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromXML</h4>
<pre>public&nbsp;void&nbsp;fromXML(org.jdom.Element&nbsp;doc)
             throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">XMLSerializable</a></code></strong></div>
<div class="block">Restore the object from the JDOM Element returned by <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a>.
 <p/>
 This method generally should not be called directly. Instead, you should usually call
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element)"><code>XMLSerializer.classFromXML(org.jdom.Element)</code></a>
 or <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializer.html#classFromXML(org.jdom.Element, java.lang.Class)"><code>XMLSerializer.classFromXML(org.jdom.Element, Class)</code></a>
 which calls this method internally.
 <p/>
 It is optional to implement this method. Instead of implementing an empty constructor and implementing
 this method properly, the implementation may instead
 throw an UnsupportedOperationException and implement a constructor
 that takes a single <code>Element</code> as a parameter.
 This allows for cleaner code such as support for final fields in the XMLSerializable class.
 <p/>
 The element parameter should not be modified since it may be reused.  If you need a modified version of it,
 take a copy with <code>Element.clone()</code>.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#fromXML(org.jdom.Element)">fromXML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>doc</code> - representation from a previous call to <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if the Element can't be converted into this type of object</dd></dl>
</li>
</ul>
<a name="getDescription()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">PluginDocument</a></code></strong></div>
<div class="block">A human readable single line of text describing this document,
 to appear in the description field in the document table.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#getDescription()">getDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getDescription()">getDescription</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a one line simple text description of the document, or null if no description is available.</dd></dl>
</li>
</ul>
<a name="toHTML()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>toHTML</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toHTML()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">PluginDocument</a></code></strong></div>
<div class="block">Provide a textual view of document. The contents should be at most 30K-60K characters. One reason is that this
 text is rendered using JTextPane, which is horrendously slow when handling large text. The more fundamental
 reason is that large amounts of text are generally not useful for a user. The whole idea of using a tool such as
 Geneious is to provide specialized viewers which make the large amount of bioinformatics data perceivable and
 comprehensible to a human user.
 <p/>
 Return an HTML body fragment (i.e. a fragment which sits inside a &lt;body&gt;&lt;/body&gt; tag). (Header,stylesheet and
 annotations are automatically added by GENEious code).
 <p/>
 Can optionally return an entire html tag set, as long as the first six characters are <code>&lt;html></code></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#toHTML()">toHTML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a></code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#toHTML()">toHTML</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>an HTML fragment of tag set, or null to have no text view.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DefaultPhylogenyDocument.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultSameTaxaTreesDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/DefaultPhylogenyDocument.html" target="_top">Frames</a></li>
<li><a href="DefaultPhylogenyDocument.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
