<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>EndGapsManager (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="EndGapsManager (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/EndGapsManager.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsHelper.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.Builder.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" target="_top">Frames</a></li>
<li><a href="EndGapsManager.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations</div>
<h2 title="Class EndGapsManager" class="title">Class EndGapsManager</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.EndGapsManager</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">EndGapsManager</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Given a set of aligned sequences, provides methods for quickly finding all sequences
 that intersect a given base number, ignoring those sequences in end gap regions.
 On alignments/contigs with a reference sequence, the reference sequence is
 returned as an intersecting sequence just like reads.
 <p/>
 Some <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>DefaultAlignmentDocuments</code></a> may provide <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceListOnDisk.AlignmentData</code></a>
 via 
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/DefaultAlignmentDocument.html#getAlignmentDataForSequencesNotInMemory()"><code>DefaultAlignmentDocument.getAlignmentDataForSequencesNotInMemory()</code></a>
 which in turn can provide a pre-built EndGapsManager via
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.AlignmentData.html#getEndGapsManager()"><code>AlignmentData.getEndGapsManager()</code></a></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.Builder.html" title="class in com.biomatters.geneious.publicapi.implementations">EndGapsManager.Builder</a></strong></code>
<div class="block">Used for building an <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>EndGapsManager</code></a> which is too large to fit in memory</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)">EndGapsManager</a></strong>(org.jdom.Element&nbsp;element,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;associatedSequences)</code>
<div class="block">Recreates an EndGapsManager from XML.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(com.biomatters.geneious.publicapi.implementations.EndGapsManager, com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence)">EndGapsManager</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" title="class in com.biomatters.geneious.publicapi.implementations">EndGapsManager</a>&nbsp;endGapsManager,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&nbsp;referenceSequence)</code>
<div class="block">Creates a new EndGapsManager identical to the provided one except for the addition of a reference sequence at index 0.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(java.util.List, int, int)">EndGapsManager</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&gt;&nbsp;sequences,
              int&nbsp;minimumHashSize,
              int&nbsp;numberOfSequences)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, int)">EndGapsManager</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
              int&nbsp;minimumHashSize)</code>
<div class="block">Creates an EndGapsManager for the sequences in the given alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence[], int)">EndGapsManager</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>[]&nbsp;sequences,
              int&nbsp;minimumHashSize)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence[], int, int)">EndGapsManager</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>[]&nbsp;sequences,
              int&nbsp;minimumHashSize,
              int&nbsp;numberOfSequences)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getHashSize()">getHashSize</a></strong>()</code>
<div class="block">Gets the hash size used for this manager.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getNumberOfColumns()">getNumberOfColumns</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)">getPotentialSequencesCoveringArray</a></strong>(int&nbsp;residueIndex)</code>
<div class="block">Returns all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap, however, the returned list may also include sequences that do not cover the given residueIndex.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int, int)">getPotentialSequencesCoveringArray</a></strong>(int&nbsp;residueIndex,
                                  int&nbsp;circularAlignmentLength)</code>
<div class="block">Returns all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap, however, the returned list may also include sequences that do not cover the given residueIndex.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSamePotentialSequencesLowerBound(int)">getSamePotentialSequencesLowerBound</a></strong>(int&nbsp;residueIndex)</code>
<div class="block">Gets a lower bound such that all calls to <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a>
 with an index between LowerBound and residueIndex inclusive would return the same array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSamePotentialSequencesUpperBoundExclusive(int)">getSamePotentialSequencesUpperBoundExclusive</a></strong>(int&nbsp;residueIndex)</code>
<div class="block">Gets an upper bounds such at all calls to <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a>
 with an index between residueIndex and this upper bound exclusive would return the same array.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequence(int)">getSequence</a></strong>(int&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequenceCount()">getSequenceCount</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequences()">getSequences</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequencesCoveringArray(int)">getSequencesCoveringArray</a></strong>(int&nbsp;residueIndex)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequencesCoveringIterable(int)">getSequencesCoveringIterable</a></strong>(int&nbsp;residueIndex)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequencesCoveringIterator(int)">getSequencesCoveringIterator</a></strong>(int&nbsp;residueIndex)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Like <code>XMLSerializable.OldVersionCompatible#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code>
 but this class isn't really XML serialiable.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#toXmlExcludingSequences(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXmlExcludingSequences</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                       <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Converts this EndGapsManager to XML, excluding the associated sequences, which must be stored independently and provided again during serialization using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)"><code>EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#toXmlExcludingSequences(java.lang.String)">toXmlExcludingSequences</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName)</code>
<div class="block">Converts this EndGapsManager to XML, excluding the associated sequences, which must be stored independently and provided again during serialization using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)"><code>EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)</code></a></div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="EndGapsManager(com.biomatters.geneious.publicapi.implementations.EndGapsManager, com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EndGapsManager</h4>
<pre>public&nbsp;EndGapsManager(<a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" title="class in com.biomatters.geneious.publicapi.implementations">EndGapsManager</a>&nbsp;endGapsManager,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&nbsp;referenceSequence)</pre>
<div class="block">Creates a new EndGapsManager identical to the provided one except for the addition of a reference sequence at index 0.
 If the provided EndGapsManager already has a reference sequence then the previous reference sequence is excluded from this new end gaps manager.
 An EndGapsManager created using this method is not serializable using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#toXmlExcludingSequences(java.lang.String)"><code>toXmlExcludingSequences(String)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>endGapsManager</code> - an end gaps manager</dd><dd><code>referenceSequence</code> - the new reference sequence or null for no reference sequence</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EndGapsManager</h4>
<pre>public&nbsp;EndGapsManager(org.jdom.Element&nbsp;element,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceListOnDisk.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceListOnDisk</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&gt;&nbsp;associatedSequences)</pre>
<div class="block">Recreates an EndGapsManager from XML.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - n element previously returned from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#toXmlExcludingSequences(java.lang.String)"><code>toXmlExcludingSequences(String)</code></a></dd><dd><code>associatedSequences</code> - the sequence associated with the EndGapsManager</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="EndGapsManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence[], int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EndGapsManager</h4>
<pre>public&nbsp;EndGapsManager(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>[]&nbsp;sequences,
              int&nbsp;minimumHashSize)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - the sequences the consider end gaps on. This array is not copied and this class relies on the caller not changing the contents of this array after constructing this class.</dd><dd><code>minimumHashSize</code> - Suggested value of 256. This means that for every 256 residues we maintain an index of all possible sequences that could cover that range. Lower values make lookups faster, but lower values take longer to construct and uses more memory. The implementation of this class may use a higher hash size than the provided one and the size it selects will be returned from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getHashSize()"><code>getHashSize()</code></a></dd></dl>
</li>
</ul>
<a name="EndGapsManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence[], int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EndGapsManager</h4>
<pre>public&nbsp;EndGapsManager(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>[]&nbsp;sequences,
              int&nbsp;minimumHashSize,
              int&nbsp;numberOfSequences)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - the sequences to consider end gaps on. This array is not copied and this class relies on the caller not changing the contents of this array after constructing this class.</dd><dd><code>minimumHashSize</code> - Suggested value of 256. This means that for every 256 residues we maintain an index of all possible sequences that could cover that range. Lower values make lookups faster, but lower values take longer to construct and uses more memory. The implementation of this class may use a higher hash size than the provided one and the size it selects will be returned from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getHashSize()"><code>getHashSize()</code></a></dd><dd><code>numberOfSequences</code> - usually this is sequences.length, but sometimes the caller may want to use a subset of the sequences.</dd></dl>
</li>
</ul>
<a name="EndGapsManager(java.util.List, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EndGapsManager</h4>
<pre>public&nbsp;EndGapsManager(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&gt;&nbsp;sequences,
              int&nbsp;minimumHashSize,
              int&nbsp;numberOfSequences)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - the sequences to consider end gaps on. This list is not copied and this class relies on the caller not changing the contents of this list after constructing this class.</dd><dd><code>minimumHashSize</code> - Suggested value of 256. This means that for every 256 residues we maintain an index of all possible sequences that could cover that range. Lower values make lookups faster, but lower values take longer to construct and uses more memory. The implementation of this class may use a higher hash size than the provided one and the size it selects will be returned from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getHashSize()"><code>getHashSize()</code></a></dd><dd><code>numberOfSequences</code> - usually this is sequences.length, but sometimes the caller may want to use a subset of the sequences.</dd></dl>
</li>
</ul>
<a name="EndGapsManager(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>EndGapsManager</h4>
<pre>public&nbsp;EndGapsManager(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignment,
              int&nbsp;minimumHashSize)</pre>
<div class="block">Creates an EndGapsManager for the sequences in the given alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - the alignment containing the sequences to create the EndGapsManager on.</dd><dd><code>minimumHashSize</code> - Suggested value of 256. This means that for every 256 residues we maintain an index of all possible sequences that could cover that range. Lower values make lookups faster, but lower values take longer to construct and uses more memory. The implementation of this class may use a higher hash size than the provided one and the size it selects will be returned from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getHashSize()"><code>getHashSize()</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block">Like <code>XMLSerializable.OldVersionCompatible#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code>
 but this class isn't really XML serialiable. Instead it provides a partial XML serialization method</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - as defined by <code>XMLSerializable.OldVersionCompatible#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code></dd>
<dt><span class="strong">Returns:</span></dt><dd>as defined by <code>XMLSerializable.OldVersionCompatible#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)</code></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="toXmlExcludingSequences(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXmlExcludingSequences</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXmlExcludingSequences(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName)
                                         throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Converts this EndGapsManager to XML, excluding the associated sequences, which must be stored independently and provided again during serialization using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)"><code>EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>elementName</code> - the name of the element to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>XML for this EndGapsManager. Note that this uses <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME"><code>PluginDocument.FILE_DATA_ATTRIBUTE_NAME</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if it can't be serialized (e.g. can't write to disk)</dd>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true" title="class or interface in java.lang">UnsupportedOperationException</a></code> - if this EndGapsManager was created using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(com.biomatters.geneious.publicapi.implementations.EndGapsManager, com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence)"><code>EndGapsManager(EndGapsManager, com.biomatters.geneious.publicapi.documents.sequence.SequenceCharSequence)</code></a></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="toXmlExcludingSequences(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXmlExcludingSequences</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXmlExcludingSequences(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                                       <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)
                                         throws <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></pre>
<div class="block">Converts this EndGapsManager to XML, excluding the associated sequences, which must be stored independently and provided again during serialization using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)"><code>EndGapsManager(org.jdom.Element, com.biomatters.geneious.publicapi.documents.sequence.SequenceListOnDisk)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - </dd><dd><code>progressListener</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>XML for this EndGapsManager. Note that this uses <a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html#FILE_DATA_ATTRIBUTE_NAME"><code>PluginDocument.FILE_DATA_ATTRIBUTE_NAME</code></a></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents">XMLSerializationException</a></code> - if it can't be serialized (e.g. can't write to disk)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getSequencesCoveringArray(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesCoveringArray</h4>
<pre>public&nbsp;int[]&nbsp;getSequencesCoveringArray(int&nbsp;residueIndex)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the 0-based index to get sequences indices for</dd>
<dt><span class="strong">Returns:</span></dt><dd>all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap.</dd></dl>
</li>
</ul>
<a name="getPotentialSequencesCoveringArray(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPotentialSequencesCoveringArray</h4>
<pre>public&nbsp;int[]&nbsp;getPotentialSequencesCoveringArray(int&nbsp;residueIndex)</pre>
<div class="block">Returns all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap, however, the returned list may also include sequences that do not cover the given residueIndex.
 Most usages should use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequencesCoveringArray(int)"><code>getSequencesCoveringArray(int)</code></a> instead unless they are very concerned about performance and don't
 want to allocate the extra array required to return the results.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the 0-based index to get sequences indices for</dd>
<dt><span class="strong">Returns:</span></dt><dd>all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap. However, the returned list may also include sequences that do not cover the given residueIndex.</dd></dl>
</li>
</ul>
<a name="getPotentialSequencesCoveringArray(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPotentialSequencesCoveringArray</h4>
<pre>public&nbsp;int[]&nbsp;getPotentialSequencesCoveringArray(int&nbsp;residueIndex,
                                       int&nbsp;circularAlignmentLength)</pre>
<div class="block">Returns all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap, however, the returned list may also include sequences that do not cover the given residueIndex.
 Most usages should use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequencesCoveringArray(int)"><code>getSequencesCoveringArray(int)</code></a> instead unless they are very concerned about performance and don't
 want to allocate the extra array required to return the results.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the 0-based index to get sequences indices for</dd><dd><code>circularAlignmentLength</code> - the circular length of the alignment if it is circular, or 0 if it is not circular (as defined by <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html#getCircularLength()"><code>SequenceAlignmentDocument.getCircularLength()</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap. However, the returned list may also include sequences that do not cover the given residueIndex.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="getSamePotentialSequencesLowerBound(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSamePotentialSequencesLowerBound</h4>
<pre>public&nbsp;int&nbsp;getSamePotentialSequencesLowerBound(int&nbsp;residueIndex)</pre>
<div class="block">Gets a lower bound such that all calls to <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a>
 with an index between LowerBound and residueIndex inclusive would return the same array.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the residue index to get a lower bound for.</dd>
<dt><span class="strong">Returns:</span></dt><dd>an index<=residueIndex which indicates a lower bound 'LowerBound' for which all calls to <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a>
 with an index between LowerBound and residueIndex inclusive would return the same array.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSamePotentialSequencesUpperBoundExclusive(int)"><code>(int)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a></dd></dl>
</li>
</ul>
<a name="getSamePotentialSequencesUpperBoundExclusive(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSamePotentialSequencesUpperBoundExclusive</h4>
<pre>public&nbsp;int&nbsp;getSamePotentialSequencesUpperBoundExclusive(int&nbsp;residueIndex)</pre>
<div class="block">Gets an upper bounds such at all calls to <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a>
 with an index between residueIndex and this upper bound exclusive would return the same array.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the residue index to get an upper bound for.</dd>
<dt><span class="strong">Returns:</span></dt><dd>and index>residueIndex which indicates an upper bound for which all calls to <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a>
 with an index between residueIndex and this upper bound exclusive would return the same array.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSamePotentialSequencesLowerBound(int)"><code>getSamePotentialSequencesLowerBound(int)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getPotentialSequencesCoveringArray(int)"><code>getPotentialSequencesCoveringArray(int)</code></a></dd></dl>
</li>
</ul>
<a name="getHashSize()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHashSize</h4>
<pre>public&nbsp;int&nbsp;getHashSize()</pre>
<div class="block">Gets the hash size used for this manager.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the hash size used for this end gaps manager. This will be greater than or equal to the hashSize passed to the constructor.</dd></dl>
</li>
</ul>
<a name="getSequencesCoveringIterable(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesCoveringIterable</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;getSequencesCoveringIterable(int&nbsp;residueIndex)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the 0-based index to get sequences indices for</dd>
<dt><span class="strong">Returns:</span></dt><dd>all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap.</dd></dl>
</li>
</ul>
<a name="getSequencesCoveringIterator(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequencesCoveringIterator</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;getSequencesCoveringIterator(int&nbsp;residueIndex)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>residueIndex</code> - the 0-based index to get sequences indices for</dd>
<dt><span class="strong">Returns:</span></dt><dd>all sequences which contain the given index, excluding those sequences for which residueIndex is an end gap.</dd></dl>
</li>
</ul>
<a name="getSequenceCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceCount</h4>
<pre>public&nbsp;int&nbsp;getSequenceCount()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences for which end gaps are being managed on.</dd></dl>
</li>
</ul>
<a name="getNumberOfColumns()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfColumns</h4>
<pre>public&nbsp;int&nbsp;getNumberOfColumns()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of columns in the alignment (i.e. the <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true#length()" title="class or interface in java.lang"><code>CharSequence.length()</code></a> of all sequences)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.610 (Geneious 6.1.0)</dd></dl>
</li>
</ul>
<a name="getSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&nbsp;getSequence(int&nbsp;index)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>index</code> - the index of the sequence to get in the range [0, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequenceCount()"><code>getSequenceCount()</code></a>)</dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence at this index.</dd></dl>
</li>
</ul>
<a name="getSequences()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getSequences</h4>
<pre>public&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceCharSequence</a>&gt;&nbsp;getSequences()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>all sequences (plus potentially some more if <a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.html#getSequenceCount()"><code>getSequenceCount()</code></a> < than the returned list length) managed by this end gaps manager. The returned value is the stored internal list so any usage of this method must not modify the contents of this list.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/EndGapsManager.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsHelper.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EndGapsManager.Builder.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/EndGapsManager.html" target="_top">Frames</a></li>
<li><a href="EndGapsManager.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
