<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:55 NZDT 2016 -->
<title>Uses of Package com.biomatters.geneious.publicapi.implementations (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Package com.biomatters.geneious.publicapi.implementations (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Uses of Package com.biomatters.geneious.publicapi.implementations" class="title">Uses of Package<br>com.biomatters.geneious.publicapi.implementations</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#com.biomatters.geneious.publicapi.documents.sequence">com.biomatters.geneious.publicapi.documents.sequence</a></td>
<td class="colLast">
<div class="block">Provides interfaces specifying the types of
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>sequence</code></a> and
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>alignment</code></a>
 documents in Geneious, together with concrete classes used by sequence documents
 such as <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotation</code></a>
 and <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceCharSequence.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceCharSequence</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.biomatters.geneious.publicapi.documents.types">com.biomatters.geneious.publicapi.documents.types</a></td>
<td class="colLast">
<div class="block">Provides interfaces specifying the types of documents available in Geneious.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.biomatters.geneious.publicapi.implementations">com.biomatters.geneious.publicapi.implementations</a></td>
<td class="colLast">
<div class="block">Provides implementations for many of the document interfaces defined
 in the package com.biomatters.geneious.publicapi.documents.types
 together with some utility classes for dealing with some of these
 document types.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.biomatters.geneious.publicapi.implementations.sequence">com.biomatters.geneious.publicapi.implementations.sequence</a></td>
<td class="colLast">
<div class="block">Provides <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a>
 implementations for nucelotide, amino acid, and nucleotide graph (chromatogram) sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.biomatters.geneious.publicapi.plugin">com.biomatters.geneious.publicapi.plugin</a></td>
<td class="colLast">
<div class="block">Provides the <a href="../../../../../com/biomatters/geneious/publicapi/plugin/GeneiousPlugin.html" title="class in com.biomatters.geneious.publicapi.plugin"><code>GeneiousPlugin</code></a> interface and plugin
 related interfaces.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.biomatters.geneious.publicapi.utilities">com.biomatters.geneious.publicapi.utilities</a></td>
<td class="colLast">
<div class="block">Provides various utility methods and classes, none of which
 are necessary for creating a plugin, but many plugins implentations
 will find these methods useful.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.biomatters.geneious.publicapi.documents.sequence">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a> used by <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/package-summary.html">com.biomatters.geneious.publicapi.documents.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/DefaultAlignmentDocument.html#com.biomatters.geneious.publicapi.documents.sequence">DefaultAlignmentDocument</a>
<div class="block">An implementation of a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/EndGapsManager.html#com.biomatters.geneious.publicapi.documents.sequence">EndGapsManager</a>
<div class="block">Given a set of aligned sequences, provides methods for quickly finding all sequences
 that intersect a given base number, ignoring those sequences in end gap regions.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/PairedReadManager.html#com.biomatters.geneious.publicapi.documents.sequence">PairedReadManager</a>
<div class="block">Keeps track of which sequences in a list of sequences are paired together and their paired distances.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/SequenceGapInformation.html#com.biomatters.geneious.publicapi.documents.sequence">SequenceGapInformation</a>
<div class="block">Precalculates information about the location of gaps ('-') in a CharSequence,
 and can efficiently calculate translate between indices in the gapped and ungapped
 sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.biomatters.geneious.publicapi.documents.types">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a> used by <a href="../../../../../com/biomatters/geneious/publicapi/documents/types/package-summary.html">com.biomatters.geneious.publicapi.documents.types</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/DefaultAlignmentDocument.html#com.biomatters.geneious.publicapi.documents.types">DefaultAlignmentDocument</a>
<div class="block">An implementation of a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.biomatters.geneious.publicapi.implementations">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a> used by <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/DefaultAlignmentDocument.html#com.biomatters.geneious.publicapi.implementations">DefaultAlignmentDocument</a>
<div class="block">An implementation of a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/DefaultSequenceGraphFactories.SequenceAlignmentScorer.html#com.biomatters.geneious.publicapi.implementations">DefaultSequenceGraphFactories.SequenceAlignmentScorer</a>
<div class="block">Provides methods to supply a score for a given set of sequence residues and provide
 a color for a given score.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/DefaultSequenceGraphFactories.SingleSequenceScorer.html#com.biomatters.geneious.publicapi.implementations">DefaultSequenceGraphFactories.SingleSequenceScorer</a>
<div class="block">Provides methods to supply a score for a given sequence residue and provide a color for a given score.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/EndGapsHelper.html#com.biomatters.geneious.publicapi.implementations">EndGapsHelper</a>
<div class="block">Calculates, and provides a method to determine whether residues in a
 set of sequences are end gaps.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/EndGapsManager.html#com.biomatters.geneious.publicapi.implementations">EndGapsManager</a>
<div class="block">Given a set of aligned sequences, provides methods for quickly finding all sequences
 that intersect a given base number, ignoring those sequences in end gap regions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/EValue.html#com.biomatters.geneious.publicapi.implementations">EValue</a>
<div class="block">Just like a double, but the toString() method displays it as a nicely formatted e value</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/PairedReadManager.html#com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>
<div class="block">Keeps track of which sequences in a list of sequences are paired together and their paired distances.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/PairedReadManager.Orientation.html#com.biomatters.geneious.publicapi.implementations">PairedReadManager.Orientation</a>
<div class="block">Represents the relative orientation for a set of interlaced reads.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/Percentage.html#com.biomatters.geneious.publicapi.implementations">Percentage</a>
<div class="block">Just like a double, but the toString() method displays it as a percentage
 with the specified number of decimal places.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/SequenceExtractionUtilities.ExtractionOptions.html#com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>
<div class="block">Represents options controlling how sequence extraction works in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>SequenceExtractionUtilities.extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/SequenceGapInformation.html#com.biomatters.geneious.publicapi.implementations">SequenceGapInformation</a>
<div class="block">Precalculates information about the location of gaps ('-') in a CharSequence,
 and can efficiently calculate translate between indices in the gapped and ungapped
 sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.biomatters.geneious.publicapi.implementations.sequence">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a> used by <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/package-summary.html">com.biomatters.geneious.publicapi.implementations.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/SequenceGapInformation.html#com.biomatters.geneious.publicapi.implementations.sequence">SequenceGapInformation</a>
<div class="block">Precalculates information about the location of gaps ('-') in a CharSequence,
 and can efficiently calculate translate between indices in the gapped and ungapped
 sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/SequenceGapInformation.Provider.html#com.biomatters.geneious.publicapi.implementations.sequence">SequenceGapInformation.Provider</a>
<div class="block">An interface that <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocuments</code></a> can optionally to implement to indicate they may provide a (potentially) pre-built SequenceGapInformation</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.biomatters.geneious.publicapi.plugin">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a> used by <a href="../../../../../com/biomatters/geneious/publicapi/plugin/package-summary.html">com.biomatters.geneious.publicapi.plugin</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/EndGapsManager.html#com.biomatters.geneious.publicapi.plugin">EndGapsManager</a>
<div class="block">Given a set of aligned sequences, provides methods for quickly finding all sequences
 that intersect a given base number, ignoring those sequences in end gap regions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/PairedReadManager.html#com.biomatters.geneious.publicapi.plugin">PairedReadManager</a>
<div class="block">Keeps track of which sequences in a list of sequences are paired together and their paired distances.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.biomatters.geneious.publicapi.utilities">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/package-summary.html">com.biomatters.geneious.publicapi.implementations</a> used by <a href="../../../../../com/biomatters/geneious/publicapi/utilities/package-summary.html">com.biomatters.geneious.publicapi.utilities</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/biomatters/geneious/publicapi/implementations/class-use/DefaultAlignmentDocument.html#com.biomatters.geneious.publicapi.utilities">DefaultAlignmentDocument</a>
<div class="block">An implementation of a <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAlignmentDocument</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
