<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>PairedReadManager (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="PairedReadManager (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/PairedReadManager.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EValue.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Builder.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" target="_top">Frames</a></li>
<li><a href="PairedReadManager.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations</div>
<h2 title="Class PairedReadManager" class="title">Class PairedReadManager</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.PairedReadManager</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></dd>
</dl>
<hr>
<br>
<pre>public final class <span class="strong">PairedReadManager</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a>, <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></pre>
<div class="block">Keeps track of which sequences in a list of sequences are paired together and their paired distances. For more information,
 see <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a>
 <p/>
 While sequences can be queried individually for their expected distances (using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getMateIndex(int)"><code>getMateIndex(int)</code></a>),
 often if the whole set of sequences
 have a common orientation and distance then <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getInterlacedExpectedDistance()"><code>getInterlacedExpectedDistance()</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getInterlacedOrientation()"><code>getInterlacedOrientation()</code></a> are useful.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>PairedReads</code></a></dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Builder.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager.Builder</a></strong></code>
<div class="block">Allows for the construction of a PairedReadManager where the data resides on disk instead of in memory.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Orientation.html" title="enum in com.biomatters.geneious.publicapi.implementations">PairedReadManager.Orientation</a></strong></code>
<div class="block">Represents the relative orientation for a set of interlaced reads.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(org.jdom.Element)">PairedReadManager</a></strong>(org.jdom.Element&nbsp;element)</code>
<div class="block">Reconstructs a PairedReadManager previously serialized to XML using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toXml(java.lang.String)"><code>toXml(String)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">PairedReadManager</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)</code>
<div class="block">Reconstructs a PairedReadManager previously serialized using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)"><code>toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(int)">PairedReadManager</a></strong>(int&nbsp;numberOfSequences)</code>
<div class="block">Constructs a new paired read manager with this many sequences, none of which are initially paired.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#addSequence()">addSequence</a></strong>()</code>
<div class="block">Adds a sequence to be managed by this paired read manager.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#finalize()">finalize</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></strong>()</code>
<div class="block">Marks this instance as <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()"><code>setCopyOnWrite()</code></a> and returns this.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getInterlacedExpectedDistance()">getInterlacedExpectedDistance</a></strong>()</code>
<div class="block">returns the expected distance (positive) of all reads in this data set if they are all interlaced with a common orientation or 0 if that are not interlaced with a common orientations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Orientation.html" title="enum in com.biomatters.geneious.publicapi.implementations">PairedReadManager.Orientation</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getInterlacedOrientation()">getInterlacedOrientation</a></strong>()</code>
<div class="block">returns the orientation of all reads in this data set if they are all interlaced with a common orientation or null if that are not interlaced with a common orientations.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getMateIndex(int)">getMateIndex</a></strong>(int&nbsp;sequenceIndex)</code>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getNumberOfSequences()">getNumberOfSequences</a></strong>()</code>
<div class="block">Get the number of sequences managed by this paired read manger.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getNumberOfSequencesWithMates()">getNumberOfSequencesWithMates</a></strong>()</code>
<div class="block">Returns the number of sequences which have mates.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</code>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#hasAnyPairedReads()">hasAnyPairedReads</a></strong>()</code>
<div class="block">Query whether there there is at least 1 pair of reads managed by this PairedReadManager.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#hasInterlacedMates()">hasInterlacedMates</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#isInterlacedWithAllReadsHavingSameExpectedDistance()">isInterlacedWithAllReadsHavingSameExpectedDistance</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#removeSequence(int)">removeSequence</a></strong>(int&nbsp;indexToRemove)</code>
<div class="block">Removes a sequence from this PairedReadManager, unpairing it with its mate if necessary.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#reorderSequences(int[])">reorderSequences</a></strong>(int[]&nbsp;ordering)</code>
<div class="block">Reorders the sequences in this paired read manager, as defined by <a href="../../../../../com/biomatters/geneious/publicapi/utilities/GeneralUtilities.html#reorder(int[], int[])"><code>GeneralUtilities.reorder(int[], int[])</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()">setCopyOnWrite</a></strong>()</code>
<div class="block">Flags this PairedReadManager to be duplicated when it is modified to using one of the methods <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setMates(int, int, int, boolean)"><code>setMates(int, int, int, boolean)</code></a> ,
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#removeSequence(int)"><code>removeSequence(int)</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#addSequence()"><code>addSequence()</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#reorderSequences(int[])"><code>reorderSequences(int[])</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setInterlaced(int, com.biomatters.geneious.publicapi.implementations.PairedReadManager.Orientation)">setInterlaced</a></strong>(int&nbsp;expectedDistance,
             <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Orientation.html" title="enum in com.biomatters.geneious.publicapi.implementations">PairedReadManager.Orientation</a>&nbsp;orientation)</code>
<div class="block">Marks all the reads in the data set as being interlaced with the given orientation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setMates(int, int, int)">setMates</a></strong>(int&nbsp;sequenceIndex,
        int&nbsp;mateIndex,
        int&nbsp;expectedMateDistance)</code>
<div class="block">Records the 2nd sequence as being paired with the first sequence with the given distance.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toOutputStream(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)">toOutputStream</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
              <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;objectOutputStream)</code>
<div class="block">Serializes this PairedReadManager to an output stream so that it can be reconstructed later using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)">toOutputStream</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;objectOutputStream)</code>
<div class="block">Serializes this PairedReadManager to an output stream so that it can be reconstructed later using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;majorVersion,
     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</code>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.jdom.Element</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toXml(java.lang.String)">toXml</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName)</code>
<div class="block">Serializes this PairedReadManager to XML so that it can be reconstructed later using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(org.jdom.Element)"><code>PairedReadManager(org.jdom.Element)</code></a></div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="PairedReadManager(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PairedReadManager</h4>
<pre>public&nbsp;PairedReadManager(int&nbsp;numberOfSequences)</pre>
<div class="block">Constructs a new paired read manager with this many sequences, none of which are initially paired.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numberOfSequences</code> - the number of potentially paired reads to manage.</dd></dl>
</li>
</ul>
<a name="PairedReadManager(org.jdom.Element)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PairedReadManager</h4>
<pre>public&nbsp;PairedReadManager(org.jdom.Element&nbsp;element)</pre>
<div class="block">Reconstructs a PairedReadManager previously serialized to XML using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toXml(java.lang.String)"><code>toXml(String)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>element</code> - an element previously returned from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toXml(java.lang.String)"><code>toXml(String)</code></a>.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toXml(java.lang.String)"><code>toXml(String)</code></a></dd></dl>
</li>
</ul>
<a name="PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>PairedReadManager</h4>
<pre>public&nbsp;PairedReadManager(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectInputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectInputStream</a>&nbsp;in)
                  throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Reconstructs a PairedReadManager previously serialized using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)"><code>toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>in</code> - the object stream to read from</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if anything goes wrong.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)"><code>toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)</code></a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="setInterlaced(int, com.biomatters.geneious.publicapi.implementations.PairedReadManager.Orientation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setInterlaced</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;setInterlaced(int&nbsp;expectedDistance,
                              <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Orientation.html" title="enum in com.biomatters.geneious.publicapi.implementations">PairedReadManager.Orientation</a>&nbsp;orientation)</pre>
<div class="block">Marks all the reads in the data set as being interlaced with the given orientation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>expectedDistance</code> - the expected distance between the outer ends of the reads. Must be positive.</dd><dd><code>orientation</code> - the relative orientation of the reads</dd>
<dt><span class="strong">Returns:</span></dt><dd>either this <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>PairedReadManager</code></a> or if this PairedReadManager has had <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()"><code>setCopyOnWrite()</code></a> called on it, a copy of it with the new mate set. The copy will not be marked as copyOnWrite.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="getInterlacedOrientation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInterlacedOrientation</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Orientation.html" title="enum in com.biomatters.geneious.publicapi.implementations">PairedReadManager.Orientation</a>&nbsp;getInterlacedOrientation()</pre>
<div class="block">returns the orientation of all reads in this data set if they are all interlaced with a common orientation or null if that are not interlaced with a common orientations.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the orientation of all reads in this data set if they are all interlaced with a common orientation or null if that are not interlaced with a common orientations.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getInterlacedExpectedDistance()"><code>getInterlacedExpectedDistance()</code></a></dd></dl>
</li>
</ul>
<a name="getInterlacedExpectedDistance()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInterlacedExpectedDistance</h4>
<pre>public&nbsp;int&nbsp;getInterlacedExpectedDistance()</pre>
<div class="block">returns the expected distance (positive) of all reads in this data set if they are all interlaced with a common orientation or 0 if that are not interlaced with a common orientations.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the expected distance (positive) of all reads in this data set if they are all interlaced with a common orientation or 0 if that are not interlaced with a common orientations.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getInterlacedOrientation()"><code>getInterlacedOrientation()</code></a></dd></dl>
</li>
</ul>
<a name="setMates(int, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMates</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;setMates(int&nbsp;sequenceIndex,
                         int&nbsp;mateIndex,
                         int&nbsp;expectedMateDistance)</pre>
<div class="block">Records the 2nd sequence as being paired with the first sequence with the given distance.
 <p/>
 If the mate sequence has not already been paired with this sequence, then the mate's expectedMateDistance will be set to -expectedMateDistance</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the sequence to set the mate for.</dd><dd><code>mateIndex</code> - the index of its mate, or -1 if its mate is not in this contig.</dd><dd><code>expectedMateDistance</code> - the expected mate distance. See <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a> for a definition of what this means
 expected distance can be correctly set for each of them since both values should be either positive or negative in this case. (see <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getMateExpectedDistance(int)"><code>getMateExpectedDistance(int)</code></a>)</dd>
<dt><span class="strong">Returns:</span></dt><dd>either this <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>PairedReadManager</code></a> or if this PairedReadManager has had <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()"><code>setCopyOnWrite()</code></a> called on it, a copy of it with the new mate set. The copy will not be marked as copyOnWrite.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if this sequence already has a different mate or if its mate is paired to a different sequence.</dd></dl>
</li>
</ul>
<a name="getMateExpectedDistance(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateExpectedDistance</h4>
<pre>public&nbsp;int&nbsp;getMateExpectedDistance(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the expected mate distance (also known as predicted median insert size) of this nucleotide sequence from its mate. This is defined
 as the length of the 2 reads plus the distance between them.
 For example if the 2 reads both have length 30 with a 40 bp gap between them, then the
 expected mate distance is 100.
 If the mate is expected to appear after this sequence with this sequence in its current orientation, then
 the expected distance will be positive. If the mate is expected to appear before this sequence with this sequence in its current orientation, then
 the expected distance will be negative.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)">getMateExpectedDistance</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence for which to get the mate index of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the expected mate distance or 0 if this read is not paired. Note that a read may still have a mate distance!=0 but its mate may not be present in this document (in which case <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a> will be -1.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)"><code>PairedReads.getMateIndex(int)</code></a></dd></dl>
</li>
</ul>
<a name="getMateIndex(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMateIndex</h4>
<pre>public&nbsp;int&nbsp;getMateIndex(int&nbsp;sequenceIndex)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">PairedReads</a></code></strong></div>
<div class="block">Gets the index of the mate (if any) of the nucleotide sequence at this index.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateIndex(int)">getMateIndex</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>sequenceIndex</code> - the index of the nucleotide sequence to get the mate index of</dd>
<dt><span class="strong">Returns:</span></dt><dd>an index of the mate of the sequence, or -1 if the sequence has no mate.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getMateExpectedDistance(int)"><code>PairedReads.getMateExpectedDistance(int)</code></a></dd></dl>
</li>
</ul>
<a name="hasInterlacedMates()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasInterlacedMates</h4>
<pre>public&nbsp;boolean&nbsp;hasInterlacedMates()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if either there are no paired reads or if all of the paired reads are interlaced (i.e 0 and 1 are paired, 2 and 3 are paired)</dd></dl>
</li>
</ul>
<a name="isInterlacedWithAllReadsHavingSameExpectedDistance()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isInterlacedWithAllReadsHavingSameExpectedDistance</h4>
<pre>public&nbsp;boolean&nbsp;isInterlacedWithAllReadsHavingSameExpectedDistance()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if all reads have the same mate distance or if none of the reads are paired. This means we are using only a very small amount of memory. This method is intended only for testing purposes.</dd></dl>
</li>
</ul>
<a name="reorderSequences(int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderSequences</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;reorderSequences(int[]&nbsp;ordering)</pre>
<div class="block">Reorders the sequences in this paired read manager, as defined by <a href="../../../../../com/biomatters/geneious/publicapi/utilities/GeneralUtilities.html#reorder(int[], int[])"><code>GeneralUtilities.reorder(int[], int[])</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>ordering</code> - the reodering array as specified by <a href="../../../../../com/biomatters/geneious/publicapi/utilities/GeneralUtilities.html#reorder(int[], int[])"><code>GeneralUtilities.reorder(int[], int[])</code></a>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>either this <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>PairedReadManager</code></a> or if this PairedReadManager has had <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()"><code>setCopyOnWrite()</code></a> called on it, a copy of it with the sequences reordered. The copy will not be marked as copyOnWrite.</dd></dl>
</li>
</ul>
<a name="addSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addSequence</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;addSequence()</pre>
<div class="block">Adds a sequence to be managed by this paired read manager. The new sequence is initially unpaired.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>either this <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>PairedReadManager</code></a> or if this PairedReadManager has had <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()"><code>setCopyOnWrite()</code></a> called on it, a copy of it with the sequence added. The copy will not be marked as copyOnWrite.</dd></dl>
</li>
</ul>
<a name="getNumberOfSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfSequences</h4>
<pre>public&nbsp;int&nbsp;getNumberOfSequences()</pre>
<div class="block">Get the number of sequences managed by this paired read manger. This is equal to the value passed to the constructor (<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(int)"><code>PairedReadManager(int)</code></a>) adjusted by any changes
 made using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#addSequence()"><code>addSequence()</code></a> or <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#removeSequence(int)"><code>removeSequence(int)</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences managed by this paired read manger.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
<a name="toOutputStream(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toOutputStream</h4>
<pre>public&nbsp;void&nbsp;toOutputStream(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;objectOutputStream)
                    throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Serializes this PairedReadManager to an output stream so that it can be reconstructed later using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>objectOutputStream</code> - the stream to write to</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if an error is encountered when writing to the stream</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)</code></a></dd></dl>
</li>
</ul>
<a name="toOutputStream(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectOutputStream)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toOutputStream</h4>
<pre>public&nbsp;void&nbsp;toOutputStream(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;version,
                  <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/GeneiousObjectOutputStream.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">GeneiousObjectOutputStream</a>&nbsp;objectOutputStream)
                    throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Serializes this PairedReadManager to an output stream so that it can be reconstructed later using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>version</code> - the version of Geneious to ensure the data is compatible with</dd><dd><code>objectOutputStream</code> - the stream to write to</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if an error is encountered when writing to the stream</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)"><code>PairedReadManager(com.biomatters.geneious.publicapi.documents.sequence.GeneiousObjectInputStream)</code></a></dd></dl>
</li>
</ul>
<a name="getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVersionSupport</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;getVersionSupport(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html" title="enum in com.biomatters.geneious.publicapi.documents">XMLSerializable.VersionSupportType</a>&nbsp;versionType)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Gets either the latest version this class can serialize to via <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a>
 or the most recent version in which the XML returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)"><code>XMLSerializable.OldVersionCompatible.toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)</code></a> last changed.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)">getVersionSupport</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>versionType</code> - specifies which version property to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>the major version of Geneious corresponding to the versionType for this classes toXML method.</dd></dl>
</li>
</ul>
<a name="toXml(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXml</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXml(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;elementName)</pre>
<div class="block">Serializes this PairedReadManager to XML so that it can be reconstructed later using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#PairedReadManager(org.jdom.Element)"><code>PairedReadManager(org.jdom.Element)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>elementName</code> - the name of root element to return</dd>
<dt><span class="strong">Returns:</span></dt><dd>some XML representing this PairedReadManager</dd></dl>
</li>
</ul>
<a name="toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXML</h4>
<pre>public&nbsp;org.jdom.Element&nbsp;toXML(<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.MajorVersion.html" title="enum in com.biomatters.geneious.publicapi.plugin">Geneious.MajorVersion</a>&nbsp;majorVersion,
                     <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true" title="class or interface in jebl.util">ProgressListener</a>&nbsp;progressListener)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">XMLSerializable.OldVersionCompatible</a></code></strong></div>
<div class="block">Serializes this class to XML format, potentially to a format readable by an earlier version of Geneious. It is acceptable for the XML to include unnecessary tags that will be ignored by the earlier version. For
 example if the implementation has only extended the XML since the earlier version, then the XML returned may be identical to the XML returned for the current version.<p/>
 See <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> for a more
 detailed description of what it means to serialize to XML.
 <p/>
 All classes that implement this method must also implement <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.html#toXML()"><code>XMLSerializable.toXML()</code></a> and should delegate back to
 this method using <a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a> and <a href="http://jebl.sourceforge.net/doc/api/jebl/util/ProgressListener.html?is-external=true#EMPTY" title="class or interface in jebl.util"><code>ProgressListener.EMPTY</code></a> as parameters.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#toXML(com.biomatters.geneious.publicapi.plugin.Geneious.MajorVersion, jebl.util.ProgressListener)">toXML</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html" title="interface in com.biomatters.geneious.publicapi.documents">XMLSerializable.OldVersionCompatible</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>majorVersion</code> - the major version of Geneious to serialize to XML for. For example "6.0" but not "6.0.0". This must be a version
 returned greater or equal to a version returned from
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.OldVersionCompatible.html#getVersionSupport(com.biomatters.geneious.publicapi.documents.XMLSerializable.VersionSupportType)"><code>getVersionSupport</code></a>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializable.VersionSupportType.html#OldestVersionSerializableTo"><code>VersionSupportType.OldestVersionSerializableTo</code></a>)
 and must never be greater than the current version (<a href="../../../../../com/biomatters/geneious/publicapi/plugin/Geneious.html#getMajorVersion()"><code>Geneious.getMajorVersion()</code></a>)</dd><dd><code>progressListener</code> - for reporting progress and cancelling</dd>
<dt><span class="strong">Returns:</span></dt><dd>object encoded as a JDOM element</dd></dl>
</li>
</ul>
<a name="removeSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeSequence</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;removeSequence(int&nbsp;indexToRemove)</pre>
<div class="block">Removes a sequence from this PairedReadManager, unpairing it with its mate if necessary. If the sequence to be removed is not the last sequence,
 then all other reads will need to be shuffled down which is inefficient when removing many sequences. Instead in this case, the sequences
 should be reordered first using <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#reorderSequences(int[])"><code>reorderSequences(int[])</code></a> and then the final sequences deleted in reverse order.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>indexToRemove</code> - the 0-based index of the sequence to remove.</dd>
<dt><span class="strong">Returns:</span></dt><dd>either this <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations"><code>PairedReadManager</code></a> or if this PairedReadManager has had <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()"><code>setCopyOnWrite()</code></a> called on it, a copy of it with the sequence removed. The copy will not be marked as copyOnWrite.</dd></dl>
</li>
</ul>
<a name="hasAnyPairedReads()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasAnyPairedReads</h4>
<pre>public&nbsp;boolean&nbsp;hasAnyPairedReads()</pre>
<div class="block">Query whether there there is at least 1 pair of reads managed by this PairedReadManager.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if at there is at least 1 pair of reads managed by this PairedReadManager.</dd></dl>
</li>
</ul>
<a name="setCopyOnWrite()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCopyOnWrite</h4>
<pre>public&nbsp;void&nbsp;setCopyOnWrite()</pre>
<div class="block">Flags this PairedReadManager to be duplicated when it is modified to using one of the methods <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setMates(int, int, int, boolean)"><code>setMates(int, int, int, boolean)</code></a> ,
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#removeSequence(int)"><code>removeSequence(int)</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#addSequence()"><code>addSequence()</code></a>, <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#reorderSequences(int[])"><code>reorderSequences(int[])</code></a>. When it has been marked as copy
 on write, those methods will return a copy of the PairedReadManager that includes the change. The copy will not be
 marked as copyOnWrite.</div>
</li>
</ul>
<a name="getImmutablePairedReadManager()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getImmutablePairedReadManager</h4>
<pre>public&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" title="class in com.biomatters.geneious.publicapi.implementations">PairedReadManager</a>&nbsp;getImmutablePairedReadManager()</pre>
<div class="block">Marks this instance as <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#setCopyOnWrite()"><code>setCopyOnWrite()</code></a> and returns this.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html#getImmutablePairedReadManager()">getImmutablePairedReadManager</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/PairedReads.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">PairedReads</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>this</dd></dl>
</li>
</ul>
<a name="finalize()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>finalize</h4>
<pre>protected&nbsp;void&nbsp;finalize()
                 throws <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Throwable.html?is-external=true" title="class or interface in java.lang">Throwable</a></pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a></code>&nbsp;in class&nbsp;<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Throwable.html?is-external=true" title="class or interface in java.lang">Throwable</a></code></dd></dl>
</li>
</ul>
<a name="getNumberOfSequencesWithMates()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getNumberOfSequencesWithMates</h4>
<pre>public&nbsp;int&nbsp;getNumberOfSequencesWithMates()</pre>
<div class="block">Returns the number of sequences which have mates. If all sequences are paired, this is equal to <a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.html#getNumberOfSequences()"><code>getNumberOfSequences()</code></a></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of sequences which have mates.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.40 (Geneious 5.4.0)</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/PairedReadManager.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/EValue.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/PairedReadManager.Builder.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/PairedReadManager.html" target="_top">Frames</a></li>
<li><a href="PairedReadManager.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
