<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Feb 29 17:24:46 NZDT 2016 -->
<title>SequenceExtractionUtilities (Geneious Public API)</title>
<meta name="date" content="2016-02-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceExtractionUtilities (Geneious Public API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceExtractionUtilities.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/Percentage.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html" target="_top">Frames</a></li>
<li><a href="SequenceExtractionUtilities.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.biomatters.geneious.publicapi.implementations</div>
<h2 title="Class SequenceExtractionUtilities" class="title">Class SequenceExtractionUtilities</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">SequenceExtractionUtilities</span>
extends <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Provides methods for extracting a portion of a sequence to a new sequence document. Also provides methods
 for reverse complementing and translation.
 <p/>
 Although this class will accept any <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceDocument</code></a> as input,
 the extracted region may not include all information that was present on some SequenceDocument sub-classes. For example
 <a href="../../../../../com/biomatters/geneious/publicapi/documents/types/MolecularStructureDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.types"><code>MolecularStructureDocument</code></a> information will not be included in extracted regions. The following information
 will be included in extracted regions:
 <ul>
     <li>Nucleotides / amino acids returned from <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html#getCharSequence()"><code>SequenceDocument.getCharSequence()</code></a></li>
     <li><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence"><code>SequenceAnnotations</code></a></li>
     <li><a href="../../../../../com/biomatters/geneious/publicapi/documents/DocumentField.html" title="class in com.biomatters.geneious.publicapi.documents"><code>DocumentFields</code></a></li>
     <li><a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideGraph.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideGraphs</code></a></li>
 </ul></div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html" title="class in com.biomatters.geneious.publicapi.utilities"><code>SequenceUtilities</code></a></dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a></strong></code>
<div class="block">Represents options controlling how sequence extraction works in <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a>.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX">REVERSED_NAME_SUFFIX</a></strong></code>
<div class="block">The suffix that is appended to sequence names when they are reversed.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGapRemoval(java.lang.CharSequence, java.util.List)">adjustAnnotationsForGapRemoval</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceWithGaps,
                              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for removing all the gaps in the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGapRemoval(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.List)">adjustAnnotationsForGapRemoval</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceWithGaps,
                              <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for removing all the gaps in the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGaps(java.lang.CharSequence, java.util.List)">adjustAnnotationsForGaps</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceWithGaps,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for adding gaps into the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.List)">adjustAnnotationsForGaps</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceWithGaps,
                        <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</code>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for adding gaps into the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#copyAllDocumentFields(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument)">copyAllDocumentFields</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;source,
                     <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;target)</code>
<div class="block">Copy all document fields (such as organism, taxonomy and common name) from any SequenceDocument to a DefaultSequenceDocument.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#copyDocumentFields(com.biomatters.geneious.publicapi.documents.PluginDocument, com.biomatters.geneious.publicapi.documents.AbstractPluginDocument, boolean)">copyDocumentFields</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>&nbsp;source,
                  <a href="../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a>&nbsp;target,
                  boolean&nbsp;isTranslation)</code>
<div class="block">Copy important document fields (such as organism, taxonomy and common name) from any PluginDocument to any AbstractPluginDocument</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#copyDocumentFields(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument, boolean)">copyDocumentFields</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;source,
                  <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;target,
                  boolean&nbsp;isTranslation)</code>
<div class="block">Copy important document fields (such as organism, taxonomy and common name) from any SequenceDocument to any DefaultSequenceDocument</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extract(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">extract</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
       <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)</code>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.utilities.Interval...)">extractIntervals</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>...&nbsp;intervalsToExtract)</code>
<div class="block">A convenience method for extracting intervals from a sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval...)">extractIntervals</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>...&nbsp;intervalsToExtract)</code>
<div class="block">A convenience method for extracting intervals from a sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">extractIntervals</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)</code>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, jebl.evolution.sequences.GeneticCode, int, boolean, java.util.List, boolean, boolean)">extractIntervals</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;overrideName,
                <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a>&nbsp;geneticCode,
                int&nbsp;translationFrame,
                boolean&nbsp;residuesOnly,
                <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;intervals,
                boolean&nbsp;allowReverseComplement,
                boolean&nbsp;removeGaps)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a> instead.</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervalsOrThrow(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">extractIntervalsOrThrow</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                       <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)</code>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractOrThrow(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">extractOrThrow</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
              <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)</code>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#getComplementWithoutReverseSequenceName(java.lang.String)">getComplementWithoutReverseSequenceName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName)</code>
<div class="block">Appends the " (complement without reverse)" suffix to the sequence name (or removes if it is already there).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#getExtractionName(java.lang.String, java.lang.String, java.lang.String, boolean)">getExtractionName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;extractionType,
                 <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;annotationName,
                 <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName,
                 boolean&nbsp;appendExtractionType)</code>
<div class="block">Create a name for an extracted region</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#getGeneticCode(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">getGeneticCode</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#getReversedSequenceName(java.lang.String)">getReversedSequenceName</a></strong>(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName)</code>
<div class="block">Appends the " (reversed)" suffix to the sequence name (or removes if it is already there).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#isReversedSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">isReversedSequence</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Determines if a SequenceDocument is reversed or not based on whether its name ends with <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>REVERSED_NAME_SUFFIX</code></a> or not.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">removeGaps</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</code>
<div class="block">Equivalent to calling <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)"><code>removeGaps(sequence, true)</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">removeGaps</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
          boolean&nbsp;mustGeneratedDefaultSequenceDocument)</code>
<div class="block">creates a sequence the same as the given sequence but with gap characters removed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument)">reverseComplement</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignmentToReverse)</code>
<div class="block">Creates a new alignment that is the reverse complement of the given alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">reverseComplement</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceToReverse)</code>
<div class="block">Convenience method for calling <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)"><code>reverseComplement(sequenceToReverse,true)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">reverseComplement</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceToReverse,
                 boolean&nbsp;appendReversedToName)</code>
<div class="block">Create a sequence that is the reverse complement another sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultNucleotideSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean, boolean)">reverseComplement</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceToReverse,
                 boolean&nbsp;appendReversedToName,
                 boolean&nbsp;includeOriginalBaseNumbersAnnotation)</code>
<div class="block">Create a sequence that is the reverse complement another sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#translate(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">translate</a></strong>(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
         <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a> instead.</i></div>
</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="REVERSED_NAME_SUFFIX">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>REVERSED_NAME_SUFFIX</h4>
<pre>public static final&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> REVERSED_NAME_SUFFIX</pre>
<div class="block">The suffix that is appended to sequence names when they are reversed. As of Geneious 3.5 and later, this is " (reversed)"</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#getReversedSequenceName(java.lang.String)"><code>getReversedSequenceName(String)</code></a>, 
<a href="../../../../../constant-values.html#com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.REVERSED_NAME_SUFFIX">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getReversedSequenceName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReversedSequenceName</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getReversedSequenceName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName)</pre>
<div class="block">Appends the " (reversed)" suffix to the sequence name (or removes if it is already there).
 For example:<br>
  <code>getReversedSequenceName("test") returns "test (reversed)"</code><br>
  <code>getReversedSequenceName("test (reversed)") returns "test"</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceName</code> - the sequence index to change the name of.</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>sequenceName</code> with " (reversed)" appended (or removed if it was already there)</dd></dl>
</li>
</ul>
<a name="getComplementWithoutReverseSequenceName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComplementWithoutReverseSequenceName</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getComplementWithoutReverseSequenceName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName)</pre>
<div class="block">Appends the " (complement without reverse)" suffix to the sequence name (or removes if it is already there). Also
 takes into account existing " (reversed") suffix and uses " (reverse without complement)" instead.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceName</code> - the sequence index to change the name of.</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>sequenceName</code> with " (complement without reverse)" appended (or removed if it was already there)</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.600 (Geneious 6.0.0)</dd></dl>
</li>
</ul>
<a name="isReversedSequence(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReversedSequence</h4>
<pre>public static&nbsp;boolean&nbsp;isReversedSequence(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Determines if a SequenceDocument is reversed or not based on whether its name ends with <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>REVERSED_NAME_SUFFIX</code></a> or not.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to test.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if <code>sequence</code> is reversed, false otherwise</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.30 (Geneious 5.3.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#REVERSED_NAME_SUFFIX"><code>REVERSED_NAME_SUFFIX</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#getReversedSequenceName(java.lang.String)"><code>getReversedSequenceName(String)</code></a></dd></dl>
</li>
</ul>
<a name="translate(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>translate</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;translate(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                          <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a> instead.</i></div>
<div class="block">if the supplied annotation specifies a genetic code, then the interval specified
 by this annotation will be translated. If not, this function returns null</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - </dd><dd><code>annotation</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>the translated sequence string</dd></dl>
</li>
</ul>
<a name="removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeGaps</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;removeGaps(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence)</pre>
<div class="block">Equivalent to calling <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)"><code>removeGaps(sequence, true)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to remove gaps from</dd>
<dt><span class="strong">Returns:</span></dt><dd>a sequence without gaps which will always been an instance of <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultSequenceDocument</code></a></dd></dl>
</li>
</ul>
<a name="removeGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeGaps</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;removeGaps(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                          boolean&nbsp;mustGeneratedDefaultSequenceDocument)</pre>
<div class="block">creates a sequence the same as the given sequence but with gap characters removed.
 All internal information such as annotations are adjusted accordingly.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to remove gaps from</dd><dd><code>mustGeneratedDefaultSequenceDocument</code> - true to ensure the returned sequence will be an instance of a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultSequenceDocument</code></a>. If this is false, the returned sequence may be any SequenceDocument implementation.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a sequence without gaps</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/utilities/SequenceUtilities.html#createSequenceCopyAdjustedForGapInsertion(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.CharSequence)"><code>SequenceUtilities.createSequenceCopyAdjustedForGapInsertion(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, CharSequence)</code></a></dd></dl>
</li>
</ul>
<a name="extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotationInterval...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractIntervals</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;extractIntervals(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                                       <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>...&nbsp;intervalsToExtract)</pre>
<div class="block">A convenience method for extracting intervals from a sequence.

 This method is equivalent to calling
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, jebl.evolution.sequences.GeneticCode, int, boolean, java.util.List, boolean, boolean)"><code>extractIntervals(sequence,sequence,null,null,0,false, Arrays.asList(intervalsToExtract),true,false)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to extract intervals from</dd><dd><code>intervalsToExtract</code> - the intervals to extract.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the extracted intervals concatenated together as a new sequence.</dd></dl>
</li>
</ul>
<a name="extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.utilities.Interval...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractIntervals</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;extractIntervals(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequence,
                                       <a href="../../../../../com/biomatters/geneious/publicapi/utilities/Interval.html" title="class in com.biomatters.geneious.publicapi.utilities">Interval</a>...&nbsp;intervalsToExtract)</pre>
<div class="block">A convenience method for extracting intervals from a sequence.

 This method is equivalent to calling
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, jebl.evolution.sequences.GeneticCode, int, boolean, java.util.List, boolean, boolean)"><code>extractIntervals(sequence,sequence,null,null,0,false, Arrays.asList(intervalsToExtract),true,false)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - the sequence to extract intervals from</dd><dd><code>intervalsToExtract</code> - the intervals to extract.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the extracted intervals concatenated together as a new sequence.</dd></dl>
</li>
</ul>
<a name="adjustAnnotationsForGapRemoval(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>adjustAnnotationsForGapRemoval</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;adjustAnnotationsForGapRemoval(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceWithGaps,
                                                      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for removing all the gaps in the sequence. Opposite of
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGaps(java.lang.CharSequence, java.util.List)"><code>adjustAnnotationsForGaps(CharSequence, java.util.List)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceWithGaps</code> - sequence that may include gap characters</dd><dd><code>annotations</code> - the original list of sequence annotations</dd>
<dt><span class="strong">Returns:</span></dt><dd>the adjusted sequence annotations</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.List)"><code>adjustAnnotationsForGaps(SequenceDocument, java.util.List)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html#adjustAnnotationsForGapRemoval(java.util.List)"><code>SequenceGapInformation.adjustAnnotationsForGapRemoval(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="adjustAnnotationsForGapRemoval(java.lang.CharSequence, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>adjustAnnotationsForGapRemoval</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;adjustAnnotationsForGapRemoval(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceWithGaps,
                                                      <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for removing all the gaps in the sequence. Opposite of
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGaps(java.lang.CharSequence, java.util.List)"><code>adjustAnnotationsForGaps(CharSequence, java.util.List)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceWithGaps</code> - sequence that may include gap characters</dd><dd><code>annotations</code> - the original list of sequence annotations</dd>
<dt><span class="strong">Returns:</span></dt><dd>the adjusted sequence annotations</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGaps(java.lang.CharSequence, java.util.List)"><code>adjustAnnotationsForGaps(CharSequence, java.util.List)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html#adjustAnnotationsForGapRemoval(java.util.List)"><code>SequenceGapInformation.adjustAnnotationsForGapRemoval(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="adjustAnnotationsForGaps(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>adjustAnnotationsForGaps</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;adjustAnnotationsForGaps(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceWithGaps,
                                                <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for adding gaps into the sequence. Opposite of
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGapRemoval(java.lang.CharSequence, java.util.List)"><code>adjustAnnotationsForGapRemoval(CharSequence, java.util.List)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceWithGaps</code> - sequence with gaps</dd><dd><code>annotations</code> - annotations against the sequence string with gaps removed</dd>
<dt><span class="strong">Returns:</span></dt><dd>annotations against the sequence string with gaps included</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd>
<dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGapRemoval(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.util.List)"><code>adjustAnnotationsForGapRemoval(SequenceDocument, java.util.List)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html#adjustAnnotationsForGapInsertion(java.util.List)"><code>SequenceGapInformation.adjustAnnotationsForGapInsertion(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="adjustAnnotationsForGaps(java.lang.CharSequence, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>adjustAnnotationsForGaps</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;adjustAnnotationsForGaps(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/CharSequence.html?is-external=true" title="class or interface in java.lang">CharSequence</a>&nbsp;sequenceWithGaps,
                                                <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&gt;&nbsp;annotations)</pre>
<div class="block">Creates a new list of annotations by adjusting their locations
 to compensate for adding gaps into the sequence. Opposite of
 <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGapRemoval(java.lang.CharSequence, java.util.List)"><code>adjustAnnotationsForGapRemoval(CharSequence, java.util.List)</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceWithGaps</code> - sequence string with gaps</dd><dd><code>annotations</code> - annotations against the sequence string with gaps removed</dd>
<dt><span class="strong">Returns:</span></dt><dd>annotations against the sequence string with gaps included</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#adjustAnnotationsForGapRemoval(java.lang.CharSequence, java.util.List)"><code>adjustAnnotationsForGapRemoval(CharSequence, java.util.List)</code></a>, 
<a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceGapInformation.html#adjustAnnotationsForGapInsertion(java.util.List)"><code>SequenceGapInformation.adjustAnnotationsForGapInsertion(java.util.List)</code></a></dd></dl>
</li>
</ul>
<a name="reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplement</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;reverseComplement(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceToReverse)</pre>
<div class="block">Convenience method for calling <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)"><code>reverseComplement(sequenceToReverse,true)</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceToReverse</code> - the sequence to create the reverse complement of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the reverse complement of  <code>sequenceToReverse</code></dd></dl>
</li>
</ul>
<a name="reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceAlignmentDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplement</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;reverseComplement(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAlignmentDocument.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAlignmentDocument</a>&nbsp;alignmentToReverse)</pre>
<div class="block">Creates a new alignment that is the reverse complement of the given alignment.
 Any documents referenced by the original alignment will also be referenced by the returned alignment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignmentToReverse</code> - the alignment to create the reverse complement of.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the reverse complement of <code>alignmentToReverse</code></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if any of the sequences in <code>alignmentToReverse</code> are not <a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/NucleotideSequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence"><code>NucleotideSequenceDocuments</code></a>.</dd></dl>
</li>
</ul>
<a name="reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplement</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;reverseComplement(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceToReverse,
                                        boolean&nbsp;appendReversedToName)</pre>
<div class="block">Create a sequence that is the reverse complement another sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceToReverse</code> - the sequence to create the reverse complement of.</dd><dd><code>appendReversedToName</code> - true to append " (reversed)" to the name of <code>sequenceToReverse</code> to form the name of the returned sequence. false to give the returned sequence the same name as <code>sequenceToReverse</code>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the reverse complement of  <code>sequenceToReverse</code></dd></dl>
</li>
</ul>
<a name="reverseComplement(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplement</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultNucleotideSequence.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultNucleotideSequence</a>&nbsp;reverseComplement(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sequenceToReverse,
                                          boolean&nbsp;appendReversedToName,
                                          boolean&nbsp;includeOriginalBaseNumbersAnnotation)</pre>
<div class="block">Create a sequence that is the reverse complement another sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceToReverse</code> - the sequence to create the reverse complement of.</dd><dd><code>appendReversedToName</code> - true to append " (reversed)" to the name of <code>sequenceToReverse</code> to form the name of the returned sequence. false to give the returned sequence the same name as <code>sequenceToReverse</code>.</dd><dd><code>includeOriginalBaseNumbersAnnotation</code> - true to include an annotation (which is invisible to end users) indicating the original numbering of this sequence. false to exclude it. Usually this should be true, except in cases where we are dealing with enormous amounts of data (e.g. next gen sequencing) in which case we want to exclude it for performance reasons.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the reverse complement of  <code>sequenceToReverse</code></dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.11 (Geneious 5.0)</dd></dl>
</li>
</ul>
<a name="getGeneticCode(com.biomatters.geneious.publicapi.documents.sequence.SequenceAnnotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneticCode</h4>
<pre>public static&nbsp;<a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a>&nbsp;getGeneticCode(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotation.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotation</a>&nbsp;annotation)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotation</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>the genetic code specified by this annotation, or null if it does not specify one</dd></dl>
</li>
</ul>
<a name="copyAllDocumentFields(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>copyAllDocumentFields</h4>
<pre>public static&nbsp;void&nbsp;copyAllDocumentFields(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;source,
                         <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;target)</pre>
<div class="block">Copy all document fields (such as organism, taxonomy and common name) from any SequenceDocument to a DefaultSequenceDocument.
 This should only be called on documents that are identical apart from gaps since some fields (for example accession) would
 be incorrect on a modified/partial instance of the document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>source</code> - to copy document fields from</dd><dd><code>target</code> - to copy document fields to</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.714 (Geneious 7.1.4)</dd></dl>
</li>
</ul>
<a name="copyDocumentFields(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.sequence.DefaultSequenceDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>copyDocumentFields</h4>
<pre>public static&nbsp;void&nbsp;copyDocumentFields(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;source,
                      <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;target,
                      boolean&nbsp;isTranslation)</pre>
<div class="block">Copy important document fields (such as organism, taxonomy and common name) from any SequenceDocument to any DefaultSequenceDocument</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>source</code> - to copy document fields from</dd><dd><code>target</code> - to copy document fields to</dd><dd><code>isTranslation</code> - whether the target is a translation of the source. If it is not a translation, the molecule type field will be copied.</dd></dl>
</li>
</ul>
<a name="copyDocumentFields(com.biomatters.geneious.publicapi.documents.PluginDocument, com.biomatters.geneious.publicapi.documents.AbstractPluginDocument, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>copyDocumentFields</h4>
<pre>public static&nbsp;void&nbsp;copyDocumentFields(<a href="../../../../../com/biomatters/geneious/publicapi/documents/PluginDocument.html" title="interface in com.biomatters.geneious.publicapi.documents">PluginDocument</a>&nbsp;source,
                      <a href="../../../../../com/biomatters/geneious/publicapi/documents/AbstractPluginDocument.html" title="class in com.biomatters.geneious.publicapi.documents">AbstractPluginDocument</a>&nbsp;target,
                      boolean&nbsp;isTranslation)</pre>
<div class="block">Copy important document fields (such as organism, taxonomy and common name) from any PluginDocument to any AbstractPluginDocument</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>source</code> - to copy document fields from</dd><dd><code>target</code> - to copy document fields to</dd><dd><code>isTranslation</code> - whether the target is a translation of the source. If it is not a translation, the molecule type field will be copied.</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.900 (Geneious 9.0.0)</dd></dl>
</li>
</ul>
<a name="extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, java.lang.String, jebl.evolution.sequences.GeneticCode, int, boolean, java.util.List, boolean, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractIntervals</h4>
<pre><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;extractIntervals(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                                                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;overrideName,
                                                  <a href="http://jebl.sourceforge.net/doc/api/jebl/evolution/sequences/GeneticCode.html?is-external=true" title="class or interface in jebl.evolution.sequences">GeneticCode</a>&nbsp;geneticCode,
                                                  int&nbsp;translationFrame,
                                                  boolean&nbsp;residuesOnly,
                                                  <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceAnnotationInterval.html" title="class in com.biomatters.geneious.publicapi.documents.sequence">SequenceAnnotationInterval</a>&gt;&nbsp;intervals,
                                                  boolean&nbsp;allowReverseComplement,
                                                  boolean&nbsp;removeGaps)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a> instead.</i></div>
<div class="block">Extracts a group of intervals to a new sequence document</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sourceDocument</code> - To extract from</dd><dd><code>geneticCode</code> - translates from nucleotide's to amino acids if specified, or performs no translation if this parameter is null</dd><dd><code>translationFrame</code> - offset for each interval (typically 0, 1 or 2 for DNA). If not translating, this should be 0.</dd><dd><code>residuesOnly</code> - whether to extract only residues or to include the annotations.</dd><dd><code>intervals</code> - to extract</dd><dd><code>allowReverseComplement</code> - if false, automatic reverse complementing will not happen. If true,
 intervals of the form (from, to) where from >= to will be reverse complemented.</dd><dd><code>overrideName</code> - name of new sequence. If not given will be derived from seqeunce name and type of
 extraction</dd><dd><code>removeGaps</code> - if true remove any gaps</dd>
<dt><span class="strong">Returns:</span></dt><dd>the newly generated sequence document.</dd></dl>
</li>
</ul>
<a name="extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractIntervals</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;extractIntervals(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                                       <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)</pre>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sourceDocument</code> - The sequence to extract from</dd><dd><code>extractionOptions</code> - the options to use when extracting.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the extracted sequence document.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang">RuntimeException</a></code> - whose cause is an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a> if there is insufficient memory to load the annotations in tracks</dd></dl>
</li>
</ul>
<a name="extractIntervalsOrThrow(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractIntervalsOrThrow</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence">DefaultSequenceDocument</a>&nbsp;extractIntervalsOrThrow(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                                              <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)
                                                       throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sourceDocument</code> - The sequence to extract from</dd><dd><code>extractionOptions</code> - the options to use when extracting.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the extracted sequence document.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if there is insufficient memory to load the annotations in tracks</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.800 (Geneious 8.0.0)</dd></dl>
</li>
</ul>
<a name="extractOrThrow(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractOrThrow</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;extractOrThrow(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                              <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)
                                       throws <a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></pre>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document. This differs from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a>
 only in that the returned document will not necessarily be a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultSequenceDocument</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sourceDocument</code> - The sequence to extract from</dd><dd><code>extractionOptions</code> - the options to use when extracting. The same extractionOptions should be reused for calls
 that perform many extractions on the same sequence as this allows caching of sequence properties and therefore improved performance.
 Methods like <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html#setIntervals(java.util.List)"><code>SequenceExtractionUtilities.ExtractionOptions.setIntervals(java.util.List)</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html#setOverrideName(java.lang.String)"><code>SequenceExtractionUtilities.ExtractionOptions.setOverrideName(String)</code></a>
 can be used between each call to this method.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the extracted sequence document, or null if <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html#progressListener"><code>SequenceExtractionUtilities.ExtractionOptions.progressListener</code></a> was canceled.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../com/biomatters/geneious/publicapi/plugin/DocumentOperationException.html" title="class in com.biomatters.geneious.publicapi.plugin">DocumentOperationException</a></code> - if there is insufficient memory to load the annotations in tracks</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.800 (Geneious 8.0.0)</dd></dl>
</li>
</ul>
<a name="extract(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extract</h4>
<pre>public static&nbsp;<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;extract(<a href="../../../../../com/biomatters/geneious/publicapi/documents/sequence/SequenceDocument.html" title="interface in com.biomatters.geneious.publicapi.documents.sequence">SequenceDocument</a>&nbsp;sourceDocument,
                       <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations">SequenceExtractionUtilities.ExtractionOptions</a>&nbsp;extractionOptions)</pre>
<div class="block">Extracts one or more intervals from a sequence as a new sequence document. This differs from <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html#extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)"><code>extractIntervals(com.biomatters.geneious.publicapi.documents.sequence.SequenceDocument, com.biomatters.geneious.publicapi.implementations.SequenceExtractionUtilities.ExtractionOptions)</code></a>
 only in that the returned document will not necessarily be a <a href="../../../../../com/biomatters/geneious/publicapi/implementations/sequence/DefaultSequenceDocument.html" title="class in com.biomatters.geneious.publicapi.implementations.sequence"><code>DefaultSequenceDocument</code></a></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sourceDocument</code> - The sequence to extract from</dd><dd><code>extractionOptions</code> - the options to use when extracting. The same extractionOptions should be reused for calls
 that perform many extractions on the same sequence as this allows caching of sequence properties and therefore improved performance.
 Methods like <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html#setIntervals(java.util.List)"><code>SequenceExtractionUtilities.ExtractionOptions.setIntervals(java.util.List)</code></a> and <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html#setOverrideName(java.lang.String)"><code>SequenceExtractionUtilities.ExtractionOptions.setOverrideName(String)</code></a>
 can be used between each call to this method.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the extracted sequence document, or null if <a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html#progressListener"><code>SequenceExtractionUtilities.ExtractionOptions.progressListener</code></a> was canceled.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang">RuntimeException</a></code> - whose cause is an <a href="../../../../../com/biomatters/geneious/publicapi/documents/XMLSerializationException.html" title="class in com.biomatters.geneious.publicapi.documents"><code>XMLSerializationException</code></a> if there is insufficient memory to load the annotations in tracks</dd><dt><span class="strong">Since:</span></dt>
  <dd>API 4.14 (Geneious 5.1)</dd></dl>
</li>
</ul>
<a name="getExtractionName(java.lang.String, java.lang.String, java.lang.String, boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getExtractionName</h4>
<pre>public static&nbsp;<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getExtractionName(<a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;extractionType,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;annotationName,
                       <a href="http://java.sun.com/j2se/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequenceName,
                       boolean&nbsp;appendExtractionType)</pre>
<div class="block">Create a name for an extracted region</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>extractionType</code> - eg. "reverse complement", "translation" or "extraction". Can be others.</dd><dd><code>annotationName</code> - the name of the annotation being extracted, or null if not extracting an annotation</dd><dd><code>sequenceName</code> - the name of the sequence being extracted from</dd><dd><code>appendExtractionType</code> - true if the extraction type string should be appended. eg. this would be false if the entire sequence is being extracted from an alignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>a name for the resulting extracted sequence (or alignment).</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceExtractionUtilities.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/Percentage.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.ExtractionOptions.html" title="class in com.biomatters.geneious.publicapi.implementations"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/biomatters/geneious/publicapi/implementations/SequenceExtractionUtilities.html" target="_top">Frames</a></li>
<li><a href="SequenceExtractionUtilities.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
